{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "0807aa44-7dd4-4fce-bdcd-8420508deaa1",
   "metadata": {
    "tags": [
     "parameters",
     "remove-input"
    ]
   },
   "outputs": [],
   "source": [
    "# Notebook parameters. Values here are for development only and\n",
    "# will be overridden when running via snakemake and papermill.\n",
    "\n",
    "config_file = \"../../../config/agam.yaml\"\n",
    "# config_file = \"../../../config/afun.yaml\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "01e7c9f6-baee-47dc-9aff-67aaa272787c",
   "metadata": {
    "tags": [
     "remove-input"
    ]
   },
   "outputs": [
    {
     "data": {
      "application/javascript": [
       "'use strict';\n",
       "(function(root) {\n",
       "  function now() {\n",
       "    return new Date();\n",
       "  }\n",
       "\n",
       "  const force = true;\n",
       "\n",
       "  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n",
       "    root._bokeh_onload_callbacks = [];\n",
       "    root._bokeh_is_loading = undefined;\n",
       "  }\n",
       "\n",
       "const JS_MIME_TYPE = 'application/javascript';\n",
       "  const HTML_MIME_TYPE = 'text/html';\n",
       "  const EXEC_MIME_TYPE = 'application/vnd.bokehjs_exec.v0+json';\n",
       "  const CLASS_NAME = 'output_bokeh rendered_html';\n",
       "\n",
       "  /**\n",
       "   * Render data to the DOM node\n",
       "   */\n",
       "  function render(props, node) {\n",
       "    const script = document.createElement(\"script\");\n",
       "    node.appendChild(script);\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when an output is cleared or removed\n",
       "   */\n",
       "  function handleClearOutput(event, handle) {\n",
       "    function drop(id) {\n",
       "      const view = Bokeh.index.get_by_id(id)\n",
       "      if (view != null) {\n",
       "        view.model.document.clear()\n",
       "        Bokeh.index.delete(view)\n",
       "      }\n",
       "    }\n",
       "\n",
       "    const cell = handle.cell;\n",
       "\n",
       "    const id = cell.output_area._bokeh_element_id;\n",
       "    const server_id = cell.output_area._bokeh_server_id;\n",
       "\n",
       "    // Clean up Bokeh references\n",
       "    if (id != null) {\n",
       "      drop(id)\n",
       "    }\n",
       "\n",
       "    if (server_id !== undefined) {\n",
       "      // Clean up Bokeh references\n",
       "      const cmd_clean = \"from bokeh.io.state import curstate; print(curstate().uuid_to_server['\" + server_id + \"'].get_sessions()[0].document.roots[0]._id)\";\n",
       "      cell.notebook.kernel.execute(cmd_clean, {\n",
       "        iopub: {\n",
       "          output: function(msg) {\n",
       "            const id = msg.content.text.trim()\n",
       "            drop(id)\n",
       "          }\n",
       "        }\n",
       "      });\n",
       "      // Destroy server and session\n",
       "      const cmd_destroy = \"import bokeh.io.notebook as ion; ion.destroy_server('\" + server_id + \"')\";\n",
       "      cell.notebook.kernel.execute(cmd_destroy);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when a new output is added\n",
       "   */\n",
       "  function handleAddOutput(event, handle) {\n",
       "    const output_area = handle.output_area;\n",
       "    const output = handle.output;\n",
       "\n",
       "    // limit handleAddOutput to display_data with EXEC_MIME_TYPE content only\n",
       "    if ((output.output_type != \"display_data\") || (!Object.prototype.hasOwnProperty.call(output.data, EXEC_MIME_TYPE))) {\n",
       "      return\n",
       "    }\n",
       "\n",
       "    const toinsert = output_area.element.find(\".\" + CLASS_NAME.split(' ')[0]);\n",
       "\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"id\"] !== undefined) {\n",
       "      toinsert[toinsert.length - 1].firstChild.textContent = output.data[JS_MIME_TYPE];\n",
       "      // store reference to embed id on output_area\n",
       "      output_area._bokeh_element_id = output.metadata[EXEC_MIME_TYPE][\"id\"];\n",
       "    }\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"server_id\"] !== undefined) {\n",
       "      const bk_div = document.createElement(\"div\");\n",
       "      bk_div.innerHTML = output.data[HTML_MIME_TYPE];\n",
       "      const script_attrs = bk_div.children[0].attributes;\n",
       "      for (let i = 0; i < script_attrs.length; i++) {\n",
       "        toinsert[toinsert.length - 1].firstChild.setAttribute(script_attrs[i].name, script_attrs[i].value);\n",
       "        toinsert[toinsert.length - 1].firstChild.textContent = bk_div.children[0].textContent\n",
       "      }\n",
       "      // store reference to server id on output_area\n",
       "      output_area._bokeh_server_id = output.metadata[EXEC_MIME_TYPE][\"server_id\"];\n",
       "    }\n",
       "  }\n",
       "\n",
       "  function register_renderer(events, OutputArea) {\n",
       "\n",
       "    function append_mime(data, metadata, element) {\n",
       "      // create a DOM node to render to\n",
       "      const toinsert = this.create_output_subarea(\n",
       "        metadata,\n",
       "        CLASS_NAME,\n",
       "        EXEC_MIME_TYPE\n",
       "      );\n",
       "      this.keyboard_manager.register_events(toinsert);\n",
       "      // Render to node\n",
       "      const props = {data: data, metadata: metadata[EXEC_MIME_TYPE]};\n",
       "      render(props, toinsert[toinsert.length - 1]);\n",
       "      element.append(toinsert);\n",
       "      return toinsert\n",
       "    }\n",
       "\n",
       "    /* Handle when an output is cleared or removed */\n",
       "    events.on('clear_output.CodeCell', handleClearOutput);\n",
       "    events.on('delete.Cell', handleClearOutput);\n",
       "\n",
       "    /* Handle when a new output is added */\n",
       "    events.on('output_added.OutputArea', handleAddOutput);\n",
       "\n",
       "    /**\n",
       "     * Register the mime type and append_mime function with output_area\n",
       "     */\n",
       "    OutputArea.prototype.register_mime_type(EXEC_MIME_TYPE, append_mime, {\n",
       "      /* Is output safe? */\n",
       "      safe: true,\n",
       "      /* Index of renderer in `output_area.display_order` */\n",
       "      index: 0\n",
       "    });\n",
       "  }\n",
       "\n",
       "  // register the mime type if in Jupyter Notebook environment and previously unregistered\n",
       "  if (root.Jupyter !== undefined) {\n",
       "    const events = require('base/js/events');\n",
       "    const OutputArea = require('notebook/js/outputarea').OutputArea;\n",
       "\n",
       "    if (OutputArea.prototype.mime_types().indexOf(EXEC_MIME_TYPE) == -1) {\n",
       "      register_renderer(events, OutputArea);\n",
       "    }\n",
       "  }\n",
       "  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n",
       "    root._bokeh_timeout = Date.now() + 5000;\n",
       "    root._bokeh_failed_load = false;\n",
       "  }\n",
       "\n",
       "  const NB_LOAD_WARNING = {'data': {'text/html':\n",
       "     \"<div style='background-color: #fdd'>\\n\"+\n",
       "     \"<p>\\n\"+\n",
       "     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n",
       "     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n",
       "     \"</p>\\n\"+\n",
       "     \"<ul>\\n\"+\n",
       "     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n",
       "     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n",
       "     \"</ul>\\n\"+\n",
       "     \"<code>\\n\"+\n",
       "     \"from bokeh.resources import INLINE\\n\"+\n",
       "     \"output_notebook(resources=INLINE)\\n\"+\n",
       "     \"</code>\\n\"+\n",
       "     \"</div>\"}};\n",
       "\n",
       "  function display_loaded(error = null) {\n",
       "    const el = document.getElementById(null);\n",
       "    if (el != null) {\n",
       "      const html = (() => {\n",
       "        if (typeof root.Bokeh === \"undefined\") {\n",
       "          if (error == null) {\n",
       "            return \"BokehJS is loading ...\";\n",
       "          } else {\n",
       "            return \"BokehJS failed to load.\";\n",
       "          }\n",
       "        } else {\n",
       "          const prefix = `BokehJS ${root.Bokeh.version}`;\n",
       "          if (error == null) {\n",
       "            return `${prefix} successfully loaded.`;\n",
       "          } else {\n",
       "            return `${prefix} <b>encountered errors</b> while loading and may not function as expected.`;\n",
       "          }\n",
       "        }\n",
       "      })();\n",
       "      el.innerHTML = html;\n",
       "\n",
       "      if (error != null) {\n",
       "        const wrapper = document.createElement(\"div\");\n",
       "        wrapper.style.overflow = \"auto\";\n",
       "        wrapper.style.height = \"5em\";\n",
       "        wrapper.style.resize = \"vertical\";\n",
       "        const content = document.createElement(\"div\");\n",
       "        content.style.fontFamily = \"monospace\";\n",
       "        content.style.whiteSpace = \"pre-wrap\";\n",
       "        content.style.backgroundColor = \"rgb(255, 221, 221)\";\n",
       "        content.textContent = error.stack ?? error.toString();\n",
       "        wrapper.append(content);\n",
       "        el.append(wrapper);\n",
       "      }\n",
       "    } else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(() => display_loaded(error), 100);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  function run_callbacks() {\n",
       "    try {\n",
       "      root._bokeh_onload_callbacks.forEach(function(callback) {\n",
       "        if (callback != null)\n",
       "          callback();\n",
       "      });\n",
       "    } finally {\n",
       "      delete root._bokeh_onload_callbacks\n",
       "    }\n",
       "    console.debug(\"Bokeh: all callbacks have finished\");\n",
       "  }\n",
       "\n",
       "  function load_libs(css_urls, js_urls, callback) {\n",
       "    if (css_urls == null) css_urls = [];\n",
       "    if (js_urls == null) js_urls = [];\n",
       "\n",
       "    root._bokeh_onload_callbacks.push(callback);\n",
       "    if (root._bokeh_is_loading > 0) {\n",
       "      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n",
       "      return null;\n",
       "    }\n",
       "    if (js_urls == null || js_urls.length === 0) {\n",
       "      run_callbacks();\n",
       "      return null;\n",
       "    }\n",
       "    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n",
       "    root._bokeh_is_loading = css_urls.length + js_urls.length;\n",
       "\n",
       "    function on_load() {\n",
       "      root._bokeh_is_loading--;\n",
       "      if (root._bokeh_is_loading === 0) {\n",
       "        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n",
       "        run_callbacks()\n",
       "      }\n",
       "    }\n",
       "\n",
       "    function on_error(url) {\n",
       "      console.error(\"failed to load \" + url);\n",
       "    }\n",
       "\n",
       "    for (let i = 0; i < css_urls.length; i++) {\n",
       "      const url = css_urls[i];\n",
       "      const element = document.createElement(\"link\");\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error.bind(null, url);\n",
       "      element.rel = \"stylesheet\";\n",
       "      element.type = \"text/css\";\n",
       "      element.href = url;\n",
       "      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n",
       "      document.body.appendChild(element);\n",
       "    }\n",
       "\n",
       "    for (let i = 0; i < js_urls.length; i++) {\n",
       "      const url = js_urls[i];\n",
       "      const element = document.createElement('script');\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error.bind(null, url);\n",
       "      element.async = false;\n",
       "      element.src = url;\n",
       "      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n",
       "      document.head.appendChild(element);\n",
       "    }\n",
       "  };\n",
       "\n",
       "  function inject_raw_css(css) {\n",
       "    const element = document.createElement(\"style\");\n",
       "    element.appendChild(document.createTextNode(css));\n",
       "    document.body.appendChild(element);\n",
       "  }\n",
       "\n",
       "  const js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-3.6.3.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-3.6.3.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-3.6.3.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-3.6.3.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-mathjax-3.6.3.min.js\"];\n",
       "  const css_urls = [];\n",
       "\n",
       "  const inline_js = [    function(Bokeh) {\n",
       "      Bokeh.set_log_level(\"info\");\n",
       "    },\n",
       "function(Bokeh) {\n",
       "    }\n",
       "  ];\n",
       "\n",
       "  function run_inline_js() {\n",
       "    if (root.Bokeh !== undefined || force === true) {\n",
       "      try {\n",
       "            for (let i = 0; i < inline_js.length; i++) {\n",
       "      inline_js[i].call(root, root.Bokeh);\n",
       "    }\n",
       "\n",
       "      } catch (error) {throw error;\n",
       "      }} else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(run_inline_js, 100);\n",
       "    } else if (!root._bokeh_failed_load) {\n",
       "      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n",
       "      root._bokeh_failed_load = true;\n",
       "    } else if (force !== true) {\n",
       "      const cell = $(document.getElementById(null)).parents('.cell').data().cell;\n",
       "      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n",
       "    }\n",
       "  }\n",
       "\n",
       "  if (root._bokeh_is_loading === 0) {\n",
       "    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n",
       "    run_inline_js();\n",
       "  } else {\n",
       "    load_libs(css_urls, js_urls, function() {\n",
       "      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n",
       "      run_inline_js();\n",
       "    });\n",
       "  }\n",
       "}(window));"
      ],
      "application/vnd.bokehjs_load.v0+json": "'use strict';\n(function(root) {\n  function now() {\n    return new Date();\n  }\n\n  const force = true;\n\n  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n    root._bokeh_onload_callbacks = [];\n    root._bokeh_is_loading = undefined;\n  }\n\n\n  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n    root._bokeh_timeout = Date.now() + 5000;\n    root._bokeh_failed_load = false;\n  }\n\n  const NB_LOAD_WARNING = {'data': {'text/html':\n     \"<div style='background-color: #fdd'>\\n\"+\n     \"<p>\\n\"+\n     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n     \"</p>\\n\"+\n     \"<ul>\\n\"+\n     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n     \"</ul>\\n\"+\n     \"<code>\\n\"+\n     \"from bokeh.resources import INLINE\\n\"+\n     \"output_notebook(resources=INLINE)\\n\"+\n     \"</code>\\n\"+\n     \"</div>\"}};\n\n  function display_loaded(error = null) {\n    const el = document.getElementById(null);\n    if (el != null) {\n      const html = (() => {\n        if (typeof root.Bokeh === \"undefined\") {\n          if (error == null) {\n            return \"BokehJS is loading ...\";\n          } else {\n            return \"BokehJS failed to load.\";\n          }\n        } else {\n          const prefix = `BokehJS ${root.Bokeh.version}`;\n          if (error == null) {\n            return `${prefix} successfully loaded.`;\n          } else {\n            return `${prefix} <b>encountered errors</b> while loading and may not function as expected.`;\n          }\n        }\n      })();\n      el.innerHTML = html;\n\n      if (error != null) {\n        const wrapper = document.createElement(\"div\");\n        wrapper.style.overflow = \"auto\";\n        wrapper.style.height = \"5em\";\n        wrapper.style.resize = \"vertical\";\n        const content = document.createElement(\"div\");\n        content.style.fontFamily = \"monospace\";\n        content.style.whiteSpace = \"pre-wrap\";\n        content.style.backgroundColor = \"rgb(255, 221, 221)\";\n        content.textContent = error.stack ?? error.toString();\n        wrapper.append(content);\n        el.append(wrapper);\n      }\n    } else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(() => display_loaded(error), 100);\n    }\n  }\n\n  function run_callbacks() {\n    try {\n      root._bokeh_onload_callbacks.forEach(function(callback) {\n        if (callback != null)\n          callback();\n      });\n    } finally {\n      delete root._bokeh_onload_callbacks\n    }\n    console.debug(\"Bokeh: all callbacks have finished\");\n  }\n\n  function load_libs(css_urls, js_urls, callback) {\n    if (css_urls == null) css_urls = [];\n    if (js_urls == null) js_urls = [];\n\n    root._bokeh_onload_callbacks.push(callback);\n    if (root._bokeh_is_loading > 0) {\n      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n      return null;\n    }\n    if (js_urls == null || js_urls.length === 0) {\n      run_callbacks();\n      return null;\n    }\n    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n    root._bokeh_is_loading = css_urls.length + js_urls.length;\n\n    function on_load() {\n      root._bokeh_is_loading--;\n      if (root._bokeh_is_loading === 0) {\n        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n        run_callbacks()\n      }\n    }\n\n    function on_error(url) {\n      console.error(\"failed to load \" + url);\n    }\n\n    for (let i = 0; i < css_urls.length; i++) {\n      const url = css_urls[i];\n      const element = document.createElement(\"link\");\n      element.onload = on_load;\n      element.onerror = on_error.bind(null, url);\n      element.rel = \"stylesheet\";\n      element.type = \"text/css\";\n      element.href = url;\n      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n      document.body.appendChild(element);\n    }\n\n    for (let i = 0; i < js_urls.length; i++) {\n      const url = js_urls[i];\n      const element = document.createElement('script');\n      element.onload = on_load;\n      element.onerror = on_error.bind(null, url);\n      element.async = false;\n      element.src = url;\n      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n      document.head.appendChild(element);\n    }\n  };\n\n  function inject_raw_css(css) {\n    const element = document.createElement(\"style\");\n    element.appendChild(document.createTextNode(css));\n    document.body.appendChild(element);\n  }\n\n  const js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-3.6.3.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-3.6.3.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-3.6.3.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-3.6.3.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-mathjax-3.6.3.min.js\"];\n  const css_urls = [];\n\n  const inline_js = [    function(Bokeh) {\n      Bokeh.set_log_level(\"info\");\n    },\nfunction(Bokeh) {\n    }\n  ];\n\n  function run_inline_js() {\n    if (root.Bokeh !== undefined || force === true) {\n      try {\n            for (let i = 0; i < inline_js.length; i++) {\n      inline_js[i].call(root, root.Bokeh);\n    }\n\n      } catch (error) {throw error;\n      }} else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(run_inline_js, 100);\n    } else if (!root._bokeh_failed_load) {\n      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n      root._bokeh_failed_load = true;\n    } else if (force !== true) {\n      const cell = $(document.getElementById(null)).parents('.cell').data().cell;\n      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n    }\n  }\n\n  if (root._bokeh_is_loading === 0) {\n    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n    run_inline_js();\n  } else {\n    load_libs(css_urls, js_urls, function() {\n      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n      run_inline_js();\n    });\n  }\n}(window));"
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from bokeh.io import output_notebook\n",
    "import malariagen_data\n",
    "from IPython.display import Markdown\n",
    "from selection_atlas.setup import AtlasSetup\n",
    "from selection_atlas.page_utils import AtlasPageUtils\n",
    "\n",
    "# Initialise the atlas setup.\n",
    "setup = AtlasSetup(config_file)\n",
    "page_utils = AtlasPageUtils(setup=setup)\n",
    "\n",
    "# N.B., do not add the \"remove-output\" tag to this cell!!! If you do,\n",
    "# the bokeh javascript libraries will not get loaded in the generated\n",
    "# HTML page. The call to output_notebook() injects javascript in the\n",
    "# cell output which triggers the bokeh javascript libraries to be loaded\n",
    "# in the page.\n",
    "output_notebook(hide_banner=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0442c1a1-7dd1-4b5b-85e1-a69d2fb803d5",
   "metadata": {},
   "source": [
    "# Methods"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "77233d5d-c5ff-439d-9a99-94721d64a36b",
   "metadata": {},
   "source": [
    "## Data sources"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "713f44fc-33f1-4cd0-8534-0067bf3d50af",
   "metadata": {
    "tags": [
     "remove-input"
    ]
   },
   "outputs": [
    {
     "data": {
      "application/javascript": [
       "'use strict';\n",
       "(function(root) {\n",
       "  function now() {\n",
       "    return new Date();\n",
       "  }\n",
       "\n",
       "  const force = true;\n",
       "\n",
       "  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n",
       "    root._bokeh_onload_callbacks = [];\n",
       "    root._bokeh_is_loading = undefined;\n",
       "  }\n",
       "\n",
       "const JS_MIME_TYPE = 'application/javascript';\n",
       "  const HTML_MIME_TYPE = 'text/html';\n",
       "  const EXEC_MIME_TYPE = 'application/vnd.bokehjs_exec.v0+json';\n",
       "  const CLASS_NAME = 'output_bokeh rendered_html';\n",
       "\n",
       "  /**\n",
       "   * Render data to the DOM node\n",
       "   */\n",
       "  function render(props, node) {\n",
       "    const script = document.createElement(\"script\");\n",
       "    node.appendChild(script);\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when an output is cleared or removed\n",
       "   */\n",
       "  function handleClearOutput(event, handle) {\n",
       "    function drop(id) {\n",
       "      const view = Bokeh.index.get_by_id(id)\n",
       "      if (view != null) {\n",
       "        view.model.document.clear()\n",
       "        Bokeh.index.delete(view)\n",
       "      }\n",
       "    }\n",
       "\n",
       "    const cell = handle.cell;\n",
       "\n",
       "    const id = cell.output_area._bokeh_element_id;\n",
       "    const server_id = cell.output_area._bokeh_server_id;\n",
       "\n",
       "    // Clean up Bokeh references\n",
       "    if (id != null) {\n",
       "      drop(id)\n",
       "    }\n",
       "\n",
       "    if (server_id !== undefined) {\n",
       "      // Clean up Bokeh references\n",
       "      const cmd_clean = \"from bokeh.io.state import curstate; print(curstate().uuid_to_server['\" + server_id + \"'].get_sessions()[0].document.roots[0]._id)\";\n",
       "      cell.notebook.kernel.execute(cmd_clean, {\n",
       "        iopub: {\n",
       "          output: function(msg) {\n",
       "            const id = msg.content.text.trim()\n",
       "            drop(id)\n",
       "          }\n",
       "        }\n",
       "      });\n",
       "      // Destroy server and session\n",
       "      const cmd_destroy = \"import bokeh.io.notebook as ion; ion.destroy_server('\" + server_id + \"')\";\n",
       "      cell.notebook.kernel.execute(cmd_destroy);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when a new output is added\n",
       "   */\n",
       "  function handleAddOutput(event, handle) {\n",
       "    const output_area = handle.output_area;\n",
       "    const output = handle.output;\n",
       "\n",
       "    // limit handleAddOutput to display_data with EXEC_MIME_TYPE content only\n",
       "    if ((output.output_type != \"display_data\") || (!Object.prototype.hasOwnProperty.call(output.data, EXEC_MIME_TYPE))) {\n",
       "      return\n",
       "    }\n",
       "\n",
       "    const toinsert = output_area.element.find(\".\" + CLASS_NAME.split(' ')[0]);\n",
       "\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"id\"] !== undefined) {\n",
       "      toinsert[toinsert.length - 1].firstChild.textContent = output.data[JS_MIME_TYPE];\n",
       "      // store reference to embed id on output_area\n",
       "      output_area._bokeh_element_id = output.metadata[EXEC_MIME_TYPE][\"id\"];\n",
       "    }\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"server_id\"] !== undefined) {\n",
       "      const bk_div = document.createElement(\"div\");\n",
       "      bk_div.innerHTML = output.data[HTML_MIME_TYPE];\n",
       "      const script_attrs = bk_div.children[0].attributes;\n",
       "      for (let i = 0; i < script_attrs.length; i++) {\n",
       "        toinsert[toinsert.length - 1].firstChild.setAttribute(script_attrs[i].name, script_attrs[i].value);\n",
       "        toinsert[toinsert.length - 1].firstChild.textContent = bk_div.children[0].textContent\n",
       "      }\n",
       "      // store reference to server id on output_area\n",
       "      output_area._bokeh_server_id = output.metadata[EXEC_MIME_TYPE][\"server_id\"];\n",
       "    }\n",
       "  }\n",
       "\n",
       "  function register_renderer(events, OutputArea) {\n",
       "\n",
       "    function append_mime(data, metadata, element) {\n",
       "      // create a DOM node to render to\n",
       "      const toinsert = this.create_output_subarea(\n",
       "        metadata,\n",
       "        CLASS_NAME,\n",
       "        EXEC_MIME_TYPE\n",
       "      );\n",
       "      this.keyboard_manager.register_events(toinsert);\n",
       "      // Render to node\n",
       "      const props = {data: data, metadata: metadata[EXEC_MIME_TYPE]};\n",
       "      render(props, toinsert[toinsert.length - 1]);\n",
       "      element.append(toinsert);\n",
       "      return toinsert\n",
       "    }\n",
       "\n",
       "    /* Handle when an output is cleared or removed */\n",
       "    events.on('clear_output.CodeCell', handleClearOutput);\n",
       "    events.on('delete.Cell', handleClearOutput);\n",
       "\n",
       "    /* Handle when a new output is added */\n",
       "    events.on('output_added.OutputArea', handleAddOutput);\n",
       "\n",
       "    /**\n",
       "     * Register the mime type and append_mime function with output_area\n",
       "     */\n",
       "    OutputArea.prototype.register_mime_type(EXEC_MIME_TYPE, append_mime, {\n",
       "      /* Is output safe? */\n",
       "      safe: true,\n",
       "      /* Index of renderer in `output_area.display_order` */\n",
       "      index: 0\n",
       "    });\n",
       "  }\n",
       "\n",
       "  // register the mime type if in Jupyter Notebook environment and previously unregistered\n",
       "  if (root.Jupyter !== undefined) {\n",
       "    const events = require('base/js/events');\n",
       "    const OutputArea = require('notebook/js/outputarea').OutputArea;\n",
       "\n",
       "    if (OutputArea.prototype.mime_types().indexOf(EXEC_MIME_TYPE) == -1) {\n",
       "      register_renderer(events, OutputArea);\n",
       "    }\n",
       "  }\n",
       "  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n",
       "    root._bokeh_timeout = Date.now() + 5000;\n",
       "    root._bokeh_failed_load = false;\n",
       "  }\n",
       "\n",
       "  const NB_LOAD_WARNING = {'data': {'text/html':\n",
       "     \"<div style='background-color: #fdd'>\\n\"+\n",
       "     \"<p>\\n\"+\n",
       "     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n",
       "     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n",
       "     \"</p>\\n\"+\n",
       "     \"<ul>\\n\"+\n",
       "     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n",
       "     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n",
       "     \"</ul>\\n\"+\n",
       "     \"<code>\\n\"+\n",
       "     \"from bokeh.resources import INLINE\\n\"+\n",
       "     \"output_notebook(resources=INLINE)\\n\"+\n",
       "     \"</code>\\n\"+\n",
       "     \"</div>\"}};\n",
       "\n",
       "  function display_loaded(error = null) {\n",
       "    const el = document.getElementById(null);\n",
       "    if (el != null) {\n",
       "      const html = (() => {\n",
       "        if (typeof root.Bokeh === \"undefined\") {\n",
       "          if (error == null) {\n",
       "            return \"BokehJS is loading ...\";\n",
       "          } else {\n",
       "            return \"BokehJS failed to load.\";\n",
       "          }\n",
       "        } else {\n",
       "          const prefix = `BokehJS ${root.Bokeh.version}`;\n",
       "          if (error == null) {\n",
       "            return `${prefix} successfully loaded.`;\n",
       "          } else {\n",
       "            return `${prefix} <b>encountered errors</b> while loading and may not function as expected.`;\n",
       "          }\n",
       "        }\n",
       "      })();\n",
       "      el.innerHTML = html;\n",
       "\n",
       "      if (error != null) {\n",
       "        const wrapper = document.createElement(\"div\");\n",
       "        wrapper.style.overflow = \"auto\";\n",
       "        wrapper.style.height = \"5em\";\n",
       "        wrapper.style.resize = \"vertical\";\n",
       "        const content = document.createElement(\"div\");\n",
       "        content.style.fontFamily = \"monospace\";\n",
       "        content.style.whiteSpace = \"pre-wrap\";\n",
       "        content.style.backgroundColor = \"rgb(255, 221, 221)\";\n",
       "        content.textContent = error.stack ?? error.toString();\n",
       "        wrapper.append(content);\n",
       "        el.append(wrapper);\n",
       "      }\n",
       "    } else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(() => display_loaded(error), 100);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  function run_callbacks() {\n",
       "    try {\n",
       "      root._bokeh_onload_callbacks.forEach(function(callback) {\n",
       "        if (callback != null)\n",
       "          callback();\n",
       "      });\n",
       "    } finally {\n",
       "      delete root._bokeh_onload_callbacks\n",
       "    }\n",
       "    console.debug(\"Bokeh: all callbacks have finished\");\n",
       "  }\n",
       "\n",
       "  function load_libs(css_urls, js_urls, callback) {\n",
       "    if (css_urls == null) css_urls = [];\n",
       "    if (js_urls == null) js_urls = [];\n",
       "\n",
       "    root._bokeh_onload_callbacks.push(callback);\n",
       "    if (root._bokeh_is_loading > 0) {\n",
       "      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n",
       "      return null;\n",
       "    }\n",
       "    if (js_urls == null || js_urls.length === 0) {\n",
       "      run_callbacks();\n",
       "      return null;\n",
       "    }\n",
       "    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n",
       "    root._bokeh_is_loading = css_urls.length + js_urls.length;\n",
       "\n",
       "    function on_load() {\n",
       "      root._bokeh_is_loading--;\n",
       "      if (root._bokeh_is_loading === 0) {\n",
       "        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n",
       "        run_callbacks()\n",
       "      }\n",
       "    }\n",
       "\n",
       "    function on_error(url) {\n",
       "      console.error(\"failed to load \" + url);\n",
       "    }\n",
       "\n",
       "    for (let i = 0; i < css_urls.length; i++) {\n",
       "      const url = css_urls[i];\n",
       "      const element = document.createElement(\"link\");\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error.bind(null, url);\n",
       "      element.rel = \"stylesheet\";\n",
       "      element.type = \"text/css\";\n",
       "      element.href = url;\n",
       "      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n",
       "      document.body.appendChild(element);\n",
       "    }\n",
       "\n",
       "    for (let i = 0; i < js_urls.length; i++) {\n",
       "      const url = js_urls[i];\n",
       "      const element = document.createElement('script');\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error.bind(null, url);\n",
       "      element.async = false;\n",
       "      element.src = url;\n",
       "      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n",
       "      document.head.appendChild(element);\n",
       "    }\n",
       "  };\n",
       "\n",
       "  function inject_raw_css(css) {\n",
       "    const element = document.createElement(\"style\");\n",
       "    element.appendChild(document.createTextNode(css));\n",
       "    document.body.appendChild(element);\n",
       "  }\n",
       "\n",
       "  const js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-3.6.3.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-3.6.3.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-3.6.3.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-3.6.3.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-mathjax-3.6.3.min.js\"];\n",
       "  const css_urls = [];\n",
       "\n",
       "  const inline_js = [    function(Bokeh) {\n",
       "      Bokeh.set_log_level(\"info\");\n",
       "    },\n",
       "function(Bokeh) {\n",
       "    }\n",
       "  ];\n",
       "\n",
       "  function run_inline_js() {\n",
       "    if (root.Bokeh !== undefined || force === true) {\n",
       "      try {\n",
       "            for (let i = 0; i < inline_js.length; i++) {\n",
       "      inline_js[i].call(root, root.Bokeh);\n",
       "    }\n",
       "\n",
       "      } catch (error) {throw error;\n",
       "      }} else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(run_inline_js, 100);\n",
       "    } else if (!root._bokeh_failed_load) {\n",
       "      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n",
       "      root._bokeh_failed_load = true;\n",
       "    } else if (force !== true) {\n",
       "      const cell = $(document.getElementById(null)).parents('.cell').data().cell;\n",
       "      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n",
       "    }\n",
       "  }\n",
       "\n",
       "  if (root._bokeh_is_loading === 0) {\n",
       "    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n",
       "    run_inline_js();\n",
       "  } else {\n",
       "    load_libs(css_urls, js_urls, function() {\n",
       "      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n",
       "      run_inline_js();\n",
       "    });\n",
       "  }\n",
       "}(window));"
      ],
      "application/vnd.bokehjs_load.v0+json": "'use strict';\n(function(root) {\n  function now() {\n    return new Date();\n  }\n\n  const force = true;\n\n  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n    root._bokeh_onload_callbacks = [];\n    root._bokeh_is_loading = undefined;\n  }\n\n\n  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n    root._bokeh_timeout = Date.now() + 5000;\n    root._bokeh_failed_load = false;\n  }\n\n  const NB_LOAD_WARNING = {'data': {'text/html':\n     \"<div style='background-color: #fdd'>\\n\"+\n     \"<p>\\n\"+\n     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n     \"</p>\\n\"+\n     \"<ul>\\n\"+\n     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n     \"</ul>\\n\"+\n     \"<code>\\n\"+\n     \"from bokeh.resources import INLINE\\n\"+\n     \"output_notebook(resources=INLINE)\\n\"+\n     \"</code>\\n\"+\n     \"</div>\"}};\n\n  function display_loaded(error = null) {\n    const el = document.getElementById(null);\n    if (el != null) {\n      const html = (() => {\n        if (typeof root.Bokeh === \"undefined\") {\n          if (error == null) {\n            return \"BokehJS is loading ...\";\n          } else {\n            return \"BokehJS failed to load.\";\n          }\n        } else {\n          const prefix = `BokehJS ${root.Bokeh.version}`;\n          if (error == null) {\n            return `${prefix} successfully loaded.`;\n          } else {\n            return `${prefix} <b>encountered errors</b> while loading and may not function as expected.`;\n          }\n        }\n      })();\n      el.innerHTML = html;\n\n      if (error != null) {\n        const wrapper = document.createElement(\"div\");\n        wrapper.style.overflow = \"auto\";\n        wrapper.style.height = \"5em\";\n        wrapper.style.resize = \"vertical\";\n        const content = document.createElement(\"div\");\n        content.style.fontFamily = \"monospace\";\n        content.style.whiteSpace = \"pre-wrap\";\n        content.style.backgroundColor = \"rgb(255, 221, 221)\";\n        content.textContent = error.stack ?? error.toString();\n        wrapper.append(content);\n        el.append(wrapper);\n      }\n    } else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(() => display_loaded(error), 100);\n    }\n  }\n\n  function run_callbacks() {\n    try {\n      root._bokeh_onload_callbacks.forEach(function(callback) {\n        if (callback != null)\n          callback();\n      });\n    } finally {\n      delete root._bokeh_onload_callbacks\n    }\n    console.debug(\"Bokeh: all callbacks have finished\");\n  }\n\n  function load_libs(css_urls, js_urls, callback) {\n    if (css_urls == null) css_urls = [];\n    if (js_urls == null) js_urls = [];\n\n    root._bokeh_onload_callbacks.push(callback);\n    if (root._bokeh_is_loading > 0) {\n      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n      return null;\n    }\n    if (js_urls == null || js_urls.length === 0) {\n      run_callbacks();\n      return null;\n    }\n    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n    root._bokeh_is_loading = css_urls.length + js_urls.length;\n\n    function on_load() {\n      root._bokeh_is_loading--;\n      if (root._bokeh_is_loading === 0) {\n        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n        run_callbacks()\n      }\n    }\n\n    function on_error(url) {\n      console.error(\"failed to load \" + url);\n    }\n\n    for (let i = 0; i < css_urls.length; i++) {\n      const url = css_urls[i];\n      const element = document.createElement(\"link\");\n      element.onload = on_load;\n      element.onerror = on_error.bind(null, url);\n      element.rel = \"stylesheet\";\n      element.type = \"text/css\";\n      element.href = url;\n      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n      document.body.appendChild(element);\n    }\n\n    for (let i = 0; i < js_urls.length; i++) {\n      const url = js_urls[i];\n      const element = document.createElement('script');\n      element.onload = on_load;\n      element.onerror = on_error.bind(null, url);\n      element.async = false;\n      element.src = url;\n      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n      document.head.appendChild(element);\n    }\n  };\n\n  function inject_raw_css(css) {\n    const element = document.createElement(\"style\");\n    element.appendChild(document.createTextNode(css));\n    document.body.appendChild(element);\n  }\n\n  const js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-3.6.3.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-3.6.3.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-3.6.3.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-3.6.3.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-mathjax-3.6.3.min.js\"];\n  const css_urls = [];\n\n  const inline_js = [    function(Bokeh) {\n      Bokeh.set_log_level(\"info\");\n    },\nfunction(Bokeh) {\n    }\n  ];\n\n  function run_inline_js() {\n    if (root.Bokeh !== undefined || force === true) {\n      try {\n            for (let i = 0; i < inline_js.length; i++) {\n      inline_js[i].call(root, root.Bokeh);\n    }\n\n      } catch (error) {throw error;\n      }} else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(run_inline_js, 100);\n    } else if (!root._bokeh_failed_load) {\n      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n      root._bokeh_failed_load = true;\n    } else if (force !== true) {\n      const cell = $(document.getElementById(null)).parents('.cell').data().cell;\n      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n    }\n  }\n\n  if (root._bokeh_is_loading === 0) {\n    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n    run_inline_js();\n  } else {\n    load_libs(css_urls, js_urls, function() {\n      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n      run_inline_js();\n    });\n  }\n}(window));"
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                     \r"
     ]
    }
   ],
   "source": [
    "df_samples = setup.sample_metadata()\n",
    "countries = df_samples[\"country\"].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a8c4f186-a0a3-4e00-9b49-246d238aa5ed",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/markdown": [
       "\n",
       "This report analyses genome variation data from the \n",
       "{term}`Malaria Vector Genome Observatory`. See Table 1 below for a \n",
       "complete list of the sample sets used in the current analysis version, \n",
       "with information about the corresponding contributors, data releases \n",
       "and citations. These sample sets provide data for a total of\n",
       "4,878 mosquitoes sampled from 25 countries. \n"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Markdown(f\"\"\"\n",
    "This report analyses genome variation data from the \n",
    "{{term}}`Malaria Vector Genome Observatory`. See Table 1 below for a \n",
    "complete list of the sample sets used in the current analysis version, \n",
    "with information about the corresponding contributors, data releases \n",
    "and citations. These sample sets provide data for a total of\n",
    "{len(df_samples):,} mosquitoes sampled from {len(countries)} countries. \n",
    "\"\"\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "35ee5710-6493-4eb7-97cb-5df742053204",
   "metadata": {
    "tags": [
     "remove-input"
    ]
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "#T_277a7 th {\n",
       "  text-align: left;\n",
       "}\n",
       "#T_277a7 td {\n",
       "  text-align: left;\n",
       "  font-weight: normal;\n",
       "}\n",
       "</style>\n",
       "<table id=\"T_277a7\">\n",
       "  <caption>Table 1. Data sources included in the current analysis version.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th id=\"T_277a7_level0_col0\" class=\"col_heading level0 col0\" >Sample Set</th>\n",
       "      <th id=\"T_277a7_level0_col1\" class=\"col_heading level0 col1\" >Study</th>\n",
       "      <th id=\"T_277a7_level0_col2\" class=\"col_heading level0 col2\" >Contributor</th>\n",
       "      <th id=\"T_277a7_level0_col3\" class=\"col_heading level0 col3\" >Data Release</th>\n",
       "      <th id=\"T_277a7_level0_col4\" class=\"col_heading level0 col4\" >Citation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row0_col0\" class=\"data row0 col0\" >AG1000G-AO</td>\n",
       "      <td id=\"T_277a7_row0_col1\" class=\"data row0 col1\" ><a href=\"https://www.malariagen.net/partner_study/AG1000G-AO\" target=\"_blank\">AG1000G-AO</a></td>\n",
       "      <td id=\"T_277a7_row0_col2\" class=\"data row0 col2\" >Joao Pinto</td>\n",
       "      <td id=\"T_277a7_row0_col3\" class=\"data row0 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.0.html\" target=\"_blank\">Ag3.0</a></td>\n",
       "      <td id=\"T_277a7_row0_col4\" class=\"data row0 col4\" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC6026373/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2017</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row1_col0\" class=\"data row1 col0\" >AG1000G-BF-A</td>\n",
       "      <td id=\"T_277a7_row1_col1\" class=\"data row1 col1\" ><a href=\"https://www.malariagen.net/partner_study/AG1000G-BF-1\" target=\"_blank\">AG1000G-BF-1</a></td>\n",
       "      <td id=\"T_277a7_row1_col2\" class=\"data row1 col2\" >Austin Burt</td>\n",
       "      <td id=\"T_277a7_row1_col3\" class=\"data row1 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.0.html\" target=\"_blank\">Ag3.0</a></td>\n",
       "      <td id=\"T_277a7_row1_col4\" class=\"data row1 col4\" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC6026373/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2017</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row2_col0\" class=\"data row2 col0\" >AG1000G-BF-B</td>\n",
       "      <td id=\"T_277a7_row2_col1\" class=\"data row2 col1\" ><a href=\"https://www.malariagen.net/partner_study/AG1000G-BF-1\" target=\"_blank\">AG1000G-BF-1</a></td>\n",
       "      <td id=\"T_277a7_row2_col2\" class=\"data row2 col2\" >Austin Burt</td>\n",
       "      <td id=\"T_277a7_row2_col3\" class=\"data row2 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.0.html\" target=\"_blank\">Ag3.0</a></td>\n",
       "      <td id=\"T_277a7_row2_col4\" class=\"data row2 col4\" ><a href='https://www.malariagen.net/data_package/ag30-anopheles-gambiae-data-resource/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2021</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row3_col0\" class=\"data row3 col0\" >AG1000G-BF-C</td>\n",
       "      <td id=\"T_277a7_row3_col1\" class=\"data row3 col1\" ><a href=\"https://www.malariagen.net/partner_study/AG1000G-BF-2\" target=\"_blank\">AG1000G-BF-2</a></td>\n",
       "      <td id=\"T_277a7_row3_col2\" class=\"data row3 col2\" >Nora Besansky</td>\n",
       "      <td id=\"T_277a7_row3_col3\" class=\"data row3 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.0.html\" target=\"_blank\">Ag3.0</a></td>\n",
       "      <td id=\"T_277a7_row3_col4\" class=\"data row3 col4\" ><a href='https://www.malariagen.net/data_package/ag30-anopheles-gambiae-data-resource/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2021</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row4_col0\" class=\"data row4 col0\" >AG1000G-CD</td>\n",
       "      <td id=\"T_277a7_row4_col1\" class=\"data row4 col1\" ><a href=\"https://www.malariagen.net/partner_study/AG1000G-CD\" target=\"_blank\">AG1000G-CD</a></td>\n",
       "      <td id=\"T_277a7_row4_col2\" class=\"data row4 col2\" >David Weetman</td>\n",
       "      <td id=\"T_277a7_row4_col3\" class=\"data row4 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.0.html\" target=\"_blank\">Ag3.0</a></td>\n",
       "      <td id=\"T_277a7_row4_col4\" class=\"data row4 col4\" ><a href='https://www.malariagen.net/data_package/ag30-anopheles-gambiae-data-resource/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2021</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row5_col0\" class=\"data row5 col0\" >AG1000G-CF</td>\n",
       "      <td id=\"T_277a7_row5_col1\" class=\"data row5 col1\" ><a href=\"https://www.malariagen.net/partner_study/AG1000G-CF\" target=\"_blank\">AG1000G-CF</a></td>\n",
       "      <td id=\"T_277a7_row5_col2\" class=\"data row5 col2\" >Alessandra della Torre</td>\n",
       "      <td id=\"T_277a7_row5_col3\" class=\"data row5 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.0.html\" target=\"_blank\">Ag3.0</a></td>\n",
       "      <td id=\"T_277a7_row5_col4\" class=\"data row5 col4\" ><a href='https://www.malariagen.net/data_package/ag30-anopheles-gambiae-data-resource/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2021</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row6_col0\" class=\"data row6 col0\" >AG1000G-CI</td>\n",
       "      <td id=\"T_277a7_row6_col1\" class=\"data row6 col1\" ><a href=\"https://www.malariagen.net/partner_study/AG1000G-CI\" target=\"_blank\">AG1000G-CI</a></td>\n",
       "      <td id=\"T_277a7_row6_col2\" class=\"data row6 col2\" >David Weetman</td>\n",
       "      <td id=\"T_277a7_row6_col3\" class=\"data row6 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.0.html\" target=\"_blank\">Ag3.0</a></td>\n",
       "      <td id=\"T_277a7_row6_col4\" class=\"data row6 col4\" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC7605271/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2020</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row7_col0\" class=\"data row7 col0\" >AG1000G-CM-A</td>\n",
       "      <td id=\"T_277a7_row7_col1\" class=\"data row7 col1\" ><a href=\"https://www.malariagen.net/partner_study/AG1000G-CM-1\" target=\"_blank\">AG1000G-CM-1</a></td>\n",
       "      <td id=\"T_277a7_row7_col2\" class=\"data row7 col2\" >Nora Besansky</td>\n",
       "      <td id=\"T_277a7_row7_col3\" class=\"data row7 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.0.html\" target=\"_blank\">Ag3.0</a></td>\n",
       "      <td id=\"T_277a7_row7_col4\" class=\"data row7 col4\" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC6026373/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2017</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row8_col0\" class=\"data row8 col0\" >AG1000G-CM-B</td>\n",
       "      <td id=\"T_277a7_row8_col1\" class=\"data row8 col1\" ><a href=\"https://www.malariagen.net/partner_study/AG1000G-CM-2\" target=\"_blank\">AG1000G-CM-2</a></td>\n",
       "      <td id=\"T_277a7_row8_col2\" class=\"data row8 col2\" >Nora Besansky</td>\n",
       "      <td id=\"T_277a7_row8_col3\" class=\"data row8 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.0.html\" target=\"_blank\">Ag3.0</a></td>\n",
       "      <td id=\"T_277a7_row8_col4\" class=\"data row8 col4\" ><a href='https://www.malariagen.net/data_package/ag30-anopheles-gambiae-data-resource/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2021</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row9_col0\" class=\"data row9 col0\" >AG1000G-CM-C</td>\n",
       "      <td id=\"T_277a7_row9_col1\" class=\"data row9 col1\" ><a href=\"https://www.malariagen.net/partner_study/AG1000G-CM-3\" target=\"_blank\">AG1000G-CM-3</a></td>\n",
       "      <td id=\"T_277a7_row9_col2\" class=\"data row9 col2\" >Brad White</td>\n",
       "      <td id=\"T_277a7_row9_col3\" class=\"data row9 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.0.html\" target=\"_blank\">Ag3.0</a></td>\n",
       "      <td id=\"T_277a7_row9_col4\" class=\"data row9 col4\" ><a href='https://www.malariagen.net/data_package/ag30-anopheles-gambiae-data-resource/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2021</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row10_col0\" class=\"data row10 col0\" >AG1000G-FR</td>\n",
       "      <td id=\"T_277a7_row10_col1\" class=\"data row10 col1\" ><a href=\"https://www.malariagen.net/partner_study/AG1000G-FR\" target=\"_blank\">AG1000G-FR</a></td>\n",
       "      <td id=\"T_277a7_row10_col2\" class=\"data row10 col2\" >Igor Sharakhov</td>\n",
       "      <td id=\"T_277a7_row10_col3\" class=\"data row10 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.0.html\" target=\"_blank\">Ag3.0</a></td>\n",
       "      <td id=\"T_277a7_row10_col4\" class=\"data row10 col4\" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC7605271/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2020</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row11_col0\" class=\"data row11 col0\" >AG1000G-GA-A</td>\n",
       "      <td id=\"T_277a7_row11_col1\" class=\"data row11 col1\" ><a href=\"https://www.malariagen.net/partner_study/AG1000G-GA-1\" target=\"_blank\">AG1000G-GA-1</a></td>\n",
       "      <td id=\"T_277a7_row11_col2\" class=\"data row11 col2\" >Joao Pinto</td>\n",
       "      <td id=\"T_277a7_row11_col3\" class=\"data row11 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.0.html\" target=\"_blank\">Ag3.0</a></td>\n",
       "      <td id=\"T_277a7_row11_col4\" class=\"data row11 col4\" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC6026373/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2017</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row12_col0\" class=\"data row12 col0\" >AG1000G-GH</td>\n",
       "      <td id=\"T_277a7_row12_col1\" class=\"data row12 col1\" ><a href=\"https://www.malariagen.net/partner_study/AG1000G-GH\" target=\"_blank\">AG1000G-GH</a></td>\n",
       "      <td id=\"T_277a7_row12_col2\" class=\"data row12 col2\" >David Weetman</td>\n",
       "      <td id=\"T_277a7_row12_col3\" class=\"data row12 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.0.html\" target=\"_blank\">Ag3.0</a></td>\n",
       "      <td id=\"T_277a7_row12_col4\" class=\"data row12 col4\" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC7605271/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2020</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row13_col0\" class=\"data row13 col0\" >AG1000G-GM-A</td>\n",
       "      <td id=\"T_277a7_row13_col1\" class=\"data row13 col1\" ><a href=\"https://www.malariagen.net/partner_study/AG1000G-GM-1\" target=\"_blank\">AG1000G-GM-1</a></td>\n",
       "      <td id=\"T_277a7_row13_col2\" class=\"data row13 col2\" >Martin Donnelly</td>\n",
       "      <td id=\"T_277a7_row13_col3\" class=\"data row13 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.0.html\" target=\"_blank\">Ag3.0</a></td>\n",
       "      <td id=\"T_277a7_row13_col4\" class=\"data row13 col4\" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC7605271/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2020</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row14_col0\" class=\"data row14 col0\" >AG1000G-GM-B</td>\n",
       "      <td id=\"T_277a7_row14_col1\" class=\"data row14 col1\" ><a href=\"https://www.malariagen.net/partner_study/AG1000G-GM-2\" target=\"_blank\">AG1000G-GM-2</a></td>\n",
       "      <td id=\"T_277a7_row14_col2\" class=\"data row14 col2\" >Beniamino Caputo</td>\n",
       "      <td id=\"T_277a7_row14_col3\" class=\"data row14 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.0.html\" target=\"_blank\">Ag3.0</a></td>\n",
       "      <td id=\"T_277a7_row14_col4\" class=\"data row14 col4\" ><a href='https://www.malariagen.net/data_package/ag30-anopheles-gambiae-data-resource/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2021</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row15_col0\" class=\"data row15 col0\" >AG1000G-GM-C</td>\n",
       "      <td id=\"T_277a7_row15_col1\" class=\"data row15 col1\" ><a href=\"https://www.malariagen.net/partner_study/AG1000G-GM-3\" target=\"_blank\">AG1000G-GM-3</a></td>\n",
       "      <td id=\"T_277a7_row15_col2\" class=\"data row15 col2\" >Charles Godfray</td>\n",
       "      <td id=\"T_277a7_row15_col3\" class=\"data row15 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.0.html\" target=\"_blank\">Ag3.0</a></td>\n",
       "      <td id=\"T_277a7_row15_col4\" class=\"data row15 col4\" ><a href='https://www.malariagen.net/data_package/ag30-anopheles-gambiae-data-resource/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2021</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row16_col0\" class=\"data row16 col0\" >AG1000G-GN-A</td>\n",
       "      <td id=\"T_277a7_row16_col1\" class=\"data row16 col1\" ><a href=\"https://www.malariagen.net/partner_study/AG1000G-GN-ML\" target=\"_blank\">AG1000G-GN-ML</a></td>\n",
       "      <td id=\"T_277a7_row16_col2\" class=\"data row16 col2\" >Ken Vernick</td>\n",
       "      <td id=\"T_277a7_row16_col3\" class=\"data row16 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.0.html\" target=\"_blank\">Ag3.0</a></td>\n",
       "      <td id=\"T_277a7_row16_col4\" class=\"data row16 col4\" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC6026373/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2017</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row17_col0\" class=\"data row17 col0\" >AG1000G-GN-B</td>\n",
       "      <td id=\"T_277a7_row17_col1\" class=\"data row17 col1\" ><a href=\"https://www.malariagen.net/partner_study/AG1000G-GN-ML\" target=\"_blank\">AG1000G-GN-ML</a></td>\n",
       "      <td id=\"T_277a7_row17_col2\" class=\"data row17 col2\" >Ken Vernick</td>\n",
       "      <td id=\"T_277a7_row17_col3\" class=\"data row17 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.0.html\" target=\"_blank\">Ag3.0</a></td>\n",
       "      <td id=\"T_277a7_row17_col4\" class=\"data row17 col4\" ><a href='https://www.malariagen.net/data_package/ag30-anopheles-gambiae-data-resource/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2021</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row18_col0\" class=\"data row18 col0\" >AG1000G-GQ</td>\n",
       "      <td id=\"T_277a7_row18_col1\" class=\"data row18 col1\" ><a href=\"https://www.malariagen.net/partner_study/AG1000G-GQ\" target=\"_blank\">AG1000G-GQ</a></td>\n",
       "      <td id=\"T_277a7_row18_col2\" class=\"data row18 col2\" >Igor Sharakhov</td>\n",
       "      <td id=\"T_277a7_row18_col3\" class=\"data row18 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.0.html\" target=\"_blank\">Ag3.0</a></td>\n",
       "      <td id=\"T_277a7_row18_col4\" class=\"data row18 col4\" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC7605271/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2020</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row19_col0\" class=\"data row19 col0\" >AG1000G-GW</td>\n",
       "      <td id=\"T_277a7_row19_col1\" class=\"data row19 col1\" ><a href=\"https://www.malariagen.net/partner_study/AG1000G-GW\" target=\"_blank\">AG1000G-GW</a></td>\n",
       "      <td id=\"T_277a7_row19_col2\" class=\"data row19 col2\" >Joao Pinto</td>\n",
       "      <td id=\"T_277a7_row19_col3\" class=\"data row19 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.0.html\" target=\"_blank\">Ag3.0</a></td>\n",
       "      <td id=\"T_277a7_row19_col4\" class=\"data row19 col4\" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC6026373/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2017</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row20_col0\" class=\"data row20 col0\" >AG1000G-KE</td>\n",
       "      <td id=\"T_277a7_row20_col1\" class=\"data row20 col1\" ><a href=\"https://www.malariagen.net/partner_study/AG1000G-KE\" target=\"_blank\">AG1000G-KE</a></td>\n",
       "      <td id=\"T_277a7_row20_col2\" class=\"data row20 col2\" >Janet Midega</td>\n",
       "      <td id=\"T_277a7_row20_col3\" class=\"data row20 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.0.html\" target=\"_blank\">Ag3.0</a></td>\n",
       "      <td id=\"T_277a7_row20_col4\" class=\"data row20 col4\" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC6026373/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2017</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row21_col0\" class=\"data row21 col0\" >AG1000G-ML-A</td>\n",
       "      <td id=\"T_277a7_row21_col1\" class=\"data row21 col1\" ><a href=\"https://www.malariagen.net/partner_study/AG1000G-ML-1\" target=\"_blank\">AG1000G-ML-1</a></td>\n",
       "      <td id=\"T_277a7_row21_col2\" class=\"data row21 col2\" >Austin Burt</td>\n",
       "      <td id=\"T_277a7_row21_col3\" class=\"data row21 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.0.html\" target=\"_blank\">Ag3.0</a></td>\n",
       "      <td id=\"T_277a7_row21_col4\" class=\"data row21 col4\" ><a href='https://www.malariagen.net/data_package/ag30-anopheles-gambiae-data-resource/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2021</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row22_col0\" class=\"data row22 col0\" >AG1000G-ML-B</td>\n",
       "      <td id=\"T_277a7_row22_col1\" class=\"data row22 col1\" ><a href=\"https://www.malariagen.net/partner_study/AG1000G-ML-2\" target=\"_blank\">AG1000G-ML-2</a></td>\n",
       "      <td id=\"T_277a7_row22_col2\" class=\"data row22 col2\" >Nora Besansky</td>\n",
       "      <td id=\"T_277a7_row22_col3\" class=\"data row22 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.0.html\" target=\"_blank\">Ag3.0</a></td>\n",
       "      <td id=\"T_277a7_row22_col4\" class=\"data row22 col4\" ><a href='https://www.malariagen.net/data_package/ag30-anopheles-gambiae-data-resource/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2021</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row23_col0\" class=\"data row23 col0\" >AG1000G-MW</td>\n",
       "      <td id=\"T_277a7_row23_col1\" class=\"data row23 col1\" ><a href=\"https://www.malariagen.net/partner_study/AG1000G-MW\" target=\"_blank\">AG1000G-MW</a></td>\n",
       "      <td id=\"T_277a7_row23_col2\" class=\"data row23 col2\" >Martin Donnelly</td>\n",
       "      <td id=\"T_277a7_row23_col3\" class=\"data row23 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.0.html\" target=\"_blank\">Ag3.0</a></td>\n",
       "      <td id=\"T_277a7_row23_col4\" class=\"data row23 col4\" ><a href='https://www.malariagen.net/data_package/ag30-anopheles-gambiae-data-resource/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2021</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row24_col0\" class=\"data row24 col0\" >AG1000G-MZ</td>\n",
       "      <td id=\"T_277a7_row24_col1\" class=\"data row24 col1\" ><a href=\"https://www.malariagen.net/partner_study/AG1000G-MZ\" target=\"_blank\">AG1000G-MZ</a></td>\n",
       "      <td id=\"T_277a7_row24_col2\" class=\"data row24 col2\" >Joao Pinto</td>\n",
       "      <td id=\"T_277a7_row24_col3\" class=\"data row24 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.0.html\" target=\"_blank\">Ag3.0</a></td>\n",
       "      <td id=\"T_277a7_row24_col4\" class=\"data row24 col4\" ><a href='https://www.malariagen.net/data_package/ag30-anopheles-gambiae-data-resource/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2021</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row25_col0\" class=\"data row25 col0\" >AG1000G-TZ</td>\n",
       "      <td id=\"T_277a7_row25_col1\" class=\"data row25 col1\" ><a href=\"https://www.malariagen.net/partner_study/AG1000G-TZ\" target=\"_blank\">AG1000G-TZ</a></td>\n",
       "      <td id=\"T_277a7_row25_col2\" class=\"data row25 col2\" >Bilali Kabula, David Weetman</td>\n",
       "      <td id=\"T_277a7_row25_col3\" class=\"data row25 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.0.html\" target=\"_blank\">Ag3.0</a></td>\n",
       "      <td id=\"T_277a7_row25_col4\" class=\"data row25 col4\" ><a href='https://www.malariagen.net/data_package/ag30-anopheles-gambiae-data-resource/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2021</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row26_col0\" class=\"data row26 col0\" >AG1000G-UG</td>\n",
       "      <td id=\"T_277a7_row26_col1\" class=\"data row26 col1\" ><a href=\"https://www.malariagen.net/partner_study/AG1000G-UG\" target=\"_blank\">AG1000G-UG</a></td>\n",
       "      <td id=\"T_277a7_row26_col2\" class=\"data row26 col2\" >Martin Donnelly</td>\n",
       "      <td id=\"T_277a7_row26_col3\" class=\"data row26 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.0.html\" target=\"_blank\">Ag3.0</a></td>\n",
       "      <td id=\"T_277a7_row26_col4\" class=\"data row26 col4\" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC6026373/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2017</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row27_col0\" class=\"data row27 col0\" >1237-VO-BJ-DJOGBENOU-VMF00050</td>\n",
       "      <td id=\"T_277a7_row27_col1\" class=\"data row27 col1\" ><a href=\"https://www.malariagen.net/partner_study/1237-VO-BJ-DJOGBENOU\" target=\"_blank\">1237-VO-BJ-DJOGBENOU</a></td>\n",
       "      <td id=\"T_277a7_row27_col2\" class=\"data row27 col2\" >Luc Djogbenou</td>\n",
       "      <td id=\"T_277a7_row27_col3\" class=\"data row27 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.2.html\" target=\"_blank\">Ag3.2</a></td>\n",
       "      <td id=\"T_277a7_row27_col4\" class=\"data row27 col4\" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC10432508/' target='_blank'>Lucas et al. 2023</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row28_col0\" class=\"data row28 col0\" >1237-VO-BJ-DJOGBENOU-VMF00067</td>\n",
       "      <td id=\"T_277a7_row28_col1\" class=\"data row28 col1\" ><a href=\"https://www.malariagen.net/partner_study/1237-VO-BJ-DJOGBENOU\" target=\"_blank\">1237-VO-BJ-DJOGBENOU</a></td>\n",
       "      <td id=\"T_277a7_row28_col2\" class=\"data row28 col2\" >Luc Djogbenou</td>\n",
       "      <td id=\"T_277a7_row28_col3\" class=\"data row28 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.2.html\" target=\"_blank\">Ag3.2</a></td>\n",
       "      <td id=\"T_277a7_row28_col4\" class=\"data row28 col4\" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC10432508/' target='_blank'>Lucas et al. 2023</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row29_col0\" class=\"data row29 col0\" >1244-VO-GH-YAWSON-VMF00051</td>\n",
       "      <td id=\"T_277a7_row29_col1\" class=\"data row29 col1\" ><a href=\"https://www.malariagen.net/partner_study/1244-VO-GH-YAWSON\" target=\"_blank\">1244-VO-GH-YAWSON</a></td>\n",
       "      <td id=\"T_277a7_row29_col2\" class=\"data row29 col2\" >Alexander Egyir-Yawson</td>\n",
       "      <td id=\"T_277a7_row29_col3\" class=\"data row29 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.2.html\" target=\"_blank\">Ag3.2</a></td>\n",
       "      <td id=\"T_277a7_row29_col4\" class=\"data row29 col4\" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC10432508/' target='_blank'>Lucas et al. 2023</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row30_col0\" class=\"data row30 col0\" >1245-VO-CI-CONSTANT-VMF00054</td>\n",
       "      <td id=\"T_277a7_row30_col1\" class=\"data row30 col1\" ><a href=\"https://www.malariagen.net/partner_study/1245-VO-CI-CONSTANT\" target=\"_blank\">1245-VO-CI-CONSTANT</a></td>\n",
       "      <td id=\"T_277a7_row30_col2\" class=\"data row30 col2\" >Edi Constant</td>\n",
       "      <td id=\"T_277a7_row30_col3\" class=\"data row30 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.2.html\" target=\"_blank\">Ag3.2</a></td>\n",
       "      <td id=\"T_277a7_row30_col4\" class=\"data row30 col4\" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC10432508/' target='_blank'>Lucas et al. 2023</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row31_col0\" class=\"data row31 col0\" >1253-VO-TG-DJOGBENOU-VMF00052</td>\n",
       "      <td id=\"T_277a7_row31_col1\" class=\"data row31 col1\" ><a href=\"https://www.malariagen.net/partner_study/1253-VO-TG-DJOGBENOU\" target=\"_blank\">1253-VO-TG-DJOGBENOU</a></td>\n",
       "      <td id=\"T_277a7_row31_col2\" class=\"data row31 col2\" >Luc Djogbenou</td>\n",
       "      <td id=\"T_277a7_row31_col3\" class=\"data row31 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.2.html\" target=\"_blank\">Ag3.2</a></td>\n",
       "      <td id=\"T_277a7_row31_col4\" class=\"data row31 col4\" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC10432508/' target='_blank'>Lucas et al. 2023</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row32_col0\" class=\"data row32 col0\" >1178-VO-UG-LAWNICZAK-VMF00025</td>\n",
       "      <td id=\"T_277a7_row32_col1\" class=\"data row32 col1\" ><a href=\"https://www.malariagen.net/partner_study/1178-VO-UG-LAWNICZAK\" target=\"_blank\">1178-VO-UG-LAWNICZAK</a></td>\n",
       "      <td id=\"T_277a7_row32_col2\" class=\"data row32 col2\" >Mara Lawniczak</td>\n",
       "      <td id=\"T_277a7_row32_col3\" class=\"data row32 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.3.html\" target=\"_blank\">Ag3.3</a></td>\n",
       "      <td id=\"T_277a7_row32_col4\" class=\"data row32 col4\" ></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row33_col0\" class=\"data row33 col0\" >1244-VO-GH-YAWSON-VMF00149</td>\n",
       "      <td id=\"T_277a7_row33_col1\" class=\"data row33 col1\" ><a href=\"https://www.malariagen.net/partner_study/1244-VO-GH-YAWSON\" target=\"_blank\">1244-VO-GH-YAWSON</a></td>\n",
       "      <td id=\"T_277a7_row33_col2\" class=\"data row33 col2\" >Alexander Egyir-Yawson</td>\n",
       "      <td id=\"T_277a7_row33_col3\" class=\"data row33 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.4.html\" target=\"_blank\">Ag3.4</a></td>\n",
       "      <td id=\"T_277a7_row33_col4\" class=\"data row33 col4\" >Nagi et al. in prep.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row34_col0\" class=\"data row34 col0\" >crawford-2016</td>\n",
       "      <td id=\"T_277a7_row34_col1\" class=\"data row34 col1\" ><a href=\"https://doi.org/10.1111/mec.13572\" target=\"_blank\">crawford-2016</a></td>\n",
       "      <td id=\"T_277a7_row34_col2\" class=\"data row34 col2\" >Jacob E Crawford</td>\n",
       "      <td id=\"T_277a7_row34_col3\" class=\"data row34 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.7.html\" target=\"_blank\">Ag3.7</a></td>\n",
       "      <td id=\"T_277a7_row34_col4\" class=\"data row34 col4\" ><a href='https://pubmed.ncbi.nlm.nih.gov/26846876/' target='_blank'>Crawford et al. 2016</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row35_col0\" class=\"data row35 col0\" >tennessen-2021</td>\n",
       "      <td id=\"T_277a7_row35_col1\" class=\"data row35 col1\" ><a href=\"https://doi.org/10.1111/mec.15756\" target=\"_blank\">tennessen-2021</a></td>\n",
       "      <td id=\"T_277a7_row35_col2\" class=\"data row35 col2\" >Jacob Tennessen</td>\n",
       "      <td id=\"T_277a7_row35_col3\" class=\"data row35 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.8.html\" target=\"_blank\">Ag3.8</a></td>\n",
       "      <td id=\"T_277a7_row35_col4\" class=\"data row35 col4\" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC7858241/' target='_blank'>Tennessen et al. 2021</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row36_col0\" class=\"data row36 col0\" >bergey-2019</td>\n",
       "      <td id=\"T_277a7_row36_col1\" class=\"data row36 col1\" ><a href=\"https://doi.org/10.1111/eva.12878\" target=\"_blank\">bergey-2019</a></td>\n",
       "      <td id=\"T_277a7_row36_col2\" class=\"data row36 col2\" >Christina Bergey</td>\n",
       "      <td id=\"T_277a7_row36_col3\" class=\"data row36 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.9.html\" target=\"_blank\">Ag3.9</a></td>\n",
       "      <td id=\"T_277a7_row36_col4\" class=\"data row36 col4\" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC6976967/' target='_blank'>Bergey et al. 2019</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row37_col0\" class=\"data row37 col0\" >campos-2021</td>\n",
       "      <td id=\"T_277a7_row37_col1\" class=\"data row37 col1\" ><a href=\"https://doi.org/10.1038/s42003-021-02168-0\" target=\"_blank\">campos-2021</a></td>\n",
       "      <td id=\"T_277a7_row37_col2\" class=\"data row37 col2\" >Greg Lanzaro</td>\n",
       "      <td id=\"T_277a7_row37_col3\" class=\"data row37 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.9.html\" target=\"_blank\">Ag3.9</a></td>\n",
       "      <td id=\"T_277a7_row37_col4\" class=\"data row37 col4\" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC8155153/' target='_blank'>Campos et al. 2021</a>, <a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC8477601/' target='_blank'>Lanzaro et al. 2021</a></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_277a7_row38_col0\" class=\"data row38 col0\" >fontaine-2015-rebuild</td>\n",
       "      <td id=\"T_277a7_row38_col1\" class=\"data row38 col1\" ><a href=\"https://doi.org/10.1126/science.1258524\" target=\"_blank\">fontaine-2015-rebuild</a></td>\n",
       "      <td id=\"T_277a7_row38_col2\" class=\"data row38 col2\" >Michael.C.Fontaine</td>\n",
       "      <td id=\"T_277a7_row38_col3\" class=\"data row38 col3\" ><a href=\"https://malariagen.github.io/vector-data/ag3/ag3.10.html\" target=\"_blank\">Ag3.10</a></td>\n",
       "      <td id=\"T_277a7_row38_col4\" class=\"data row38 col4\" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC4380269/' target='_blank'>Fontaine et al. 2014</a></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x7ff9ee1efb60>"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "page_utils.style_data_sources(\n",
    "    df_samples=df_samples,\n",
    "    caption=\"Table 1. Data sources included in the current analysis version.\",\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "97167c53-42fb-4dd4-8046-d078d4473457",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/markdown": [
       "\n",
       "Sample metadata, unphased SNP calls, and phased SNP haplotypes were retrieved from \n",
       "the {term}`Malaria Vector Genome Observatory` cloud data repository hosted in \n",
       "Google Cloud Storage (GCS) via the {term}`MalariaGEN Python API` version \n",
       "15.0.1.\n"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Markdown(f\"\"\"\n",
    "Sample metadata, unphased SNP calls, and phased SNP haplotypes were retrieved from \n",
    "the {{term}}`Malaria Vector Genome Observatory` cloud data repository hosted in \n",
    "Google Cloud Storage (GCS) via the {{term}}`MalariaGEN Python API` version \n",
    "{malariagen_data.__version__}.\n",
    "\"\"\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "32b058d5-be5e-4c11-a74f-90081d26975c",
   "metadata": {
    "tags": [
     "remove-input"
    ]
   },
   "source": [
    "## Sample inclusion and grouping into cohorts\n",
    "\n",
    "Samples were considered for inclusion if they met the following criteria:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "474d0b6f-2d84-4191-abde-719f9462fef3",
   "metadata": {
    "tags": [
     "remove-input"
    ]
   },
   "outputs": [],
   "source": [
    "def human_readable_list(x):\n",
    "    x = [str(v) for v in x]\n",
    "    if len(x) > 1:\n",
    "        output = \", \".join(x[:-1]) + \" or \" + x[-1]\n",
    "    else:\n",
    "        output = x[0]\n",
    "    return output\n",
    "\n",
    "\n",
    "readable_taxa = human_readable_list(setup.taxa)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "68959f83-5785-4cec-9ee7-d491b20f2437",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/markdown": [
       "\n",
       "* Gender assigned as female via comparison of sequence coverage on autosomes and sex \n",
       "  chromosomes. \n",
       "* Taxon assigned as gambiae, coluzzii, arabiensis or bissau via principle components analysis of \n",
       "  genomic data from Chromosome 3 and comparison with reference samples \n",
       "  with known taxon assignments.\n"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Markdown(f\"\"\"\n",
    "* Gender assigned as female via comparison of sequence coverage on autosomes and sex \n",
    "  chromosomes. \n",
    "* Taxon assigned as {readable_taxa} via principle components analysis of \n",
    "  genomic data from Chromosome 3 and comparison with reference samples \n",
    "  with known taxon assignments.\n",
    "\"\"\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "fc0de9ab-618d-49b1-a12c-9e567fc84f05",
   "metadata": {
    "tags": [
     "remove-input"
    ]
   },
   "outputs": [],
   "source": [
    "gdf_cohorts = page_utils.gdf_cohorts"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9c79abe0-46e4-4228-a9ea-5472e2fedfc4",
   "metadata": {},
   "source": [
    "After filtering according to these inclusion criteria, samples were grouped into cohorts by taxon, location of sampling and date of sampling. Samples were grouped spatially if their collection locations were within the same level 2 administrative unit, according to geoBoundaries version 5.0.0. Samples were grouped temporally if their collection dates were within the same quarter (3 month period) where possible, except in a small number of cases where metadata were only available on year of collection."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "8c6a207c-33e9-49d0-b202-3d5cfb2aa807",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/markdown": [
       "\n",
       "Cohorts were excluded from the analysis if the sample size was \n",
       "less than 15. Cohorts with more than \n",
       "100 samples were randomly downsampled for\n",
       "computational efficiency. Cohorts were also excluded from the \n",
       "analysis if they failed H12 or G123 window size calibration\n",
       "(see below). After applying these filters, a total of \n",
       "61 cohorts were retained for analysis (Table 2).\n"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Markdown(f\"\"\"\n",
    "Cohorts were excluded from the analysis if the sample size was \n",
    "less than {setup.min_cohort_size}. Cohorts with more than \n",
    "{setup.max_cohort_size} samples were randomly downsampled for\n",
    "computational efficiency. Cohorts were also excluded from the \n",
    "analysis if they failed H12 or G123 window size calibration\n",
    "(see below). After applying these filters, a total of \n",
    "{len(gdf_cohorts)} cohorts were retained for analysis (Table 2).\n",
    "\"\"\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "25d5ff7c-1524-47f1-9a78-43a3c73e9956",
   "metadata": {
    "tags": [
     "remove-input"
    ]
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       "</style>\n",
       "<table id=\"T_455ae\">\n",
       "  <caption>Table 2. Cohorts selected for genome-wide selection scan analyses.</caption>\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th id=\"T_455ae_level0_col0\" class=\"col_heading level0 col0\" >Cohort</th>\n",
       "      <th id=\"T_455ae_level0_col1\" class=\"col_heading level0 col1\" >Country</th>\n",
       "      <th id=\"T_455ae_level0_col2\" class=\"col_heading level0 col2\" >Region</th>\n",
       "      <th id=\"T_455ae_level0_col3\" class=\"col_heading level0 col3\" >District</th>\n",
       "      <th id=\"T_455ae_level0_col4\" class=\"col_heading level0 col4\" >Taxon</th>\n",
       "      <th id=\"T_455ae_level0_col5\" class=\"col_heading level0 col5\" >Year</th>\n",
       "      <th id=\"T_455ae_level0_col6\" class=\"col_heading level0 col6\" >Quarter</th>\n",
       "      <th id=\"T_455ae_level0_col7\" class=\"col_heading level0 col7\" >Sample Size</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row0_col0\" class=\"data row0 col0\" ><a href=\"cohort/AO-LUA_Luanda_colu_2009_Q2.html\">Angola / Luanda / coluzzii / 2009 / Q2</a></td>\n",
       "      <td id=\"T_455ae_row0_col1\" class=\"data row0 col1\" >Angola</td>\n",
       "      <td id=\"T_455ae_row0_col2\" class=\"data row0 col2\" >Luanda</td>\n",
       "      <td id=\"T_455ae_row0_col3\" class=\"data row0 col3\" >Luanda</td>\n",
       "      <td id=\"T_455ae_row0_col4\" class=\"data row0 col4\" >coluzzii</td>\n",
       "      <td id=\"T_455ae_row0_col5\" class=\"data row0 col5\" >2009</td>\n",
       "      <td id=\"T_455ae_row0_col6\" class=\"data row0 col6\" >2</td>\n",
       "      <td id=\"T_455ae_row0_col7\" class=\"data row0 col7\" >77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row1_col0\" class=\"data row1 col0\" ><a href=\"cohort/BF-02_Comoe_colu_2011.html\">Burkina Faso / Comoe / coluzzii / 2011</a></td>\n",
       "      <td id=\"T_455ae_row1_col1\" class=\"data row1 col1\" >Burkina Faso</td>\n",
       "      <td id=\"T_455ae_row1_col2\" class=\"data row1 col2\" >Cascades</td>\n",
       "      <td id=\"T_455ae_row1_col3\" class=\"data row1 col3\" >Comoe</td>\n",
       "      <td id=\"T_455ae_row1_col4\" class=\"data row1 col4\" >coluzzii</td>\n",
       "      <td id=\"T_455ae_row1_col5\" class=\"data row1 col5\" >2011</td>\n",
       "      <td id=\"T_455ae_row1_col6\" class=\"data row1 col6\" ></td>\n",
       "      <td id=\"T_455ae_row1_col7\" class=\"data row1 col7\" >18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row2_col0\" class=\"data row2 col0\" ><a href=\"cohort/BF-02_Comoe_colu_2012.html\">Burkina Faso / Comoe / coluzzii / 2012</a></td>\n",
       "      <td id=\"T_455ae_row2_col1\" class=\"data row2 col1\" >Burkina Faso</td>\n",
       "      <td id=\"T_455ae_row2_col2\" class=\"data row2 col2\" >Cascades</td>\n",
       "      <td id=\"T_455ae_row2_col3\" class=\"data row2 col3\" >Comoe</td>\n",
       "      <td id=\"T_455ae_row2_col4\" class=\"data row2 col4\" >coluzzii</td>\n",
       "      <td id=\"T_455ae_row2_col5\" class=\"data row2 col5\" >2012</td>\n",
       "      <td id=\"T_455ae_row2_col6\" class=\"data row2 col6\" ></td>\n",
       "      <td id=\"T_455ae_row2_col7\" class=\"data row2 col7\" >63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row3_col0\" class=\"data row3 col0\" ><a href=\"cohort/BF-02_Comoe_colu_2015.html\">Burkina Faso / Comoe / coluzzii / 2015</a></td>\n",
       "      <td id=\"T_455ae_row3_col1\" class=\"data row3 col1\" >Burkina Faso</td>\n",
       "      <td id=\"T_455ae_row3_col2\" class=\"data row3 col2\" >Cascades</td>\n",
       "      <td id=\"T_455ae_row3_col3\" class=\"data row3 col3\" >Comoe</td>\n",
       "      <td id=\"T_455ae_row3_col4\" class=\"data row3 col4\" >coluzzii</td>\n",
       "      <td id=\"T_455ae_row3_col5\" class=\"data row3 col5\" >2015</td>\n",
       "      <td id=\"T_455ae_row3_col6\" class=\"data row3 col6\" ></td>\n",
       "      <td id=\"T_455ae_row3_col7\" class=\"data row3 col7\" >33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row4_col0\" class=\"data row4 col0\" ><a href=\"cohort/BF-02_Comoe_colu_2016.html\">Burkina Faso / Comoe / coluzzii / 2016</a></td>\n",
       "      <td id=\"T_455ae_row4_col1\" class=\"data row4 col1\" >Burkina Faso</td>\n",
       "      <td id=\"T_455ae_row4_col2\" class=\"data row4 col2\" >Cascades</td>\n",
       "      <td id=\"T_455ae_row4_col3\" class=\"data row4 col3\" >Comoe</td>\n",
       "      <td id=\"T_455ae_row4_col4\" class=\"data row4 col4\" >coluzzii</td>\n",
       "      <td id=\"T_455ae_row4_col5\" class=\"data row4 col5\" >2016</td>\n",
       "      <td id=\"T_455ae_row4_col6\" class=\"data row4 col6\" ></td>\n",
       "      <td id=\"T_455ae_row4_col7\" class=\"data row4 col7\" >53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row5_col0\" class=\"data row5 col0\" ><a href=\"cohort/BF-09_Houet_colu_2012_Q3.html\">Burkina Faso / Houet / coluzzii / 2012 / Q3</a></td>\n",
       "      <td id=\"T_455ae_row5_col1\" class=\"data row5 col1\" >Burkina Faso</td>\n",
       "      <td id=\"T_455ae_row5_col2\" class=\"data row5 col2\" >Hauts-Bassins</td>\n",
       "      <td id=\"T_455ae_row5_col3\" class=\"data row5 col3\" >Houet</td>\n",
       "      <td id=\"T_455ae_row5_col4\" class=\"data row5 col4\" >coluzzii</td>\n",
       "      <td id=\"T_455ae_row5_col5\" class=\"data row5 col5\" >2012</td>\n",
       "      <td id=\"T_455ae_row5_col6\" class=\"data row5 col6\" >3</td>\n",
       "      <td id=\"T_455ae_row5_col7\" class=\"data row5 col7\" >78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row6_col0\" class=\"data row6 col0\" ><a href=\"cohort/BF-09_Houet_colu_2014_Q3.html\">Burkina Faso / Houet / coluzzii / 2014 / Q3</a></td>\n",
       "      <td id=\"T_455ae_row6_col1\" class=\"data row6 col1\" >Burkina Faso</td>\n",
       "      <td id=\"T_455ae_row6_col2\" class=\"data row6 col2\" >Hauts-Bassins</td>\n",
       "      <td id=\"T_455ae_row6_col3\" class=\"data row6 col3\" >Houet</td>\n",
       "      <td id=\"T_455ae_row6_col4\" class=\"data row6 col4\" >coluzzii</td>\n",
       "      <td id=\"T_455ae_row6_col5\" class=\"data row6 col5\" >2014</td>\n",
       "      <td id=\"T_455ae_row6_col6\" class=\"data row6 col6\" >3</td>\n",
       "      <td id=\"T_455ae_row6_col7\" class=\"data row6 col7\" >32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row7_col0\" class=\"data row7 col0\" ><a href=\"cohort/BF-09_Houet_gamb_2012_Q3.html\">Burkina Faso / Houet / gambiae / 2012 / Q3</a></td>\n",
       "      <td id=\"T_455ae_row7_col1\" class=\"data row7 col1\" >Burkina Faso</td>\n",
       "      <td id=\"T_455ae_row7_col2\" class=\"data row7 col2\" >Hauts-Bassins</td>\n",
       "      <td id=\"T_455ae_row7_col3\" class=\"data row7 col3\" >Houet</td>\n",
       "      <td id=\"T_455ae_row7_col4\" class=\"data row7 col4\" >gambiae</td>\n",
       "      <td id=\"T_455ae_row7_col5\" class=\"data row7 col5\" >2012</td>\n",
       "      <td id=\"T_455ae_row7_col6\" class=\"data row7 col6\" >3</td>\n",
       "      <td id=\"T_455ae_row7_col7\" class=\"data row7 col7\" >73</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row8_col0\" class=\"data row8 col0\" ><a href=\"cohort/BF-09_Houet_gamb_2014_Q3.html\">Burkina Faso / Houet / gambiae / 2014 / Q3</a></td>\n",
       "      <td id=\"T_455ae_row8_col1\" class=\"data row8 col1\" >Burkina Faso</td>\n",
       "      <td id=\"T_455ae_row8_col2\" class=\"data row8 col2\" >Hauts-Bassins</td>\n",
       "      <td id=\"T_455ae_row8_col3\" class=\"data row8 col3\" >Houet</td>\n",
       "      <td id=\"T_455ae_row8_col4\" class=\"data row8 col4\" >gambiae</td>\n",
       "      <td id=\"T_455ae_row8_col5\" class=\"data row8 col5\" >2014</td>\n",
       "      <td id=\"T_455ae_row8_col6\" class=\"data row8 col6\" >3</td>\n",
       "      <td id=\"T_455ae_row8_col7\" class=\"data row8 col7\" >41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row9_col0\" class=\"data row9 col0\" ><a href=\"cohort/BJ-DO_Djougou_colu_2017_Q2.html\">Benin / Djougou / coluzzii / 2017 / Q2</a></td>\n",
       "      <td id=\"T_455ae_row9_col1\" class=\"data row9 col1\" >Benin</td>\n",
       "      <td id=\"T_455ae_row9_col2\" class=\"data row9 col2\" >Donga</td>\n",
       "      <td id=\"T_455ae_row9_col3\" class=\"data row9 col3\" >Djougou</td>\n",
       "      <td id=\"T_455ae_row9_col4\" class=\"data row9 col4\" >coluzzii</td>\n",
       "      <td id=\"T_455ae_row9_col5\" class=\"data row9 col5\" >2017</td>\n",
       "      <td id=\"T_455ae_row9_col6\" class=\"data row9 col6\" >2</td>\n",
       "      <td id=\"T_455ae_row9_col7\" class=\"data row9 col7\" >78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row10_col0\" class=\"data row10 col0\" ><a href=\"cohort/BJ-DO_Djougou_gamb_2017_Q2.html\">Benin / Djougou / gambiae / 2017 / Q2</a></td>\n",
       "      <td id=\"T_455ae_row10_col1\" class=\"data row10 col1\" >Benin</td>\n",
       "      <td id=\"T_455ae_row10_col2\" class=\"data row10 col2\" >Donga</td>\n",
       "      <td id=\"T_455ae_row10_col3\" class=\"data row10 col3\" >Djougou</td>\n",
       "      <td id=\"T_455ae_row10_col4\" class=\"data row10 col4\" >gambiae</td>\n",
       "      <td id=\"T_455ae_row10_col5\" class=\"data row10 col5\" >2017</td>\n",
       "      <td id=\"T_455ae_row10_col6\" class=\"data row10 col6\" >2</td>\n",
       "      <td id=\"T_455ae_row10_col7\" class=\"data row10 col7\" >30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row11_col0\" class=\"data row11 col0\" ><a href=\"cohort/BJ-DO_Djougou_gamb_2017_Q3.html\">Benin / Djougou / gambiae / 2017 / Q3</a></td>\n",
       "      <td id=\"T_455ae_row11_col1\" class=\"data row11 col1\" >Benin</td>\n",
       "      <td id=\"T_455ae_row11_col2\" class=\"data row11 col2\" >Donga</td>\n",
       "      <td id=\"T_455ae_row11_col3\" class=\"data row11 col3\" >Djougou</td>\n",
       "      <td id=\"T_455ae_row11_col4\" class=\"data row11 col4\" >gambiae</td>\n",
       "      <td id=\"T_455ae_row11_col5\" class=\"data row11 col5\" >2017</td>\n",
       "      <td id=\"T_455ae_row11_col6\" class=\"data row11 col6\" >3</td>\n",
       "      <td id=\"T_455ae_row11_col7\" class=\"data row11 col7\" >34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row12_col0\" class=\"data row12 col0\" ><a href=\"cohort/BJ-OU_Avrankou_colu_2017_Q3.html\">Benin / Avrankou / coluzzii / 2017 / Q3</a></td>\n",
       "      <td id=\"T_455ae_row12_col1\" class=\"data row12 col1\" >Benin</td>\n",
       "      <td id=\"T_455ae_row12_col2\" class=\"data row12 col2\" >Oueme</td>\n",
       "      <td id=\"T_455ae_row12_col3\" class=\"data row12 col3\" >Avrankou</td>\n",
       "      <td id=\"T_455ae_row12_col4\" class=\"data row12 col4\" >coluzzii</td>\n",
       "      <td id=\"T_455ae_row12_col5\" class=\"data row12 col5\" >2017</td>\n",
       "      <td id=\"T_455ae_row12_col6\" class=\"data row12 col6\" >3</td>\n",
       "      <td id=\"T_455ae_row12_col7\" class=\"data row12 col7\" >88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row13_col0\" class=\"data row13 col0\" ><a href=\"cohort/CD-NU_Gbadolite_gamb_2015_Q3.html\">Democratic Republic of the Congo / Gbadolite / gambiae / 2015 / Q3</a></td>\n",
       "      <td id=\"T_455ae_row13_col1\" class=\"data row13 col1\" >Democratic Republic of the Congo</td>\n",
       "      <td id=\"T_455ae_row13_col2\" class=\"data row13 col2\" >Nord-Ubangi</td>\n",
       "      <td id=\"T_455ae_row13_col3\" class=\"data row13 col3\" >Gbadolite</td>\n",
       "      <td id=\"T_455ae_row13_col4\" class=\"data row13 col4\" >gambiae</td>\n",
       "      <td id=\"T_455ae_row13_col5\" class=\"data row13 col5\" >2015</td>\n",
       "      <td id=\"T_455ae_row13_col6\" class=\"data row13 col6\" >3</td>\n",
       "      <td id=\"T_455ae_row13_col7\" class=\"data row13 col7\" >44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row14_col0\" class=\"data row14 col0\" ><a href=\"cohort/CF-BGF_Bangui_gamb_1994_Q1.html\">Central African Republic / Bangui / gambiae / 1994 / Q1</a></td>\n",
       "      <td id=\"T_455ae_row14_col1\" class=\"data row14 col1\" >Central African Republic</td>\n",
       "      <td id=\"T_455ae_row14_col2\" class=\"data row14 col2\" >Bangui</td>\n",
       "      <td id=\"T_455ae_row14_col3\" class=\"data row14 col3\" >Bangui</td>\n",
       "      <td id=\"T_455ae_row14_col4\" class=\"data row14 col4\" >gambiae</td>\n",
       "      <td id=\"T_455ae_row14_col5\" class=\"data row14 col5\" >1994</td>\n",
       "      <td id=\"T_455ae_row14_col6\" class=\"data row14 col6\" >1</td>\n",
       "      <td id=\"T_455ae_row14_col7\" class=\"data row14 col7\" >53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row15_col0\" class=\"data row15 col0\" ><a href=\"cohort/CI-CM_Sud-Comoe_gamb_2017_Q3.html\">Cote d'Ivoire / Sud-Comoe / gambiae / 2017 / Q3</a></td>\n",
       "      <td id=\"T_455ae_row15_col1\" class=\"data row15 col1\" >Cote d'Ivoire</td>\n",
       "      <td id=\"T_455ae_row15_col2\" class=\"data row15 col2\" >Comoe</td>\n",
       "      <td id=\"T_455ae_row15_col3\" class=\"data row15 col3\" >Sud-Comoe</td>\n",
       "      <td id=\"T_455ae_row15_col4\" class=\"data row15 col4\" >gambiae</td>\n",
       "      <td id=\"T_455ae_row15_col5\" class=\"data row15 col5\" >2017</td>\n",
       "      <td id=\"T_455ae_row15_col6\" class=\"data row15 col6\" >3</td>\n",
       "      <td id=\"T_455ae_row15_col7\" class=\"data row15 col7\" >37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row16_col0\" class=\"data row16 col0\" ><a href=\"cohort/CI-LG_Agneby-Tiassa_colu_2012.html\">Cote d'Ivoire / Agneby-Tiassa / coluzzii / 2012</a></td>\n",
       "      <td id=\"T_455ae_row16_col1\" class=\"data row16 col1\" >Cote d'Ivoire</td>\n",
       "      <td id=\"T_455ae_row16_col2\" class=\"data row16 col2\" >Lagunes</td>\n",
       "      <td id=\"T_455ae_row16_col3\" class=\"data row16 col3\" >Agneby-Tiassa</td>\n",
       "      <td id=\"T_455ae_row16_col4\" class=\"data row16 col4\" >coluzzii</td>\n",
       "      <td id=\"T_455ae_row16_col5\" class=\"data row16 col5\" >2012</td>\n",
       "      <td id=\"T_455ae_row16_col6\" class=\"data row16 col6\" ></td>\n",
       "      <td id=\"T_455ae_row16_col7\" class=\"data row16 col7\" >80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row17_col0\" class=\"data row17 col0\" ><a href=\"cohort/CM-EN_Mayo-Kani_gamb_2005.html\">Cameroon / Mayo-Kani / gambiae / 2005</a></td>\n",
       "      <td id=\"T_455ae_row17_col1\" class=\"data row17 col1\" >Cameroon</td>\n",
       "      <td id=\"T_455ae_row17_col2\" class=\"data row17 col2\" >Far North</td>\n",
       "      <td id=\"T_455ae_row17_col3\" class=\"data row17 col3\" >Mayo-Kani</td>\n",
       "      <td id=\"T_455ae_row17_col4\" class=\"data row17 col4\" >gambiae</td>\n",
       "      <td id=\"T_455ae_row17_col5\" class=\"data row17 col5\" >2005</td>\n",
       "      <td id=\"T_455ae_row17_col6\" class=\"data row17 col6\" ></td>\n",
       "      <td id=\"T_455ae_row17_col7\" class=\"data row17 col7\" >18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row18_col0\" class=\"data row18 col0\" ><a href=\"cohort/CM-ES_Haut-Nyong_gamb_2009_Q3.html\">Cameroon / Haut-Nyong / gambiae / 2009 / Q3</a></td>\n",
       "      <td id=\"T_455ae_row18_col1\" class=\"data row18 col1\" >Cameroon</td>\n",
       "      <td id=\"T_455ae_row18_col2\" class=\"data row18 col2\" >East</td>\n",
       "      <td id=\"T_455ae_row18_col3\" class=\"data row18 col3\" >Haut-Nyong</td>\n",
       "      <td id=\"T_455ae_row18_col4\" class=\"data row18 col4\" >gambiae</td>\n",
       "      <td id=\"T_455ae_row18_col5\" class=\"data row18 col5\" >2009</td>\n",
       "      <td id=\"T_455ae_row18_col6\" class=\"data row18 col6\" >3</td>\n",
       "      <td id=\"T_455ae_row18_col7\" class=\"data row18 col7\" >95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row19_col0\" class=\"data row19 col0\" ><a href=\"cohort/CM-ES_Lom-Et-Djerem_gamb_2009_Q3.html\">Cameroon / Lom-Et-Djerem / gambiae / 2009 / Q3</a></td>\n",
       "      <td id=\"T_455ae_row19_col1\" class=\"data row19 col1\" >Cameroon</td>\n",
       "      <td id=\"T_455ae_row19_col2\" class=\"data row19 col2\" >East</td>\n",
       "      <td id=\"T_455ae_row19_col3\" class=\"data row19 col3\" >Lom-Et-Djerem</td>\n",
       "      <td id=\"T_455ae_row19_col4\" class=\"data row19 col4\" >gambiae</td>\n",
       "      <td id=\"T_455ae_row19_col5\" class=\"data row19 col5\" >2009</td>\n",
       "      <td id=\"T_455ae_row19_col6\" class=\"data row19 col6\" >3</td>\n",
       "      <td id=\"T_455ae_row19_col7\" class=\"data row19 col7\" >163</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row20_col0\" class=\"data row20 col0\" ><a href=\"cohort/GA-1_Libreville_gamb_2000_Q4.html\">Gabon / Libreville / gambiae / 2000 / Q4</a></td>\n",
       "      <td id=\"T_455ae_row20_col1\" class=\"data row20 col1\" >Gabon</td>\n",
       "      <td id=\"T_455ae_row20_col2\" class=\"data row20 col2\" >Estuaire</td>\n",
       "      <td id=\"T_455ae_row20_col3\" class=\"data row20 col3\" >Libreville</td>\n",
       "      <td id=\"T_455ae_row20_col4\" class=\"data row20 col4\" >gambiae</td>\n",
       "      <td id=\"T_455ae_row20_col5\" class=\"data row20 col5\" >2000</td>\n",
       "      <td id=\"T_455ae_row20_col6\" class=\"data row20 col6\" >4</td>\n",
       "      <td id=\"T_455ae_row20_col7\" class=\"data row20 col7\" >69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row21_col0\" class=\"data row21 col0\" ><a href=\"cohort/GH-AA_Ablekuma-Central-Municipal_colu_2018_Q1.html\">Ghana / Ablekuma Central Municipal / coluzzii / 2018 / Q1</a></td>\n",
       "      <td id=\"T_455ae_row21_col1\" class=\"data row21 col1\" >Ghana</td>\n",
       "      <td id=\"T_455ae_row21_col2\" class=\"data row21 col2\" >Greater Accra Region</td>\n",
       "      <td id=\"T_455ae_row21_col3\" class=\"data row21 col3\" >Ablekuma Central Municipal</td>\n",
       "      <td id=\"T_455ae_row21_col4\" class=\"data row21 col4\" >coluzzii</td>\n",
       "      <td id=\"T_455ae_row21_col5\" class=\"data row21 col5\" >2018</td>\n",
       "      <td id=\"T_455ae_row21_col6\" class=\"data row21 col6\" >1</td>\n",
       "      <td id=\"T_455ae_row21_col7\" class=\"data row21 col7\" >266</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row22_col0\" class=\"data row22 col0\" ><a href=\"cohort/GH-AA_La-Nkwantanang-Madina_gamb_2017_Q4.html\">Ghana / La-Nkwantanang-Madina / gambiae / 2017 / Q4</a></td>\n",
       "      <td id=\"T_455ae_row22_col1\" class=\"data row22 col1\" >Ghana</td>\n",
       "      <td id=\"T_455ae_row22_col2\" class=\"data row22 col2\" >Greater Accra Region</td>\n",
       "      <td id=\"T_455ae_row22_col3\" class=\"data row22 col3\" >La-Nkwantanang-Madina</td>\n",
       "      <td id=\"T_455ae_row22_col4\" class=\"data row22 col4\" >gambiae</td>\n",
       "      <td id=\"T_455ae_row22_col5\" class=\"data row22 col5\" >2017</td>\n",
       "      <td id=\"T_455ae_row22_col6\" class=\"data row22 col6\" >4</td>\n",
       "      <td id=\"T_455ae_row22_col7\" class=\"data row22 col7\" >200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row23_col0\" class=\"data row23 col0\" ><a href=\"cohort/GH-AH_Adansi-Akrofuom_colu_2018_Q4.html\">Ghana / Adansi Akrofuom / coluzzii / 2018 / Q4</a></td>\n",
       "      <td id=\"T_455ae_row23_col1\" class=\"data row23 col1\" >Ghana</td>\n",
       "      <td id=\"T_455ae_row23_col2\" class=\"data row23 col2\" >Ashanti Region</td>\n",
       "      <td id=\"T_455ae_row23_col3\" class=\"data row23 col3\" >Adansi Akrofuom</td>\n",
       "      <td id=\"T_455ae_row23_col4\" class=\"data row23 col4\" >coluzzii</td>\n",
       "      <td id=\"T_455ae_row23_col5\" class=\"data row23 col5\" >2018</td>\n",
       "      <td id=\"T_455ae_row23_col6\" class=\"data row23 col6\" >4</td>\n",
       "      <td id=\"T_455ae_row23_col7\" class=\"data row23 col7\" >64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row24_col0\" class=\"data row24 col0\" ><a href=\"cohort/GH-AH_Adansi-South_colu_2018_Q4.html\">Ghana / Adansi South / coluzzii / 2018 / Q4</a></td>\n",
       "      <td id=\"T_455ae_row24_col1\" class=\"data row24 col1\" >Ghana</td>\n",
       "      <td id=\"T_455ae_row24_col2\" class=\"data row24 col2\" >Ashanti Region</td>\n",
       "      <td id=\"T_455ae_row24_col3\" class=\"data row24 col3\" >Adansi South</td>\n",
       "      <td id=\"T_455ae_row24_col4\" class=\"data row24 col4\" >coluzzii</td>\n",
       "      <td id=\"T_455ae_row24_col5\" class=\"data row24 col5\" >2018</td>\n",
       "      <td id=\"T_455ae_row24_col6\" class=\"data row24 col6\" >4</td>\n",
       "      <td id=\"T_455ae_row24_col7\" class=\"data row24 col7\" >36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row25_col0\" class=\"data row25 col0\" ><a href=\"cohort/GH-AH_Adansi-South_gamb_2018_Q4.html\">Ghana / Adansi South / gambiae / 2018 / Q4</a></td>\n",
       "      <td id=\"T_455ae_row25_col1\" class=\"data row25 col1\" >Ghana</td>\n",
       "      <td id=\"T_455ae_row25_col2\" class=\"data row25 col2\" >Ashanti Region</td>\n",
       "      <td id=\"T_455ae_row25_col3\" class=\"data row25 col3\" >Adansi South</td>\n",
       "      <td id=\"T_455ae_row25_col4\" class=\"data row25 col4\" >gambiae</td>\n",
       "      <td id=\"T_455ae_row25_col5\" class=\"data row25 col5\" >2018</td>\n",
       "      <td id=\"T_455ae_row25_col6\" class=\"data row25 col6\" >4</td>\n",
       "      <td id=\"T_455ae_row25_col7\" class=\"data row25 col7\" >29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row26_col0\" class=\"data row26 col0\" ><a href=\"cohort/GH-AH_Amansie-Central_colu_2018_Q4.html\">Ghana / Amansie Central / coluzzii / 2018 / Q4</a></td>\n",
       "      <td id=\"T_455ae_row26_col1\" class=\"data row26 col1\" >Ghana</td>\n",
       "      <td id=\"T_455ae_row26_col2\" class=\"data row26 col2\" >Ashanti Region</td>\n",
       "      <td id=\"T_455ae_row26_col3\" class=\"data row26 col3\" >Amansie Central</td>\n",
       "      <td id=\"T_455ae_row26_col4\" class=\"data row26 col4\" >coluzzii</td>\n",
       "      <td id=\"T_455ae_row26_col5\" class=\"data row26 col5\" >2018</td>\n",
       "      <td id=\"T_455ae_row26_col6\" class=\"data row26 col6\" >4</td>\n",
       "      <td id=\"T_455ae_row26_col7\" class=\"data row26 col7\" >69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row27_col0\" class=\"data row27 col0\" ><a href=\"cohort/GH-AH_Bekwai-Municipal_colu_2018_Q4.html\">Ghana / Bekwai Municipal / coluzzii / 2018 / Q4</a></td>\n",
       "      <td id=\"T_455ae_row27_col1\" class=\"data row27 col1\" >Ghana</td>\n",
       "      <td id=\"T_455ae_row27_col2\" class=\"data row27 col2\" >Ashanti Region</td>\n",
       "      <td id=\"T_455ae_row27_col3\" class=\"data row27 col3\" >Bekwai Municipal</td>\n",
       "      <td id=\"T_455ae_row27_col4\" class=\"data row27 col4\" >coluzzii</td>\n",
       "      <td id=\"T_455ae_row27_col5\" class=\"data row27 col5\" >2018</td>\n",
       "      <td id=\"T_455ae_row27_col6\" class=\"data row27 col6\" >4</td>\n",
       "      <td id=\"T_455ae_row27_col7\" class=\"data row27 col7\" >53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row28_col0\" class=\"data row28 col0\" ><a href=\"cohort/GH-CP_Twifo-Atti-Morkwa_colu_2012_Q3.html\">Ghana / Twifo Atti-Morkwa / coluzzii / 2012 / Q3</a></td>\n",
       "      <td id=\"T_455ae_row28_col1\" class=\"data row28 col1\" >Ghana</td>\n",
       "      <td id=\"T_455ae_row28_col2\" class=\"data row28 col2\" >Central Region</td>\n",
       "      <td id=\"T_455ae_row28_col3\" class=\"data row28 col3\" >Twifo Atti-Morkwa</td>\n",
       "      <td id=\"T_455ae_row28_col4\" class=\"data row28 col4\" >coluzzii</td>\n",
       "      <td id=\"T_455ae_row28_col5\" class=\"data row28 col5\" >2012</td>\n",
       "      <td id=\"T_455ae_row28_col6\" class=\"data row28 col6\" >3</td>\n",
       "      <td id=\"T_455ae_row28_col7\" class=\"data row28 col7\" >25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row29_col0\" class=\"data row29 col0\" ><a href=\"cohort/GH-CP_Upper-Denkyira-East-Municipal_colu_2018_Q4.html\">Ghana / Upper Denkyira East Municipal / coluzzii / 2018 / Q4</a></td>\n",
       "      <td id=\"T_455ae_row29_col1\" class=\"data row29 col1\" >Ghana</td>\n",
       "      <td id=\"T_455ae_row29_col2\" class=\"data row29 col2\" >Central Region</td>\n",
       "      <td id=\"T_455ae_row29_col3\" class=\"data row29 col3\" >Upper Denkyira East Municipal</td>\n",
       "      <td id=\"T_455ae_row29_col4\" class=\"data row29 col4\" >coluzzii</td>\n",
       "      <td id=\"T_455ae_row29_col5\" class=\"data row29 col5\" >2018</td>\n",
       "      <td id=\"T_455ae_row29_col6\" class=\"data row29 col6\" >4</td>\n",
       "      <td id=\"T_455ae_row29_col7\" class=\"data row29 col7\" >23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row30_col0\" class=\"data row30 col0\" ><a href=\"cohort/GH-CP_Upper-Denkyira-West_colu_2018_Q4.html\">Ghana / Upper Denkyira West / coluzzii / 2018 / Q4</a></td>\n",
       "      <td id=\"T_455ae_row30_col1\" class=\"data row30 col1\" >Ghana</td>\n",
       "      <td id=\"T_455ae_row30_col2\" class=\"data row30 col2\" >Central Region</td>\n",
       "      <td id=\"T_455ae_row30_col3\" class=\"data row30 col3\" >Upper Denkyira West</td>\n",
       "      <td id=\"T_455ae_row30_col4\" class=\"data row30 col4\" >coluzzii</td>\n",
       "      <td id=\"T_455ae_row30_col5\" class=\"data row30 col5\" >2018</td>\n",
       "      <td id=\"T_455ae_row30_col6\" class=\"data row30 col6\" >4</td>\n",
       "      <td id=\"T_455ae_row30_col7\" class=\"data row30 col7\" >118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row31_col0\" class=\"data row31 col0\" ><a href=\"cohort/GH-EP_New-Juaben-South-Municipal_gamb_2012_Q4.html\">Ghana / New Juaben South Municipal / gambiae / 2012 / Q4</a></td>\n",
       "      <td id=\"T_455ae_row31_col1\" class=\"data row31 col1\" >Ghana</td>\n",
       "      <td id=\"T_455ae_row31_col2\" class=\"data row31 col2\" >Eastern Region</td>\n",
       "      <td id=\"T_455ae_row31_col3\" class=\"data row31 col3\" >New Juaben South Municipal</td>\n",
       "      <td id=\"T_455ae_row31_col4\" class=\"data row31 col4\" >gambiae</td>\n",
       "      <td id=\"T_455ae_row31_col5\" class=\"data row31 col5\" >2012</td>\n",
       "      <td id=\"T_455ae_row31_col6\" class=\"data row31 col6\" >4</td>\n",
       "      <td id=\"T_455ae_row31_col7\" class=\"data row31 col7\" >23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row32_col0\" class=\"data row32 col0\" ><a href=\"cohort/GH-WP_Effia-Kwesimintsim-Municipal_colu_2012_Q3.html\">Ghana / Effia Kwesimintsim Municipal / coluzzii / 2012 / Q3</a></td>\n",
       "      <td id=\"T_455ae_row32_col1\" class=\"data row32 col1\" >Ghana</td>\n",
       "      <td id=\"T_455ae_row32_col2\" class=\"data row32 col2\" >Western Region</td>\n",
       "      <td id=\"T_455ae_row32_col3\" class=\"data row32 col3\" >Effia Kwesimintsim Municipal</td>\n",
       "      <td id=\"T_455ae_row32_col4\" class=\"data row32 col4\" >coluzzii</td>\n",
       "      <td id=\"T_455ae_row32_col5\" class=\"data row32 col5\" >2012</td>\n",
       "      <td id=\"T_455ae_row32_col6\" class=\"data row32 col6\" >3</td>\n",
       "      <td id=\"T_455ae_row32_col7\" class=\"data row32 col7\" >24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row33_col0\" class=\"data row33 col0\" ><a href=\"cohort/GM-M_Lower-Fuladu-West_colu_2012_Q4.html\">Gambia, The / Lower Fuladu West / coluzzii / 2012 / Q4</a></td>\n",
       "      <td id=\"T_455ae_row33_col1\" class=\"data row33 col1\" >Gambia, The</td>\n",
       "      <td id=\"T_455ae_row33_col2\" class=\"data row33 col2\" >Janjanbureh</td>\n",
       "      <td id=\"T_455ae_row33_col3\" class=\"data row33 col3\" >Lower Fuladu West</td>\n",
       "      <td id=\"T_455ae_row33_col4\" class=\"data row33 col4\" >coluzzii</td>\n",
       "      <td id=\"T_455ae_row33_col5\" class=\"data row33 col5\" >2012</td>\n",
       "      <td id=\"T_455ae_row33_col6\" class=\"data row33 col6\" >4</td>\n",
       "      <td id=\"T_455ae_row33_col7\" class=\"data row33 col7\" >172</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row34_col0\" class=\"data row34 col0\" ><a href=\"cohort/GM-N_Central-Badibu_biss_2011_Q3.html\">Gambia, The / Central Badibu / bissau / 2011 / Q3</a></td>\n",
       "      <td id=\"T_455ae_row34_col1\" class=\"data row34 col1\" >Gambia, The</td>\n",
       "      <td id=\"T_455ae_row34_col2\" class=\"data row34 col2\" >Kerewan</td>\n",
       "      <td id=\"T_455ae_row34_col3\" class=\"data row34 col3\" >Central Badibu</td>\n",
       "      <td id=\"T_455ae_row34_col4\" class=\"data row34 col4\" >bissau</td>\n",
       "      <td id=\"T_455ae_row34_col5\" class=\"data row34 col5\" >2011</td>\n",
       "      <td id=\"T_455ae_row34_col6\" class=\"data row34 col6\" >3</td>\n",
       "      <td id=\"T_455ae_row34_col7\" class=\"data row34 col7\" >52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row35_col0\" class=\"data row35 col0\" ><a href=\"cohort/GN-F_Kissidougou_gamb_2012_Q4.html\">Guinea / Kissidougou / gambiae / 2012 / Q4</a></td>\n",
       "      <td id=\"T_455ae_row35_col1\" class=\"data row35 col1\" >Guinea</td>\n",
       "      <td id=\"T_455ae_row35_col2\" class=\"data row35 col2\" >Faranah</td>\n",
       "      <td id=\"T_455ae_row35_col3\" class=\"data row35 col3\" >Kissidougou</td>\n",
       "      <td id=\"T_455ae_row35_col4\" class=\"data row35 col4\" >gambiae</td>\n",
       "      <td id=\"T_455ae_row35_col5\" class=\"data row35 col5\" >2012</td>\n",
       "      <td id=\"T_455ae_row35_col6\" class=\"data row35 col6\" >4</td>\n",
       "      <td id=\"T_455ae_row35_col7\" class=\"data row35 col7\" >51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row36_col0\" class=\"data row36 col0\" ><a href=\"cohort/GN-N_Macenta_gamb_2012_Q4.html\">Guinea / Macenta / gambiae / 2012 / Q4</a></td>\n",
       "      <td id=\"T_455ae_row36_col1\" class=\"data row36 col1\" >Guinea</td>\n",
       "      <td id=\"T_455ae_row36_col2\" class=\"data row36 col2\" >Nzerekore</td>\n",
       "      <td id=\"T_455ae_row36_col3\" class=\"data row36 col3\" >Macenta</td>\n",
       "      <td id=\"T_455ae_row36_col4\" class=\"data row36 col4\" >gambiae</td>\n",
       "      <td id=\"T_455ae_row36_col5\" class=\"data row36 col5\" >2012</td>\n",
       "      <td id=\"T_455ae_row36_col6\" class=\"data row36 col6\" >4</td>\n",
       "      <td id=\"T_455ae_row36_col7\" class=\"data row36 col7\" >51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row37_col0\" class=\"data row37 col0\" ><a href=\"cohort/GW-BM_Setor-De-Safim_biss_2010.html\">Guinea-Bissau / Setor De Safim / bissau / 2010</a></td>\n",
       "      <td id=\"T_455ae_row37_col1\" class=\"data row37 col1\" >Guinea-Bissau</td>\n",
       "      <td id=\"T_455ae_row37_col2\" class=\"data row37 col2\" >Biombo</td>\n",
       "      <td id=\"T_455ae_row37_col3\" class=\"data row37 col3\" >Setor de Safim</td>\n",
       "      <td id=\"T_455ae_row37_col4\" class=\"data row37 col4\" >bissau</td>\n",
       "      <td id=\"T_455ae_row37_col5\" class=\"data row37 col5\" >2010</td>\n",
       "      <td id=\"T_455ae_row37_col6\" class=\"data row37 col6\" ></td>\n",
       "      <td id=\"T_455ae_row37_col7\" class=\"data row37 col7\" >33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row38_col0\" class=\"data row38 col0\" ><a href=\"cohort/GW-BS_Bissau-Autonomous-Sector_biss_2010.html\">Guinea-Bissau / Bissau Autonomous Sector / bissau / 2010</a></td>\n",
       "      <td id=\"T_455ae_row38_col1\" class=\"data row38 col1\" >Guinea-Bissau</td>\n",
       "      <td id=\"T_455ae_row38_col2\" class=\"data row38 col2\" >Bissau</td>\n",
       "      <td id=\"T_455ae_row38_col3\" class=\"data row38 col3\" >Bissau Autonomous Sector</td>\n",
       "      <td id=\"T_455ae_row38_col4\" class=\"data row38 col4\" >bissau</td>\n",
       "      <td id=\"T_455ae_row38_col5\" class=\"data row38 col5\" >2010</td>\n",
       "      <td id=\"T_455ae_row38_col6\" class=\"data row38 col6\" ></td>\n",
       "      <td id=\"T_455ae_row38_col7\" class=\"data row38 col7\" >60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row39_col0\" class=\"data row39 col0\" ><a href=\"cohort/ML-2_Kangaba_gamb_2004_Q3.html\">Mali / Kangaba / gambiae / 2004 / Q3</a></td>\n",
       "      <td id=\"T_455ae_row39_col1\" class=\"data row39 col1\" >Mali</td>\n",
       "      <td id=\"T_455ae_row39_col2\" class=\"data row39 col2\" >Koulikouro</td>\n",
       "      <td id=\"T_455ae_row39_col3\" class=\"data row39 col3\" >Kangaba</td>\n",
       "      <td id=\"T_455ae_row39_col4\" class=\"data row39 col4\" >gambiae</td>\n",
       "      <td id=\"T_455ae_row39_col5\" class=\"data row39 col5\" >2004</td>\n",
       "      <td id=\"T_455ae_row39_col6\" class=\"data row39 col6\" >3</td>\n",
       "      <td id=\"T_455ae_row39_col7\" class=\"data row39 col7\" >23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row40_col0\" class=\"data row40 col0\" ><a href=\"cohort/ML-2_Kati_colu_2014_Q3.html\">Mali / Kati / coluzzii / 2014 / Q3</a></td>\n",
       "      <td id=\"T_455ae_row40_col1\" class=\"data row40 col1\" >Mali</td>\n",
       "      <td id=\"T_455ae_row40_col2\" class=\"data row40 col2\" >Koulikouro</td>\n",
       "      <td id=\"T_455ae_row40_col3\" class=\"data row40 col3\" >Kati</td>\n",
       "      <td id=\"T_455ae_row40_col4\" class=\"data row40 col4\" >coluzzii</td>\n",
       "      <td id=\"T_455ae_row40_col5\" class=\"data row40 col5\" >2014</td>\n",
       "      <td id=\"T_455ae_row40_col6\" class=\"data row40 col6\" >3</td>\n",
       "      <td id=\"T_455ae_row40_col7\" class=\"data row40 col7\" >27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row41_col0\" class=\"data row41 col0\" ><a href=\"cohort/ML-2_Kati_gamb_2014_Q3.html\">Mali / Kati / gambiae / 2014 / Q3</a></td>\n",
       "      <td id=\"T_455ae_row41_col1\" class=\"data row41 col1\" >Mali</td>\n",
       "      <td id=\"T_455ae_row41_col2\" class=\"data row41 col2\" >Koulikouro</td>\n",
       "      <td id=\"T_455ae_row41_col3\" class=\"data row41 col3\" >Kati</td>\n",
       "      <td id=\"T_455ae_row41_col4\" class=\"data row41 col4\" >gambiae</td>\n",
       "      <td id=\"T_455ae_row41_col5\" class=\"data row41 col5\" >2014</td>\n",
       "      <td id=\"T_455ae_row41_col6\" class=\"data row41 col6\" >3</td>\n",
       "      <td id=\"T_455ae_row41_col7\" class=\"data row41 col7\" >24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row42_col0\" class=\"data row42 col0\" ><a href=\"cohort/ML-3_Yanfolila_colu_2012_Q4.html\">Mali / Yanfolila / coluzzii / 2012 / Q4</a></td>\n",
       "      <td id=\"T_455ae_row42_col1\" class=\"data row42 col1\" >Mali</td>\n",
       "      <td id=\"T_455ae_row42_col2\" class=\"data row42 col2\" >Sikasso</td>\n",
       "      <td id=\"T_455ae_row42_col3\" class=\"data row42 col3\" >Yanfolila</td>\n",
       "      <td id=\"T_455ae_row42_col4\" class=\"data row42 col4\" >coluzzii</td>\n",
       "      <td id=\"T_455ae_row42_col5\" class=\"data row42 col5\" >2012</td>\n",
       "      <td id=\"T_455ae_row42_col6\" class=\"data row42 col6\" >4</td>\n",
       "      <td id=\"T_455ae_row42_col7\" class=\"data row42 col7\" >23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row43_col0\" class=\"data row43 col0\" ><a href=\"cohort/ML-3_Yanfolila_gamb_2012_Q4.html\">Mali / Yanfolila / gambiae / 2012 / Q4</a></td>\n",
       "      <td id=\"T_455ae_row43_col1\" class=\"data row43 col1\" >Mali</td>\n",
       "      <td id=\"T_455ae_row43_col2\" class=\"data row43 col2\" >Sikasso</td>\n",
       "      <td id=\"T_455ae_row43_col3\" class=\"data row43 col3\" >Yanfolila</td>\n",
       "      <td id=\"T_455ae_row43_col4\" class=\"data row43 col4\" >gambiae</td>\n",
       "      <td id=\"T_455ae_row43_col5\" class=\"data row43 col5\" >2012</td>\n",
       "      <td id=\"T_455ae_row43_col6\" class=\"data row43 col6\" >4</td>\n",
       "      <td id=\"T_455ae_row43_col7\" class=\"data row43 col7\" >53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row44_col0\" class=\"data row44 col0\" ><a href=\"cohort/ML-4_Bla_colu_2004_Q3.html\">Mali / Bla / coluzzii / 2004 / Q3</a></td>\n",
       "      <td id=\"T_455ae_row44_col1\" class=\"data row44 col1\" >Mali</td>\n",
       "      <td id=\"T_455ae_row44_col2\" class=\"data row44 col2\" >Segou</td>\n",
       "      <td id=\"T_455ae_row44_col3\" class=\"data row44 col3\" >Bla</td>\n",
       "      <td id=\"T_455ae_row44_col4\" class=\"data row44 col4\" >coluzzii</td>\n",
       "      <td id=\"T_455ae_row44_col5\" class=\"data row44 col5\" >2004</td>\n",
       "      <td id=\"T_455ae_row44_col6\" class=\"data row44 col6\" >3</td>\n",
       "      <td id=\"T_455ae_row44_col7\" class=\"data row44 col7\" >19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row45_col0\" class=\"data row45 col0\" ><a href=\"cohort/MW-S_Chikwawa_arab_2015_Q2.html\">Malawi / Chikwawa / arabiensis / 2015 / Q2</a></td>\n",
       "      <td id=\"T_455ae_row45_col1\" class=\"data row45 col1\" >Malawi</td>\n",
       "      <td id=\"T_455ae_row45_col2\" class=\"data row45 col2\" >Southern Region</td>\n",
       "      <td id=\"T_455ae_row45_col3\" class=\"data row45 col3\" >Chikwawa</td>\n",
       "      <td id=\"T_455ae_row45_col4\" class=\"data row45 col4\" >arabiensis</td>\n",
       "      <td id=\"T_455ae_row45_col5\" class=\"data row45 col5\" >2015</td>\n",
       "      <td id=\"T_455ae_row45_col6\" class=\"data row45 col6\" >2</td>\n",
       "      <td id=\"T_455ae_row45_col7\" class=\"data row45 col7\" >41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row46_col0\" class=\"data row46 col0\" ><a href=\"cohort/MZ-I_Morrumbene_gamb_2004_Q1.html\">Mozambique / Morrumbene / gambiae / 2004 / Q1</a></td>\n",
       "      <td id=\"T_455ae_row46_col1\" class=\"data row46 col1\" >Mozambique</td>\n",
       "      <td id=\"T_455ae_row46_col2\" class=\"data row46 col2\" >Inhambane</td>\n",
       "      <td id=\"T_455ae_row46_col3\" class=\"data row46 col3\" >Morrumbene</td>\n",
       "      <td id=\"T_455ae_row46_col4\" class=\"data row46 col4\" >gambiae</td>\n",
       "      <td id=\"T_455ae_row46_col5\" class=\"data row46 col5\" >2004</td>\n",
       "      <td id=\"T_455ae_row46_col6\" class=\"data row46 col6\" >1</td>\n",
       "      <td id=\"T_455ae_row46_col7\" class=\"data row46 col7\" >49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row47_col0\" class=\"data row47 col0\" ><a href=\"cohort/MZ-I_Morrumbene_gamb_2004_Q2.html\">Mozambique / Morrumbene / gambiae / 2004 / Q2</a></td>\n",
       "      <td id=\"T_455ae_row47_col1\" class=\"data row47 col1\" >Mozambique</td>\n",
       "      <td id=\"T_455ae_row47_col2\" class=\"data row47 col2\" >Inhambane</td>\n",
       "      <td id=\"T_455ae_row47_col3\" class=\"data row47 col3\" >Morrumbene</td>\n",
       "      <td id=\"T_455ae_row47_col4\" class=\"data row47 col4\" >gambiae</td>\n",
       "      <td id=\"T_455ae_row47_col5\" class=\"data row47 col5\" >2004</td>\n",
       "      <td id=\"T_455ae_row47_col6\" class=\"data row47 col6\" >2</td>\n",
       "      <td id=\"T_455ae_row47_col7\" class=\"data row47 col7\" >22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row48_col0\" class=\"data row48 col0\" ><a href=\"cohort/TG-M_Lome-Commune_gamb_2017_Q4.html\">Togo / Lome Commune / gambiae / 2017 / Q4</a></td>\n",
       "      <td id=\"T_455ae_row48_col1\" class=\"data row48 col1\" >Togo</td>\n",
       "      <td id=\"T_455ae_row48_col2\" class=\"data row48 col2\" >Maritime Region</td>\n",
       "      <td id=\"T_455ae_row48_col3\" class=\"data row48 col3\" >Lome Commune</td>\n",
       "      <td id=\"T_455ae_row48_col4\" class=\"data row48 col4\" >gambiae</td>\n",
       "      <td id=\"T_455ae_row48_col5\" class=\"data row48 col5\" >2017</td>\n",
       "      <td id=\"T_455ae_row48_col6\" class=\"data row48 col6\" >4</td>\n",
       "      <td id=\"T_455ae_row48_col7\" class=\"data row48 col7\" >179</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row49_col0\" class=\"data row49 col0\" ><a href=\"cohort/TZ-05_Muleba_arab_2015_Q1.html\">Tanzania / Muleba / arabiensis / 2015 / Q1</a></td>\n",
       "      <td id=\"T_455ae_row49_col1\" class=\"data row49 col1\" >Tanzania</td>\n",
       "      <td id=\"T_455ae_row49_col2\" class=\"data row49 col2\" >Kagera</td>\n",
       "      <td id=\"T_455ae_row49_col3\" class=\"data row49 col3\" >Muleba</td>\n",
       "      <td id=\"T_455ae_row49_col4\" class=\"data row49 col4\" >arabiensis</td>\n",
       "      <td id=\"T_455ae_row49_col5\" class=\"data row49 col5\" >2015</td>\n",
       "      <td id=\"T_455ae_row49_col6\" class=\"data row49 col6\" >1</td>\n",
       "      <td id=\"T_455ae_row49_col7\" class=\"data row49 col7\" >39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row50_col0\" class=\"data row50 col0\" ><a href=\"cohort/TZ-05_Muleba_arab_2015_Q2.html\">Tanzania / Muleba / arabiensis / 2015 / Q2</a></td>\n",
       "      <td id=\"T_455ae_row50_col1\" class=\"data row50 col1\" >Tanzania</td>\n",
       "      <td id=\"T_455ae_row50_col2\" class=\"data row50 col2\" >Kagera</td>\n",
       "      <td id=\"T_455ae_row50_col3\" class=\"data row50 col3\" >Muleba</td>\n",
       "      <td id=\"T_455ae_row50_col4\" class=\"data row50 col4\" >arabiensis</td>\n",
       "      <td id=\"T_455ae_row50_col5\" class=\"data row50 col5\" >2015</td>\n",
       "      <td id=\"T_455ae_row50_col6\" class=\"data row50 col6\" >2</td>\n",
       "      <td id=\"T_455ae_row50_col7\" class=\"data row50 col7\" >98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row51_col0\" class=\"data row51 col0\" ><a href=\"cohort/TZ-05_Muleba_gamb_2015_Q2.html\">Tanzania / Muleba / gambiae / 2015 / Q2</a></td>\n",
       "      <td id=\"T_455ae_row51_col1\" class=\"data row51 col1\" >Tanzania</td>\n",
       "      <td id=\"T_455ae_row51_col2\" class=\"data row51 col2\" >Kagera</td>\n",
       "      <td id=\"T_455ae_row51_col3\" class=\"data row51 col3\" >Muleba</td>\n",
       "      <td id=\"T_455ae_row51_col4\" class=\"data row51 col4\" >gambiae</td>\n",
       "      <td id=\"T_455ae_row51_col5\" class=\"data row51 col5\" >2015</td>\n",
       "      <td id=\"T_455ae_row51_col6\" class=\"data row51 col6\" >2</td>\n",
       "      <td id=\"T_455ae_row51_col7\" class=\"data row51 col7\" >18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row52_col0\" class=\"data row52 col0\" ><a href=\"cohort/TZ-13_Tarime_arab_2012_Q3.html\">Tanzania / Tarime / arabiensis / 2012 / Q3</a></td>\n",
       "      <td id=\"T_455ae_row52_col1\" class=\"data row52 col1\" >Tanzania</td>\n",
       "      <td id=\"T_455ae_row52_col2\" class=\"data row52 col2\" >Mara</td>\n",
       "      <td id=\"T_455ae_row52_col3\" class=\"data row52 col3\" >Tarime</td>\n",
       "      <td id=\"T_455ae_row52_col4\" class=\"data row52 col4\" >arabiensis</td>\n",
       "      <td id=\"T_455ae_row52_col5\" class=\"data row52 col5\" >2012</td>\n",
       "      <td id=\"T_455ae_row52_col6\" class=\"data row52 col6\" >3</td>\n",
       "      <td id=\"T_455ae_row52_col7\" class=\"data row52 col7\" >47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row53_col0\" class=\"data row53 col0\" ><a href=\"cohort/TZ-25_Muheza_gamb_2013_Q1.html\">Tanzania / Muheza / gambiae / 2013 / Q1</a></td>\n",
       "      <td id=\"T_455ae_row53_col1\" class=\"data row53 col1\" >Tanzania</td>\n",
       "      <td id=\"T_455ae_row53_col2\" class=\"data row53 col2\" >Tanga</td>\n",
       "      <td id=\"T_455ae_row53_col3\" class=\"data row53 col3\" >Muheza</td>\n",
       "      <td id=\"T_455ae_row53_col4\" class=\"data row53 col4\" >gambiae</td>\n",
       "      <td id=\"T_455ae_row53_col5\" class=\"data row53 col5\" >2013</td>\n",
       "      <td id=\"T_455ae_row53_col6\" class=\"data row53 col6\" >1</td>\n",
       "      <td id=\"T_455ae_row53_col7\" class=\"data row53 col7\" >32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row54_col0\" class=\"data row54 col0\" ><a href=\"cohort/TZ-26_Moshi_arab_2012_Q3.html\">Tanzania / Moshi / arabiensis / 2012 / Q3</a></td>\n",
       "      <td id=\"T_455ae_row54_col1\" class=\"data row54 col1\" >Tanzania</td>\n",
       "      <td id=\"T_455ae_row54_col2\" class=\"data row54 col2\" >Manyara</td>\n",
       "      <td id=\"T_455ae_row54_col3\" class=\"data row54 col3\" >Moshi</td>\n",
       "      <td id=\"T_455ae_row54_col4\" class=\"data row54 col4\" >arabiensis</td>\n",
       "      <td id=\"T_455ae_row54_col5\" class=\"data row54 col5\" >2012</td>\n",
       "      <td id=\"T_455ae_row54_col6\" class=\"data row54 col6\" >3</td>\n",
       "      <td id=\"T_455ae_row54_col7\" class=\"data row54 col7\" >39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row55_col0\" class=\"data row55 col0\" ><a href=\"cohort/UG-C_Kalangala_gamb_2015_Q2.html\">Uganda / Kalangala / gambiae / 2015 / Q2</a></td>\n",
       "      <td id=\"T_455ae_row55_col1\" class=\"data row55 col1\" >Uganda</td>\n",
       "      <td id=\"T_455ae_row55_col2\" class=\"data row55 col2\" >Central Region</td>\n",
       "      <td id=\"T_455ae_row55_col3\" class=\"data row55 col3\" >Kalangala</td>\n",
       "      <td id=\"T_455ae_row55_col4\" class=\"data row55 col4\" >gambiae</td>\n",
       "      <td id=\"T_455ae_row55_col5\" class=\"data row55 col5\" >2015</td>\n",
       "      <td id=\"T_455ae_row55_col6\" class=\"data row55 col6\" >2</td>\n",
       "      <td id=\"T_455ae_row55_col7\" class=\"data row55 col7\" >60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row56_col0\" class=\"data row56 col0\" ><a href=\"cohort/UG-E_Busia_gamb_2016_Q2.html\">Uganda / Busia / gambiae / 2016 / Q2</a></td>\n",
       "      <td id=\"T_455ae_row56_col1\" class=\"data row56 col1\" >Uganda</td>\n",
       "      <td id=\"T_455ae_row56_col2\" class=\"data row56 col2\" >Eastern Region</td>\n",
       "      <td id=\"T_455ae_row56_col3\" class=\"data row56 col3\" >Busia</td>\n",
       "      <td id=\"T_455ae_row56_col4\" class=\"data row56 col4\" >gambiae</td>\n",
       "      <td id=\"T_455ae_row56_col5\" class=\"data row56 col5\" >2016</td>\n",
       "      <td id=\"T_455ae_row56_col6\" class=\"data row56 col6\" >2</td>\n",
       "      <td id=\"T_455ae_row56_col7\" class=\"data row56 col7\" >24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row57_col0\" class=\"data row57 col0\" ><a href=\"cohort/UG-E_Mayuge_gamb_2017_Q2.html\">Uganda / Mayuge / gambiae / 2017 / Q2</a></td>\n",
       "      <td id=\"T_455ae_row57_col1\" class=\"data row57 col1\" >Uganda</td>\n",
       "      <td id=\"T_455ae_row57_col2\" class=\"data row57 col2\" >Eastern Region</td>\n",
       "      <td id=\"T_455ae_row57_col3\" class=\"data row57 col3\" >Mayuge</td>\n",
       "      <td id=\"T_455ae_row57_col4\" class=\"data row57 col4\" >gambiae</td>\n",
       "      <td id=\"T_455ae_row57_col5\" class=\"data row57 col5\" >2017</td>\n",
       "      <td id=\"T_455ae_row57_col6\" class=\"data row57 col6\" >2</td>\n",
       "      <td id=\"T_455ae_row57_col7\" class=\"data row57 col7\" >21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row58_col0\" class=\"data row58 col0\" ><a href=\"cohort/UG-E_Tororo_arab_2012_Q4.html\">Uganda / Tororo / arabiensis / 2012 / Q4</a></td>\n",
       "      <td id=\"T_455ae_row58_col1\" class=\"data row58 col1\" >Uganda</td>\n",
       "      <td id=\"T_455ae_row58_col2\" class=\"data row58 col2\" >Eastern Region</td>\n",
       "      <td id=\"T_455ae_row58_col3\" class=\"data row58 col3\" >Tororo</td>\n",
       "      <td id=\"T_455ae_row58_col4\" class=\"data row58 col4\" >arabiensis</td>\n",
       "      <td id=\"T_455ae_row58_col5\" class=\"data row58 col5\" >2012</td>\n",
       "      <td id=\"T_455ae_row58_col6\" class=\"data row58 col6\" >4</td>\n",
       "      <td id=\"T_455ae_row58_col7\" class=\"data row58 col7\" >81</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row59_col0\" class=\"data row59 col0\" ><a href=\"cohort/UG-E_Tororo_gamb_2012_Q4.html\">Uganda / Tororo / gambiae / 2012 / Q4</a></td>\n",
       "      <td id=\"T_455ae_row59_col1\" class=\"data row59 col1\" >Uganda</td>\n",
       "      <td id=\"T_455ae_row59_col2\" class=\"data row59 col2\" >Eastern Region</td>\n",
       "      <td id=\"T_455ae_row59_col3\" class=\"data row59 col3\" >Tororo</td>\n",
       "      <td id=\"T_455ae_row59_col4\" class=\"data row59 col4\" >gambiae</td>\n",
       "      <td id=\"T_455ae_row59_col5\" class=\"data row59 col5\" >2012</td>\n",
       "      <td id=\"T_455ae_row59_col6\" class=\"data row59 col6\" >4</td>\n",
       "      <td id=\"T_455ae_row59_col7\" class=\"data row59 col7\" >112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td id=\"T_455ae_row60_col0\" class=\"data row60 col0\" ><a href=\"cohort/UG-W_Kanungu_gamb_2012_Q4.html\">Uganda / Kanungu / gambiae / 2012 / Q4</a></td>\n",
       "      <td id=\"T_455ae_row60_col1\" class=\"data row60 col1\" >Uganda</td>\n",
       "      <td id=\"T_455ae_row60_col2\" class=\"data row60 col2\" >Western Region</td>\n",
       "      <td id=\"T_455ae_row60_col3\" class=\"data row60 col3\" >Kanungu</td>\n",
       "      <td id=\"T_455ae_row60_col4\" class=\"data row60 col4\" >gambiae</td>\n",
       "      <td id=\"T_455ae_row60_col5\" class=\"data row60 col5\" >2012</td>\n",
       "      <td id=\"T_455ae_row60_col6\" class=\"data row60 col6\" >4</td>\n",
       "      <td id=\"T_455ae_row60_col7\" class=\"data row60 col7\" >95</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x7ff9ed350f20>"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "page_utils.style_cohorts_table(\n",
    "    gdf_cohorts,\n",
    "    caption=\"Table 2. Cohorts selected for genome-wide selection scan analyses.\",\n",
    "    url_prefix=\"\",\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "908df9c7-5b95-44fd-a644-072a210e2200",
   "metadata": {},
   "source": [
    "## H12 and G123 window size calibration\n",
    "\n",
    "Both H12 ([Garud et al. 2015](https://pmc.ncbi.nlm.nih.gov/articles/PMC4338236/)) and G123 ([Harris et al. 2018](https://pmc.ncbi.nlm.nih.gov/articles/PMC6283157/)) are statistical methods for performing {term}`genome-wide selection scans` which rely on dividing data into windows along the {term}`genome`. Typically the size of these windows is set to a fixed number of polymorphic sites ({term}`SNP`s). I.e., all windows contain data  either phased haplotypes or unphased genotypes  for the same number of {term}`SNP`s. In order to detect recent selective sweeps, the size of these windows needs to be chosen so that windows are generally larger than the normal genetic distance over which linkage disequlibrium (LD) decays to background levels in the absence of recent positive selection. Therefore, in windows which are unaffected by recent selective sweeps, genetic diversity will be high and thus the values of the selection statistics will be low. Conversely, in windows affected by recent selective sweeps, linkage disequilibrium will extend over a longer genetic distance spanning multiple windows, so that genetic diversity within those windows is low and thus values of selection statistics will be high. In other words, the choice of window size affects the signal to noise ratio for selection scans using H12 and G123 statistics. If windows are too small, results are dominated by background noise. If windows are too large, noise is minimal but power to detect recent selection signals is reduced.\n",
    "\n",
    "This decision regarding an appropriate window size needs to be made independently for each cohort of samples over which a selection scan will be performed. This is because different source populations may have different demographic histories, and this in turn may alter the genetic distance over which LD decays in the absence of positive selection. Previous studies have used various demographic inference methods to try to infer key demographic parameters for each cohort being analysed, then use these parameters to inform the decision of window size. In practice, this approach presents a number of challenges. Firstly, inference of demographic parameters is difficult, and even state of the art inference methods may reach inaccurate conclusions. Secondly, running demographic inference methods can be computationally demanding, and this becomes impractical for large numbers of cohorts."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "0102b71a-3f48-45f6-8d5e-9a0cd2c321e3",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/markdown": [
       "\n",
       "For these reasons we have taken an empirical approach to window size calibration for H12 \n",
       "and G123 scans, designed to reach a good signal to noise ratio. \n",
       "\n",
       "For each cohort, we compute H12 over contig 3RL for multiple window \n",
       "sizes of 100, 200, 300, 400, 500, 600, 700, 800, 900, 1000, 1200, 1400, 1600, 1800, 2000, 2500, 3000, 3500, 4000, 4500, 5000, 6000, 7000, 8000, 9000 or 10000 SNPs. We then compute the 95th \n",
       "percentile of statistic values over all windows. We choose the smallest window size for which the \n",
       "95th percentile is below 0.08. This means that any window with a\n",
       "statistic value above this threshold will be in the top 5% of windows.\n",
       "\n",
       "Similarly, we compute G123 over contig 3RL for multiple window \n",
       "sizes of 100, 200, 300, 400, 500, 600, 700, 800, 900, 1000, 1200, 1400, 1600, 1800, 2000, 2500, 3000, 3500, 4000, 4500, 5000, 6000, 7000, 8000, 9000 or 10000 SNPs. We then compute the 95th \n",
       "percentile of statistic values over all windows. We choose the smallest window size for which the \n",
       "95th percentile is below 0.08.\n",
       "\n"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Markdown(f\"\"\"\n",
    "For these reasons we have taken an empirical approach to window size calibration for H12 \n",
    "and G123 scans, designed to reach a good signal to noise ratio. \n",
    "\n",
    "For each cohort, we compute H12 over contig {setup.h12_calibration_contig} for multiple window \n",
    "sizes of {human_readable_list(setup.h12_calibration_window_sizes)} SNPs. We then compute the 95th \n",
    "percentile of statistic values over all windows. We choose the smallest window size for which the \n",
    "95th percentile is below {setup.h12_calibration_threshold}. This means that any window with a\n",
    "statistic value above this threshold will be in the top 5% of windows.\n",
    "\n",
    "Similarly, we compute G123 over contig {setup.g123_calibration_contig} for multiple window \n",
    "sizes of {human_readable_list(setup.g123_calibration_window_sizes)} SNPs. We then compute the 95th \n",
    "percentile of statistic values over all windows. We choose the smallest window size for which the \n",
    "95th percentile is below {setup.g123_calibration_threshold}.\n",
    "\n",
    "\"\"\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fef5d695-012d-48d4-a313-755e8c726f13",
   "metadata": {},
   "source": [
    "TODO how was window-size calibration done?\n",
    "\n",
    "TODO after calibration, some cohorts removed if cannot get a window-size."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3549b2f6-a7c8-4625-9260-78846ed9ca2e",
   "metadata": {},
   "source": [
    "## H12 genome-wide selection scans\n",
    "\n",
    "TODO"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "aa16dbb2-2ed3-4f16-ad74-9b8f9821648e",
   "metadata": {},
   "source": [
    "## G123 genome-wide selection scans\n",
    "\n",
    "TODO"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "777d6d11-dd99-44c6-9a52-c68142122065",
   "metadata": {},
   "source": [
    "## IHS genome-wide selection scans\n",
    "\n",
    "TODO"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "afdda01b-9155-4981-8206-df22739fff33",
   "metadata": {},
   "source": [
    "## Automated detection of selection signals\n",
    "\n",
    "TODO"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "417e3d70-368e-49b8-a237-12c2a6f5ed06",
   "metadata": {},
   "source": [
    "## Identification of selection alerts\n",
    "\n",
    "TODO"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "66294008-9a26-4d51-9303-d534c108a4db",
   "metadata": {},
   "source": [
    "## Web report generation\n",
    "\n",
    "TODO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1a09b58a-0129-4174-af02-290fb5e6084e",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "selection-atlas",
   "language": "python",
   "name": "selection-atlas"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
