{"cells": [{"cell_type": "code", "execution_count": 1, "id": "0807aa44-7dd4-4fce-bdcd-8420508deaa1", "metadata": {"execution": {"iopub.execute_input": "2025-03-27T13:05:23.025408Z", "iopub.status.busy": "2025-03-27T13:05:23.025082Z", "iopub.status.idle": "2025-03-27T13:05:23.028358Z", "shell.execute_reply": "2025-03-27T13:05:23.027812Z"}, "papermill": {"duration": 0.008296, "end_time": "2025-03-27T13:05:23.029261", "exception": false, "start_time": "2025-03-27T13:05:23.020965", "status": "completed"}, "tags": ["parameters", "remove-input"]}, "outputs": [], "source": ["# Notebook parameters. Values here are for development only and\n", "# will be overridden when running via snakemake and papermill.\n", "\n", "config_file = \"../../../config/agam.yaml\"\n", "# config_file = \"../../../config/afun.yaml\""]}, {"cell_type": "code", "execution_count": 2, "id": "e2fa75f4", "metadata": {"execution": {"iopub.execute_input": "2025-03-27T13:05:23.034806Z", "iopub.status.busy": "2025-03-27T13:05:23.034432Z", "iopub.status.idle": "2025-03-27T13:05:23.036927Z", "shell.execute_reply": "2025-03-27T13:05:23.036413Z"}, "papermill": {"duration": 0.006008, "end_time": "2025-03-27T13:05:23.037773", "exception": false, "start_time": "2025-03-27T13:05:23.031765", "status": "completed"}, "tags": ["injected-parameters", "remove-input"]}, "outputs": [], "source": ["# Parameters\n", "config_file = \"/home/runner/work/selection-atlas/selection-atlas/config/afun.yaml\"\n"]}, {"cell_type": "code", "execution_count": 3, "id": "01e7c9f6-baee-47dc-9aff-67aaa272787c", "metadata": {"execution": {"iopub.execute_input": "2025-03-27T13:05:23.043146Z", "iopub.status.busy": "2025-03-27T13:05:23.042769Z", "iopub.status.idle": "2025-03-27T13:05:25.317539Z", "shell.execute_reply": "2025-03-27T13:05:25.317017Z"}, "papermill": {"duration": 2.278243, "end_time": "2025-03-27T13:05:25.318365", "exception": false, "start_time": "2025-03-27T13:05:23.040122", "status": "completed"}, "tags": ["remove-input"]}, "outputs": [{"data": {"application/javascript": ["'use strict';\n", "(function(root) {\n", "  function now() {\n", "    return new Date();\n", "  }\n", "\n", "  const force = true;\n", "\n", "  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n", "    root._bokeh_onload_callbacks = [];\n", "    root._bokeh_is_loading = undefined;\n", "  }\n", "\n", "const JS_MIME_TYPE = 'application/javascript';\n", "  const HTML_MIME_TYPE = 'text/html';\n", "  const EXEC_MIME_TYPE = 'application/vnd.bokehjs_exec.v0+json';\n", "  const CLASS_NAME = 'output_bokeh rendered_html';\n", "\n", "  /**\n", "   * Render data to the DOM node\n", "   */\n", "  function render(props, node) {\n", "    const script = document.createElement(\"script\");\n", "    node.appendChild(script);\n", "  }\n", "\n", "  /**\n", "   * Handle when an output is cleared or removed\n", "   */\n", "  function handleClearOutput(event, handle) {\n", "    function drop(id) {\n", "      const view = Bokeh.index.get_by_id(id)\n", "      if (view != null) {\n", "        view.model.document.clear()\n", "        Bokeh.index.delete(view)\n", "      }\n", "    }\n", "\n", "    const cell = handle.cell;\n", "\n", "    const id = cell.output_area._bokeh_element_id;\n", "    const server_id = cell.output_area._bokeh_server_id;\n", "\n", "    // Clean up Bokeh references\n", "    if (id != null) {\n", "      drop(id)\n", "    }\n", "\n", "    if (server_id !== undefined) {\n", "      // Clean up Bokeh references\n", "      const cmd_clean = \"from bokeh.io.state import curstate; print(curstate().uuid_to_server['\" + server_id + \"'].get_sessions()[0].document.roots[0]._id)\";\n", "      cell.notebook.kernel.execute(cmd_clean, {\n", "        iopub: {\n", "          output: function(msg) {\n", "            const id = msg.content.text.trim()\n", "            drop(id)\n", "          }\n", "        }\n", "      });\n", "      // Destroy server and session\n", "      const cmd_destroy = \"import bokeh.io.notebook as ion; ion.destroy_server('\" + server_id + \"')\";\n", "      cell.notebook.kernel.execute(cmd_destroy);\n", "    }\n", "  }\n", "\n", "  /**\n", "   * Handle when a new output is added\n", "   */\n", "  function handleAddOutput(event, handle) {\n", "    const output_area = handle.output_area;\n", "    const output = handle.output;\n", "\n", "    // limit handleAddOutput to display_data with EXEC_MIME_TYPE content only\n", "    if ((output.output_type != \"display_data\") || (!Object.prototype.hasOwnProperty.call(output.data, EXEC_MIME_TYPE))) {\n", "      return\n", "    }\n", "\n", "    const toinsert = output_area.element.find(\".\" + CLASS_NAME.split(' ')[0]);\n", "\n", "    if (output.metadata[EXEC_MIME_TYPE][\"id\"] !== undefined) {\n", "      toinsert[toinsert.length - 1].firstChild.textContent = output.data[JS_MIME_TYPE];\n", "      // store reference to embed id on output_area\n", "      output_area._bokeh_element_id = output.metadata[EXEC_MIME_TYPE][\"id\"];\n", "    }\n", "    if (output.metadata[EXEC_MIME_TYPE][\"server_id\"] !== undefined) {\n", "      const bk_div = document.createElement(\"div\");\n", "      bk_div.innerHTML = output.data[HTML_MIME_TYPE];\n", "      const script_attrs = bk_div.children[0].attributes;\n", "      for (let i = 0; i < script_attrs.length; i++) {\n", "        toinsert[toinsert.length - 1].firstChild.setAttribute(script_attrs[i].name, script_attrs[i].value);\n", "        toinsert[toinsert.length - 1].firstChild.textContent = bk_div.children[0].textContent\n", "      }\n", "      // store reference to server id on output_area\n", "      output_area._bokeh_server_id = output.metadata[EXEC_MIME_TYPE][\"server_id\"];\n", "    }\n", "  }\n", "\n", "  function register_renderer(events, OutputArea) {\n", "\n", "    function append_mime(data, metadata, element) {\n", "      // create a DOM node to render to\n", "      const toinsert = this.create_output_subarea(\n", "        metadata,\n", "        CLASS_NAME,\n", "        EXEC_MIME_TYPE\n", "      );\n", "      this.keyboard_manager.register_events(toinsert);\n", "      // Render to node\n", "      const props = {data: data, metadata: metadata[EXEC_MIME_TYPE]};\n", "      render(props, toinsert[toinsert.length - 1]);\n", "      element.append(toinsert);\n", "      return toinsert\n", "    }\n", "\n", "    /* Handle when an output is cleared or removed */\n", "    events.on('clear_output.CodeCell', handleClearOutput);\n", "    events.on('delete.Cell', handleClearOutput);\n", "\n", "    /* Handle when a new output is added */\n", "    events.on('output_added.OutputArea', handleAddOutput);\n", "\n", "    /**\n", "     * Register the mime type and append_mime function with output_area\n", "     */\n", "    OutputArea.prototype.register_mime_type(EXEC_MIME_TYPE, append_mime, {\n", "      /* Is output safe? */\n", "      safe: true,\n", "      /* Index of renderer in `output_area.display_order` */\n", "      index: 0\n", "    });\n", "  }\n", "\n", "  // register the mime type if in Jupyter Notebook environment and previously unregistered\n", "  if (root.Jupyter !== undefined) {\n", "    const events = require('base/js/events');\n", "    const OutputArea = require('notebook/js/outputarea').OutputArea;\n", "\n", "    if (OutputArea.prototype.mime_types().indexOf(EXEC_MIME_TYPE) == -1) {\n", "      register_renderer(events, OutputArea);\n", "    }\n", "  }\n", "  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n", "    root._bokeh_timeout = Date.now() + 5000;\n", "    root._bokeh_failed_load = false;\n", "  }\n", "\n", "  const NB_LOAD_WARNING = {'data': {'text/html':\n", "     \"<div style='background-color: #fdd'>\\n\"+\n", "     \"<p>\\n\"+\n", "     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n", "     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n", "     \"</p>\\n\"+\n", "     \"<ul>\\n\"+\n", "     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n", "     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n", "     \"</ul>\\n\"+\n", "     \"<code>\\n\"+\n", "     \"from bokeh.resources import INLINE\\n\"+\n", "     \"output_notebook(resources=INLINE)\\n\"+\n", "     \"</code>\\n\"+\n", "     \"</div>\"}};\n", "\n", "  function display_loaded(error = null) {\n", "    const el = document.getElementById(null);\n", "    if (el != null) {\n", "      const html = (() => {\n", "        if (typeof root.Bokeh === \"undefined\") {\n", "          if (error == null) {\n", "            return \"BokehJS is loading ...\";\n", "          } else {\n", "            return \"BokehJS failed to load.\";\n", "          }\n", "        } else {\n", "          const prefix = `BokehJS ${root.Bokeh.version}`;\n", "          if (error == null) {\n", "            return `${prefix} successfully loaded.`;\n", "          } else {\n", "            return `${prefix} <b>encountered errors</b> while loading and may not function as expected.`;\n", "          }\n", "        }\n", "      })();\n", "      el.innerHTML = html;\n", "\n", "      if (error != null) {\n", "        const wrapper = document.createElement(\"div\");\n", "        wrapper.style.overflow = \"auto\";\n", "        wrapper.style.height = \"5em\";\n", "        wrapper.style.resize = \"vertical\";\n", "        const content = document.createElement(\"div\");\n", "        content.style.fontFamily = \"monospace\";\n", "        content.style.whiteSpace = \"pre-wrap\";\n", "        content.style.backgroundColor = \"rgb(255, 221, 221)\";\n", "        content.textContent = error.stack ?? error.toString();\n", "        wrapper.append(content);\n", "        el.append(wrapper);\n", "      }\n", "    } else if (Date.now() < root._bokeh_timeout) {\n", "      setTimeout(() => display_loaded(error), 100);\n", "    }\n", "  }\n", "\n", "  function run_callbacks() {\n", "    try {\n", "      root._bokeh_onload_callbacks.forEach(function(callback) {\n", "        if (callback != null)\n", "          callback();\n", "      });\n", "    } finally {\n", "      delete root._bokeh_onload_callbacks\n", "    }\n", "    console.debug(\"Bokeh: all callbacks have finished\");\n", "  }\n", "\n", "  function load_libs(css_urls, js_urls, callback) {\n", "    if (css_urls == null) css_urls = [];\n", "    if (js_urls == null) js_urls = [];\n", "\n", "    root._bokeh_onload_callbacks.push(callback);\n", "    if (root._bokeh_is_loading > 0) {\n", "      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n", "      return null;\n", "    }\n", "    if (js_urls == null || js_urls.length === 0) {\n", "      run_callbacks();\n", "      return null;\n", "    }\n", "    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n", "    root._bokeh_is_loading = css_urls.length + js_urls.length;\n", "\n", "    function on_load() {\n", "      root._bokeh_is_loading--;\n", "      if (root._bokeh_is_loading === 0) {\n", "        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n", "        run_callbacks()\n", "      }\n", "    }\n", "\n", "    function on_error(url) {\n", "      console.error(\"failed to load \" + url);\n", "    }\n", "\n", "    for (let i = 0; i < css_urls.length; i++) {\n", "      const url = css_urls[i];\n", "      const element = document.createElement(\"link\");\n", "      element.onload = on_load;\n", "      element.onerror = on_error.bind(null, url);\n", "      element.rel = \"stylesheet\";\n", "      element.type = \"text/css\";\n", "      element.href = url;\n", "      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n", "      document.body.appendChild(element);\n", "    }\n", "\n", "    for (let i = 0; i < js_urls.length; i++) {\n", "      const url = js_urls[i];\n", "      const element = document.createElement('script');\n", "      element.onload = on_load;\n", "      element.onerror = on_error.bind(null, url);\n", "      element.async = false;\n", "      element.src = url;\n", "      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n", "      document.head.appendChild(element);\n", "    }\n", "  };\n", "\n", "  function inject_raw_css(css) {\n", "    const element = document.createElement(\"style\");\n", "    element.appendChild(document.createTextNode(css));\n", "    document.body.appendChild(element);\n", "  }\n", "\n", "  const js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-3.7.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-3.7.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-3.7.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-3.7.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-mathjax-3.7.0.min.js\"];\n", "  const css_urls = [];\n", "\n", "  const inline_js = [    function(Bokeh) {\n", "      Bokeh.set_log_level(\"info\");\n", "    },\n", "function(Bokeh) {\n", "    }\n", "  ];\n", "\n", "  function run_inline_js() {\n", "    if (root.Bokeh !== undefined || force === true) {\n", "      try {\n", "            for (let i = 0; i < inline_js.length; i++) {\n", "      inline_js[i].call(root, root.Bokeh);\n", "    }\n", "\n", "      } catch (error) {throw error;\n", "      }} else if (Date.now() < root._bokeh_timeout) {\n", "      setTimeout(run_inline_js, 100);\n", "    } else if (!root._bokeh_failed_load) {\n", "      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n", "      root._bokeh_failed_load = true;\n", "    } else if (force !== true) {\n", "      const cell = $(document.getElementById(null)).parents('.cell').data().cell;\n", "      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n", "    }\n", "  }\n", "\n", "  if (root._bokeh_is_loading === 0) {\n", "    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n", "    run_inline_js();\n", "  } else {\n", "    load_libs(css_urls, js_urls, function() {\n", "      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n", "      run_inline_js();\n", "    });\n", "  }\n", "}(window));"], "application/vnd.bokehjs_load.v0+json": "'use strict';\n(function(root) {\n  function now() {\n    return new Date();\n  }\n\n  const force = true;\n\n  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n    root._bokeh_onload_callbacks = [];\n    root._bokeh_is_loading = undefined;\n  }\n\n\n  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n    root._bokeh_timeout = Date.now() + 5000;\n    root._bokeh_failed_load = false;\n  }\n\n  const NB_LOAD_WARNING = {'data': {'text/html':\n     \"<div style='background-color: #fdd'>\\n\"+\n     \"<p>\\n\"+\n     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n     \"</p>\\n\"+\n     \"<ul>\\n\"+\n     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n     \"</ul>\\n\"+\n     \"<code>\\n\"+\n     \"from bokeh.resources import INLINE\\n\"+\n     \"output_notebook(resources=INLINE)\\n\"+\n     \"</code>\\n\"+\n     \"</div>\"}};\n\n  function display_loaded(error = null) {\n    const el = document.getElementById(null);\n    if (el != null) {\n      const html = (() => {\n        if (typeof root.Bokeh === \"undefined\") {\n          if (error == null) {\n            return \"BokehJS is loading ...\";\n          } else {\n            return \"BokehJS failed to load.\";\n          }\n        } else {\n          const prefix = `BokehJS ${root.Bokeh.version}`;\n          if (error == null) {\n            return `${prefix} successfully loaded.`;\n          } else {\n            return `${prefix} <b>encountered errors</b> while loading and may not function as expected.`;\n          }\n        }\n      })();\n      el.innerHTML = html;\n\n      if (error != null) {\n        const wrapper = document.createElement(\"div\");\n        wrapper.style.overflow = \"auto\";\n        wrapper.style.height = \"5em\";\n        wrapper.style.resize = \"vertical\";\n        const content = document.createElement(\"div\");\n        content.style.fontFamily = \"monospace\";\n        content.style.whiteSpace = \"pre-wrap\";\n        content.style.backgroundColor = \"rgb(255, 221, 221)\";\n        content.textContent = error.stack ?? error.toString();\n        wrapper.append(content);\n        el.append(wrapper);\n      }\n    } else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(() => display_loaded(error), 100);\n    }\n  }\n\n  function run_callbacks() {\n    try {\n      root._bokeh_onload_callbacks.forEach(function(callback) {\n        if (callback != null)\n          callback();\n      });\n    } finally {\n      delete root._bokeh_onload_callbacks\n    }\n    console.debug(\"Bokeh: all callbacks have finished\");\n  }\n\n  function load_libs(css_urls, js_urls, callback) {\n    if (css_urls == null) css_urls = [];\n    if (js_urls == null) js_urls = [];\n\n    root._bokeh_onload_callbacks.push(callback);\n    if (root._bokeh_is_loading > 0) {\n      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n      return null;\n    }\n    if (js_urls == null || js_urls.length === 0) {\n      run_callbacks();\n      return null;\n    }\n    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n    root._bokeh_is_loading = css_urls.length + js_urls.length;\n\n    function on_load() {\n      root._bokeh_is_loading--;\n      if (root._bokeh_is_loading === 0) {\n        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n        run_callbacks()\n      }\n    }\n\n    function on_error(url) {\n      console.error(\"failed to load \" + url);\n    }\n\n    for (let i = 0; i < css_urls.length; i++) {\n      const url = css_urls[i];\n      const element = document.createElement(\"link\");\n      element.onload = on_load;\n      element.onerror = on_error.bind(null, url);\n      element.rel = \"stylesheet\";\n      element.type = \"text/css\";\n      element.href = url;\n      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n      document.body.appendChild(element);\n    }\n\n    for (let i = 0; i < js_urls.length; i++) {\n      const url = js_urls[i];\n      const element = document.createElement('script');\n      element.onload = on_load;\n      element.onerror = on_error.bind(null, url);\n      element.async = false;\n      element.src = url;\n      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n      document.head.appendChild(element);\n    }\n  };\n\n  function inject_raw_css(css) {\n    const element = document.createElement(\"style\");\n    element.appendChild(document.createTextNode(css));\n    document.body.appendChild(element);\n  }\n\n  const js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-3.7.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-3.7.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-3.7.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-3.7.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-mathjax-3.7.0.min.js\"];\n  const css_urls = [];\n\n  const inline_js = [    function(Bokeh) {\n      Bokeh.set_log_level(\"info\");\n    },\nfunction(Bokeh) {\n    }\n  ];\n\n  function run_inline_js() {\n    if (root.Bokeh !== undefined || force === true) {\n      try {\n            for (let i = 0; i < inline_js.length; i++) {\n      inline_js[i].call(root, root.Bokeh);\n    }\n\n      } catch (error) {throw error;\n      }} else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(run_inline_js, 100);\n    } else if (!root._bokeh_failed_load) {\n      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n      root._bokeh_failed_load = true;\n    } else if (force !== true) {\n      const cell = $(document.getElementById(null)).parents('.cell').data().cell;\n      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n    }\n  }\n\n  if (root._bokeh_is_loading === 0) {\n    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n    run_inline_js();\n  } else {\n    load_libs(css_urls, js_urls, function() {\n      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n      run_inline_js();\n    });\n  }\n}(window));"}, "metadata": {}, "output_type": "display_data"}], "source": ["from bokeh.io import output_notebook\n", "import malariagen_data\n", "from IPython.display import Markdown\n", "from selection_atlas.setup import AtlasSetup\n", "from selection_atlas.page_utils import AtlasPageUtils\n", "\n", "# Initialise the atlas setup.\n", "setup = AtlasSetup(config_file)\n", "page_utils = AtlasPageUtils(setup=setup)\n", "\n", "# N.B., do not add the \"remove-output\" tag to this cell!!! If you do,\n", "# the bokeh javascript libraries will not get loaded in the generated\n", "# HTML page. The call to output_notebook() injects javascript in the\n", "# cell output which triggers the bokeh javascript libraries to be loaded\n", "# in the page.\n", "output_notebook(hide_banner=True)"]}, {"cell_type": "markdown", "id": "0442c1a1-7dd1-4b5b-85e1-a69d2fb803d5", "metadata": {"papermill": {"duration": 0.002568, "end_time": "2025-03-27T13:05:25.323920", "exception": false, "start_time": "2025-03-27T13:05:25.321352", "status": "completed"}, "tags": []}, "source": ["# Methods"]}, {"cell_type": "markdown", "id": "77233d5d-c5ff-439d-9a99-94721d64a36b", "metadata": {"papermill": {"duration": 0.002505, "end_time": "2025-03-27T13:05:25.329139", "exception": false, "start_time": "2025-03-27T13:05:25.326634", "status": "completed"}, "tags": []}, "source": ["## Data sources"]}, {"cell_type": "code", "execution_count": 4, "id": "713f44fc-33f1-4cd0-8534-0067bf3d50af", "metadata": {"execution": {"iopub.execute_input": "2025-03-27T13:05:25.335719Z", "iopub.status.busy": "2025-03-27T13:05:25.334860Z", "iopub.status.idle": "2025-03-27T13:05:28.266411Z", "shell.execute_reply": "2025-03-27T13:05:28.265923Z"}, "papermill": {"duration": 2.935584, "end_time": "2025-03-27T13:05:28.267343", "exception": false, "start_time": "2025-03-27T13:05:25.331759", "status": "completed"}, "tags": ["remove-input"]}, "outputs": [{"data": {"application/javascript": ["'use strict';\n", "(function(root) {\n", "  function now() {\n", "    return new Date();\n", "  }\n", "\n", "  const force = true;\n", "\n", "  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n", "    root._bokeh_onload_callbacks = [];\n", "    root._bokeh_is_loading = undefined;\n", "  }\n", "\n", "const JS_MIME_TYPE = 'application/javascript';\n", "  const HTML_MIME_TYPE = 'text/html';\n", "  const EXEC_MIME_TYPE = 'application/vnd.bokehjs_exec.v0+json';\n", "  const CLASS_NAME = 'output_bokeh rendered_html';\n", "\n", "  /**\n", "   * Render data to the DOM node\n", "   */\n", "  function render(props, node) {\n", "    const script = document.createElement(\"script\");\n", "    node.appendChild(script);\n", "  }\n", "\n", "  /**\n", "   * Handle when an output is cleared or removed\n", "   */\n", "  function handleClearOutput(event, handle) {\n", "    function drop(id) {\n", "      const view = Bokeh.index.get_by_id(id)\n", "      if (view != null) {\n", "        view.model.document.clear()\n", "        Bokeh.index.delete(view)\n", "      }\n", "    }\n", "\n", "    const cell = handle.cell;\n", "\n", "    const id = cell.output_area._bokeh_element_id;\n", "    const server_id = cell.output_area._bokeh_server_id;\n", "\n", "    // Clean up Bokeh references\n", "    if (id != null) {\n", "      drop(id)\n", "    }\n", "\n", "    if (server_id !== undefined) {\n", "      // Clean up Bokeh references\n", "      const cmd_clean = \"from bokeh.io.state import curstate; print(curstate().uuid_to_server['\" + server_id + \"'].get_sessions()[0].document.roots[0]._id)\";\n", "      cell.notebook.kernel.execute(cmd_clean, {\n", "        iopub: {\n", "          output: function(msg) {\n", "            const id = msg.content.text.trim()\n", "            drop(id)\n", "          }\n", "        }\n", "      });\n", "      // Destroy server and session\n", "      const cmd_destroy = \"import bokeh.io.notebook as ion; ion.destroy_server('\" + server_id + \"')\";\n", "      cell.notebook.kernel.execute(cmd_destroy);\n", "    }\n", "  }\n", "\n", "  /**\n", "   * Handle when a new output is added\n", "   */\n", "  function handleAddOutput(event, handle) {\n", "    const output_area = handle.output_area;\n", "    const output = handle.output;\n", "\n", "    // limit handleAddOutput to display_data with EXEC_MIME_TYPE content only\n", "    if ((output.output_type != \"display_data\") || (!Object.prototype.hasOwnProperty.call(output.data, EXEC_MIME_TYPE))) {\n", "      return\n", "    }\n", "\n", "    const toinsert = output_area.element.find(\".\" + CLASS_NAME.split(' ')[0]);\n", "\n", "    if (output.metadata[EXEC_MIME_TYPE][\"id\"] !== undefined) {\n", "      toinsert[toinsert.length - 1].firstChild.textContent = output.data[JS_MIME_TYPE];\n", "      // store reference to embed id on output_area\n", "      output_area._bokeh_element_id = output.metadata[EXEC_MIME_TYPE][\"id\"];\n", "    }\n", "    if (output.metadata[EXEC_MIME_TYPE][\"server_id\"] !== undefined) {\n", "      const bk_div = document.createElement(\"div\");\n", "      bk_div.innerHTML = output.data[HTML_MIME_TYPE];\n", "      const script_attrs = bk_div.children[0].attributes;\n", "      for (let i = 0; i < script_attrs.length; i++) {\n", "        toinsert[toinsert.length - 1].firstChild.setAttribute(script_attrs[i].name, script_attrs[i].value);\n", "        toinsert[toinsert.length - 1].firstChild.textContent = bk_div.children[0].textContent\n", "      }\n", "      // store reference to server id on output_area\n", "      output_area._bokeh_server_id = output.metadata[EXEC_MIME_TYPE][\"server_id\"];\n", "    }\n", "  }\n", "\n", "  function register_renderer(events, OutputArea) {\n", "\n", "    function append_mime(data, metadata, element) {\n", "      // create a DOM node to render to\n", "      const toinsert = this.create_output_subarea(\n", "        metadata,\n", "        CLASS_NAME,\n", "        EXEC_MIME_TYPE\n", "      );\n", "      this.keyboard_manager.register_events(toinsert);\n", "      // Render to node\n", "      const props = {data: data, metadata: metadata[EXEC_MIME_TYPE]};\n", "      render(props, toinsert[toinsert.length - 1]);\n", "      element.append(toinsert);\n", "      return toinsert\n", "    }\n", "\n", "    /* Handle when an output is cleared or removed */\n", "    events.on('clear_output.CodeCell', handleClearOutput);\n", "    events.on('delete.Cell', handleClearOutput);\n", "\n", "    /* Handle when a new output is added */\n", "    events.on('output_added.OutputArea', handleAddOutput);\n", "\n", "    /**\n", "     * Register the mime type and append_mime function with output_area\n", "     */\n", "    OutputArea.prototype.register_mime_type(EXEC_MIME_TYPE, append_mime, {\n", "      /* Is output safe? */\n", "      safe: true,\n", "      /* Index of renderer in `output_area.display_order` */\n", "      index: 0\n", "    });\n", "  }\n", "\n", "  // register the mime type if in Jupyter Notebook environment and previously unregistered\n", "  if (root.Jupyter !== undefined) {\n", "    const events = require('base/js/events');\n", "    const OutputArea = require('notebook/js/outputarea').OutputArea;\n", "\n", "    if (OutputArea.prototype.mime_types().indexOf(EXEC_MIME_TYPE) == -1) {\n", "      register_renderer(events, OutputArea);\n", "    }\n", "  }\n", "  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n", "    root._bokeh_timeout = Date.now() + 5000;\n", "    root._bokeh_failed_load = false;\n", "  }\n", "\n", "  const NB_LOAD_WARNING = {'data': {'text/html':\n", "     \"<div style='background-color: #fdd'>\\n\"+\n", "     \"<p>\\n\"+\n", "     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n", "     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n", "     \"</p>\\n\"+\n", "     \"<ul>\\n\"+\n", "     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n", "     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n", "     \"</ul>\\n\"+\n", "     \"<code>\\n\"+\n", "     \"from bokeh.resources import INLINE\\n\"+\n", "     \"output_notebook(resources=INLINE)\\n\"+\n", "     \"</code>\\n\"+\n", "     \"</div>\"}};\n", "\n", "  function display_loaded(error = null) {\n", "    const el = document.getElementById(null);\n", "    if (el != null) {\n", "      const html = (() => {\n", "        if (typeof root.Bokeh === \"undefined\") {\n", "          if (error == null) {\n", "            return \"BokehJS is loading ...\";\n", "          } else {\n", "            return \"BokehJS failed to load.\";\n", "          }\n", "        } else {\n", "          const prefix = `BokehJS ${root.Bokeh.version}`;\n", "          if (error == null) {\n", "            return `${prefix} successfully loaded.`;\n", "          } else {\n", "            return `${prefix} <b>encountered errors</b> while loading and may not function as expected.`;\n", "          }\n", "        }\n", "      })();\n", "      el.innerHTML = html;\n", "\n", "      if (error != null) {\n", "        const wrapper = document.createElement(\"div\");\n", "        wrapper.style.overflow = \"auto\";\n", "        wrapper.style.height = \"5em\";\n", "        wrapper.style.resize = \"vertical\";\n", "        const content = document.createElement(\"div\");\n", "        content.style.fontFamily = \"monospace\";\n", "        content.style.whiteSpace = \"pre-wrap\";\n", "        content.style.backgroundColor = \"rgb(255, 221, 221)\";\n", "        content.textContent = error.stack ?? error.toString();\n", "        wrapper.append(content);\n", "        el.append(wrapper);\n", "      }\n", "    } else if (Date.now() < root._bokeh_timeout) {\n", "      setTimeout(() => display_loaded(error), 100);\n", "    }\n", "  }\n", "\n", "  function run_callbacks() {\n", "    try {\n", "      root._bokeh_onload_callbacks.forEach(function(callback) {\n", "        if (callback != null)\n", "          callback();\n", "      });\n", "    } finally {\n", "      delete root._bokeh_onload_callbacks\n", "    }\n", "    console.debug(\"Bokeh: all callbacks have finished\");\n", "  }\n", "\n", "  function load_libs(css_urls, js_urls, callback) {\n", "    if (css_urls == null) css_urls = [];\n", "    if (js_urls == null) js_urls = [];\n", "\n", "    root._bokeh_onload_callbacks.push(callback);\n", "    if (root._bokeh_is_loading > 0) {\n", "      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n", "      return null;\n", "    }\n", "    if (js_urls == null || js_urls.length === 0) {\n", "      run_callbacks();\n", "      return null;\n", "    }\n", "    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n", "    root._bokeh_is_loading = css_urls.length + js_urls.length;\n", "\n", "    function on_load() {\n", "      root._bokeh_is_loading--;\n", "      if (root._bokeh_is_loading === 0) {\n", "        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n", "        run_callbacks()\n", "      }\n", "    }\n", "\n", "    function on_error(url) {\n", "      console.error(\"failed to load \" + url);\n", "    }\n", "\n", "    for (let i = 0; i < css_urls.length; i++) {\n", "      const url = css_urls[i];\n", "      const element = document.createElement(\"link\");\n", "      element.onload = on_load;\n", "      element.onerror = on_error.bind(null, url);\n", "      element.rel = \"stylesheet\";\n", "      element.type = \"text/css\";\n", "      element.href = url;\n", "      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n", "      document.body.appendChild(element);\n", "    }\n", "\n", "    for (let i = 0; i < js_urls.length; i++) {\n", "      const url = js_urls[i];\n", "      const element = document.createElement('script');\n", "      element.onload = on_load;\n", "      element.onerror = on_error.bind(null, url);\n", "      element.async = false;\n", "      element.src = url;\n", "      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n", "      document.head.appendChild(element);\n", "    }\n", "  };\n", "\n", "  function inject_raw_css(css) {\n", "    const element = document.createElement(\"style\");\n", "    element.appendChild(document.createTextNode(css));\n", "    document.body.appendChild(element);\n", "  }\n", "\n", "  const js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-3.7.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-3.7.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-3.7.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-3.7.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-mathjax-3.7.0.min.js\"];\n", "  const css_urls = [];\n", "\n", "  const inline_js = [    function(Bokeh) {\n", "      Bokeh.set_log_level(\"info\");\n", "    },\n", "function(Bokeh) {\n", "    }\n", "  ];\n", "\n", "  function run_inline_js() {\n", "    if (root.Bokeh !== undefined || force === true) {\n", "      try {\n", "            for (let i = 0; i < inline_js.length; i++) {\n", "      inline_js[i].call(root, root.Bokeh);\n", "    }\n", "\n", "      } catch (error) {throw error;\n", "      }} else if (Date.now() < root._bokeh_timeout) {\n", "      setTimeout(run_inline_js, 100);\n", "    } else if (!root._bokeh_failed_load) {\n", "      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n", "      root._bokeh_failed_load = true;\n", "    } else if (force !== true) {\n", "      const cell = $(document.getElementById(null)).parents('.cell').data().cell;\n", "      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n", "    }\n", "  }\n", "\n", "  if (root._bokeh_is_loading === 0) {\n", "    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n", "    run_inline_js();\n", "  } else {\n", "    load_libs(css_urls, js_urls, function() {\n", "      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n", "      run_inline_js();\n", "    });\n", "  }\n", "}(window));"], "application/vnd.bokehjs_load.v0+json": "'use strict';\n(function(root) {\n  function now() {\n    return new Date();\n  }\n\n  const force = true;\n\n  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n    root._bokeh_onload_callbacks = [];\n    root._bokeh_is_loading = undefined;\n  }\n\n\n  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n    root._bokeh_timeout = Date.now() + 5000;\n    root._bokeh_failed_load = false;\n  }\n\n  const NB_LOAD_WARNING = {'data': {'text/html':\n     \"<div style='background-color: #fdd'>\\n\"+\n     \"<p>\\n\"+\n     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n     \"</p>\\n\"+\n     \"<ul>\\n\"+\n     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n     \"</ul>\\n\"+\n     \"<code>\\n\"+\n     \"from bokeh.resources import INLINE\\n\"+\n     \"output_notebook(resources=INLINE)\\n\"+\n     \"</code>\\n\"+\n     \"</div>\"}};\n\n  function display_loaded(error = null) {\n    const el = document.getElementById(null);\n    if (el != null) {\n      const html = (() => {\n        if (typeof root.Bokeh === \"undefined\") {\n          if (error == null) {\n            return \"BokehJS is loading ...\";\n          } else {\n            return \"BokehJS failed to load.\";\n          }\n        } else {\n          const prefix = `BokehJS ${root.Bokeh.version}`;\n          if (error == null) {\n            return `${prefix} successfully loaded.`;\n          } else {\n            return `${prefix} <b>encountered errors</b> while loading and may not function as expected.`;\n          }\n        }\n      })();\n      el.innerHTML = html;\n\n      if (error != null) {\n        const wrapper = document.createElement(\"div\");\n        wrapper.style.overflow = \"auto\";\n        wrapper.style.height = \"5em\";\n        wrapper.style.resize = \"vertical\";\n        const content = document.createElement(\"div\");\n        content.style.fontFamily = \"monospace\";\n        content.style.whiteSpace = \"pre-wrap\";\n        content.style.backgroundColor = \"rgb(255, 221, 221)\";\n        content.textContent = error.stack ?? error.toString();\n        wrapper.append(content);\n        el.append(wrapper);\n      }\n    } else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(() => display_loaded(error), 100);\n    }\n  }\n\n  function run_callbacks() {\n    try {\n      root._bokeh_onload_callbacks.forEach(function(callback) {\n        if (callback != null)\n          callback();\n      });\n    } finally {\n      delete root._bokeh_onload_callbacks\n    }\n    console.debug(\"Bokeh: all callbacks have finished\");\n  }\n\n  function load_libs(css_urls, js_urls, callback) {\n    if (css_urls == null) css_urls = [];\n    if (js_urls == null) js_urls = [];\n\n    root._bokeh_onload_callbacks.push(callback);\n    if (root._bokeh_is_loading > 0) {\n      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n      return null;\n    }\n    if (js_urls == null || js_urls.length === 0) {\n      run_callbacks();\n      return null;\n    }\n    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n    root._bokeh_is_loading = css_urls.length + js_urls.length;\n\n    function on_load() {\n      root._bokeh_is_loading--;\n      if (root._bokeh_is_loading === 0) {\n        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n        run_callbacks()\n      }\n    }\n\n    function on_error(url) {\n      console.error(\"failed to load \" + url);\n    }\n\n    for (let i = 0; i < css_urls.length; i++) {\n      const url = css_urls[i];\n      const element = document.createElement(\"link\");\n      element.onload = on_load;\n      element.onerror = on_error.bind(null, url);\n      element.rel = \"stylesheet\";\n      element.type = \"text/css\";\n      element.href = url;\n      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n      document.body.appendChild(element);\n    }\n\n    for (let i = 0; i < js_urls.length; i++) {\n      const url = js_urls[i];\n      const element = document.createElement('script');\n      element.onload = on_load;\n      element.onerror = on_error.bind(null, url);\n      element.async = false;\n      element.src = url;\n      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n      document.head.appendChild(element);\n    }\n  };\n\n  function inject_raw_css(css) {\n    const element = document.createElement(\"style\");\n    element.appendChild(document.createTextNode(css));\n    document.body.appendChild(element);\n  }\n\n  const js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-3.7.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-3.7.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-3.7.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-3.7.0.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-mathjax-3.7.0.min.js\"];\n  const css_urls = [];\n\n  const inline_js = [    function(Bokeh) {\n      Bokeh.set_log_level(\"info\");\n    },\nfunction(Bokeh) {\n    }\n  ];\n\n  function run_inline_js() {\n    if (root.Bokeh !== undefined || force === true) {\n      try {\n            for (let i = 0; i < inline_js.length; i++) {\n      inline_js[i].call(root, root.Bokeh);\n    }\n\n      } catch (error) {throw error;\n      }} else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(run_inline_js, 100);\n    } else if (!root._bokeh_failed_load) {\n      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n      root._bokeh_failed_load = true;\n    } else if (force !== true) {\n      const cell = $(document.getElementById(null)).parents('.cell').data().cell;\n      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n    }\n  }\n\n  if (root._bokeh_is_loading === 0) {\n    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n    run_inline_js();\n  } else {\n    load_libs(css_urls, js_urls, function() {\n      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n      run_inline_js();\n    });\n  }\n}(window));"}, "metadata": {}, "output_type": "display_data"}], "source": ["df_samples = setup.sample_metadata()\n", "countries = df_samples[\"country\"].unique()"]}, {"cell_type": "markdown", "id": "28582ded-19c5-4a68-9467-dc67de5236a6", "metadata": {}, "source": ["\n", "This report analyses genome variation data from the \n", "{term}`Malaria Vector Genome Observatory`. See Table 1 below for a \n", "complete list of the sample sets used in the current analysis version, \n", "with information about the corresponding contributors, data releases \n", "and citations. These sample sets provide data for a total of\n", "656 mosquitoes sampled from 13 countries. \n"]}, {"cell_type": "code", "execution_count": 6, "id": "35ee5710-6493-4eb7-97cb-5df742053204", "metadata": {"execution": {"iopub.execute_input": "2025-03-27T13:05:28.287238Z", "iopub.status.busy": "2025-03-27T13:05:28.286859Z", "iopub.status.idle": "2025-03-27T13:05:28.566573Z", "shell.execute_reply": "2025-03-27T13:05:28.566052Z"}, "papermill": {"duration": 0.283939, "end_time": "2025-03-27T13:05:28.567391", "exception": false, "start_time": "2025-03-27T13:05:28.283452", "status": "completed"}, "tags": ["remove-input"]}, "outputs": [{"data": {"text/html": ["<style type=\"text/css\">\n", "#T_bab30 th {\n", "  text-align: left;\n", "}\n", "#T_bab30 td {\n", "  text-align: left;\n", "  font-weight: normal;\n", "}\n", "</style>\n", "<table id=\"T_bab30\">\n", "  <caption>Table 1. Data sources included in the current analysis version.</caption>\n", "  <thead>\n", "    <tr>\n", "      <th id=\"T_bab30_level0_col0\" class=\"col_heading level0 col0\" >Sample Set</th>\n", "      <th id=\"T_bab30_level0_col1\" class=\"col_heading level0 col1\" >Study</th>\n", "      <th id=\"T_bab30_level0_col2\" class=\"col_heading level0 col2\" >Contributor</th>\n", "      <th id=\"T_bab30_level0_col3\" class=\"col_heading level0 col3\" >Data Release</th>\n", "      <th id=\"T_bab30_level0_col4\" class=\"col_heading level0 col4\" >Citation</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <td id=\"T_bab30_row0_col0\" class=\"data row0 col0\" >1229-VO-GH-DADZIE-VMF00095</td>\n", "      <td id=\"T_bab30_row0_col1\" class=\"data row0 col1\" ><a href=\"https://www.malariagen.net/partner_study/1229-VO-GH-DADZIE\" target=\"_blank\">1229-VO-GH-DADZIE</a></td>\n", "      <td id=\"T_bab30_row0_col2\" class=\"data row0 col2\" >Samuel Dadzie</td>\n", "      <td id=\"T_bab30_row0_col3\" class=\"data row0 col3\" ><a href=\"https://malariagen.github.io/vector-data/af1/af1.0.html\" target=\"_blank\">Af1.0</a></td>\n", "      <td id=\"T_bab30_row0_col4\" class=\"data row0 col4\" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC11702533/' target='_blank'>Bodd\u00e9, Nwezeobi et al. 2024</a></td>\n", "    </tr>\n", "    <tr>\n", "      <td id=\"T_bab30_row1_col0\" class=\"data row1 col0\" >1230-VO-GA-CF-AYALA-VMF00045</td>\n", "      <td id=\"T_bab30_row1_col1\" class=\"data row1 col1\" ><a href=\"https://www.malariagen.net/partner_study/1230-VO-MULTI-AYALA\" target=\"_blank\">1230-VO-MULTI-AYALA</a></td>\n", "      <td id=\"T_bab30_row1_col2\" class=\"data row1 col2\" >Diego Ayala</td>\n", "      <td id=\"T_bab30_row1_col3\" class=\"data row1 col3\" ><a href=\"https://malariagen.github.io/vector-data/af1/af1.0.html\" target=\"_blank\">Af1.0</a></td>\n", "      <td id=\"T_bab30_row1_col4\" class=\"data row1 col4\" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC11702533/' target='_blank'>Bodd\u00e9, Nwezeobi et al. 2024</a></td>\n", "    </tr>\n", "    <tr>\n", "      <td id=\"T_bab30_row2_col0\" class=\"data row2 col0\" >1231-VO-MULTI-WONDJI-VMF00043</td>\n", "      <td id=\"T_bab30_row2_col1\" class=\"data row2 col1\" ><a href=\"https://www.malariagen.net/partner_study/1231-VO-MULTI-WONDJI\" target=\"_blank\">1231-VO-MULTI-WONDJI</a></td>\n", "      <td id=\"T_bab30_row2_col2\" class=\"data row2 col2\" >Charles Wondji</td>\n", "      <td id=\"T_bab30_row2_col3\" class=\"data row2 col3\" ><a href=\"https://malariagen.github.io/vector-data/af1/af1.0.html\" target=\"_blank\">Af1.0</a></td>\n", "      <td id=\"T_bab30_row2_col4\" class=\"data row2 col4\" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC11702533/' target='_blank'>Bodd\u00e9, Nwezeobi et al. 2024</a></td>\n", "    </tr>\n", "    <tr>\n", "      <td id=\"T_bab30_row3_col0\" class=\"data row3 col0\" >1232-VO-KE-OCHOMO-VMF00044</td>\n", "      <td id=\"T_bab30_row3_col1\" class=\"data row3 col1\" ><a href=\"https://www.malariagen.net/partner_study/1232-VO-KE-OCHOMO\" target=\"_blank\">1232-VO-KE-OCHOMO</a></td>\n", "      <td id=\"T_bab30_row3_col2\" class=\"data row3 col2\" >Eric Ochomo</td>\n", "      <td id=\"T_bab30_row3_col3\" class=\"data row3 col3\" ><a href=\"https://malariagen.github.io/vector-data/af1/af1.0.html\" target=\"_blank\">Af1.0</a></td>\n", "      <td id=\"T_bab30_row3_col4\" class=\"data row3 col4\" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC11702533/' target='_blank'>Bodd\u00e9, Nwezeobi et al. 2024</a></td>\n", "    </tr>\n", "    <tr>\n", "      <td id=\"T_bab30_row4_col0\" class=\"data row4 col0\" >1235-VO-MZ-PAAIJMANS-VMF00094</td>\n", "      <td id=\"T_bab30_row4_col1\" class=\"data row4 col1\" ><a href=\"https://www.malariagen.net/partner_study/1235-VO-MZ-PAAIJMANS\" target=\"_blank\">1235-VO-MZ-PAAIJMANS</a></td>\n", "      <td id=\"T_bab30_row4_col2\" class=\"data row4 col2\" >Krijn Paaijmans</td>\n", "      <td id=\"T_bab30_row4_col3\" class=\"data row4 col3\" ><a href=\"https://malariagen.github.io/vector-data/af1/af1.0.html\" target=\"_blank\">Af1.0</a></td>\n", "      <td id=\"T_bab30_row4_col4\" class=\"data row4 col4\" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC11702533/' target='_blank'>Bodd\u00e9, Nwezeobi et al. 2024</a></td>\n", "    </tr>\n", "    <tr>\n", "      <td id=\"T_bab30_row5_col0\" class=\"data row5 col0\" >1236-VO-TZ-OKUMU-VMF00090</td>\n", "      <td id=\"T_bab30_row5_col1\" class=\"data row5 col1\" ><a href=\"https://www.malariagen.net/partner_study/1236-VO-TZ-OKUMU\" target=\"_blank\">1236-VO-TZ-OKUMU</a></td>\n", "      <td id=\"T_bab30_row5_col2\" class=\"data row5 col2\" >Fredros Okumu</td>\n", "      <td id=\"T_bab30_row5_col3\" class=\"data row5 col3\" ><a href=\"https://malariagen.github.io/vector-data/af1/af1.0.html\" target=\"_blank\">Af1.0</a></td>\n", "      <td id=\"T_bab30_row5_col4\" class=\"data row5 col4\" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC11702533/' target='_blank'>Bodd\u00e9, Nwezeobi et al. 2024</a></td>\n", "    </tr>\n", "    <tr>\n", "      <td id=\"T_bab30_row6_col0\" class=\"data row6 col0\" >1240-VO-CD-KOEKEMOER-VMF00099</td>\n", "      <td id=\"T_bab30_row6_col1\" class=\"data row6 col1\" ><a href=\"https://www.malariagen.net/partner_study/1240-VO-MULTI-KOEKEMOER\" target=\"_blank\">1240-VO-MULTI-KOEKEMOER</a></td>\n", "      <td id=\"T_bab30_row6_col2\" class=\"data row6 col2\" >Lizette Koekemoer</td>\n", "      <td id=\"T_bab30_row6_col3\" class=\"data row6 col3\" ><a href=\"https://malariagen.github.io/vector-data/af1/af1.0.html\" target=\"_blank\">Af1.0</a></td>\n", "      <td id=\"T_bab30_row6_col4\" class=\"data row6 col4\" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC11702533/' target='_blank'>Bodd\u00e9, Nwezeobi et al. 2024</a></td>\n", "    </tr>\n", "    <tr>\n", "      <td id=\"T_bab30_row7_col0\" class=\"data row7 col0\" >1240-VO-MZ-KOEKEMOER-VMF00101</td>\n", "      <td id=\"T_bab30_row7_col1\" class=\"data row7 col1\" ><a href=\"https://www.malariagen.net/partner_study/1240-VO-MULTI-KOEKEMOER\" target=\"_blank\">1240-VO-MULTI-KOEKEMOER</a></td>\n", "      <td id=\"T_bab30_row7_col2\" class=\"data row7 col2\" >Lizette Koekemoer</td>\n", "      <td id=\"T_bab30_row7_col3\" class=\"data row7 col3\" ><a href=\"https://malariagen.github.io/vector-data/af1/af1.0.html\" target=\"_blank\">Af1.0</a></td>\n", "      <td id=\"T_bab30_row7_col4\" class=\"data row7 col4\" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC11702533/' target='_blank'>Bodd\u00e9, Nwezeobi et al. 2024</a></td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n"], "text/plain": ["<pandas.io.formats.style.Styler at 0x7f55708ec200>"]}, "execution_count": 6, "metadata": {}, "output_type": "execute_result"}], "source": ["page_utils.style_data_sources(\n", "    df_samples=df_samples,\n", "    caption=\"Table 1. Data sources included in the current analysis version.\",\n", ")"]}, {"cell_type": "markdown", "id": "5eb43cf7-20ed-4b71-bae5-b4874071e3d4", "metadata": {}, "source": ["\n", "Sample metadata, unphased SNP calls, and phased SNP haplotypes were retrieved from \n", "the {term}`Malaria Vector Genome Observatory` cloud data repository hosted in \n", "Google Cloud Storage (GCS) via the {term}`MalariaGEN Python API` version \n", "15.0.1.\n"]}, {"cell_type": "markdown", "id": "32b058d5-be5e-4c11-a74f-90081d26975c", "metadata": {"papermill": {"duration": 0.003211, "end_time": "2025-03-27T13:05:28.586078", "exception": false, "start_time": "2025-03-27T13:05:28.582867", "status": "completed"}, "tags": ["remove-input"]}, "source": ["## Sample inclusion and grouping into cohorts\n", "\n", "Samples were considered for inclusion if they met the following criteria:"]}, {"cell_type": "code", "execution_count": 8, "id": "474d0b6f-2d84-4191-abde-719f9462fef3", "metadata": {"execution": {"iopub.execute_input": "2025-03-27T13:05:28.593123Z", "iopub.status.busy": "2025-03-27T13:05:28.592916Z", "iopub.status.idle": "2025-03-27T13:05:28.596347Z", "shell.execute_reply": "2025-03-27T13:05:28.595779Z"}, "papermill": {"duration": 0.007844, "end_time": "2025-03-27T13:05:28.597116", "exception": false, "start_time": "2025-03-27T13:05:28.589272", "status": "completed"}, "tags": ["remove-input"]}, "outputs": [], "source": ["def human_readable_list(x):\n", "    x = [str(v) for v in x]\n", "    if len(x) > 1:\n", "        output = \", \".join(x[:-1]) + \" or \" + x[-1]\n", "    else:\n", "        output = x[0]\n", "    return output\n", "\n", "\n", "readable_taxa = human_readable_list(setup.taxa)"]}, {"cell_type": "markdown", "id": "dd27a48c-782d-445e-97a4-7340bfc0f5dd", "metadata": {}, "source": ["\n", "* Gender assigned as female via comparison of sequence coverage on autosomes and sex \n", "  chromosomes. \n", "* Taxon assigned as funestus via principle components analysis of \n", "  genomic data from Chromosome 3 and comparison with reference samples \n", "  with known taxon assignments.\n"]}, {"cell_type": "code", "execution_count": 10, "id": "fc0de9ab-618d-49b1-a12c-9e567fc84f05", "metadata": {"execution": {"iopub.execute_input": "2025-03-27T13:05:28.615902Z", "iopub.status.busy": "2025-03-27T13:05:28.615620Z", "iopub.status.idle": "2025-03-27T13:05:28.648529Z", "shell.execute_reply": "2025-03-27T13:05:28.648043Z"}, "papermill": {"duration": 0.037425, "end_time": "2025-03-27T13:05:28.649305", "exception": false, "start_time": "2025-03-27T13:05:28.611880", "status": "completed"}, "tags": ["remove-input"]}, "outputs": [], "source": ["gdf_cohorts = page_utils.gdf_cohorts"]}, {"cell_type": "markdown", "id": "9c79abe0-46e4-4228-a9ea-5472e2fedfc4", "metadata": {"papermill": {"duration": 0.003392, "end_time": "2025-03-27T13:05:28.656137", "exception": false, "start_time": "2025-03-27T13:05:28.652745", "status": "completed"}, "tags": []}, "source": ["After filtering according to these inclusion criteria, samples were grouped into cohorts by taxon, location of sampling and date of sampling. Samples were grouped spatially if their collection locations were within the same level 2 administrative unit, according to geoBoundaries version 5.0.0. Samples were grouped temporally if their collection dates were within the same quarter (3 month period) where possible, except in a small number of cases where metadata were only available on year of collection."]}, {"cell_type": "markdown", "id": "58aeb693-8d2a-4322-96c4-f71d6d22c85c", "metadata": {}, "source": ["\n", "Cohorts were excluded from the analysis if the sample size was \n", "less than 15. Cohorts with more than \n", "100 samples were randomly downsampled for\n", "computational efficiency. Cohorts were also excluded from the \n", "analysis if they failed H12 or G123 window size calibration\n", "(see below). After applying these filters, a total of \n", "15 cohorts were retained for analysis (Table 2).\n"]}, {"cell_type": "code", "execution_count": 12, "id": "25d5ff7c-1524-47f1-9a78-43a3c73e9956", "metadata": {"execution": {"iopub.execute_input": "2025-03-27T13:05:28.676258Z", "iopub.status.busy": "2025-03-27T13:05:28.675927Z", "iopub.status.idle": "2025-03-27T13:05:28.685232Z", "shell.execute_reply": "2025-03-27T13:05:28.684686Z"}, "papermill": {"duration": 0.01392, "end_time": "2025-03-27T13:05:28.685923", "exception": false, "start_time": "2025-03-27T13:05:28.672003", "status": "completed"}, "tags": ["remove-input"]}, "outputs": [{"data": {"text/html": ["<style type=\"text/css\">\n", "</style>\n", "<table id=\"T_77d53\">\n", "  <caption>Table 2. Cohorts selected for genome-wide selection scan analyses.</caption>\n", "  <thead>\n", "    <tr>\n", "      <th id=\"T_77d53_level0_col0\" class=\"col_heading level0 col0\" >Cohort</th>\n", "      <th id=\"T_77d53_level0_col1\" class=\"col_heading level0 col1\" >Country</th>\n", "      <th id=\"T_77d53_level0_col2\" class=\"col_heading level0 col2\" >Region</th>\n", "      <th id=\"T_77d53_level0_col3\" class=\"col_heading level0 col3\" >District</th>\n", "      <th id=\"T_77d53_level0_col4\" class=\"col_heading level0 col4\" >Taxon</th>\n", "      <th id=\"T_77d53_level0_col5\" class=\"col_heading level0 col5\" >Year</th>\n", "      <th id=\"T_77d53_level0_col6\" class=\"col_heading level0 col6\" >Quarter</th>\n", "      <th id=\"T_77d53_level0_col7\" class=\"col_heading level0 col7\" >Sample Size</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <td id=\"T_77d53_row0_col0\" class=\"data row0 col0\" ><a href=\"cohort/BJ-AQ_Ouidah_fune_2014_Q2.html\">Benin / Ouidah / funestus / 2014 / Q2</a></td>\n", "      <td id=\"T_77d53_row0_col1\" class=\"data row0 col1\" >Benin</td>\n", "      <td id=\"T_77d53_row0_col2\" class=\"data row0 col2\" >Atlanique</td>\n", "      <td id=\"T_77d53_row0_col3\" class=\"data row0 col3\" >Ouidah</td>\n", "      <td id=\"T_77d53_row0_col4\" class=\"data row0 col4\" >funestus</td>\n", "      <td id=\"T_77d53_row0_col5\" class=\"data row0 col5\" >2014</td>\n", "      <td id=\"T_77d53_row0_col6\" class=\"data row0 col6\" >2</td>\n", "      <td id=\"T_77d53_row0_col7\" class=\"data row0 col7\" >37</td>\n", "    </tr>\n", "    <tr>\n", "      <td id=\"T_77d53_row1_col0\" class=\"data row1 col0\" ><a href=\"cohort/CD-HU_Watsa_fune_2017_Q4.html\">Democratic Republic of the Congo / Watsa / funestus / 2017 / Q4</a></td>\n", "      <td id=\"T_77d53_row1_col1\" class=\"data row1 col1\" >Democratic Republic of the Congo</td>\n", "      <td id=\"T_77d53_row1_col2\" class=\"data row1 col2\" >Upper Uele</td>\n", "      <td id=\"T_77d53_row1_col3\" class=\"data row1 col3\" >Watsa</td>\n", "      <td id=\"T_77d53_row1_col4\" class=\"data row1 col4\" >funestus</td>\n", "      <td id=\"T_77d53_row1_col5\" class=\"data row1 col5\" >2017</td>\n", "      <td id=\"T_77d53_row1_col6\" class=\"data row1 col6\" >4</td>\n", "      <td id=\"T_77d53_row1_col7\" class=\"data row1 col7\" >43</td>\n", "    </tr>\n", "    <tr>\n", "      <td id=\"T_77d53_row2_col0\" class=\"data row2 col0\" ><a href=\"cohort/CD-KN_Kinshasa_fune_2015_Q2.html\">Democratic Republic of the Congo / Kinshasa / funestus / 2015 / Q2</a></td>\n", "      <td id=\"T_77d53_row2_col1\" class=\"data row2 col1\" >Democratic Republic of the Congo</td>\n", "      <td id=\"T_77d53_row2_col2\" class=\"data row2 col2\" >Kinshasa</td>\n", "      <td id=\"T_77d53_row2_col3\" class=\"data row2 col3\" >Kinshasa</td>\n", "      <td id=\"T_77d53_row2_col4\" class=\"data row2 col4\" >funestus</td>\n", "      <td id=\"T_77d53_row2_col5\" class=\"data row2 col5\" >2015</td>\n", "      <td id=\"T_77d53_row2_col6\" class=\"data row2 col6\" >2</td>\n", "      <td id=\"T_77d53_row2_col7\" class=\"data row2 col7\" >34</td>\n", "    </tr>\n", "    <tr>\n", "      <td id=\"T_77d53_row3_col0\" class=\"data row3 col0\" ><a href=\"cohort/CM-AD_Mayo-Banyo_fune_2014_Q3.html\">Cameroon / Mayo-Banyo / funestus / 2014 / Q3</a></td>\n", "      <td id=\"T_77d53_row3_col1\" class=\"data row3 col1\" >Cameroon</td>\n", "      <td id=\"T_77d53_row3_col2\" class=\"data row3 col2\" >Adamaoua</td>\n", "      <td id=\"T_77d53_row3_col3\" class=\"data row3 col3\" >Mayo-Banyo</td>\n", "      <td id=\"T_77d53_row3_col4\" class=\"data row3 col4\" >funestus</td>\n", "      <td id=\"T_77d53_row3_col5\" class=\"data row3 col5\" >2014</td>\n", "      <td id=\"T_77d53_row3_col6\" class=\"data row3 col6\" >3</td>\n", "      <td id=\"T_77d53_row3_col7\" class=\"data row3 col7\" >45</td>\n", "    </tr>\n", "    <tr>\n", "      <td id=\"T_77d53_row4_col0\" class=\"data row4 col0\" ><a href=\"cohort/GA-2_Mpassa_fune_2017_Q4.html\">Gabon / Mpassa / funestus / 2017 / Q4</a></td>\n", "      <td id=\"T_77d53_row4_col1\" class=\"data row4 col1\" >Gabon</td>\n", "      <td id=\"T_77d53_row4_col2\" class=\"data row4 col2\" >Haut-Ogoou\u00e9</td>\n", "      <td id=\"T_77d53_row4_col3\" class=\"data row4 col3\" >Mpassa</td>\n", "      <td id=\"T_77d53_row4_col4\" class=\"data row4 col4\" >funestus</td>\n", "      <td id=\"T_77d53_row4_col5\" class=\"data row4 col5\" >2017</td>\n", "      <td id=\"T_77d53_row4_col6\" class=\"data row4 col6\" >4</td>\n", "      <td id=\"T_77d53_row4_col7\" class=\"data row4 col7\" >26</td>\n", "    </tr>\n", "    <tr>\n", "      <td id=\"T_77d53_row5_col0\" class=\"data row5 col0\" ><a href=\"cohort/GH-AH_Adansi-Akrofuom_fune_2014_Q1.html\">Ghana / Adansi Akrofuom / funestus / 2014 / Q1</a></td>\n", "      <td id=\"T_77d53_row5_col1\" class=\"data row5 col1\" >Ghana</td>\n", "      <td id=\"T_77d53_row5_col2\" class=\"data row5 col2\" >Ashanti Region</td>\n", "      <td id=\"T_77d53_row5_col3\" class=\"data row5 col3\" >Adansi Akrofuom</td>\n", "      <td id=\"T_77d53_row5_col4\" class=\"data row5 col4\" >funestus</td>\n", "      <td id=\"T_77d53_row5_col5\" class=\"data row5 col5\" >2014</td>\n", "      <td id=\"T_77d53_row5_col6\" class=\"data row5 col6\" >1</td>\n", "      <td id=\"T_77d53_row5_col7\" class=\"data row5 col7\" >31</td>\n", "    </tr>\n", "    <tr>\n", "      <td id=\"T_77d53_row6_col0\" class=\"data row6 col0\" ><a href=\"cohort/KE-03_Mt.-Elgon_fune_2016_Q4.html\">Kenya / Mt. Elgon / funestus / 2016 / Q4</a></td>\n", "      <td id=\"T_77d53_row6_col1\" class=\"data row6 col1\" >Kenya</td>\n", "      <td id=\"T_77d53_row6_col2\" class=\"data row6 col2\" >Bungoma</td>\n", "      <td id=\"T_77d53_row6_col3\" class=\"data row6 col3\" >Mt. Elgon</td>\n", "      <td id=\"T_77d53_row6_col4\" class=\"data row6 col4\" >funestus</td>\n", "      <td id=\"T_77d53_row6_col5\" class=\"data row6 col5\" >2016</td>\n", "      <td id=\"T_77d53_row6_col6\" class=\"data row6 col6\" >4</td>\n", "      <td id=\"T_77d53_row6_col7\" class=\"data row6 col7\" >21</td>\n", "    </tr>\n", "    <tr>\n", "      <td id=\"T_77d53_row7_col0\" class=\"data row7 col0\" ><a href=\"cohort/KE-17_Nyando_fune_2014_Q2.html\">Kenya / Nyando / funestus / 2014 / Q2</a></td>\n", "      <td id=\"T_77d53_row7_col1\" class=\"data row7 col1\" >Kenya</td>\n", "      <td id=\"T_77d53_row7_col2\" class=\"data row7 col2\" >Kisumu</td>\n", "      <td id=\"T_77d53_row7_col3\" class=\"data row7 col3\" >Nyando</td>\n", "      <td id=\"T_77d53_row7_col4\" class=\"data row7 col4\" >funestus</td>\n", "      <td id=\"T_77d53_row7_col5\" class=\"data row7 col5\" >2014</td>\n", "      <td id=\"T_77d53_row7_col6\" class=\"data row7 col6\" >2</td>\n", "      <td id=\"T_77d53_row7_col7\" class=\"data row7 col7\" >35</td>\n", "    </tr>\n", "    <tr>\n", "      <td id=\"T_77d53_row8_col0\" class=\"data row8 col0\" ><a href=\"cohort/MW-S_Chikwawa_fune_2014_Q1.html\">Malawi / Chikwawa / funestus / 2014 / Q1</a></td>\n", "      <td id=\"T_77d53_row8_col1\" class=\"data row8 col1\" >Malawi</td>\n", "      <td id=\"T_77d53_row8_col2\" class=\"data row8 col2\" >Southern Region</td>\n", "      <td id=\"T_77d53_row8_col3\" class=\"data row8 col3\" >Chikwawa</td>\n", "      <td id=\"T_77d53_row8_col4\" class=\"data row8 col4\" >funestus</td>\n", "      <td id=\"T_77d53_row8_col5\" class=\"data row8 col5\" >2014</td>\n", "      <td id=\"T_77d53_row8_col6\" class=\"data row8 col6\" >1</td>\n", "      <td id=\"T_77d53_row8_col7\" class=\"data row8 col7\" >18</td>\n", "    </tr>\n", "    <tr>\n", "      <td id=\"T_77d53_row9_col0\" class=\"data row9 col0\" ><a href=\"cohort/MZ-L_Manhi\u00e7a_fune_2016_Q2.html\">Mozambique / Manhi\u00e7a / funestus / 2016 / Q2</a></td>\n", "      <td id=\"T_77d53_row9_col1\" class=\"data row9 col1\" >Mozambique</td>\n", "      <td id=\"T_77d53_row9_col2\" class=\"data row9 col2\" >Maputo</td>\n", "      <td id=\"T_77d53_row9_col3\" class=\"data row9 col3\" >Manhi\u00e7a</td>\n", "      <td id=\"T_77d53_row9_col4\" class=\"data row9 col4\" >funestus</td>\n", "      <td id=\"T_77d53_row9_col5\" class=\"data row9 col5\" >2016</td>\n", "      <td id=\"T_77d53_row9_col6\" class=\"data row9 col6\" >2</td>\n", "      <td id=\"T_77d53_row9_col7\" class=\"data row9 col7\" >22</td>\n", "    </tr>\n", "    <tr>\n", "      <td id=\"T_77d53_row10_col0\" class=\"data row10 col0\" ><a href=\"cohort/MZ-L_Manhi\u00e7a_fune_2018_Q1.html\">Mozambique / Manhi\u00e7a / funestus / 2018 / Q1</a></td>\n", "      <td id=\"T_77d53_row10_col1\" class=\"data row10 col1\" >Mozambique</td>\n", "      <td id=\"T_77d53_row10_col2\" class=\"data row10 col2\" >Maputo</td>\n", "      <td id=\"T_77d53_row10_col3\" class=\"data row10 col3\" >Manhi\u00e7a</td>\n", "      <td id=\"T_77d53_row10_col4\" class=\"data row10 col4\" >funestus</td>\n", "      <td id=\"T_77d53_row10_col5\" class=\"data row10 col5\" >2018</td>\n", "      <td id=\"T_77d53_row10_col6\" class=\"data row10 col6\" >1</td>\n", "      <td id=\"T_77d53_row10_col7\" class=\"data row10 col7\" >58</td>\n", "    </tr>\n", "    <tr>\n", "      <td id=\"T_77d53_row11_col0\" class=\"data row11 col0\" ><a href=\"cohort/MZ-P_Palma_fune_2015_Q3.html\">Mozambique / Palma / funestus / 2015 / Q3</a></td>\n", "      <td id=\"T_77d53_row11_col1\" class=\"data row11 col1\" >Mozambique</td>\n", "      <td id=\"T_77d53_row11_col2\" class=\"data row11 col2\" >Cabo Delgado</td>\n", "      <td id=\"T_77d53_row11_col3\" class=\"data row11 col3\" >Palma</td>\n", "      <td id=\"T_77d53_row11_col4\" class=\"data row11 col4\" >funestus</td>\n", "      <td id=\"T_77d53_row11_col5\" class=\"data row11 col5\" >2015</td>\n", "      <td id=\"T_77d53_row11_col6\" class=\"data row11 col6\" >3</td>\n", "      <td id=\"T_77d53_row11_col7\" class=\"data row11 col7\" >40</td>\n", "    </tr>\n", "    <tr>\n", "      <td id=\"T_77d53_row12_col0\" class=\"data row12 col0\" ><a href=\"cohort/NG-OG_Remo-North_fune_2015_Q1.html\">Nigeria / Remo North / funestus / 2015 / Q1</a></td>\n", "      <td id=\"T_77d53_row12_col1\" class=\"data row12 col1\" >Nigeria</td>\n", "      <td id=\"T_77d53_row12_col2\" class=\"data row12 col2\" >Ogun</td>\n", "      <td id=\"T_77d53_row12_col3\" class=\"data row12 col3\" >Remo North</td>\n", "      <td id=\"T_77d53_row12_col4\" class=\"data row12 col4\" >funestus</td>\n", "      <td id=\"T_77d53_row12_col5\" class=\"data row12 col5\" >2015</td>\n", "      <td id=\"T_77d53_row12_col6\" class=\"data row12 col6\" >1</td>\n", "      <td id=\"T_77d53_row12_col7\" class=\"data row12 col7\" >41</td>\n", "    </tr>\n", "    <tr>\n", "      <td id=\"T_77d53_row13_col0\" class=\"data row13 col0\" ><a href=\"cohort/UG-E_Tororo_fune_2014_Q2.html\">Uganda / Tororo / funestus / 2014 / Q2</a></td>\n", "      <td id=\"T_77d53_row13_col1\" class=\"data row13 col1\" >Uganda</td>\n", "      <td id=\"T_77d53_row13_col2\" class=\"data row13 col2\" >Eastern Region</td>\n", "      <td id=\"T_77d53_row13_col3\" class=\"data row13 col3\" >Tororo</td>\n", "      <td id=\"T_77d53_row13_col4\" class=\"data row13 col4\" >funestus</td>\n", "      <td id=\"T_77d53_row13_col5\" class=\"data row13 col5\" >2014</td>\n", "      <td id=\"T_77d53_row13_col6\" class=\"data row13 col6\" >2</td>\n", "      <td id=\"T_77d53_row13_col7\" class=\"data row13 col7\" >49</td>\n", "    </tr>\n", "    <tr>\n", "      <td id=\"T_77d53_row14_col0\" class=\"data row14 col0\" ><a href=\"cohort/ZM-03_Sinda_fune_2016_Q3.html\">Zambia / Sinda / funestus / 2016 / Q3</a></td>\n", "      <td id=\"T_77d53_row14_col1\" class=\"data row14 col1\" >Zambia</td>\n", "      <td id=\"T_77d53_row14_col2\" class=\"data row14 col2\" >Eastern</td>\n", "      <td id=\"T_77d53_row14_col3\" class=\"data row14 col3\" >Sinda</td>\n", "      <td id=\"T_77d53_row14_col4\" class=\"data row14 col4\" >funestus</td>\n", "      <td id=\"T_77d53_row14_col5\" class=\"data row14 col5\" >2016</td>\n", "      <td id=\"T_77d53_row14_col6\" class=\"data row14 col6\" >3</td>\n", "      <td id=\"T_77d53_row14_col7\" class=\"data row14 col7\" >43</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n"], "text/plain": ["<pandas.io.formats.style.Styler at 0x7f551d265400>"]}, "execution_count": 12, "metadata": {}, "output_type": "execute_result"}], "source": ["page_utils.style_cohorts_table(\n", "    gdf_cohorts,\n", "    caption=\"Table 2. Cohorts selected for genome-wide selection scan analyses.\",\n", "    url_prefix=\"\",\n", ")"]}, {"cell_type": "markdown", "id": "908df9c7-5b95-44fd-a644-072a210e2200", "metadata": {"papermill": {"duration": 0.003584, "end_time": "2025-03-27T13:05:28.693226", "exception": false, "start_time": "2025-03-27T13:05:28.689642", "status": "completed"}, "tags": []}, "source": ["## H12 and G123 window size calibration\n", "\n", "Both H12 ([Garud et al. 2015](https://pmc.ncbi.nlm.nih.gov/articles/PMC4338236/)) and G123 ([Harris et al. 2018](https://pmc.ncbi.nlm.nih.gov/articles/PMC6283157/)) are statistical methods for performing {term}`genome-wide selection scan`s which rely on dividing data into windows along the {term}`genome`. Typically the size of these windows is set to a fixed number of polymorphic sites ({term}`SNP`s). I.e., all windows contain data \u2500 either phased haplotypes or unphased genotypes \u2500 for the same number of {term}`SNP`s. In order to detect recent selective sweeps, the size of these windows needs to be chosen so that windows are generally larger than the normal genetic distance over which linkage disequlibrium (LD) decays to background levels in the absence of recent positive selection. Therefore, in windows which are unaffected by recent selective sweeps, genetic diversity will be high and thus the values of the selection statistics will be low. Conversely, in windows affected by recent selective sweeps, linkage disequilibrium will extend over a longer genetic distance spanning multiple windows, so that genetic diversity within those windows is low and thus values of selection statistics will be high. In other words, the choice of window size affects the signal to noise ratio for selection scans using H12 and G123 statistics. If windows are too small, results are dominated by background noise. If windows are too large, noise is minimal but power to detect recent selection signals is reduced.\n", "\n", "This decision regarding an appropriate window size needs to be made independently for each cohort of samples over which a selection scan will be performed. This is because different source populations may have different demographic histories, and this in turn may alter the genetic distance over which LD decays in the absence of positive selection. Previous studies have used various demographic inference methods to try to infer key demographic parameters for each cohort being analysed, then use these parameters to inform the decision of window size. In practice, this approach presents a number of challenges. Firstly, inference of demographic parameters is difficult, and even state of the art inference methods may reach inaccurate conclusions. Secondly, running demographic inference methods can be computationally demanding, and this becomes impractical for large numbers of cohorts."]}, {"cell_type": "markdown", "id": "5c8fab7e-af63-4345-bb23-87ebb5c164c3", "metadata": {}, "source": ["\n", "For these reasons we have taken an empirical approach to window size calibration for H12 \n", "and G123 scans, designed to reach a good signal to noise ratio. \n", "\n", "For each cohort, we compute H12 over contig 3RL for multiple window \n", "sizes of 100, 200, 300, 400, 500, 600, 700, 800, 900, 1000, 1200, 1400, 1600, 1800, 2000, 2500, 3000, 3500, 4000, 4500, 5000, 6000, 7000, 8000, 9000 or 10000 SNPs. We then compute the 95th \n", "percentile of statistic values over all windows. We choose the smallest window size for which the \n", "95th percentile is below 0.08. This means that any window with a\n", "statistic value above this threshold will be in the top 5% of windows.\n", "\n", "Similarly, we compute G123 over contig 3RL for multiple window \n", "sizes of 100, 200, 300, 400, 500, 600, 700, 800, 900, 1000, 1200, 1400, 1600, 1800, 2000, 2500, 3000, 3500, 4000, 4500, 5000, 6000, 7000, 8000, 9000 or 10000 SNPs. We then compute the 95th \n", "percentile of statistic values over all windows. We choose the smallest window size for which the \n", "95th percentile is below 0.08.\n", "\n"]}, {"cell_type": "markdown", "id": "fef5d695-012d-48d4-a313-755e8c726f13", "metadata": {"papermill": {"duration": 0.003722, "end_time": "2025-03-27T13:05:28.713404", "exception": false, "start_time": "2025-03-27T13:05:28.709682", "status": "completed"}, "tags": []}, "source": ["TODO how was window-size calibration done?\n", "\n", "TODO after calibration, some cohorts removed if cannot get a window-size."]}, {"cell_type": "markdown", "id": "3549b2f6-a7c8-4625-9260-78846ed9ca2e", "metadata": {"papermill": {"duration": 0.003806, "end_time": "2025-03-27T13:05:28.720908", "exception": false, "start_time": "2025-03-27T13:05:28.717102", "status": "completed"}, "tags": []}, "source": ["## H12 genome-wide selection scans\n", "\n", "TODO"]}, {"cell_type": "markdown", "id": "aa16dbb2-2ed3-4f16-ad74-9b8f9821648e", "metadata": {"papermill": {"duration": 0.003702, "end_time": "2025-03-27T13:05:28.728345", "exception": false, "start_time": "2025-03-27T13:05:28.724643", "status": "completed"}, "tags": []}, "source": ["## G123 genome-wide selection scans\n", "\n", "TODO"]}, {"cell_type": "markdown", "id": "777d6d11-dd99-44c6-9a52-c68142122065", "metadata": {"papermill": {"duration": 0.003738, "end_time": "2025-03-27T13:05:28.735807", "exception": false, "start_time": "2025-03-27T13:05:28.732069", "status": "completed"}, "tags": []}, "source": ["## IHS genome-wide selection scans\n", "\n", "TODO"]}, {"cell_type": "markdown", "id": "afdda01b-9155-4981-8206-df22739fff33", "metadata": {"papermill": {"duration": 0.003687, "end_time": "2025-03-27T13:05:28.743212", "exception": false, "start_time": "2025-03-27T13:05:28.739525", "status": "completed"}, "tags": []}, "source": ["## Automated detection of selection signals\n", "\n", "TODO"]}, {"cell_type": "markdown", "id": "417e3d70-368e-49b8-a237-12c2a6f5ed06", "metadata": {"papermill": {"duration": 0.003734, "end_time": "2025-03-27T13:05:28.750657", "exception": false, "start_time": "2025-03-27T13:05:28.746923", "status": "completed"}, "tags": []}, "source": ["## Identification of selection alerts\n", "\n", "TODO"]}, {"cell_type": "markdown", "id": "66294008-9a26-4d51-9303-d534c108a4db", "metadata": {"papermill": {"duration": 0.003628, "end_time": "2025-03-27T13:05:28.757990", "exception": false, "start_time": "2025-03-27T13:05:28.754362", "status": "completed"}, "tags": []}, "source": ["## Web report generation\n", "\n", "TODO"]}, {"cell_type": "code", "execution_count": null, "id": "1a09b58a-0129-4174-af02-290fb5e6084e", "metadata": {"papermill": {"duration": 0.003747, "end_time": "2025-03-27T13:05:28.765424", "exception": false, "start_time": "2025-03-27T13:05:28.761677", "status": "completed"}, "tags": []}, "outputs": [], "source": []}], "metadata": {"environment": {"kernel": "selection-atlas", "name": "workbench-notebooks.m128", "type": "gcloud", "uri": "us-docker.pkg.dev/deeplearning-platform-release/gcr.io/workbench-notebooks:m128"}, "kernelspec": {"display_name": "selection-atlas", "language": "python", "name": "selection-atlas"}, "language_info": {"codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "mimetype": "text/x-python", "name": "python", "nbconvert_exporter": "python", "pygments_lexer": "ipython3", "version": "3.12.9"}, "papermill": {"default_parameters": {}, "duration": 7.340944, "end_time": "2025-03-27T13:05:29.384559", "environment_variables": {}, "exception": null, "input_path": "/home/runner/work/selection-atlas/selection-atlas/workflow/site/notebooks/methods-page.ipynb", "output_path": "/home/runner/work/selection-atlas/selection-atlas/results/afun/2025.03.05/site/notebooks/methods.ipynb", "parameters": {"config_file": "/home/runner/work/selection-atlas/selection-atlas/config/afun.yaml"}, "start_time": "2025-03-27T13:05:22.043615", "version": "2.6.0"}}, "nbformat": 4, "nbformat_minor": 5}