
<!DOCTYPE html>


<html lang="en" data-content_root="./" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Data sources &#8212; Malaria Vector Selection Atlas — *Anopheles gambiae* complex</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  <!--
    this give us a css class that will be invisible only if js is disabled
  -->
  <noscript>
    <style>
      .pst-js-only { display: none !important; }

    </style>
  </noscript>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="_static/styles/theme.css?digest=8878045cc6db502f8baf" rel="stylesheet" />
<link href="_static/styles/pydata-sphinx-theme.css?digest=8878045cc6db502f8baf" rel="stylesheet" />

    <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=03e43079" />
    <link rel="stylesheet" type="text/css" href="_static/styles/sphinx-book-theme.css?v=a3416100" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-design.min.css?v=95c83b7e" />
    <link rel="stylesheet" type="text/css" href="_static/custom.css?v=e02c4bef" />
  
  <!-- So that users can add custom icons -->
  <script src="_static/scripts/fontawesome.js?digest=8878045cc6db502f8baf"></script>
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="_static/scripts/bootstrap.js?digest=8878045cc6db502f8baf" />
<link rel="preload" as="script" href="_static/scripts/pydata-sphinx-theme.js?digest=8878045cc6db502f8baf" />

    <script src="_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="_static/doctools.js?v=9a2dae69"></script>
    <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="_static/copybutton.js?v=f281be69"></script>
    <script src="_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="_static/togglebutton.js?v=4a39c7ea"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'data-sources';</script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <link rel="icon" href="_static/favicon.ico"/>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Methods" href="methods.html" />
    <link rel="prev" title="Uganda / Kanungu / gambiae / 2012 / Q4" href="cohort/UG-W_Kanungu_gamb_2012_Q4.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  <meta name="docsearch:version" content="" />
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <dialog id="pst-search-dialog">
    
<form class="bd-search d-flex align-items-center"
      action="search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         placeholder="Search this book..."
         aria-label="Search this book..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form>
  </dialog>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <dialog id="pst-primary-sidebar-modal"></dialog>
      <div id="pst-primary-sidebar" class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="index.html">
  
  
  
  
  
    
    
      
    
    
    <img src="_static/logo-agam.png" class="logo__image only-light" alt="Malaria Vector Selection Atlas — *Anopheles gambiae* complex - Home"/>
    <img src="_static/logo-agam.png" class="logo__image only-dark pst-js-only" alt="Malaria Vector Selection Atlas — *Anopheles gambiae* complex - Home"/>
  
  
</a></div>
        <div class="sidebar-primary-item">

<button class="btn search-button-field search-button__button pst-js-only" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
 <i class="fa-solid fa-magnifying-glass"></i>
 <span class="search-button__default-text">Search</span>
 <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
</button></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="index.html">
                    Malaria Vector Selection Atlas — Anopheles gambiae complex
                </a>
            </li>
        </ul>
        <p aria-level="2" class="caption" role="heading"><span class="caption-text">Selection alerts</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="ir-alerts.html">Insecticide resistance alerts</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="alert/SA-AGAM-01.html">Alert SA-AGAM-01 (<em>Vgsc</em>)</a></li>
<li class="toctree-l2"><a class="reference internal" href="alert/SA-AGAM-02.html">Alert SA-AGAM-02 (<em>Ace1</em>)</a></li>
<li class="toctree-l2"><a class="reference internal" href="alert/SA-AGAM-03.html">Alert SA-AGAM-03 (<em>Rdl</em>)</a></li>
<li class="toctree-l2"><a class="reference internal" href="alert/SA-AGAM-04.html">Alert SA-AGAM-04 (<em>Cyp6aa/Cyp6p</em>)</a></li>
<li class="toctree-l2"><a class="reference internal" href="alert/SA-AGAM-05.html">Alert SA-AGAM-05 (<em>Cyp9k1</em>)</a></li>
<li class="toctree-l2"><a class="reference internal" href="alert/SA-AGAM-06.html">Alert SA-AGAM-06 (<em>Gste</em>)</a></li>
<li class="toctree-l2"><a class="reference internal" href="alert/SA-AGAM-07.html">Alert SA-AGAM-07 (<em>Coeae1f/Coeae2f</em>)</a></li>
<li class="toctree-l2"><a class="reference internal" href="alert/SA-AGAM-08.html">Alert SA-AGAM-08 (<em>Keap1</em>)</a></li>
<li class="toctree-l2"><a class="reference internal" href="alert/SA-AGAM-09.html">Alert SA-AGAM-09 (<em>Dgk</em>)</a></li>
</ul>
</details></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">By chromosome</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="contig/2RL.html">Chromosome 2RL</a></li>
<li class="toctree-l1"><a class="reference internal" href="contig/3RL.html">Chromosome 3RL</a></li>
<li class="toctree-l1"><a class="reference internal" href="contig/X.html">Chromosome X</a></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">By country</span></p>
<ul class="nav bd-sidenav">
<li class="toctree-l1 has-children"><a class="reference internal" href="country/AO.html">Angola</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="cohort/AO-LUA_Luanda_colu_2009_Q2.html">Angola / Luanda / coluzzii / 2009 / Q2</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="country/BF.html">Burkina Faso</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="cohort/BF-02_Comoe_colu_2011.html">Burkina Faso / Comoe / coluzzii / 2011</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/BF-02_Comoe_colu_2012.html">Burkina Faso / Comoe / coluzzii / 2012</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/BF-02_Comoe_colu_2015.html">Burkina Faso / Comoe / coluzzii / 2015</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/BF-02_Comoe_colu_2016.html">Burkina Faso / Comoe / coluzzii / 2016</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/BF-09_Houet_colu_2012_Q3.html">Burkina Faso / Houet / coluzzii / 2012 / Q3</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/BF-09_Houet_colu_2014_Q3.html">Burkina Faso / Houet / coluzzii / 2014 / Q3</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/BF-09_Houet_gamb_2012_Q3.html">Burkina Faso / Houet / gambiae / 2012 / Q3</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/BF-09_Houet_gamb_2014_Q3.html">Burkina Faso / Houet / gambiae / 2014 / Q3</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="country/BJ.html">Benin</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="cohort/BJ-DO_Djougou_colu_2017_Q2.html">Benin / Djougou / coluzzii / 2017 / Q2</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/BJ-DO_Djougou_gamb_2017_Q2.html">Benin / Djougou / gambiae / 2017 / Q2</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/BJ-DO_Djougou_gamb_2017_Q3.html">Benin / Djougou / gambiae / 2017 / Q3</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/BJ-OU_Avrankou_colu_2017_Q3.html">Benin / Avrankou / coluzzii / 2017 / Q3</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="country/CD.html">Democratic Republic of the Congo</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="cohort/CD-NU_Gbadolite_gamb_2015_Q3.html">Democratic Republic of the Congo / Gbadolite / gambiae / 2015 / Q3</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="country/CF.html">Central African Republic</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="cohort/CF-BGF_Bangui_gamb_1994_Q1.html">Central African Republic / Bangui / gambiae / 1994 / Q1</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="country/CI.html">Cote d’Ivoire</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="cohort/CI-CM_Sud-Comoe_gamb_2017_Q3.html">Cote d’Ivoire / Sud-Comoe / gambiae / 2017 / Q3</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/CI-LG_Agneby-Tiassa_colu_2012.html">Cote d’Ivoire / Agneby-Tiassa / coluzzii / 2012</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="country/CM.html">Cameroon</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="cohort/CM-EN_Mayo-Kani_gamb_2005.html">Cameroon / Mayo-Kani / gambiae / 2005</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/CM-ES_Haut-Nyong_gamb_2009_Q3.html">Cameroon / Haut-Nyong / gambiae / 2009 / Q3</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/CM-ES_Lom-Et-Djerem_gamb_2009_Q3.html">Cameroon / Lom-Et-Djerem / gambiae / 2009 / Q3</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="country/GA.html">Gabon</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="cohort/GA-1_Libreville_gamb_2000_Q4.html">Gabon / Libreville / gambiae / 2000 / Q4</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="country/GH.html">Ghana</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="cohort/GH-AA_Ablekuma-Central-Municipal_colu_2018_Q1.html">Ghana / Ablekuma Central Municipal / coluzzii / 2018 / Q1</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/GH-AA_La-Nkwantanang-Madina_gamb_2017_Q4.html">Ghana / La-Nkwantanang-Madina / gambiae / 2017 / Q4</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/GH-AH_Adansi-Akrofuom_colu_2018_Q4.html">Ghana / Adansi Akrofuom / coluzzii / 2018 / Q4</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/GH-AH_Adansi-South_colu_2018_Q4.html">Ghana / Adansi South / coluzzii / 2018 / Q4</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/GH-AH_Adansi-South_gamb_2018_Q4.html">Ghana / Adansi South / gambiae / 2018 / Q4</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/GH-AH_Amansie-Central_colu_2018_Q4.html">Ghana / Amansie Central / coluzzii / 2018 / Q4</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/GH-AH_Bekwai-Municipal_colu_2018_Q4.html">Ghana / Bekwai Municipal / coluzzii / 2018 / Q4</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/GH-CP_Twifo-Atti-Morkwa_colu_2012_Q3.html">Ghana / Twifo Atti-Morkwa / coluzzii / 2012 / Q3</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/GH-CP_Upper-Denkyira-East-Municipal_colu_2018_Q4.html">Ghana / Upper Denkyira East Municipal / coluzzii / 2018 / Q4</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/GH-CP_Upper-Denkyira-West_colu_2018_Q4.html">Ghana / Upper Denkyira West / coluzzii / 2018 / Q4</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/GH-EP_New-Juaben-South-Municipal_gamb_2012_Q4.html">Ghana / New Juaben South Municipal / gambiae / 2012 / Q4</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/GH-WP_Effia-Kwesimintsim-Municipal_colu_2012_Q3.html">Ghana / Effia Kwesimintsim Municipal / coluzzii / 2012 / Q3</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="country/GM.html">Gambia, The</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="cohort/GM-M_Lower-Fuladu-West_colu_2012_Q4.html">Gambia, The / Lower Fuladu West / coluzzii / 2012 / Q4</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/GM-N_Central-Badibu_biss_2011_Q3.html">Gambia, The / Central Badibu / bissau / 2011 / Q3</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="country/GN.html">Guinea</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="cohort/GN-F_Kissidougou_gamb_2012_Q4.html">Guinea / Kissidougou / gambiae / 2012 / Q4</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/GN-N_Macenta_gamb_2012_Q4.html">Guinea / Macenta / gambiae / 2012 / Q4</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="country/GW.html">Guinea-Bissau</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="cohort/GW-BM_Setor-De-Safim_biss_2010.html">Guinea-Bissau / Setor De Safim / bissau / 2010</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/GW-BS_Bissau-Autonomous-Sector_biss_2010.html">Guinea-Bissau / Bissau Autonomous Sector / bissau / 2010</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="country/ML.html">Mali</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="cohort/ML-2_Kangaba_gamb_2004_Q3.html">Mali / Kangaba / gambiae / 2004 / Q3</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/ML-2_Kati_colu_2014_Q3.html">Mali / Kati / coluzzii / 2014 / Q3</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/ML-2_Kati_gamb_2014_Q3.html">Mali / Kati / gambiae / 2014 / Q3</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/ML-3_Yanfolila_colu_2012_Q4.html">Mali / Yanfolila / coluzzii / 2012 / Q4</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/ML-3_Yanfolila_gamb_2012_Q4.html">Mali / Yanfolila / gambiae / 2012 / Q4</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/ML-4_Bla_colu_2004_Q3.html">Mali / Bla / coluzzii / 2004 / Q3</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="country/MW.html">Malawi</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="cohort/MW-S_Chikwawa_arab_2015_Q2.html">Malawi / Chikwawa / arabiensis / 2015 / Q2</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="country/MZ.html">Mozambique</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="cohort/MZ-I_Morrumbene_gamb_2004_Q1.html">Mozambique / Morrumbene / gambiae / 2004 / Q1</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/MZ-I_Morrumbene_gamb_2004_Q2.html">Mozambique / Morrumbene / gambiae / 2004 / Q2</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="country/TG.html">Togo</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="cohort/TG-M_Lome-Commune_gamb_2017_Q4.html">Togo / Lome Commune / gambiae / 2017 / Q4</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="country/TZ.html">Tanzania</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="cohort/TZ-05_Muleba_arab_2015_Q1.html">Tanzania / Muleba / arabiensis / 2015 / Q1</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/TZ-05_Muleba_arab_2015_Q2.html">Tanzania / Muleba / arabiensis / 2015 / Q2</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/TZ-05_Muleba_gamb_2015_Q2.html">Tanzania / Muleba / gambiae / 2015 / Q2</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/TZ-13_Tarime_arab_2012_Q3.html">Tanzania / Tarime / arabiensis / 2012 / Q3</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/TZ-25_Muheza_gamb_2013_Q1.html">Tanzania / Muheza / gambiae / 2013 / Q1</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/TZ-26_Moshi_arab_2012_Q3.html">Tanzania / Moshi / arabiensis / 2012 / Q3</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="country/UG.html">Uganda</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="cohort/UG-C_Kalangala_gamb_2015_Q2.html">Uganda / Kalangala / gambiae / 2015 / Q2</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/UG-E_Busia_gamb_2016_Q2.html">Uganda / Busia / gambiae / 2016 / Q2</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/UG-E_Mayuge_gamb_2017_Q2.html">Uganda / Mayuge / gambiae / 2017 / Q2</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/UG-E_Tororo_arab_2012_Q4.html">Uganda / Tororo / arabiensis / 2012 / Q4</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/UG-E_Tororo_gamb_2012_Q4.html">Uganda / Tororo / gambiae / 2012 / Q4</a></li>
<li class="toctree-l2"><a class="reference internal" href="cohort/UG-W_Kanungu_gamb_2012_Q4.html">Uganda / Kanungu / gambiae / 2012 / Q4</a></li>
</ul>
</details></li>
</ul>
<p aria-level="2" class="caption" role="heading"><span class="caption-text">About</span></p>
<ul class="current nav bd-sidenav">
<li class="toctree-l1 current active"><a class="current reference internal" href="#">Data sources</a></li>
<li class="toctree-l1"><a class="reference internal" href="methods.html">Methods</a></li>
<li class="toctree-l1"><a class="reference internal" href="faq.html">FAQ</a></li>
<li class="toctree-l1"><a class="reference internal" href="glossary.html">Glossary</a></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
      <div class="sidebar-primary-item">
<div id="ethical-ad-placement"
      class="flat"
      data-ea-publisher="readthedocs"
      data-ea-type="readthedocs-sidebar"
      data-ea-manual="true">
</div></div>
  </div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-source-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Source repositories">
    <i class="fab fa-github"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="https://github.com/anopheles-genomic-surveillance/selection-atlas" target="_blank"
   class="btn btn-sm btn-source-repository-button dropdown-item"
   title="Source repository"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="btn__text-container">Repository</span>
</a>
</li>
      
      
      
      
      <li><a href="https://github.com/anopheles-genomic-surveillance/selection-atlas/issues/new?title=Issue%20on%20page%20%2Fdata-sources.html&body=Your%20issue%20content%20here." target="_blank"
   class="btn btn-sm btn-source-issues-button dropdown-item"
   title="Open an issue"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="btn__text-container">Open issue</span>
</a>
</li>
      
  </ul>
</div>






<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="_sources/data-sources.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button pst-js-only" aria-label="Color mode" data-bs-title="Color mode"  data-bs-placement="bottom" data-bs-toggle="tooltip">
  <i class="theme-switch fa-solid fa-sun                fa-lg" data-mode="light" title="Light"></i>
  <i class="theme-switch fa-solid fa-moon               fa-lg" data-mode="dark"  title="Dark"></i>
  <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"  title="System Settings"></i>
</button>


<button class="btn btn-sm pst-navbar-icon search-button search-button__button pst-js-only" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
</button>

</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Data sources</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <div class="cell tag_parameters tag_remove-input docutils container">
</div>
<div class="cell tag_injected-parameters tag_remove-input docutils container">
</div>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<script type="application/javascript">'use strict';
(function(root) {
  function now() {
    return new Date();
  }

  const force = true;

  if (typeof root._bokeh_onload_callbacks === "undefined" || force === true) {
    root._bokeh_onload_callbacks = [];
    root._bokeh_is_loading = undefined;
  }

const JS_MIME_TYPE = 'application/javascript';
  const HTML_MIME_TYPE = 'text/html';
  const EXEC_MIME_TYPE = 'application/vnd.bokehjs_exec.v0+json';
  const CLASS_NAME = 'output_bokeh rendered_html';

  /**
   * Render data to the DOM node
   */
  function render(props, node) {
    const script = document.createElement("script");
    node.appendChild(script);
  }

  /**
   * Handle when an output is cleared or removed
   */
  function handleClearOutput(event, handle) {
    function drop(id) {
      const view = Bokeh.index.get_by_id(id)
      if (view != null) {
        view.model.document.clear()
        Bokeh.index.delete(view)
      }
    }

    const cell = handle.cell;

    const id = cell.output_area._bokeh_element_id;
    const server_id = cell.output_area._bokeh_server_id;

    // Clean up Bokeh references
    if (id != null) {
      drop(id)
    }

    if (server_id !== undefined) {
      // Clean up Bokeh references
      const cmd_clean = "from bokeh.io.state import curstate; print(curstate().uuid_to_server['" + server_id + "'].get_sessions()[0].document.roots[0]._id)";
      cell.notebook.kernel.execute(cmd_clean, {
        iopub: {
          output: function(msg) {
            const id = msg.content.text.trim()
            drop(id)
          }
        }
      });
      // Destroy server and session
      const cmd_destroy = "import bokeh.io.notebook as ion; ion.destroy_server('" + server_id + "')";
      cell.notebook.kernel.execute(cmd_destroy);
    }
  }

  /**
   * Handle when a new output is added
   */
  function handleAddOutput(event, handle) {
    const output_area = handle.output_area;
    const output = handle.output;

    // limit handleAddOutput to display_data with EXEC_MIME_TYPE content only
    if ((output.output_type != "display_data") || (!Object.prototype.hasOwnProperty.call(output.data, EXEC_MIME_TYPE))) {
      return
    }

    const toinsert = output_area.element.find("." + CLASS_NAME.split(' ')[0]);

    if (output.metadata[EXEC_MIME_TYPE]["id"] !== undefined) {
      toinsert[toinsert.length - 1].firstChild.textContent = output.data[JS_MIME_TYPE];
      // store reference to embed id on output_area
      output_area._bokeh_element_id = output.metadata[EXEC_MIME_TYPE]["id"];
    }
    if (output.metadata[EXEC_MIME_TYPE]["server_id"] !== undefined) {
      const bk_div = document.createElement("div");
      bk_div.innerHTML = output.data[HTML_MIME_TYPE];
      const script_attrs = bk_div.children[0].attributes;
      for (let i = 0; i < script_attrs.length; i++) {
        toinsert[toinsert.length - 1].firstChild.setAttribute(script_attrs[i].name, script_attrs[i].value);
        toinsert[toinsert.length - 1].firstChild.textContent = bk_div.children[0].textContent
      }
      // store reference to server id on output_area
      output_area._bokeh_server_id = output.metadata[EXEC_MIME_TYPE]["server_id"];
    }
  }

  function register_renderer(events, OutputArea) {

    function append_mime(data, metadata, element) {
      // create a DOM node to render to
      const toinsert = this.create_output_subarea(
        metadata,
        CLASS_NAME,
        EXEC_MIME_TYPE
      );
      this.keyboard_manager.register_events(toinsert);
      // Render to node
      const props = {data: data, metadata: metadata[EXEC_MIME_TYPE]};
      render(props, toinsert[toinsert.length - 1]);
      element.append(toinsert);
      return toinsert
    }

    /* Handle when an output is cleared or removed */
    events.on('clear_output.CodeCell', handleClearOutput);
    events.on('delete.Cell', handleClearOutput);

    /* Handle when a new output is added */
    events.on('output_added.OutputArea', handleAddOutput);

    /**
     * Register the mime type and append_mime function with output_area
     */
    OutputArea.prototype.register_mime_type(EXEC_MIME_TYPE, append_mime, {
      /* Is output safe? */
      safe: true,
      /* Index of renderer in `output_area.display_order` */
      index: 0
    });
  }

  // register the mime type if in Jupyter Notebook environment and previously unregistered
  if (root.Jupyter !== undefined) {
    const events = require('base/js/events');
    const OutputArea = require('notebook/js/outputarea').OutputArea;

    if (OutputArea.prototype.mime_types().indexOf(EXEC_MIME_TYPE) == -1) {
      register_renderer(events, OutputArea);
    }
  }
  if (typeof (root._bokeh_timeout) === "undefined" || force === true) {
    root._bokeh_timeout = Date.now() + 5000;
    root._bokeh_failed_load = false;
  }

  const NB_LOAD_WARNING = {'data': {'text/html':
     "<div style='background-color: #fdd'>\n"+
     "<p>\n"+
     "BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \n"+
     "may be due to a slow or bad network connection. Possible fixes:\n"+
     "</p>\n"+
     "<ul>\n"+
     "<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\n"+
     "<li>use INLINE resources instead, as so:</li>\n"+
     "</ul>\n"+
     "<code>\n"+
     "from bokeh.resources import INLINE\n"+
     "output_notebook(resources=INLINE)\n"+
     "</code>\n"+
     "</div>"}};

  function display_loaded(error = null) {
    const el = document.getElementById(null);
    if (el != null) {
      const html = (() => {
        if (typeof root.Bokeh === "undefined") {
          if (error == null) {
            return "BokehJS is loading ...";
          } else {
            return "BokehJS failed to load.";
          }
        } else {
          const prefix = `BokehJS ${root.Bokeh.version}`;
          if (error == null) {
            return `${prefix} successfully loaded.`;
          } else {
            return `${prefix} <b>encountered errors</b> while loading and may not function as expected.`;
          }
        }
      })();
      el.innerHTML = html;

      if (error != null) {
        const wrapper = document.createElement("div");
        wrapper.style.overflow = "auto";
        wrapper.style.height = "5em";
        wrapper.style.resize = "vertical";
        const content = document.createElement("div");
        content.style.fontFamily = "monospace";
        content.style.whiteSpace = "pre-wrap";
        content.style.backgroundColor = "rgb(255, 221, 221)";
        content.textContent = error.stack ?? error.toString();
        wrapper.append(content);
        el.append(wrapper);
      }
    } else if (Date.now() < root._bokeh_timeout) {
      setTimeout(() => display_loaded(error), 100);
    }
  }

  function run_callbacks() {
    try {
      root._bokeh_onload_callbacks.forEach(function(callback) {
        if (callback != null)
          callback();
      });
    } finally {
      delete root._bokeh_onload_callbacks
    }
    console.debug("Bokeh: all callbacks have finished");
  }

  function load_libs(css_urls, js_urls, callback) {
    if (css_urls == null) css_urls = [];
    if (js_urls == null) js_urls = [];

    root._bokeh_onload_callbacks.push(callback);
    if (root._bokeh_is_loading > 0) {
      console.debug("Bokeh: BokehJS is being loaded, scheduling callback at", now());
      return null;
    }
    if (js_urls == null || js_urls.length === 0) {
      run_callbacks();
      return null;
    }
    console.debug("Bokeh: BokehJS not loaded, scheduling load and callback at", now());
    root._bokeh_is_loading = css_urls.length + js_urls.length;

    function on_load() {
      root._bokeh_is_loading--;
      if (root._bokeh_is_loading === 0) {
        console.debug("Bokeh: all BokehJS libraries/stylesheets loaded");
        run_callbacks()
      }
    }

    function on_error(url) {
      console.error("failed to load " + url);
    }

    for (let i = 0; i < css_urls.length; i++) {
      const url = css_urls[i];
      const element = document.createElement("link");
      element.onload = on_load;
      element.onerror = on_error.bind(null, url);
      element.rel = "stylesheet";
      element.type = "text/css";
      element.href = url;
      console.debug("Bokeh: injecting link tag for BokehJS stylesheet: ", url);
      document.body.appendChild(element);
    }

    for (let i = 0; i < js_urls.length; i++) {
      const url = js_urls[i];
      const element = document.createElement('script');
      element.onload = on_load;
      element.onerror = on_error.bind(null, url);
      element.async = false;
      element.src = url;
      console.debug("Bokeh: injecting script tag for BokehJS library: ", url);
      document.head.appendChild(element);
    }
  };

  function inject_raw_css(css) {
    const element = document.createElement("style");
    element.appendChild(document.createTextNode(css));
    document.body.appendChild(element);
  }

  const js_urls = ["https://cdn.bokeh.org/bokeh/release/bokeh-3.6.3.min.js", "https://cdn.bokeh.org/bokeh/release/bokeh-gl-3.6.3.min.js", "https://cdn.bokeh.org/bokeh/release/bokeh-widgets-3.6.3.min.js", "https://cdn.bokeh.org/bokeh/release/bokeh-tables-3.6.3.min.js", "https://cdn.bokeh.org/bokeh/release/bokeh-mathjax-3.6.3.min.js"];
  const css_urls = [];

  const inline_js = [    function(Bokeh) {
      Bokeh.set_log_level("info");
    },
function(Bokeh) {
    }
  ];

  function run_inline_js() {
    if (root.Bokeh !== undefined || force === true) {
      try {
            for (let i = 0; i < inline_js.length; i++) {
      inline_js[i].call(root, root.Bokeh);
    }

      } catch (error) {throw error;
      }} else if (Date.now() < root._bokeh_timeout) {
      setTimeout(run_inline_js, 100);
    } else if (!root._bokeh_failed_load) {
      console.log("Bokeh: BokehJS failed to load within specified timeout.");
      root._bokeh_failed_load = true;
    } else if (force !== true) {
      const cell = $(document.getElementById(null)).parents('.cell').data().cell;
      cell.output_area.append_execute_result(NB_LOAD_WARNING)
    }
  }

  if (root._bokeh_is_loading === 0) {
    console.debug("Bokeh: BokehJS loaded, going straight to plotting");
    run_inline_js();
  } else {
    load_libs(css_urls, js_urls, function() {
      console.debug("Bokeh: BokehJS plotting callback run at", now());
      run_inline_js();
    });
  }
}(window));</script></div>
</div>
<section class="tex2jax_ignore mathjax_ignore" id="data-sources">
<h1>Data sources<a class="headerlink" href="#data-sources" title="Link to this heading">#</a></h1>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<script type="application/javascript">'use strict';
(function(root) {
  function now() {
    return new Date();
  }

  const force = true;

  if (typeof root._bokeh_onload_callbacks === "undefined" || force === true) {
    root._bokeh_onload_callbacks = [];
    root._bokeh_is_loading = undefined;
  }

const JS_MIME_TYPE = 'application/javascript';
  const HTML_MIME_TYPE = 'text/html';
  const EXEC_MIME_TYPE = 'application/vnd.bokehjs_exec.v0+json';
  const CLASS_NAME = 'output_bokeh rendered_html';

  /**
   * Render data to the DOM node
   */
  function render(props, node) {
    const script = document.createElement("script");
    node.appendChild(script);
  }

  /**
   * Handle when an output is cleared or removed
   */
  function handleClearOutput(event, handle) {
    function drop(id) {
      const view = Bokeh.index.get_by_id(id)
      if (view != null) {
        view.model.document.clear()
        Bokeh.index.delete(view)
      }
    }

    const cell = handle.cell;

    const id = cell.output_area._bokeh_element_id;
    const server_id = cell.output_area._bokeh_server_id;

    // Clean up Bokeh references
    if (id != null) {
      drop(id)
    }

    if (server_id !== undefined) {
      // Clean up Bokeh references
      const cmd_clean = "from bokeh.io.state import curstate; print(curstate().uuid_to_server['" + server_id + "'].get_sessions()[0].document.roots[0]._id)";
      cell.notebook.kernel.execute(cmd_clean, {
        iopub: {
          output: function(msg) {
            const id = msg.content.text.trim()
            drop(id)
          }
        }
      });
      // Destroy server and session
      const cmd_destroy = "import bokeh.io.notebook as ion; ion.destroy_server('" + server_id + "')";
      cell.notebook.kernel.execute(cmd_destroy);
    }
  }

  /**
   * Handle when a new output is added
   */
  function handleAddOutput(event, handle) {
    const output_area = handle.output_area;
    const output = handle.output;

    // limit handleAddOutput to display_data with EXEC_MIME_TYPE content only
    if ((output.output_type != "display_data") || (!Object.prototype.hasOwnProperty.call(output.data, EXEC_MIME_TYPE))) {
      return
    }

    const toinsert = output_area.element.find("." + CLASS_NAME.split(' ')[0]);

    if (output.metadata[EXEC_MIME_TYPE]["id"] !== undefined) {
      toinsert[toinsert.length - 1].firstChild.textContent = output.data[JS_MIME_TYPE];
      // store reference to embed id on output_area
      output_area._bokeh_element_id = output.metadata[EXEC_MIME_TYPE]["id"];
    }
    if (output.metadata[EXEC_MIME_TYPE]["server_id"] !== undefined) {
      const bk_div = document.createElement("div");
      bk_div.innerHTML = output.data[HTML_MIME_TYPE];
      const script_attrs = bk_div.children[0].attributes;
      for (let i = 0; i < script_attrs.length; i++) {
        toinsert[toinsert.length - 1].firstChild.setAttribute(script_attrs[i].name, script_attrs[i].value);
        toinsert[toinsert.length - 1].firstChild.textContent = bk_div.children[0].textContent
      }
      // store reference to server id on output_area
      output_area._bokeh_server_id = output.metadata[EXEC_MIME_TYPE]["server_id"];
    }
  }

  function register_renderer(events, OutputArea) {

    function append_mime(data, metadata, element) {
      // create a DOM node to render to
      const toinsert = this.create_output_subarea(
        metadata,
        CLASS_NAME,
        EXEC_MIME_TYPE
      );
      this.keyboard_manager.register_events(toinsert);
      // Render to node
      const props = {data: data, metadata: metadata[EXEC_MIME_TYPE]};
      render(props, toinsert[toinsert.length - 1]);
      element.append(toinsert);
      return toinsert
    }

    /* Handle when an output is cleared or removed */
    events.on('clear_output.CodeCell', handleClearOutput);
    events.on('delete.Cell', handleClearOutput);

    /* Handle when a new output is added */
    events.on('output_added.OutputArea', handleAddOutput);

    /**
     * Register the mime type and append_mime function with output_area
     */
    OutputArea.prototype.register_mime_type(EXEC_MIME_TYPE, append_mime, {
      /* Is output safe? */
      safe: true,
      /* Index of renderer in `output_area.display_order` */
      index: 0
    });
  }

  // register the mime type if in Jupyter Notebook environment and previously unregistered
  if (root.Jupyter !== undefined) {
    const events = require('base/js/events');
    const OutputArea = require('notebook/js/outputarea').OutputArea;

    if (OutputArea.prototype.mime_types().indexOf(EXEC_MIME_TYPE) == -1) {
      register_renderer(events, OutputArea);
    }
  }
  if (typeof (root._bokeh_timeout) === "undefined" || force === true) {
    root._bokeh_timeout = Date.now() + 5000;
    root._bokeh_failed_load = false;
  }

  const NB_LOAD_WARNING = {'data': {'text/html':
     "<div style='background-color: #fdd'>\n"+
     "<p>\n"+
     "BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \n"+
     "may be due to a slow or bad network connection. Possible fixes:\n"+
     "</p>\n"+
     "<ul>\n"+
     "<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\n"+
     "<li>use INLINE resources instead, as so:</li>\n"+
     "</ul>\n"+
     "<code>\n"+
     "from bokeh.resources import INLINE\n"+
     "output_notebook(resources=INLINE)\n"+
     "</code>\n"+
     "</div>"}};

  function display_loaded(error = null) {
    const el = document.getElementById(null);
    if (el != null) {
      const html = (() => {
        if (typeof root.Bokeh === "undefined") {
          if (error == null) {
            return "BokehJS is loading ...";
          } else {
            return "BokehJS failed to load.";
          }
        } else {
          const prefix = `BokehJS ${root.Bokeh.version}`;
          if (error == null) {
            return `${prefix} successfully loaded.`;
          } else {
            return `${prefix} <b>encountered errors</b> while loading and may not function as expected.`;
          }
        }
      })();
      el.innerHTML = html;

      if (error != null) {
        const wrapper = document.createElement("div");
        wrapper.style.overflow = "auto";
        wrapper.style.height = "5em";
        wrapper.style.resize = "vertical";
        const content = document.createElement("div");
        content.style.fontFamily = "monospace";
        content.style.whiteSpace = "pre-wrap";
        content.style.backgroundColor = "rgb(255, 221, 221)";
        content.textContent = error.stack ?? error.toString();
        wrapper.append(content);
        el.append(wrapper);
      }
    } else if (Date.now() < root._bokeh_timeout) {
      setTimeout(() => display_loaded(error), 100);
    }
  }

  function run_callbacks() {
    try {
      root._bokeh_onload_callbacks.forEach(function(callback) {
        if (callback != null)
          callback();
      });
    } finally {
      delete root._bokeh_onload_callbacks
    }
    console.debug("Bokeh: all callbacks have finished");
  }

  function load_libs(css_urls, js_urls, callback) {
    if (css_urls == null) css_urls = [];
    if (js_urls == null) js_urls = [];

    root._bokeh_onload_callbacks.push(callback);
    if (root._bokeh_is_loading > 0) {
      console.debug("Bokeh: BokehJS is being loaded, scheduling callback at", now());
      return null;
    }
    if (js_urls == null || js_urls.length === 0) {
      run_callbacks();
      return null;
    }
    console.debug("Bokeh: BokehJS not loaded, scheduling load and callback at", now());
    root._bokeh_is_loading = css_urls.length + js_urls.length;

    function on_load() {
      root._bokeh_is_loading--;
      if (root._bokeh_is_loading === 0) {
        console.debug("Bokeh: all BokehJS libraries/stylesheets loaded");
        run_callbacks()
      }
    }

    function on_error(url) {
      console.error("failed to load " + url);
    }

    for (let i = 0; i < css_urls.length; i++) {
      const url = css_urls[i];
      const element = document.createElement("link");
      element.onload = on_load;
      element.onerror = on_error.bind(null, url);
      element.rel = "stylesheet";
      element.type = "text/css";
      element.href = url;
      console.debug("Bokeh: injecting link tag for BokehJS stylesheet: ", url);
      document.body.appendChild(element);
    }

    for (let i = 0; i < js_urls.length; i++) {
      const url = js_urls[i];
      const element = document.createElement('script');
      element.onload = on_load;
      element.onerror = on_error.bind(null, url);
      element.async = false;
      element.src = url;
      console.debug("Bokeh: injecting script tag for BokehJS library: ", url);
      document.head.appendChild(element);
    }
  };

  function inject_raw_css(css) {
    const element = document.createElement("style");
    element.appendChild(document.createTextNode(css));
    document.body.appendChild(element);
  }

  const js_urls = ["https://cdn.bokeh.org/bokeh/release/bokeh-3.6.3.min.js", "https://cdn.bokeh.org/bokeh/release/bokeh-gl-3.6.3.min.js", "https://cdn.bokeh.org/bokeh/release/bokeh-widgets-3.6.3.min.js", "https://cdn.bokeh.org/bokeh/release/bokeh-tables-3.6.3.min.js", "https://cdn.bokeh.org/bokeh/release/bokeh-mathjax-3.6.3.min.js"];
  const css_urls = [];

  const inline_js = [    function(Bokeh) {
      Bokeh.set_log_level("info");
    },
function(Bokeh) {
    }
  ];

  function run_inline_js() {
    if (root.Bokeh !== undefined || force === true) {
      try {
            for (let i = 0; i < inline_js.length; i++) {
      inline_js[i].call(root, root.Bokeh);
    }

      } catch (error) {throw error;
      }} else if (Date.now() < root._bokeh_timeout) {
      setTimeout(run_inline_js, 100);
    } else if (!root._bokeh_failed_load) {
      console.log("Bokeh: BokehJS failed to load within specified timeout.");
      root._bokeh_failed_load = true;
    } else if (force !== true) {
      const cell = $(document.getElementById(null)).parents('.cell').data().cell;
      cell.output_area.append_execute_result(NB_LOAD_WARNING)
    }
  }

  if (root._bokeh_is_loading === 0) {
    console.debug("Bokeh: BokehJS loaded, going straight to plotting");
    run_inline_js();
  } else {
    load_libs(css_urls, js_urls, function() {
      console.debug("Bokeh: BokehJS plotting callback run at", now());
      run_inline_js();
    });
  }
}(window));</script></div>
</div>
<div class="cell tag_remove-input docutils container">
</div>
<p>This report analyses data from the <a class="reference internal" href="glossary.html#term-Malaria-Vector-Genome-Observatory"><span class="xref std std-term">Malaria Vector Genome Observatory</span></a>. The
current analysis version is agam/2025.03.05 which includes
data for a total of 4,878 mosquitoes sampled from 25 countries. Please
see the table below for more information about the sample sets and releases from which the data
were obtained.</p>
<div class="cell tag_remove-input docutils container">
<div class="cell_output docutils container">
<div class="output text_html"><style type="text/css">
#T_78415 th {
  text-align: left;
}
#T_78415 td {
  text-align: left;
  font-weight: normal;
}
</style>
<table id="T_78415">
  <caption>Table 1. Sample sets included in the current analysis version.</caption>
  <thead>
    <tr>
      <th id="T_78415_level0_col0" class="col_heading level0 col0" >Sample Set</th>
      <th id="T_78415_level0_col1" class="col_heading level0 col1" >Study</th>
      <th id="T_78415_level0_col2" class="col_heading level0 col2" >Contributor</th>
      <th id="T_78415_level0_col3" class="col_heading level0 col3" >Data Release</th>
      <th id="T_78415_level0_col4" class="col_heading level0 col4" >Citation</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td id="T_78415_row0_col0" class="data row0 col0" >AG1000G-AO</td>
      <td id="T_78415_row0_col1" class="data row0 col1" ><a href="https://www.malariagen.net/partner_study/AG1000G-AO" target="_blank">AG1000G-AO</a></td>
      <td id="T_78415_row0_col2" class="data row0 col2" >Joao Pinto</td>
      <td id="T_78415_row0_col3" class="data row0 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.0.html" target="_blank">Ag3.0</a></td>
      <td id="T_78415_row0_col4" class="data row0 col4" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC6026373/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2017</a></td>
    </tr>
    <tr>
      <td id="T_78415_row1_col0" class="data row1 col0" >AG1000G-BF-A</td>
      <td id="T_78415_row1_col1" class="data row1 col1" ><a href="https://www.malariagen.net/partner_study/AG1000G-BF-1" target="_blank">AG1000G-BF-1</a></td>
      <td id="T_78415_row1_col2" class="data row1 col2" >Austin Burt</td>
      <td id="T_78415_row1_col3" class="data row1 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.0.html" target="_blank">Ag3.0</a></td>
      <td id="T_78415_row1_col4" class="data row1 col4" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC6026373/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2017</a></td>
    </tr>
    <tr>
      <td id="T_78415_row2_col0" class="data row2 col0" >AG1000G-BF-B</td>
      <td id="T_78415_row2_col1" class="data row2 col1" ><a href="https://www.malariagen.net/partner_study/AG1000G-BF-1" target="_blank">AG1000G-BF-1</a></td>
      <td id="T_78415_row2_col2" class="data row2 col2" >Austin Burt</td>
      <td id="T_78415_row2_col3" class="data row2 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.0.html" target="_blank">Ag3.0</a></td>
      <td id="T_78415_row2_col4" class="data row2 col4" ><a href='https://www.malariagen.net/data_package/ag30-anopheles-gambiae-data-resource/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2021</a></td>
    </tr>
    <tr>
      <td id="T_78415_row3_col0" class="data row3 col0" >AG1000G-BF-C</td>
      <td id="T_78415_row3_col1" class="data row3 col1" ><a href="https://www.malariagen.net/partner_study/AG1000G-BF-2" target="_blank">AG1000G-BF-2</a></td>
      <td id="T_78415_row3_col2" class="data row3 col2" >Nora Besansky</td>
      <td id="T_78415_row3_col3" class="data row3 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.0.html" target="_blank">Ag3.0</a></td>
      <td id="T_78415_row3_col4" class="data row3 col4" ><a href='https://www.malariagen.net/data_package/ag30-anopheles-gambiae-data-resource/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2021</a></td>
    </tr>
    <tr>
      <td id="T_78415_row4_col0" class="data row4 col0" >AG1000G-CD</td>
      <td id="T_78415_row4_col1" class="data row4 col1" ><a href="https://www.malariagen.net/partner_study/AG1000G-CD" target="_blank">AG1000G-CD</a></td>
      <td id="T_78415_row4_col2" class="data row4 col2" >David Weetman</td>
      <td id="T_78415_row4_col3" class="data row4 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.0.html" target="_blank">Ag3.0</a></td>
      <td id="T_78415_row4_col4" class="data row4 col4" ><a href='https://www.malariagen.net/data_package/ag30-anopheles-gambiae-data-resource/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2021</a></td>
    </tr>
    <tr>
      <td id="T_78415_row5_col0" class="data row5 col0" >AG1000G-CF</td>
      <td id="T_78415_row5_col1" class="data row5 col1" ><a href="https://www.malariagen.net/partner_study/AG1000G-CF" target="_blank">AG1000G-CF</a></td>
      <td id="T_78415_row5_col2" class="data row5 col2" >Alessandra della Torre</td>
      <td id="T_78415_row5_col3" class="data row5 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.0.html" target="_blank">Ag3.0</a></td>
      <td id="T_78415_row5_col4" class="data row5 col4" ><a href='https://www.malariagen.net/data_package/ag30-anopheles-gambiae-data-resource/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2021</a></td>
    </tr>
    <tr>
      <td id="T_78415_row6_col0" class="data row6 col0" >AG1000G-CI</td>
      <td id="T_78415_row6_col1" class="data row6 col1" ><a href="https://www.malariagen.net/partner_study/AG1000G-CI" target="_blank">AG1000G-CI</a></td>
      <td id="T_78415_row6_col2" class="data row6 col2" >David Weetman</td>
      <td id="T_78415_row6_col3" class="data row6 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.0.html" target="_blank">Ag3.0</a></td>
      <td id="T_78415_row6_col4" class="data row6 col4" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC7605271/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2020</a></td>
    </tr>
    <tr>
      <td id="T_78415_row7_col0" class="data row7 col0" >AG1000G-CM-A</td>
      <td id="T_78415_row7_col1" class="data row7 col1" ><a href="https://www.malariagen.net/partner_study/AG1000G-CM-1" target="_blank">AG1000G-CM-1</a></td>
      <td id="T_78415_row7_col2" class="data row7 col2" >Nora Besansky</td>
      <td id="T_78415_row7_col3" class="data row7 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.0.html" target="_blank">Ag3.0</a></td>
      <td id="T_78415_row7_col4" class="data row7 col4" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC6026373/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2017</a></td>
    </tr>
    <tr>
      <td id="T_78415_row8_col0" class="data row8 col0" >AG1000G-CM-B</td>
      <td id="T_78415_row8_col1" class="data row8 col1" ><a href="https://www.malariagen.net/partner_study/AG1000G-CM-2" target="_blank">AG1000G-CM-2</a></td>
      <td id="T_78415_row8_col2" class="data row8 col2" >Nora Besansky</td>
      <td id="T_78415_row8_col3" class="data row8 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.0.html" target="_blank">Ag3.0</a></td>
      <td id="T_78415_row8_col4" class="data row8 col4" ><a href='https://www.malariagen.net/data_package/ag30-anopheles-gambiae-data-resource/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2021</a></td>
    </tr>
    <tr>
      <td id="T_78415_row9_col0" class="data row9 col0" >AG1000G-CM-C</td>
      <td id="T_78415_row9_col1" class="data row9 col1" ><a href="https://www.malariagen.net/partner_study/AG1000G-CM-3" target="_blank">AG1000G-CM-3</a></td>
      <td id="T_78415_row9_col2" class="data row9 col2" >Brad White</td>
      <td id="T_78415_row9_col3" class="data row9 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.0.html" target="_blank">Ag3.0</a></td>
      <td id="T_78415_row9_col4" class="data row9 col4" ><a href='https://www.malariagen.net/data_package/ag30-anopheles-gambiae-data-resource/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2021</a></td>
    </tr>
    <tr>
      <td id="T_78415_row10_col0" class="data row10 col0" >AG1000G-FR</td>
      <td id="T_78415_row10_col1" class="data row10 col1" ><a href="https://www.malariagen.net/partner_study/AG1000G-FR" target="_blank">AG1000G-FR</a></td>
      <td id="T_78415_row10_col2" class="data row10 col2" >Igor Sharakhov</td>
      <td id="T_78415_row10_col3" class="data row10 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.0.html" target="_blank">Ag3.0</a></td>
      <td id="T_78415_row10_col4" class="data row10 col4" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC7605271/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2020</a></td>
    </tr>
    <tr>
      <td id="T_78415_row11_col0" class="data row11 col0" >AG1000G-GA-A</td>
      <td id="T_78415_row11_col1" class="data row11 col1" ><a href="https://www.malariagen.net/partner_study/AG1000G-GA-1" target="_blank">AG1000G-GA-1</a></td>
      <td id="T_78415_row11_col2" class="data row11 col2" >Joao Pinto</td>
      <td id="T_78415_row11_col3" class="data row11 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.0.html" target="_blank">Ag3.0</a></td>
      <td id="T_78415_row11_col4" class="data row11 col4" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC6026373/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2017</a></td>
    </tr>
    <tr>
      <td id="T_78415_row12_col0" class="data row12 col0" >AG1000G-GH</td>
      <td id="T_78415_row12_col1" class="data row12 col1" ><a href="https://www.malariagen.net/partner_study/AG1000G-GH" target="_blank">AG1000G-GH</a></td>
      <td id="T_78415_row12_col2" class="data row12 col2" >David Weetman</td>
      <td id="T_78415_row12_col3" class="data row12 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.0.html" target="_blank">Ag3.0</a></td>
      <td id="T_78415_row12_col4" class="data row12 col4" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC7605271/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2020</a></td>
    </tr>
    <tr>
      <td id="T_78415_row13_col0" class="data row13 col0" >AG1000G-GM-A</td>
      <td id="T_78415_row13_col1" class="data row13 col1" ><a href="https://www.malariagen.net/partner_study/AG1000G-GM-1" target="_blank">AG1000G-GM-1</a></td>
      <td id="T_78415_row13_col2" class="data row13 col2" >Martin Donnelly</td>
      <td id="T_78415_row13_col3" class="data row13 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.0.html" target="_blank">Ag3.0</a></td>
      <td id="T_78415_row13_col4" class="data row13 col4" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC7605271/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2020</a></td>
    </tr>
    <tr>
      <td id="T_78415_row14_col0" class="data row14 col0" >AG1000G-GM-B</td>
      <td id="T_78415_row14_col1" class="data row14 col1" ><a href="https://www.malariagen.net/partner_study/AG1000G-GM-2" target="_blank">AG1000G-GM-2</a></td>
      <td id="T_78415_row14_col2" class="data row14 col2" >Beniamino Caputo</td>
      <td id="T_78415_row14_col3" class="data row14 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.0.html" target="_blank">Ag3.0</a></td>
      <td id="T_78415_row14_col4" class="data row14 col4" ><a href='https://www.malariagen.net/data_package/ag30-anopheles-gambiae-data-resource/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2021</a></td>
    </tr>
    <tr>
      <td id="T_78415_row15_col0" class="data row15 col0" >AG1000G-GM-C</td>
      <td id="T_78415_row15_col1" class="data row15 col1" ><a href="https://www.malariagen.net/partner_study/AG1000G-GM-3" target="_blank">AG1000G-GM-3</a></td>
      <td id="T_78415_row15_col2" class="data row15 col2" >Charles Godfray</td>
      <td id="T_78415_row15_col3" class="data row15 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.0.html" target="_blank">Ag3.0</a></td>
      <td id="T_78415_row15_col4" class="data row15 col4" ><a href='https://www.malariagen.net/data_package/ag30-anopheles-gambiae-data-resource/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2021</a></td>
    </tr>
    <tr>
      <td id="T_78415_row16_col0" class="data row16 col0" >AG1000G-GN-A</td>
      <td id="T_78415_row16_col1" class="data row16 col1" ><a href="https://www.malariagen.net/partner_study/AG1000G-GN-ML" target="_blank">AG1000G-GN-ML</a></td>
      <td id="T_78415_row16_col2" class="data row16 col2" >Ken Vernick</td>
      <td id="T_78415_row16_col3" class="data row16 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.0.html" target="_blank">Ag3.0</a></td>
      <td id="T_78415_row16_col4" class="data row16 col4" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC6026373/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2017</a></td>
    </tr>
    <tr>
      <td id="T_78415_row17_col0" class="data row17 col0" >AG1000G-GN-B</td>
      <td id="T_78415_row17_col1" class="data row17 col1" ><a href="https://www.malariagen.net/partner_study/AG1000G-GN-ML" target="_blank">AG1000G-GN-ML</a></td>
      <td id="T_78415_row17_col2" class="data row17 col2" >Ken Vernick</td>
      <td id="T_78415_row17_col3" class="data row17 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.0.html" target="_blank">Ag3.0</a></td>
      <td id="T_78415_row17_col4" class="data row17 col4" ><a href='https://www.malariagen.net/data_package/ag30-anopheles-gambiae-data-resource/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2021</a></td>
    </tr>
    <tr>
      <td id="T_78415_row18_col0" class="data row18 col0" >AG1000G-GQ</td>
      <td id="T_78415_row18_col1" class="data row18 col1" ><a href="https://www.malariagen.net/partner_study/AG1000G-GQ" target="_blank">AG1000G-GQ</a></td>
      <td id="T_78415_row18_col2" class="data row18 col2" >Igor Sharakhov</td>
      <td id="T_78415_row18_col3" class="data row18 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.0.html" target="_blank">Ag3.0</a></td>
      <td id="T_78415_row18_col4" class="data row18 col4" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC7605271/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2020</a></td>
    </tr>
    <tr>
      <td id="T_78415_row19_col0" class="data row19 col0" >AG1000G-GW</td>
      <td id="T_78415_row19_col1" class="data row19 col1" ><a href="https://www.malariagen.net/partner_study/AG1000G-GW" target="_blank">AG1000G-GW</a></td>
      <td id="T_78415_row19_col2" class="data row19 col2" >Joao Pinto</td>
      <td id="T_78415_row19_col3" class="data row19 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.0.html" target="_blank">Ag3.0</a></td>
      <td id="T_78415_row19_col4" class="data row19 col4" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC6026373/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2017</a></td>
    </tr>
    <tr>
      <td id="T_78415_row20_col0" class="data row20 col0" >AG1000G-KE</td>
      <td id="T_78415_row20_col1" class="data row20 col1" ><a href="https://www.malariagen.net/partner_study/AG1000G-KE" target="_blank">AG1000G-KE</a></td>
      <td id="T_78415_row20_col2" class="data row20 col2" >Janet Midega</td>
      <td id="T_78415_row20_col3" class="data row20 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.0.html" target="_blank">Ag3.0</a></td>
      <td id="T_78415_row20_col4" class="data row20 col4" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC6026373/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2017</a></td>
    </tr>
    <tr>
      <td id="T_78415_row21_col0" class="data row21 col0" >AG1000G-ML-A</td>
      <td id="T_78415_row21_col1" class="data row21 col1" ><a href="https://www.malariagen.net/partner_study/AG1000G-ML-1" target="_blank">AG1000G-ML-1</a></td>
      <td id="T_78415_row21_col2" class="data row21 col2" >Austin Burt</td>
      <td id="T_78415_row21_col3" class="data row21 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.0.html" target="_blank">Ag3.0</a></td>
      <td id="T_78415_row21_col4" class="data row21 col4" ><a href='https://www.malariagen.net/data_package/ag30-anopheles-gambiae-data-resource/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2021</a></td>
    </tr>
    <tr>
      <td id="T_78415_row22_col0" class="data row22 col0" >AG1000G-ML-B</td>
      <td id="T_78415_row22_col1" class="data row22 col1" ><a href="https://www.malariagen.net/partner_study/AG1000G-ML-2" target="_blank">AG1000G-ML-2</a></td>
      <td id="T_78415_row22_col2" class="data row22 col2" >Nora Besansky</td>
      <td id="T_78415_row22_col3" class="data row22 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.0.html" target="_blank">Ag3.0</a></td>
      <td id="T_78415_row22_col4" class="data row22 col4" ><a href='https://www.malariagen.net/data_package/ag30-anopheles-gambiae-data-resource/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2021</a></td>
    </tr>
    <tr>
      <td id="T_78415_row23_col0" class="data row23 col0" >AG1000G-MW</td>
      <td id="T_78415_row23_col1" class="data row23 col1" ><a href="https://www.malariagen.net/partner_study/AG1000G-MW" target="_blank">AG1000G-MW</a></td>
      <td id="T_78415_row23_col2" class="data row23 col2" >Martin Donnelly</td>
      <td id="T_78415_row23_col3" class="data row23 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.0.html" target="_blank">Ag3.0</a></td>
      <td id="T_78415_row23_col4" class="data row23 col4" ><a href='https://www.malariagen.net/data_package/ag30-anopheles-gambiae-data-resource/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2021</a></td>
    </tr>
    <tr>
      <td id="T_78415_row24_col0" class="data row24 col0" >AG1000G-MZ</td>
      <td id="T_78415_row24_col1" class="data row24 col1" ><a href="https://www.malariagen.net/partner_study/AG1000G-MZ" target="_blank">AG1000G-MZ</a></td>
      <td id="T_78415_row24_col2" class="data row24 col2" >Joao Pinto</td>
      <td id="T_78415_row24_col3" class="data row24 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.0.html" target="_blank">Ag3.0</a></td>
      <td id="T_78415_row24_col4" class="data row24 col4" ><a href='https://www.malariagen.net/data_package/ag30-anopheles-gambiae-data-resource/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2021</a></td>
    </tr>
    <tr>
      <td id="T_78415_row25_col0" class="data row25 col0" >AG1000G-TZ</td>
      <td id="T_78415_row25_col1" class="data row25 col1" ><a href="https://www.malariagen.net/partner_study/AG1000G-TZ" target="_blank">AG1000G-TZ</a></td>
      <td id="T_78415_row25_col2" class="data row25 col2" >Bilali Kabula, David Weetman</td>
      <td id="T_78415_row25_col3" class="data row25 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.0.html" target="_blank">Ag3.0</a></td>
      <td id="T_78415_row25_col4" class="data row25 col4" ><a href='https://www.malariagen.net/data_package/ag30-anopheles-gambiae-data-resource/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2021</a></td>
    </tr>
    <tr>
      <td id="T_78415_row26_col0" class="data row26 col0" >AG1000G-UG</td>
      <td id="T_78415_row26_col1" class="data row26 col1" ><a href="https://www.malariagen.net/partner_study/AG1000G-UG" target="_blank">AG1000G-UG</a></td>
      <td id="T_78415_row26_col2" class="data row26 col2" >Martin Donnelly</td>
      <td id="T_78415_row26_col3" class="data row26 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.0.html" target="_blank">Ag3.0</a></td>
      <td id="T_78415_row26_col4" class="data row26 col4" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC6026373/' target='_blank'>Anopheles gambiae 1000 Genomes Consortium 2017</a></td>
    </tr>
    <tr>
      <td id="T_78415_row27_col0" class="data row27 col0" >1237-VO-BJ-DJOGBENOU-VMF00050</td>
      <td id="T_78415_row27_col1" class="data row27 col1" ><a href="https://www.malariagen.net/partner_study/1237-VO-BJ-DJOGBENOU" target="_blank">1237-VO-BJ-DJOGBENOU</a></td>
      <td id="T_78415_row27_col2" class="data row27 col2" >Luc Djogbenou</td>
      <td id="T_78415_row27_col3" class="data row27 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.2.html" target="_blank">Ag3.2</a></td>
      <td id="T_78415_row27_col4" class="data row27 col4" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC10432508/' target='_blank'>Lucas et al. 2023</a></td>
    </tr>
    <tr>
      <td id="T_78415_row28_col0" class="data row28 col0" >1237-VO-BJ-DJOGBENOU-VMF00067</td>
      <td id="T_78415_row28_col1" class="data row28 col1" ><a href="https://www.malariagen.net/partner_study/1237-VO-BJ-DJOGBENOU" target="_blank">1237-VO-BJ-DJOGBENOU</a></td>
      <td id="T_78415_row28_col2" class="data row28 col2" >Luc Djogbenou</td>
      <td id="T_78415_row28_col3" class="data row28 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.2.html" target="_blank">Ag3.2</a></td>
      <td id="T_78415_row28_col4" class="data row28 col4" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC10432508/' target='_blank'>Lucas et al. 2023</a></td>
    </tr>
    <tr>
      <td id="T_78415_row29_col0" class="data row29 col0" >1244-VO-GH-YAWSON-VMF00051</td>
      <td id="T_78415_row29_col1" class="data row29 col1" ><a href="https://www.malariagen.net/partner_study/1244-VO-GH-YAWSON" target="_blank">1244-VO-GH-YAWSON</a></td>
      <td id="T_78415_row29_col2" class="data row29 col2" >Alexander Egyir-Yawson</td>
      <td id="T_78415_row29_col3" class="data row29 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.2.html" target="_blank">Ag3.2</a></td>
      <td id="T_78415_row29_col4" class="data row29 col4" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC10432508/' target='_blank'>Lucas et al. 2023</a></td>
    </tr>
    <tr>
      <td id="T_78415_row30_col0" class="data row30 col0" >1245-VO-CI-CONSTANT-VMF00054</td>
      <td id="T_78415_row30_col1" class="data row30 col1" ><a href="https://www.malariagen.net/partner_study/1245-VO-CI-CONSTANT" target="_blank">1245-VO-CI-CONSTANT</a></td>
      <td id="T_78415_row30_col2" class="data row30 col2" >Edi Constant</td>
      <td id="T_78415_row30_col3" class="data row30 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.2.html" target="_blank">Ag3.2</a></td>
      <td id="T_78415_row30_col4" class="data row30 col4" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC10432508/' target='_blank'>Lucas et al. 2023</a></td>
    </tr>
    <tr>
      <td id="T_78415_row31_col0" class="data row31 col0" >1253-VO-TG-DJOGBENOU-VMF00052</td>
      <td id="T_78415_row31_col1" class="data row31 col1" ><a href="https://www.malariagen.net/partner_study/1253-VO-TG-DJOGBENOU" target="_blank">1253-VO-TG-DJOGBENOU</a></td>
      <td id="T_78415_row31_col2" class="data row31 col2" >Luc Djogbenou</td>
      <td id="T_78415_row31_col3" class="data row31 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.2.html" target="_blank">Ag3.2</a></td>
      <td id="T_78415_row31_col4" class="data row31 col4" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC10432508/' target='_blank'>Lucas et al. 2023</a></td>
    </tr>
    <tr>
      <td id="T_78415_row32_col0" class="data row32 col0" >1178-VO-UG-LAWNICZAK-VMF00025</td>
      <td id="T_78415_row32_col1" class="data row32 col1" ><a href="https://www.malariagen.net/partner_study/1178-VO-UG-LAWNICZAK" target="_blank">1178-VO-UG-LAWNICZAK</a></td>
      <td id="T_78415_row32_col2" class="data row32 col2" >Mara Lawniczak</td>
      <td id="T_78415_row32_col3" class="data row32 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.3.html" target="_blank">Ag3.3</a></td>
      <td id="T_78415_row32_col4" class="data row32 col4" ></td>
    </tr>
    <tr>
      <td id="T_78415_row33_col0" class="data row33 col0" >1244-VO-GH-YAWSON-VMF00149</td>
      <td id="T_78415_row33_col1" class="data row33 col1" ><a href="https://www.malariagen.net/partner_study/1244-VO-GH-YAWSON" target="_blank">1244-VO-GH-YAWSON</a></td>
      <td id="T_78415_row33_col2" class="data row33 col2" >Alexander Egyir-Yawson</td>
      <td id="T_78415_row33_col3" class="data row33 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.4.html" target="_blank">Ag3.4</a></td>
      <td id="T_78415_row33_col4" class="data row33 col4" >Nagi et al. in prep.</td>
    </tr>
    <tr>
      <td id="T_78415_row34_col0" class="data row34 col0" >crawford-2016</td>
      <td id="T_78415_row34_col1" class="data row34 col1" ><a href="https://doi.org/10.1111/mec.13572" target="_blank">crawford-2016</a></td>
      <td id="T_78415_row34_col2" class="data row34 col2" >Jacob E Crawford</td>
      <td id="T_78415_row34_col3" class="data row34 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.7.html" target="_blank">Ag3.7</a></td>
      <td id="T_78415_row34_col4" class="data row34 col4" ><a href='https://pubmed.ncbi.nlm.nih.gov/26846876/' target='_blank'>Crawford et al. 2016</a></td>
    </tr>
    <tr>
      <td id="T_78415_row35_col0" class="data row35 col0" >tennessen-2021</td>
      <td id="T_78415_row35_col1" class="data row35 col1" ><a href="https://doi.org/10.1111/mec.15756" target="_blank">tennessen-2021</a></td>
      <td id="T_78415_row35_col2" class="data row35 col2" >Jacob Tennessen</td>
      <td id="T_78415_row35_col3" class="data row35 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.8.html" target="_blank">Ag3.8</a></td>
      <td id="T_78415_row35_col4" class="data row35 col4" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC7858241/' target='_blank'>Tennessen et al. 2021</a></td>
    </tr>
    <tr>
      <td id="T_78415_row36_col0" class="data row36 col0" >bergey-2019</td>
      <td id="T_78415_row36_col1" class="data row36 col1" ><a href="https://doi.org/10.1111/eva.12878" target="_blank">bergey-2019</a></td>
      <td id="T_78415_row36_col2" class="data row36 col2" >Christina Bergey</td>
      <td id="T_78415_row36_col3" class="data row36 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.9.html" target="_blank">Ag3.9</a></td>
      <td id="T_78415_row36_col4" class="data row36 col4" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC6976967/' target='_blank'>Bergey et al. 2019</a></td>
    </tr>
    <tr>
      <td id="T_78415_row37_col0" class="data row37 col0" >campos-2021</td>
      <td id="T_78415_row37_col1" class="data row37 col1" ><a href="https://doi.org/10.1038/s42003-021-02168-0" target="_blank">campos-2021</a></td>
      <td id="T_78415_row37_col2" class="data row37 col2" >Greg Lanzaro</td>
      <td id="T_78415_row37_col3" class="data row37 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.9.html" target="_blank">Ag3.9</a></td>
      <td id="T_78415_row37_col4" class="data row37 col4" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC8155153/' target='_blank'>Campos et al. 2021</a>, <a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC8477601/' target='_blank'>Lanzaro et al. 2021</a></td>
    </tr>
    <tr>
      <td id="T_78415_row38_col0" class="data row38 col0" >fontaine-2015-rebuild</td>
      <td id="T_78415_row38_col1" class="data row38 col1" ><a href="https://doi.org/10.1126/science.1258524" target="_blank">fontaine-2015-rebuild</a></td>
      <td id="T_78415_row38_col2" class="data row38 col2" >Michael.C.Fontaine</td>
      <td id="T_78415_row38_col3" class="data row38 col3" ><a href="https://malariagen.github.io/vector-data/ag3/ag3.10.html" target="_blank">Ag3.10</a></td>
      <td id="T_78415_row38_col4" class="data row38 col4" ><a href='https://pmc.ncbi.nlm.nih.gov/articles/PMC4380269/' target='_blank'>Fontaine et al. 2014</a></td>
    </tr>
  </tbody>
</table>
</div></div>
</div>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "selection-atlas"
        },
        kernelOptions: {
            name: "selection-atlas",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'selection-atlas'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="cohort/UG-W_Kanungu_gamb_2012_Q4.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Uganda / Kanungu / gambiae / 2012 / Q4</p>
      </div>
    </a>
    <a class="right-next"
       href="methods.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Methods</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By <a href="https://github.com/anopheles-genomic-surveillance/selection-atlas/graphs/contributors" target="_blank">Selection Atlas contributors</a>,  <a href="https://www.sanger.ac.uk/collaboration/genomic-surveillance-unit-gsu/" target="_blank">Wellcome Sanger Institute Genomic Surveillance Unit</a>,  <a href="https://www.lstmed.ac.uk/about/people/professor-martin-james-donnelly" target="_blank">Liverpool School of Tropical Medicine Vector Informatics and Genomics Group</a>.

</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2025.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
<div class="extra_footer">
  <p xmlns:cc="http://creativecommons.org/ns#" xmlns:dct="http://purl.org/dc/terms/"><span property="dct:title">The Malaria Vector Selection Atlas</span> is licensed under <a href="https://creativecommons.org/licenses/by/4.0/?ref=chooser-v1" target="_blank" rel="license noopener noreferrer" style="display:inline-block;">CC BY 4.0<img style="height:22px!important;margin-left:3px;vertical-align:text-bottom;" src="https://mirrors.creativecommons.org/presskit/icons/cc.svg?ref=chooser-v1" alt=""><img style="height:22px!important;margin-left:3px;vertical-align:text-bottom;" src="https://mirrors.creativecommons.org/presskit/icons/by.svg?ref=chooser-v1" alt=""></a></p> <p>Citation: Sanjay Nagi, Alistair Miles et al. (2025) An atlas of recent positive selection in major African malaria vectors. In prep.</p> <p>GWSS analysis version ─ agam/2025.03.05. Site version ─ <a href='https://github.com/anopheles-genomic-surveillance/selection-atlas/releases/tag/v2025.03.11-1' target='_blank'>v2025.03.11-1</a> (11 March 2025). See also <a href='https://github.com/anopheles-genomic-surveillance/selection-atlas/releases' target='_blank'>release notes</a>.</p>

</div>
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script defer src="_static/scripts/bootstrap.js?digest=8878045cc6db502f8baf"></script>
<script defer src="_static/scripts/pydata-sphinx-theme.js?digest=8878045cc6db502f8baf"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>