{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "76d2e44c",
   "metadata": {
    "tags": [
     "parameters",
     "remove-input"
    ]
   },
   "outputs": [],
   "source": [
    "# Notebook parameters. Values here are for development only and \n",
    "# will be overridden when running via snakemake and papermill.\n",
    "#cohort_id = \"CD-NU_Gbadolite_gamb_2015_Q3\"\n",
    "cohort_id = 'ML-2_Kati_colu_2014_Q3'\n",
    "# cohort_id = 'CI-LG_Agneby-Tiassa_colu_2012'\n",
    "cohorts_analysis = \"20230223\"\n",
    "contigs = ['2RL', '3RL', 'X']\n",
    "sample_sets = \"3.0\"\n",
    "min_cohort_size = 20\n",
    "max_cohort_size = 50\n",
    "h12_calibration_contig = '3L'\n",
    "use_gcs_cache = False\n",
    "dask_scheduler = \"threads\"\n",
    "analysis_version = \"2025.01.30\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "ead60adb",
   "metadata": {
    "tags": [
     "remove-input"
    ]
   },
   "outputs": [
    {
     "data": {
      "application/javascript": [
       "(function(root) {\n",
       "  function now() {\n",
       "    return new Date();\n",
       "  }\n",
       "\n",
       "  const force = true;\n",
       "\n",
       "  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n",
       "    root._bokeh_onload_callbacks = [];\n",
       "    root._bokeh_is_loading = undefined;\n",
       "  }\n",
       "\n",
       "const JS_MIME_TYPE = 'application/javascript';\n",
       "  const HTML_MIME_TYPE = 'text/html';\n",
       "  const EXEC_MIME_TYPE = 'application/vnd.bokehjs_exec.v0+json';\n",
       "  const CLASS_NAME = 'output_bokeh rendered_html';\n",
       "\n",
       "  /**\n",
       "   * Render data to the DOM node\n",
       "   */\n",
       "  function render(props, node) {\n",
       "    const script = document.createElement(\"script\");\n",
       "    node.appendChild(script);\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when an output is cleared or removed\n",
       "   */\n",
       "  function handleClearOutput(event, handle) {\n",
       "    const cell = handle.cell;\n",
       "\n",
       "    const id = cell.output_area._bokeh_element_id;\n",
       "    const server_id = cell.output_area._bokeh_server_id;\n",
       "    // Clean up Bokeh references\n",
       "    if (id != null && id in Bokeh.index) {\n",
       "      Bokeh.index[id].model.document.clear();\n",
       "      delete Bokeh.index[id];\n",
       "    }\n",
       "\n",
       "    if (server_id !== undefined) {\n",
       "      // Clean up Bokeh references\n",
       "      const cmd_clean = \"from bokeh.io.state import curstate; print(curstate().uuid_to_server['\" + server_id + \"'].get_sessions()[0].document.roots[0]._id)\";\n",
       "      cell.notebook.kernel.execute(cmd_clean, {\n",
       "        iopub: {\n",
       "          output: function(msg) {\n",
       "            const id = msg.content.text.trim();\n",
       "            if (id in Bokeh.index) {\n",
       "              Bokeh.index[id].model.document.clear();\n",
       "              delete Bokeh.index[id];\n",
       "            }\n",
       "          }\n",
       "        }\n",
       "      });\n",
       "      // Destroy server and session\n",
       "      const cmd_destroy = \"import bokeh.io.notebook as ion; ion.destroy_server('\" + server_id + \"')\";\n",
       "      cell.notebook.kernel.execute(cmd_destroy);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when a new output is added\n",
       "   */\n",
       "  function handleAddOutput(event, handle) {\n",
       "    const output_area = handle.output_area;\n",
       "    const output = handle.output;\n",
       "\n",
       "    // limit handleAddOutput to display_data with EXEC_MIME_TYPE content only\n",
       "    if ((output.output_type != \"display_data\") || (!Object.prototype.hasOwnProperty.call(output.data, EXEC_MIME_TYPE))) {\n",
       "      return\n",
       "    }\n",
       "\n",
       "    const toinsert = output_area.element.find(\".\" + CLASS_NAME.split(' ')[0]);\n",
       "\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"id\"] !== undefined) {\n",
       "      toinsert[toinsert.length - 1].firstChild.textContent = output.data[JS_MIME_TYPE];\n",
       "      // store reference to embed id on output_area\n",
       "      output_area._bokeh_element_id = output.metadata[EXEC_MIME_TYPE][\"id\"];\n",
       "    }\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"server_id\"] !== undefined) {\n",
       "      const bk_div = document.createElement(\"div\");\n",
       "      bk_div.innerHTML = output.data[HTML_MIME_TYPE];\n",
       "      const script_attrs = bk_div.children[0].attributes;\n",
       "      for (let i = 0; i < script_attrs.length; i++) {\n",
       "        toinsert[toinsert.length - 1].firstChild.setAttribute(script_attrs[i].name, script_attrs[i].value);\n",
       "        toinsert[toinsert.length - 1].firstChild.textContent = bk_div.children[0].textContent\n",
       "      }\n",
       "      // store reference to server id on output_area\n",
       "      output_area._bokeh_server_id = output.metadata[EXEC_MIME_TYPE][\"server_id\"];\n",
       "    }\n",
       "  }\n",
       "\n",
       "  function register_renderer(events, OutputArea) {\n",
       "\n",
       "    function append_mime(data, metadata, element) {\n",
       "      // create a DOM node to render to\n",
       "      const toinsert = this.create_output_subarea(\n",
       "        metadata,\n",
       "        CLASS_NAME,\n",
       "        EXEC_MIME_TYPE\n",
       "      );\n",
       "      this.keyboard_manager.register_events(toinsert);\n",
       "      // Render to node\n",
       "      const props = {data: data, metadata: metadata[EXEC_MIME_TYPE]};\n",
       "      render(props, toinsert[toinsert.length - 1]);\n",
       "      element.append(toinsert);\n",
       "      return toinsert\n",
       "    }\n",
       "\n",
       "    /* Handle when an output is cleared or removed */\n",
       "    events.on('clear_output.CodeCell', handleClearOutput);\n",
       "    events.on('delete.Cell', handleClearOutput);\n",
       "\n",
       "    /* Handle when a new output is added */\n",
       "    events.on('output_added.OutputArea', handleAddOutput);\n",
       "\n",
       "    /**\n",
       "     * Register the mime type and append_mime function with output_area\n",
       "     */\n",
       "    OutputArea.prototype.register_mime_type(EXEC_MIME_TYPE, append_mime, {\n",
       "      /* Is output safe? */\n",
       "      safe: true,\n",
       "      /* Index of renderer in `output_area.display_order` */\n",
       "      index: 0\n",
       "    });\n",
       "  }\n",
       "\n",
       "  // register the mime type if in Jupyter Notebook environment and previously unregistered\n",
       "  if (root.Jupyter !== undefined) {\n",
       "    const events = require('base/js/events');\n",
       "    const OutputArea = require('notebook/js/outputarea').OutputArea;\n",
       "\n",
       "    if (OutputArea.prototype.mime_types().indexOf(EXEC_MIME_TYPE) == -1) {\n",
       "      register_renderer(events, OutputArea);\n",
       "    }\n",
       "  }\n",
       "  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n",
       "    root._bokeh_timeout = Date.now() + 5000;\n",
       "    root._bokeh_failed_load = false;\n",
       "  }\n",
       "\n",
       "  const NB_LOAD_WARNING = {'data': {'text/html':\n",
       "     \"<div style='background-color: #fdd'>\\n\"+\n",
       "     \"<p>\\n\"+\n",
       "     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n",
       "     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n",
       "     \"</p>\\n\"+\n",
       "     \"<ul>\\n\"+\n",
       "     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n",
       "     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n",
       "     \"</ul>\\n\"+\n",
       "     \"<code>\\n\"+\n",
       "     \"from bokeh.resources import INLINE\\n\"+\n",
       "     \"output_notebook(resources=INLINE)\\n\"+\n",
       "     \"</code>\\n\"+\n",
       "     \"</div>\"}};\n",
       "\n",
       "  function display_loaded() {\n",
       "    const el = document.getElementById(null);\n",
       "    if (el != null) {\n",
       "      el.textContent = \"BokehJS is loading...\";\n",
       "    }\n",
       "    if (root.Bokeh !== undefined) {\n",
       "      if (el != null) {\n",
       "        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n",
       "      }\n",
       "    } else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(display_loaded, 100)\n",
       "    }\n",
       "  }\n",
       "\n",
       "  function run_callbacks() {\n",
       "    try {\n",
       "      root._bokeh_onload_callbacks.forEach(function(callback) {\n",
       "        if (callback != null)\n",
       "          callback();\n",
       "      });\n",
       "    } finally {\n",
       "      delete root._bokeh_onload_callbacks\n",
       "    }\n",
       "    console.debug(\"Bokeh: all callbacks have finished\");\n",
       "  }\n",
       "\n",
       "  function load_libs(css_urls, js_urls, callback) {\n",
       "    if (css_urls == null) css_urls = [];\n",
       "    if (js_urls == null) js_urls = [];\n",
       "\n",
       "    root._bokeh_onload_callbacks.push(callback);\n",
       "    if (root._bokeh_is_loading > 0) {\n",
       "      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n",
       "      return null;\n",
       "    }\n",
       "    if (js_urls == null || js_urls.length === 0) {\n",
       "      run_callbacks();\n",
       "      return null;\n",
       "    }\n",
       "    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n",
       "    root._bokeh_is_loading = css_urls.length + js_urls.length;\n",
       "\n",
       "    function on_load() {\n",
       "      root._bokeh_is_loading--;\n",
       "      if (root._bokeh_is_loading === 0) {\n",
       "        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n",
       "        run_callbacks()\n",
       "      }\n",
       "    }\n",
       "\n",
       "    function on_error(url) {\n",
       "      console.error(\"failed to load \" + url);\n",
       "    }\n",
       "\n",
       "    for (let i = 0; i < css_urls.length; i++) {\n",
       "      const url = css_urls[i];\n",
       "      const element = document.createElement(\"link\");\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error.bind(null, url);\n",
       "      element.rel = \"stylesheet\";\n",
       "      element.type = \"text/css\";\n",
       "      element.href = url;\n",
       "      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n",
       "      document.body.appendChild(element);\n",
       "    }\n",
       "\n",
       "    for (let i = 0; i < js_urls.length; i++) {\n",
       "      const url = js_urls[i];\n",
       "      const element = document.createElement('script');\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error.bind(null, url);\n",
       "      element.async = false;\n",
       "      element.src = url;\n",
       "      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n",
       "      document.head.appendChild(element);\n",
       "    }\n",
       "  };\n",
       "\n",
       "  function inject_raw_css(css) {\n",
       "    const element = document.createElement(\"style\");\n",
       "    element.appendChild(document.createTextNode(css));\n",
       "    document.body.appendChild(element);\n",
       "  }\n",
       "\n",
       "  const js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-3.1.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-3.1.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-3.1.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-mathjax-3.1.1.min.js\"];\n",
       "  const css_urls = [];\n",
       "\n",
       "  const inline_js = [    function(Bokeh) {\n",
       "      Bokeh.set_log_level(\"info\");\n",
       "    },\n",
       "function(Bokeh) {\n",
       "    }\n",
       "  ];\n",
       "\n",
       "  function run_inline_js() {\n",
       "    if (root.Bokeh !== undefined || force === true) {\n",
       "          for (let i = 0; i < inline_js.length; i++) {\n",
       "      inline_js[i].call(root, root.Bokeh);\n",
       "    }\n",
       "} else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(run_inline_js, 100);\n",
       "    } else if (!root._bokeh_failed_load) {\n",
       "      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n",
       "      root._bokeh_failed_load = true;\n",
       "    } else if (force !== true) {\n",
       "      const cell = $(document.getElementById(null)).parents('.cell').data().cell;\n",
       "      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n",
       "    }\n",
       "  }\n",
       "\n",
       "  if (root._bokeh_is_loading === 0) {\n",
       "    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n",
       "    run_inline_js();\n",
       "  } else {\n",
       "    load_libs(css_urls, js_urls, function() {\n",
       "      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n",
       "      run_inline_js();\n",
       "    });\n",
       "  }\n",
       "}(window));"
      ],
      "application/vnd.bokehjs_load.v0+json": "(function(root) {\n  function now() {\n    return new Date();\n  }\n\n  const force = true;\n\n  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n    root._bokeh_onload_callbacks = [];\n    root._bokeh_is_loading = undefined;\n  }\n\n\n  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n    root._bokeh_timeout = Date.now() + 5000;\n    root._bokeh_failed_load = false;\n  }\n\n  const NB_LOAD_WARNING = {'data': {'text/html':\n     \"<div style='background-color: #fdd'>\\n\"+\n     \"<p>\\n\"+\n     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n     \"</p>\\n\"+\n     \"<ul>\\n\"+\n     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n     \"</ul>\\n\"+\n     \"<code>\\n\"+\n     \"from bokeh.resources import INLINE\\n\"+\n     \"output_notebook(resources=INLINE)\\n\"+\n     \"</code>\\n\"+\n     \"</div>\"}};\n\n  function display_loaded() {\n    const el = document.getElementById(null);\n    if (el != null) {\n      el.textContent = \"BokehJS is loading...\";\n    }\n    if (root.Bokeh !== undefined) {\n      if (el != null) {\n        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n      }\n    } else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(display_loaded, 100)\n    }\n  }\n\n  function run_callbacks() {\n    try {\n      root._bokeh_onload_callbacks.forEach(function(callback) {\n        if (callback != null)\n          callback();\n      });\n    } finally {\n      delete root._bokeh_onload_callbacks\n    }\n    console.debug(\"Bokeh: all callbacks have finished\");\n  }\n\n  function load_libs(css_urls, js_urls, callback) {\n    if (css_urls == null) css_urls = [];\n    if (js_urls == null) js_urls = [];\n\n    root._bokeh_onload_callbacks.push(callback);\n    if (root._bokeh_is_loading > 0) {\n      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n      return null;\n    }\n    if (js_urls == null || js_urls.length === 0) {\n      run_callbacks();\n      return null;\n    }\n    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n    root._bokeh_is_loading = css_urls.length + js_urls.length;\n\n    function on_load() {\n      root._bokeh_is_loading--;\n      if (root._bokeh_is_loading === 0) {\n        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n        run_callbacks()\n      }\n    }\n\n    function on_error(url) {\n      console.error(\"failed to load \" + url);\n    }\n\n    for (let i = 0; i < css_urls.length; i++) {\n      const url = css_urls[i];\n      const element = document.createElement(\"link\");\n      element.onload = on_load;\n      element.onerror = on_error.bind(null, url);\n      element.rel = \"stylesheet\";\n      element.type = \"text/css\";\n      element.href = url;\n      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n      document.body.appendChild(element);\n    }\n\n    for (let i = 0; i < js_urls.length; i++) {\n      const url = js_urls[i];\n      const element = document.createElement('script');\n      element.onload = on_load;\n      element.onerror = on_error.bind(null, url);\n      element.async = false;\n      element.src = url;\n      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n      document.head.appendChild(element);\n    }\n  };\n\n  function inject_raw_css(css) {\n    const element = document.createElement(\"style\");\n    element.appendChild(document.createTextNode(css));\n    document.body.appendChild(element);\n  }\n\n  const js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-3.1.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-3.1.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-3.1.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-mathjax-3.1.1.min.js\"];\n  const css_urls = [];\n\n  const inline_js = [    function(Bokeh) {\n      Bokeh.set_log_level(\"info\");\n    },\nfunction(Bokeh) {\n    }\n  ];\n\n  function run_inline_js() {\n    if (root.Bokeh !== undefined || force === true) {\n          for (let i = 0; i < inline_js.length; i++) {\n      inline_js[i].call(root, root.Bokeh);\n    }\n} else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(run_inline_js, 100);\n    } else if (!root._bokeh_failed_load) {\n      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n      root._bokeh_failed_load = true;\n    } else if (force !== true) {\n      const cell = $(document.getElementById(null)).parents('.cell').data().cell;\n      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n    }\n  }\n\n  if (root._bokeh_is_loading === 0) {\n    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n    run_inline_js();\n  } else {\n    load_libs(css_urls, js_urls, function() {\n      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n      run_inline_js();\n    });\n  }\n}(window));"
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from IPython.display import Markdown, HTML\n",
    "import malariagen_data\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "from pyprojroot import here\n",
    "import yaml\n",
    "import dask\n",
    "dask.config.set(scheduler=dask_scheduler);\n",
    "from textwrap import dedent\n",
    "\n",
    "import geopandas as gpd\n",
    "import bokeh.layouts as bklay\n",
    "import bokeh.plotting as bkplt\n",
    "import bokeh.models as bkmod\n",
    "import plotly.express as px\n",
    "\n",
    "from bokeh.io import output_notebook # enables plot interface in J notebook\n",
    "\n",
    "# N.B., do not add the \"remove-output\" tag to this cell!!! If you do,\n",
    "# the bokeh javascript libraries will not get loaded in the generated\n",
    "# HTML page. The call to output_notebook() injects javascript in the\n",
    "# cell output which triggers the bokeh javascript libraries to be loaded\n",
    "# in the page.\n",
    "output_notebook(hide_banner=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3a98f014",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "80879436",
   "metadata": {
    "tags": [
     "remove-input",
     "remove-output"
    ]
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR 1: PROJ: proj_create_from_database: Open of /home/sanjaynagi/envs/selection-atlas/share/proj failed\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "cohort_size                                                          27\n",
       "country                                                            Mali\n",
       "admin1_iso                                                         ML-2\n",
       "admin1_name                                                  Koulikouro\n",
       "admin2_name                                                        Kati\n",
       "taxon                                                          coluzzii\n",
       "year                                                               2014\n",
       "quarter                                                               3\n",
       "cohort_label                         Mali / Kati / coluzzii / 2014 / Q3\n",
       "sample_query          cohort_admin2_quarter == 'ML-2_Kati_colu_2014_...\n",
       "latitude                                                      12.875556\n",
       "longitude                                                     -8.137778\n",
       "h12_window_size                                                    1000\n",
       "country_alpha2                                                       ML\n",
       "country_alpha3                                                      MLI\n",
       "shapeName                                                          Kati\n",
       "shapeISO                                                               \n",
       "shapeID                                          8926073B54119460885487\n",
       "shapeGroup                                                          MLI\n",
       "shapeType                                                          ADM2\n",
       "representative_lon                                            -8.426097\n",
       "representative_lat                                            12.598369\n",
       "geometry              POLYGON ((-9.044757152999953 12.42636384700004...\n",
       "Name: ML-2_Kati_colu_2014_Q3, dtype: object"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Load cohorts to find sample query to select samples for this cohort.\n",
    "\n",
    "df_cohorts = gpd.read_file(here() / \"build\" / analysis_version / \"final_cohorts.geojson\").set_index(\"cohort_id\")\n",
    "cohort = df_cohorts.loc[cohort_id]\n",
    "cohort"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "9ba3b9c2-1770-40d6-abdc-1bec128ba253",
   "metadata": {
    "tags": [
     "remove-input",
     "remove-output"
    ]
   },
   "outputs": [
    {
     "data": {
      "application/javascript": [
       "(function(root) {\n",
       "  function now() {\n",
       "    return new Date();\n",
       "  }\n",
       "\n",
       "  const force = true;\n",
       "\n",
       "  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n",
       "    root._bokeh_onload_callbacks = [];\n",
       "    root._bokeh_is_loading = undefined;\n",
       "  }\n",
       "\n",
       "const JS_MIME_TYPE = 'application/javascript';\n",
       "  const HTML_MIME_TYPE = 'text/html';\n",
       "  const EXEC_MIME_TYPE = 'application/vnd.bokehjs_exec.v0+json';\n",
       "  const CLASS_NAME = 'output_bokeh rendered_html';\n",
       "\n",
       "  /**\n",
       "   * Render data to the DOM node\n",
       "   */\n",
       "  function render(props, node) {\n",
       "    const script = document.createElement(\"script\");\n",
       "    node.appendChild(script);\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when an output is cleared or removed\n",
       "   */\n",
       "  function handleClearOutput(event, handle) {\n",
       "    const cell = handle.cell;\n",
       "\n",
       "    const id = cell.output_area._bokeh_element_id;\n",
       "    const server_id = cell.output_area._bokeh_server_id;\n",
       "    // Clean up Bokeh references\n",
       "    if (id != null && id in Bokeh.index) {\n",
       "      Bokeh.index[id].model.document.clear();\n",
       "      delete Bokeh.index[id];\n",
       "    }\n",
       "\n",
       "    if (server_id !== undefined) {\n",
       "      // Clean up Bokeh references\n",
       "      const cmd_clean = \"from bokeh.io.state import curstate; print(curstate().uuid_to_server['\" + server_id + \"'].get_sessions()[0].document.roots[0]._id)\";\n",
       "      cell.notebook.kernel.execute(cmd_clean, {\n",
       "        iopub: {\n",
       "          output: function(msg) {\n",
       "            const id = msg.content.text.trim();\n",
       "            if (id in Bokeh.index) {\n",
       "              Bokeh.index[id].model.document.clear();\n",
       "              delete Bokeh.index[id];\n",
       "            }\n",
       "          }\n",
       "        }\n",
       "      });\n",
       "      // Destroy server and session\n",
       "      const cmd_destroy = \"import bokeh.io.notebook as ion; ion.destroy_server('\" + server_id + \"')\";\n",
       "      cell.notebook.kernel.execute(cmd_destroy);\n",
       "    }\n",
       "  }\n",
       "\n",
       "  /**\n",
       "   * Handle when a new output is added\n",
       "   */\n",
       "  function handleAddOutput(event, handle) {\n",
       "    const output_area = handle.output_area;\n",
       "    const output = handle.output;\n",
       "\n",
       "    // limit handleAddOutput to display_data with EXEC_MIME_TYPE content only\n",
       "    if ((output.output_type != \"display_data\") || (!Object.prototype.hasOwnProperty.call(output.data, EXEC_MIME_TYPE))) {\n",
       "      return\n",
       "    }\n",
       "\n",
       "    const toinsert = output_area.element.find(\".\" + CLASS_NAME.split(' ')[0]);\n",
       "\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"id\"] !== undefined) {\n",
       "      toinsert[toinsert.length - 1].firstChild.textContent = output.data[JS_MIME_TYPE];\n",
       "      // store reference to embed id on output_area\n",
       "      output_area._bokeh_element_id = output.metadata[EXEC_MIME_TYPE][\"id\"];\n",
       "    }\n",
       "    if (output.metadata[EXEC_MIME_TYPE][\"server_id\"] !== undefined) {\n",
       "      const bk_div = document.createElement(\"div\");\n",
       "      bk_div.innerHTML = output.data[HTML_MIME_TYPE];\n",
       "      const script_attrs = bk_div.children[0].attributes;\n",
       "      for (let i = 0; i < script_attrs.length; i++) {\n",
       "        toinsert[toinsert.length - 1].firstChild.setAttribute(script_attrs[i].name, script_attrs[i].value);\n",
       "        toinsert[toinsert.length - 1].firstChild.textContent = bk_div.children[0].textContent\n",
       "      }\n",
       "      // store reference to server id on output_area\n",
       "      output_area._bokeh_server_id = output.metadata[EXEC_MIME_TYPE][\"server_id\"];\n",
       "    }\n",
       "  }\n",
       "\n",
       "  function register_renderer(events, OutputArea) {\n",
       "\n",
       "    function append_mime(data, metadata, element) {\n",
       "      // create a DOM node to render to\n",
       "      const toinsert = this.create_output_subarea(\n",
       "        metadata,\n",
       "        CLASS_NAME,\n",
       "        EXEC_MIME_TYPE\n",
       "      );\n",
       "      this.keyboard_manager.register_events(toinsert);\n",
       "      // Render to node\n",
       "      const props = {data: data, metadata: metadata[EXEC_MIME_TYPE]};\n",
       "      render(props, toinsert[toinsert.length - 1]);\n",
       "      element.append(toinsert);\n",
       "      return toinsert\n",
       "    }\n",
       "\n",
       "    /* Handle when an output is cleared or removed */\n",
       "    events.on('clear_output.CodeCell', handleClearOutput);\n",
       "    events.on('delete.Cell', handleClearOutput);\n",
       "\n",
       "    /* Handle when a new output is added */\n",
       "    events.on('output_added.OutputArea', handleAddOutput);\n",
       "\n",
       "    /**\n",
       "     * Register the mime type and append_mime function with output_area\n",
       "     */\n",
       "    OutputArea.prototype.register_mime_type(EXEC_MIME_TYPE, append_mime, {\n",
       "      /* Is output safe? */\n",
       "      safe: true,\n",
       "      /* Index of renderer in `output_area.display_order` */\n",
       "      index: 0\n",
       "    });\n",
       "  }\n",
       "\n",
       "  // register the mime type if in Jupyter Notebook environment and previously unregistered\n",
       "  if (root.Jupyter !== undefined) {\n",
       "    const events = require('base/js/events');\n",
       "    const OutputArea = require('notebook/js/outputarea').OutputArea;\n",
       "\n",
       "    if (OutputArea.prototype.mime_types().indexOf(EXEC_MIME_TYPE) == -1) {\n",
       "      register_renderer(events, OutputArea);\n",
       "    }\n",
       "  }\n",
       "  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n",
       "    root._bokeh_timeout = Date.now() + 5000;\n",
       "    root._bokeh_failed_load = false;\n",
       "  }\n",
       "\n",
       "  const NB_LOAD_WARNING = {'data': {'text/html':\n",
       "     \"<div style='background-color: #fdd'>\\n\"+\n",
       "     \"<p>\\n\"+\n",
       "     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n",
       "     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n",
       "     \"</p>\\n\"+\n",
       "     \"<ul>\\n\"+\n",
       "     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n",
       "     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n",
       "     \"</ul>\\n\"+\n",
       "     \"<code>\\n\"+\n",
       "     \"from bokeh.resources import INLINE\\n\"+\n",
       "     \"output_notebook(resources=INLINE)\\n\"+\n",
       "     \"</code>\\n\"+\n",
       "     \"</div>\"}};\n",
       "\n",
       "  function display_loaded() {\n",
       "    const el = document.getElementById(null);\n",
       "    if (el != null) {\n",
       "      el.textContent = \"BokehJS is loading...\";\n",
       "    }\n",
       "    if (root.Bokeh !== undefined) {\n",
       "      if (el != null) {\n",
       "        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n",
       "      }\n",
       "    } else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(display_loaded, 100)\n",
       "    }\n",
       "  }\n",
       "\n",
       "  function run_callbacks() {\n",
       "    try {\n",
       "      root._bokeh_onload_callbacks.forEach(function(callback) {\n",
       "        if (callback != null)\n",
       "          callback();\n",
       "      });\n",
       "    } finally {\n",
       "      delete root._bokeh_onload_callbacks\n",
       "    }\n",
       "    console.debug(\"Bokeh: all callbacks have finished\");\n",
       "  }\n",
       "\n",
       "  function load_libs(css_urls, js_urls, callback) {\n",
       "    if (css_urls == null) css_urls = [];\n",
       "    if (js_urls == null) js_urls = [];\n",
       "\n",
       "    root._bokeh_onload_callbacks.push(callback);\n",
       "    if (root._bokeh_is_loading > 0) {\n",
       "      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n",
       "      return null;\n",
       "    }\n",
       "    if (js_urls == null || js_urls.length === 0) {\n",
       "      run_callbacks();\n",
       "      return null;\n",
       "    }\n",
       "    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n",
       "    root._bokeh_is_loading = css_urls.length + js_urls.length;\n",
       "\n",
       "    function on_load() {\n",
       "      root._bokeh_is_loading--;\n",
       "      if (root._bokeh_is_loading === 0) {\n",
       "        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n",
       "        run_callbacks()\n",
       "      }\n",
       "    }\n",
       "\n",
       "    function on_error(url) {\n",
       "      console.error(\"failed to load \" + url);\n",
       "    }\n",
       "\n",
       "    for (let i = 0; i < css_urls.length; i++) {\n",
       "      const url = css_urls[i];\n",
       "      const element = document.createElement(\"link\");\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error.bind(null, url);\n",
       "      element.rel = \"stylesheet\";\n",
       "      element.type = \"text/css\";\n",
       "      element.href = url;\n",
       "      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n",
       "      document.body.appendChild(element);\n",
       "    }\n",
       "\n",
       "    for (let i = 0; i < js_urls.length; i++) {\n",
       "      const url = js_urls[i];\n",
       "      const element = document.createElement('script');\n",
       "      element.onload = on_load;\n",
       "      element.onerror = on_error.bind(null, url);\n",
       "      element.async = false;\n",
       "      element.src = url;\n",
       "      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n",
       "      document.head.appendChild(element);\n",
       "    }\n",
       "  };\n",
       "\n",
       "  function inject_raw_css(css) {\n",
       "    const element = document.createElement(\"style\");\n",
       "    element.appendChild(document.createTextNode(css));\n",
       "    document.body.appendChild(element);\n",
       "  }\n",
       "\n",
       "  const js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-3.1.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-3.1.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-3.1.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-mathjax-3.1.1.min.js\"];\n",
       "  const css_urls = [];\n",
       "\n",
       "  const inline_js = [    function(Bokeh) {\n",
       "      Bokeh.set_log_level(\"info\");\n",
       "    },\n",
       "function(Bokeh) {\n",
       "    }\n",
       "  ];\n",
       "\n",
       "  function run_inline_js() {\n",
       "    if (root.Bokeh !== undefined || force === true) {\n",
       "          for (let i = 0; i < inline_js.length; i++) {\n",
       "      inline_js[i].call(root, root.Bokeh);\n",
       "    }\n",
       "} else if (Date.now() < root._bokeh_timeout) {\n",
       "      setTimeout(run_inline_js, 100);\n",
       "    } else if (!root._bokeh_failed_load) {\n",
       "      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n",
       "      root._bokeh_failed_load = true;\n",
       "    } else if (force !== true) {\n",
       "      const cell = $(document.getElementById(null)).parents('.cell').data().cell;\n",
       "      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n",
       "    }\n",
       "  }\n",
       "\n",
       "  if (root._bokeh_is_loading === 0) {\n",
       "    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n",
       "    run_inline_js();\n",
       "  } else {\n",
       "    load_libs(css_urls, js_urls, function() {\n",
       "      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n",
       "      run_inline_js();\n",
       "    });\n",
       "  }\n",
       "}(window));"
      ],
      "application/vnd.bokehjs_load.v0+json": "(function(root) {\n  function now() {\n    return new Date();\n  }\n\n  const force = true;\n\n  if (typeof root._bokeh_onload_callbacks === \"undefined\" || force === true) {\n    root._bokeh_onload_callbacks = [];\n    root._bokeh_is_loading = undefined;\n  }\n\n\n  if (typeof (root._bokeh_timeout) === \"undefined\" || force === true) {\n    root._bokeh_timeout = Date.now() + 5000;\n    root._bokeh_failed_load = false;\n  }\n\n  const NB_LOAD_WARNING = {'data': {'text/html':\n     \"<div style='background-color: #fdd'>\\n\"+\n     \"<p>\\n\"+\n     \"BokehJS does not appear to have successfully loaded. If loading BokehJS from CDN, this \\n\"+\n     \"may be due to a slow or bad network connection. Possible fixes:\\n\"+\n     \"</p>\\n\"+\n     \"<ul>\\n\"+\n     \"<li>re-rerun `output_notebook()` to attempt to load from CDN again, or</li>\\n\"+\n     \"<li>use INLINE resources instead, as so:</li>\\n\"+\n     \"</ul>\\n\"+\n     \"<code>\\n\"+\n     \"from bokeh.resources import INLINE\\n\"+\n     \"output_notebook(resources=INLINE)\\n\"+\n     \"</code>\\n\"+\n     \"</div>\"}};\n\n  function display_loaded() {\n    const el = document.getElementById(null);\n    if (el != null) {\n      el.textContent = \"BokehJS is loading...\";\n    }\n    if (root.Bokeh !== undefined) {\n      if (el != null) {\n        el.textContent = \"BokehJS \" + root.Bokeh.version + \" successfully loaded.\";\n      }\n    } else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(display_loaded, 100)\n    }\n  }\n\n  function run_callbacks() {\n    try {\n      root._bokeh_onload_callbacks.forEach(function(callback) {\n        if (callback != null)\n          callback();\n      });\n    } finally {\n      delete root._bokeh_onload_callbacks\n    }\n    console.debug(\"Bokeh: all callbacks have finished\");\n  }\n\n  function load_libs(css_urls, js_urls, callback) {\n    if (css_urls == null) css_urls = [];\n    if (js_urls == null) js_urls = [];\n\n    root._bokeh_onload_callbacks.push(callback);\n    if (root._bokeh_is_loading > 0) {\n      console.debug(\"Bokeh: BokehJS is being loaded, scheduling callback at\", now());\n      return null;\n    }\n    if (js_urls == null || js_urls.length === 0) {\n      run_callbacks();\n      return null;\n    }\n    console.debug(\"Bokeh: BokehJS not loaded, scheduling load and callback at\", now());\n    root._bokeh_is_loading = css_urls.length + js_urls.length;\n\n    function on_load() {\n      root._bokeh_is_loading--;\n      if (root._bokeh_is_loading === 0) {\n        console.debug(\"Bokeh: all BokehJS libraries/stylesheets loaded\");\n        run_callbacks()\n      }\n    }\n\n    function on_error(url) {\n      console.error(\"failed to load \" + url);\n    }\n\n    for (let i = 0; i < css_urls.length; i++) {\n      const url = css_urls[i];\n      const element = document.createElement(\"link\");\n      element.onload = on_load;\n      element.onerror = on_error.bind(null, url);\n      element.rel = \"stylesheet\";\n      element.type = \"text/css\";\n      element.href = url;\n      console.debug(\"Bokeh: injecting link tag for BokehJS stylesheet: \", url);\n      document.body.appendChild(element);\n    }\n\n    for (let i = 0; i < js_urls.length; i++) {\n      const url = js_urls[i];\n      const element = document.createElement('script');\n      element.onload = on_load;\n      element.onerror = on_error.bind(null, url);\n      element.async = false;\n      element.src = url;\n      console.debug(\"Bokeh: injecting script tag for BokehJS library: \", url);\n      document.head.appendChild(element);\n    }\n  };\n\n  function inject_raw_css(css) {\n    const element = document.createElement(\"style\");\n    element.appendChild(document.createTextNode(css));\n    document.body.appendChild(element);\n  }\n\n  const js_urls = [\"https://cdn.bokeh.org/bokeh/release/bokeh-3.1.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-gl-3.1.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-widgets-3.1.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-tables-3.1.1.min.js\", \"https://cdn.bokeh.org/bokeh/release/bokeh-mathjax-3.1.1.min.js\"];\n  const css_urls = [];\n\n  const inline_js = [    function(Bokeh) {\n      Bokeh.set_log_level(\"info\");\n    },\nfunction(Bokeh) {\n    }\n  ];\n\n  function run_inline_js() {\n    if (root.Bokeh !== undefined || force === true) {\n          for (let i = 0; i < inline_js.length; i++) {\n      inline_js[i].call(root, root.Bokeh);\n    }\n} else if (Date.now() < root._bokeh_timeout) {\n      setTimeout(run_inline_js, 100);\n    } else if (!root._bokeh_failed_load) {\n      console.log(\"Bokeh: BokehJS failed to load within specified timeout.\");\n      root._bokeh_failed_load = true;\n    } else if (force !== true) {\n      const cell = $(document.getElementById(null)).parents('.cell').data().cell;\n      cell.output_area.append_execute_result(NB_LOAD_WARNING)\n    }\n  }\n\n  if (root._bokeh_is_loading === 0) {\n    console.debug(\"Bokeh: BokehJS loaded, going straight to plotting\");\n    run_inline_js();\n  } else {\n    load_libs(css_urls, js_urls, function() {\n      console.debug(\"Bokeh: BokehJS plotting callback run at\", now());\n      run_inline_js();\n    });\n  }\n}(window));"
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "\n",
       "            <table class=\"malariagen-ag3\">\n",
       "                <thead>\n",
       "                    <tr>\n",
       "                        <th style=\"text-align: left\" colspan=\"2\">MalariaGEN Ag3 API client</th>\n",
       "                    </tr>\n",
       "                    <tr><td colspan=\"2\" style=\"text-align: left\">\n",
       "                        Please note that data are subject to terms of use,\n",
       "                        for more information see <a href=\"https://www.malariagen.net/data\">\n",
       "                        the MalariaGEN website</a> or contact support@malariagen.net.\n",
       "                        See also the <a href=\"https://malariagen.github.io/malariagen-data-python/v14.0.0/Ag3.html\">Ag3 API docs</a>.\n",
       "                    </td></tr>\n",
       "                </thead>\n",
       "                <tbody>\n",
       "                    <tr>\n",
       "                        <th style=\"text-align: left\">\n",
       "                            Storage URL\n",
       "                        </th>\n",
       "                        <td>gs://vo_agam_release_master_us_central1</td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th style=\"text-align: left\">\n",
       "                            Data releases available\n",
       "                        </th>\n",
       "                        <td>3.0, 3.1, 3.2, 3.3, 3.4, 3.5, 3.6, 3.7, 3.8, 3.9, 3.10, 3.11, 3.12, 3.13</td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th style=\"text-align: left\">\n",
       "                            Results cache\n",
       "                        </th>\n",
       "                        <td>/home/sanjaynagi/selection-atlas/build/malariagen_data_cache</td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th style=\"text-align: left\">\n",
       "                            Cohorts analysis\n",
       "                        </th>\n",
       "                        <td>20230223</td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th style=\"text-align: left\">\n",
       "                            AIM analysis\n",
       "                        </th>\n",
       "                        <td>20220528</td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th style=\"text-align: left\">\n",
       "                            Site filters analysis\n",
       "                        </th>\n",
       "                        <td>dt_20200416</td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th style=\"text-align: left\">\n",
       "                            Software version\n",
       "                        </th>\n",
       "                        <td>malariagen_data 14.0.0</td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th style=\"text-align: left\">\n",
       "                            Client location\n",
       "                        </th>\n",
       "                        <td>Iowa, United States (Google Cloud us-central1)</td>\n",
       "                    </tr>\n",
       "                </tbody>\n",
       "            </table>\n",
       "        "
      ],
      "text/plain": [
       "<MalariaGEN Ag3 API client>\n",
       "Storage URL             : gs://vo_agam_release_master_us_central1\n",
       "Data releases available : 3.0, 3.1, 3.2, 3.3, 3.4, 3.5, 3.6, 3.7, 3.8, 3.9, 3.10, 3.11, 3.12, 3.13\n",
       "Results cache           : /home/sanjaynagi/selection-atlas/build/malariagen_data_cache\n",
       "Cohorts analysis        : 20230223\n",
       "AIM analysis            : 20220528\n",
       "Site filters analysis   : dt_20200416\n",
       "Software version        : malariagen_data 14.0.0\n",
       "Client location         : Iowa, United States (Google Cloud us-central1)\n",
       "---\n",
       "Please note that data are subject to terms of use,\n",
       "for more information see https://www.malariagen.net/data\n",
       "or contact support@malariagen.net. For API documentation see \n",
       "https://malariagen.github.io/malariagen-data-python/v14.0.0/Ag3.html"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Initialise access to MalariaGEN data.\n",
    "\n",
    "extra_params = dict()\n",
    "if use_gcs_cache:\n",
    "    extra_params[\"url\"] = \"simplecache::gs://vo_agam_release\"\n",
    "    extra_params[\"simplecache\"] = dict(cache_storage=(here() / \"gcs_cache\").as_posix())\n",
    "\n",
    "ag3 = malariagen_data.Ag3(\n",
    "    # Pin the version of the cohorts analysis for reproducibility.\n",
    "    cohorts_analysis=cohorts_analysis,\n",
    "    results_cache=(here() / \"build\" / \"malariagen_data_cache\").as_posix(),\n",
    "    **extra_params,\n",
    ")\n",
    "ag3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "4f6eba43-18e2-4185-95f1-c745d6092559",
   "metadata": {
    "tags": [
     "remove-input",
     "remove-output"
    ]
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                     \r"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_id</th>\n",
       "      <th>partner_sample_id</th>\n",
       "      <th>contributor</th>\n",
       "      <th>country</th>\n",
       "      <th>location</th>\n",
       "      <th>year</th>\n",
       "      <th>month</th>\n",
       "      <th>latitude</th>\n",
       "      <th>longitude</th>\n",
       "      <th>sex_call</th>\n",
       "      <th>...</th>\n",
       "      <th>admin1_name</th>\n",
       "      <th>admin1_iso</th>\n",
       "      <th>admin2_name</th>\n",
       "      <th>taxon</th>\n",
       "      <th>cohort_admin1_year</th>\n",
       "      <th>cohort_admin1_month</th>\n",
       "      <th>cohort_admin1_quarter</th>\n",
       "      <th>cohort_admin2_year</th>\n",
       "      <th>cohort_admin2_month</th>\n",
       "      <th>cohort_admin2_quarter</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>VBS25302-5561STDY9810072</td>\n",
       "      <td>MA11-38</td>\n",
       "      <td>Austin Burt</td>\n",
       "      <td>Mali</td>\n",
       "      <td>Tieneguebougou</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>12.81</td>\n",
       "      <td>-8.08</td>\n",
       "      <td>F</td>\n",
       "      <td>...</td>\n",
       "      <td>Koulikouro</td>\n",
       "      <td>ML-2</td>\n",
       "      <td>Kati</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>ML-2_colu_2014</td>\n",
       "      <td>ML-2_colu_2014_08</td>\n",
       "      <td>ML-2_colu_2014_Q3</td>\n",
       "      <td>ML-2_Kati_colu_2014</td>\n",
       "      <td>ML-2_Kati_colu_2014_08</td>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>VBS25303-5561STDY9810073</td>\n",
       "      <td>MA11-39</td>\n",
       "      <td>Austin Burt</td>\n",
       "      <td>Mali</td>\n",
       "      <td>Tieneguebougou</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>12.81</td>\n",
       "      <td>-8.08</td>\n",
       "      <td>F</td>\n",
       "      <td>...</td>\n",
       "      <td>Koulikouro</td>\n",
       "      <td>ML-2</td>\n",
       "      <td>Kati</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>ML-2_colu_2014</td>\n",
       "      <td>ML-2_colu_2014_08</td>\n",
       "      <td>ML-2_colu_2014_Q3</td>\n",
       "      <td>ML-2_Kati_colu_2014</td>\n",
       "      <td>ML-2_Kati_colu_2014_08</td>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>VBS25304-5561STDY9810074</td>\n",
       "      <td>MA11-40</td>\n",
       "      <td>Austin Burt</td>\n",
       "      <td>Mali</td>\n",
       "      <td>Tieneguebougou</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>12.81</td>\n",
       "      <td>-8.08</td>\n",
       "      <td>F</td>\n",
       "      <td>...</td>\n",
       "      <td>Koulikouro</td>\n",
       "      <td>ML-2</td>\n",
       "      <td>Kati</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>ML-2_colu_2014</td>\n",
       "      <td>ML-2_colu_2014_08</td>\n",
       "      <td>ML-2_colu_2014_Q3</td>\n",
       "      <td>ML-2_Kati_colu_2014</td>\n",
       "      <td>ML-2_Kati_colu_2014_08</td>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>VBS25307-5561STDY9810077</td>\n",
       "      <td>MA11-43</td>\n",
       "      <td>Austin Burt</td>\n",
       "      <td>Mali</td>\n",
       "      <td>Tieneguebougou</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>12.81</td>\n",
       "      <td>-8.08</td>\n",
       "      <td>F</td>\n",
       "      <td>...</td>\n",
       "      <td>Koulikouro</td>\n",
       "      <td>ML-2</td>\n",
       "      <td>Kati</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>ML-2_colu_2014</td>\n",
       "      <td>ML-2_colu_2014_08</td>\n",
       "      <td>ML-2_colu_2014_Q3</td>\n",
       "      <td>ML-2_Kati_colu_2014</td>\n",
       "      <td>ML-2_Kati_colu_2014_08</td>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>VBS25308-5561STDY9810078</td>\n",
       "      <td>MA11-44</td>\n",
       "      <td>Austin Burt</td>\n",
       "      <td>Mali</td>\n",
       "      <td>Tieneguebougou</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>12.81</td>\n",
       "      <td>-8.08</td>\n",
       "      <td>F</td>\n",
       "      <td>...</td>\n",
       "      <td>Koulikouro</td>\n",
       "      <td>ML-2</td>\n",
       "      <td>Kati</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>ML-2_colu_2014</td>\n",
       "      <td>ML-2_colu_2014_08</td>\n",
       "      <td>ML-2_colu_2014_Q3</td>\n",
       "      <td>ML-2_Kati_colu_2014</td>\n",
       "      <td>ML-2_Kati_colu_2014_08</td>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>VBS25309-5561STDY9810079</td>\n",
       "      <td>MA11-45</td>\n",
       "      <td>Austin Burt</td>\n",
       "      <td>Mali</td>\n",
       "      <td>Tieneguebougou</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>12.81</td>\n",
       "      <td>-8.08</td>\n",
       "      <td>F</td>\n",
       "      <td>...</td>\n",
       "      <td>Koulikouro</td>\n",
       "      <td>ML-2</td>\n",
       "      <td>Kati</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>ML-2_colu_2014</td>\n",
       "      <td>ML-2_colu_2014_08</td>\n",
       "      <td>ML-2_colu_2014_Q3</td>\n",
       "      <td>ML-2_Kati_colu_2014</td>\n",
       "      <td>ML-2_Kati_colu_2014_08</td>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>VBS25311-5561STDY9810081</td>\n",
       "      <td>MA11-47</td>\n",
       "      <td>Austin Burt</td>\n",
       "      <td>Mali</td>\n",
       "      <td>Tieneguebougou</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>12.81</td>\n",
       "      <td>-8.08</td>\n",
       "      <td>F</td>\n",
       "      <td>...</td>\n",
       "      <td>Koulikouro</td>\n",
       "      <td>ML-2</td>\n",
       "      <td>Kati</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>ML-2_colu_2014</td>\n",
       "      <td>ML-2_colu_2014_08</td>\n",
       "      <td>ML-2_colu_2014_Q3</td>\n",
       "      <td>ML-2_Kati_colu_2014</td>\n",
       "      <td>ML-2_Kati_colu_2014_08</td>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>AZ0292-C</td>\n",
       "      <td>MA7-36</td>\n",
       "      <td>Austin Burt</td>\n",
       "      <td>Mali</td>\n",
       "      <td>Kababougou</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>12.89</td>\n",
       "      <td>-8.15</td>\n",
       "      <td>F</td>\n",
       "      <td>...</td>\n",
       "      <td>Koulikouro</td>\n",
       "      <td>ML-2</td>\n",
       "      <td>Kati</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>ML-2_colu_2014</td>\n",
       "      <td>ML-2_colu_2014_08</td>\n",
       "      <td>ML-2_colu_2014_Q3</td>\n",
       "      <td>ML-2_Kati_colu_2014</td>\n",
       "      <td>ML-2_Kati_colu_2014_08</td>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>AZ0293-C</td>\n",
       "      <td>MA7-37</td>\n",
       "      <td>Austin Burt</td>\n",
       "      <td>Mali</td>\n",
       "      <td>Kababougou</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>12.89</td>\n",
       "      <td>-8.15</td>\n",
       "      <td>F</td>\n",
       "      <td>...</td>\n",
       "      <td>Koulikouro</td>\n",
       "      <td>ML-2</td>\n",
       "      <td>Kati</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>ML-2_colu_2014</td>\n",
       "      <td>ML-2_colu_2014_08</td>\n",
       "      <td>ML-2_colu_2014_Q3</td>\n",
       "      <td>ML-2_Kati_colu_2014</td>\n",
       "      <td>ML-2_Kati_colu_2014_08</td>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>AZ0294-C</td>\n",
       "      <td>MA7-4</td>\n",
       "      <td>Austin Burt</td>\n",
       "      <td>Mali</td>\n",
       "      <td>Kababougou</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>12.89</td>\n",
       "      <td>-8.15</td>\n",
       "      <td>F</td>\n",
       "      <td>...</td>\n",
       "      <td>Koulikouro</td>\n",
       "      <td>ML-2</td>\n",
       "      <td>Kati</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>ML-2_colu_2014</td>\n",
       "      <td>ML-2_colu_2014_08</td>\n",
       "      <td>ML-2_colu_2014_Q3</td>\n",
       "      <td>ML-2_Kati_colu_2014</td>\n",
       "      <td>ML-2_Kati_colu_2014_08</td>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>AZ0295-C</td>\n",
       "      <td>MA7-41</td>\n",
       "      <td>Austin Burt</td>\n",
       "      <td>Mali</td>\n",
       "      <td>Kababougou</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>12.89</td>\n",
       "      <td>-8.15</td>\n",
       "      <td>F</td>\n",
       "      <td>...</td>\n",
       "      <td>Koulikouro</td>\n",
       "      <td>ML-2</td>\n",
       "      <td>Kati</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>ML-2_colu_2014</td>\n",
       "      <td>ML-2_colu_2014_08</td>\n",
       "      <td>ML-2_colu_2014_Q3</td>\n",
       "      <td>ML-2_Kati_colu_2014</td>\n",
       "      <td>ML-2_Kati_colu_2014_08</td>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>AZ0296-C</td>\n",
       "      <td>MA7-44</td>\n",
       "      <td>Austin Burt</td>\n",
       "      <td>Mali</td>\n",
       "      <td>Kababougou</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>12.89</td>\n",
       "      <td>-8.15</td>\n",
       "      <td>F</td>\n",
       "      <td>...</td>\n",
       "      <td>Koulikouro</td>\n",
       "      <td>ML-2</td>\n",
       "      <td>Kati</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>ML-2_colu_2014</td>\n",
       "      <td>ML-2_colu_2014_08</td>\n",
       "      <td>ML-2_colu_2014_Q3</td>\n",
       "      <td>ML-2_Kati_colu_2014</td>\n",
       "      <td>ML-2_Kati_colu_2014_08</td>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>AZ0297-C</td>\n",
       "      <td>MA7-7</td>\n",
       "      <td>Austin Burt</td>\n",
       "      <td>Mali</td>\n",
       "      <td>Kababougou</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>12.89</td>\n",
       "      <td>-8.15</td>\n",
       "      <td>F</td>\n",
       "      <td>...</td>\n",
       "      <td>Koulikouro</td>\n",
       "      <td>ML-2</td>\n",
       "      <td>Kati</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>ML-2_colu_2014</td>\n",
       "      <td>ML-2_colu_2014_08</td>\n",
       "      <td>ML-2_colu_2014_Q3</td>\n",
       "      <td>ML-2_Kati_colu_2014</td>\n",
       "      <td>ML-2_Kati_colu_2014_08</td>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>AZ0298-C</td>\n",
       "      <td>MA7-8</td>\n",
       "      <td>Austin Burt</td>\n",
       "      <td>Mali</td>\n",
       "      <td>Kababougou</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>12.89</td>\n",
       "      <td>-8.15</td>\n",
       "      <td>F</td>\n",
       "      <td>...</td>\n",
       "      <td>Koulikouro</td>\n",
       "      <td>ML-2</td>\n",
       "      <td>Kati</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>ML-2_colu_2014</td>\n",
       "      <td>ML-2_colu_2014_08</td>\n",
       "      <td>ML-2_colu_2014_Q3</td>\n",
       "      <td>ML-2_Kati_colu_2014</td>\n",
       "      <td>ML-2_Kati_colu_2014_08</td>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>AZ0332-C</td>\n",
       "      <td>MA12-12</td>\n",
       "      <td>Austin Burt</td>\n",
       "      <td>Mali</td>\n",
       "      <td>Ouassorola</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>12.90</td>\n",
       "      <td>-8.16</td>\n",
       "      <td>F</td>\n",
       "      <td>...</td>\n",
       "      <td>Koulikouro</td>\n",
       "      <td>ML-2</td>\n",
       "      <td>Kati</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>ML-2_colu_2014</td>\n",
       "      <td>ML-2_colu_2014_08</td>\n",
       "      <td>ML-2_colu_2014_Q3</td>\n",
       "      <td>ML-2_Kati_colu_2014</td>\n",
       "      <td>ML-2_Kati_colu_2014_08</td>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>AZ0333-C</td>\n",
       "      <td>MA12-8</td>\n",
       "      <td>Austin Burt</td>\n",
       "      <td>Mali</td>\n",
       "      <td>Ouassorola</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>12.90</td>\n",
       "      <td>-8.16</td>\n",
       "      <td>F</td>\n",
       "      <td>...</td>\n",
       "      <td>Koulikouro</td>\n",
       "      <td>ML-2</td>\n",
       "      <td>Kati</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>ML-2_colu_2014</td>\n",
       "      <td>ML-2_colu_2014_08</td>\n",
       "      <td>ML-2_colu_2014_Q3</td>\n",
       "      <td>ML-2_Kati_colu_2014</td>\n",
       "      <td>ML-2_Kati_colu_2014_08</td>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>AZ0284-C</td>\n",
       "      <td>MA7-1</td>\n",
       "      <td>Austin Burt</td>\n",
       "      <td>Mali</td>\n",
       "      <td>Kababougou</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>12.89</td>\n",
       "      <td>-8.15</td>\n",
       "      <td>F</td>\n",
       "      <td>...</td>\n",
       "      <td>Koulikouro</td>\n",
       "      <td>ML-2</td>\n",
       "      <td>Kati</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>ML-2_colu_2014</td>\n",
       "      <td>ML-2_colu_2014_08</td>\n",
       "      <td>ML-2_colu_2014_Q3</td>\n",
       "      <td>ML-2_Kati_colu_2014</td>\n",
       "      <td>ML-2_Kati_colu_2014_08</td>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>AZ0287-C</td>\n",
       "      <td>MA7-17</td>\n",
       "      <td>Austin Burt</td>\n",
       "      <td>Mali</td>\n",
       "      <td>Kababougou</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>12.89</td>\n",
       "      <td>-8.15</td>\n",
       "      <td>F</td>\n",
       "      <td>...</td>\n",
       "      <td>Koulikouro</td>\n",
       "      <td>ML-2</td>\n",
       "      <td>Kati</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>ML-2_colu_2014</td>\n",
       "      <td>ML-2_colu_2014_08</td>\n",
       "      <td>ML-2_colu_2014_Q3</td>\n",
       "      <td>ML-2_Kati_colu_2014</td>\n",
       "      <td>ML-2_Kati_colu_2014_08</td>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>AZ0289-C</td>\n",
       "      <td>MA7-27</td>\n",
       "      <td>Austin Burt</td>\n",
       "      <td>Mali</td>\n",
       "      <td>Kababougou</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>12.89</td>\n",
       "      <td>-8.15</td>\n",
       "      <td>F</td>\n",
       "      <td>...</td>\n",
       "      <td>Koulikouro</td>\n",
       "      <td>ML-2</td>\n",
       "      <td>Kati</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>ML-2_colu_2014</td>\n",
       "      <td>ML-2_colu_2014_08</td>\n",
       "      <td>ML-2_colu_2014_Q3</td>\n",
       "      <td>ML-2_Kati_colu_2014</td>\n",
       "      <td>ML-2_Kati_colu_2014_08</td>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>AZ0290-C</td>\n",
       "      <td>MA7-28</td>\n",
       "      <td>Austin Burt</td>\n",
       "      <td>Mali</td>\n",
       "      <td>Kababougou</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>12.89</td>\n",
       "      <td>-8.15</td>\n",
       "      <td>F</td>\n",
       "      <td>...</td>\n",
       "      <td>Koulikouro</td>\n",
       "      <td>ML-2</td>\n",
       "      <td>Kati</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>ML-2_colu_2014</td>\n",
       "      <td>ML-2_colu_2014_08</td>\n",
       "      <td>ML-2_colu_2014_Q3</td>\n",
       "      <td>ML-2_Kati_colu_2014</td>\n",
       "      <td>ML-2_Kati_colu_2014_08</td>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>AZ0291-C</td>\n",
       "      <td>MA7-30</td>\n",
       "      <td>Austin Burt</td>\n",
       "      <td>Mali</td>\n",
       "      <td>Kababougou</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>12.89</td>\n",
       "      <td>-8.15</td>\n",
       "      <td>F</td>\n",
       "      <td>...</td>\n",
       "      <td>Koulikouro</td>\n",
       "      <td>ML-2</td>\n",
       "      <td>Kati</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>ML-2_colu_2014</td>\n",
       "      <td>ML-2_colu_2014_08</td>\n",
       "      <td>ML-2_colu_2014_Q3</td>\n",
       "      <td>ML-2_Kati_colu_2014</td>\n",
       "      <td>ML-2_Kati_colu_2014_08</td>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>AZ0334-C</td>\n",
       "      <td>MA13-1</td>\n",
       "      <td>Austin Burt</td>\n",
       "      <td>Mali</td>\n",
       "      <td>Ouassorola</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>12.90</td>\n",
       "      <td>-8.16</td>\n",
       "      <td>F</td>\n",
       "      <td>...</td>\n",
       "      <td>Koulikouro</td>\n",
       "      <td>ML-2</td>\n",
       "      <td>Kati</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>ML-2_colu_2014</td>\n",
       "      <td>ML-2_colu_2014_08</td>\n",
       "      <td>ML-2_colu_2014_Q3</td>\n",
       "      <td>ML-2_Kati_colu_2014</td>\n",
       "      <td>ML-2_Kati_colu_2014_08</td>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>AZ0339-C</td>\n",
       "      <td>MA13-29</td>\n",
       "      <td>Austin Burt</td>\n",
       "      <td>Mali</td>\n",
       "      <td>Ouassorola</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>12.90</td>\n",
       "      <td>-8.16</td>\n",
       "      <td>F</td>\n",
       "      <td>...</td>\n",
       "      <td>Koulikouro</td>\n",
       "      <td>ML-2</td>\n",
       "      <td>Kati</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>ML-2_colu_2014</td>\n",
       "      <td>ML-2_colu_2014_08</td>\n",
       "      <td>ML-2_colu_2014_Q3</td>\n",
       "      <td>ML-2_Kati_colu_2014</td>\n",
       "      <td>ML-2_Kati_colu_2014_08</td>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>AZ0340-C</td>\n",
       "      <td>MA13-33</td>\n",
       "      <td>Austin Burt</td>\n",
       "      <td>Mali</td>\n",
       "      <td>Ouassorola</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>12.90</td>\n",
       "      <td>-8.16</td>\n",
       "      <td>F</td>\n",
       "      <td>...</td>\n",
       "      <td>Koulikouro</td>\n",
       "      <td>ML-2</td>\n",
       "      <td>Kati</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>ML-2_colu_2014</td>\n",
       "      <td>ML-2_colu_2014_08</td>\n",
       "      <td>ML-2_colu_2014_Q3</td>\n",
       "      <td>ML-2_Kati_colu_2014</td>\n",
       "      <td>ML-2_Kati_colu_2014_08</td>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>AZ0341-C</td>\n",
       "      <td>MA13-39</td>\n",
       "      <td>Austin Burt</td>\n",
       "      <td>Mali</td>\n",
       "      <td>Ouassorola</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>12.90</td>\n",
       "      <td>-8.16</td>\n",
       "      <td>F</td>\n",
       "      <td>...</td>\n",
       "      <td>Koulikouro</td>\n",
       "      <td>ML-2</td>\n",
       "      <td>Kati</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>ML-2_colu_2014</td>\n",
       "      <td>ML-2_colu_2014_08</td>\n",
       "      <td>ML-2_colu_2014_Q3</td>\n",
       "      <td>ML-2_Kati_colu_2014</td>\n",
       "      <td>ML-2_Kati_colu_2014_08</td>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>AZ0342-C</td>\n",
       "      <td>MA13-43</td>\n",
       "      <td>Austin Burt</td>\n",
       "      <td>Mali</td>\n",
       "      <td>Ouassorola</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>12.90</td>\n",
       "      <td>-8.16</td>\n",
       "      <td>F</td>\n",
       "      <td>...</td>\n",
       "      <td>Koulikouro</td>\n",
       "      <td>ML-2</td>\n",
       "      <td>Kati</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>ML-2_colu_2014</td>\n",
       "      <td>ML-2_colu_2014_08</td>\n",
       "      <td>ML-2_colu_2014_Q3</td>\n",
       "      <td>ML-2_Kati_colu_2014</td>\n",
       "      <td>ML-2_Kati_colu_2014_08</td>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>AZ0343-C</td>\n",
       "      <td>MA13-45</td>\n",
       "      <td>Austin Burt</td>\n",
       "      <td>Mali</td>\n",
       "      <td>Ouassorola</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>12.90</td>\n",
       "      <td>-8.16</td>\n",
       "      <td>F</td>\n",
       "      <td>...</td>\n",
       "      <td>Koulikouro</td>\n",
       "      <td>ML-2</td>\n",
       "      <td>Kati</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>ML-2_colu_2014</td>\n",
       "      <td>ML-2_colu_2014_08</td>\n",
       "      <td>ML-2_colu_2014_Q3</td>\n",
       "      <td>ML-2_Kati_colu_2014</td>\n",
       "      <td>ML-2_Kati_colu_2014_08</td>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>AZ0344-C</td>\n",
       "      <td>MA13-5</td>\n",
       "      <td>Austin Burt</td>\n",
       "      <td>Mali</td>\n",
       "      <td>Ouassorola</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>12.90</td>\n",
       "      <td>-8.16</td>\n",
       "      <td>F</td>\n",
       "      <td>...</td>\n",
       "      <td>Koulikouro</td>\n",
       "      <td>ML-2</td>\n",
       "      <td>Kati</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>ML-2_colu_2014</td>\n",
       "      <td>ML-2_colu_2014_08</td>\n",
       "      <td>ML-2_colu_2014_Q3</td>\n",
       "      <td>ML-2_Kati_colu_2014</td>\n",
       "      <td>ML-2_Kati_colu_2014_08</td>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>AZ0425-CW</td>\n",
       "      <td>MA10-46</td>\n",
       "      <td>Austin Burt</td>\n",
       "      <td>Mali</td>\n",
       "      <td>Tieneguebougou</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>12.81</td>\n",
       "      <td>-8.08</td>\n",
       "      <td>F</td>\n",
       "      <td>...</td>\n",
       "      <td>Koulikouro</td>\n",
       "      <td>ML-2</td>\n",
       "      <td>Kati</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>ML-2_colu_2014</td>\n",
       "      <td>ML-2_colu_2014_08</td>\n",
       "      <td>ML-2_colu_2014_Q3</td>\n",
       "      <td>ML-2_Kati_colu_2014</td>\n",
       "      <td>ML-2_Kati_colu_2014_08</td>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>AZ0433-CW</td>\n",
       "      <td>MA11-13</td>\n",
       "      <td>Austin Burt</td>\n",
       "      <td>Mali</td>\n",
       "      <td>Tieneguebougou</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>12.81</td>\n",
       "      <td>-8.08</td>\n",
       "      <td>F</td>\n",
       "      <td>...</td>\n",
       "      <td>Koulikouro</td>\n",
       "      <td>ML-2</td>\n",
       "      <td>Kati</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>ML-2_colu_2014</td>\n",
       "      <td>ML-2_colu_2014_08</td>\n",
       "      <td>ML-2_colu_2014_Q3</td>\n",
       "      <td>ML-2_Kati_colu_2014</td>\n",
       "      <td>ML-2_Kati_colu_2014_08</td>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>AZ0434-CW</td>\n",
       "      <td>MA11-14</td>\n",
       "      <td>Austin Burt</td>\n",
       "      <td>Mali</td>\n",
       "      <td>Tieneguebougou</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>12.81</td>\n",
       "      <td>-8.08</td>\n",
       "      <td>F</td>\n",
       "      <td>...</td>\n",
       "      <td>Koulikouro</td>\n",
       "      <td>ML-2</td>\n",
       "      <td>Kati</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>ML-2_colu_2014</td>\n",
       "      <td>ML-2_colu_2014_08</td>\n",
       "      <td>ML-2_colu_2014_Q3</td>\n",
       "      <td>ML-2_Kati_colu_2014</td>\n",
       "      <td>ML-2_Kati_colu_2014_08</td>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>AZ0416-CW</td>\n",
       "      <td>MA10-32</td>\n",
       "      <td>Austin Burt</td>\n",
       "      <td>Mali</td>\n",
       "      <td>Tieneguebougou</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>12.81</td>\n",
       "      <td>-8.08</td>\n",
       "      <td>F</td>\n",
       "      <td>...</td>\n",
       "      <td>Koulikouro</td>\n",
       "      <td>ML-2</td>\n",
       "      <td>Kati</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>ML-2_colu_2014</td>\n",
       "      <td>ML-2_colu_2014_08</td>\n",
       "      <td>ML-2_colu_2014_Q3</td>\n",
       "      <td>ML-2_Kati_colu_2014</td>\n",
       "      <td>ML-2_Kati_colu_2014_08</td>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>AZ0444-CW</td>\n",
       "      <td>MA11-28</td>\n",
       "      <td>Austin Burt</td>\n",
       "      <td>Mali</td>\n",
       "      <td>Tieneguebougou</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>12.81</td>\n",
       "      <td>-8.08</td>\n",
       "      <td>F</td>\n",
       "      <td>...</td>\n",
       "      <td>Koulikouro</td>\n",
       "      <td>ML-2</td>\n",
       "      <td>Kati</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>ML-2_colu_2014</td>\n",
       "      <td>ML-2_colu_2014_08</td>\n",
       "      <td>ML-2_colu_2014_Q3</td>\n",
       "      <td>ML-2_Kati_colu_2014</td>\n",
       "      <td>ML-2_Kati_colu_2014_08</td>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>AZ0459-CW</td>\n",
       "      <td>MA11-9</td>\n",
       "      <td>Austin Burt</td>\n",
       "      <td>Mali</td>\n",
       "      <td>Tieneguebougou</td>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>12.81</td>\n",
       "      <td>-8.08</td>\n",
       "      <td>F</td>\n",
       "      <td>...</td>\n",
       "      <td>Koulikouro</td>\n",
       "      <td>ML-2</td>\n",
       "      <td>Kati</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>ML-2_colu_2014</td>\n",
       "      <td>ML-2_colu_2014_08</td>\n",
       "      <td>ML-2_colu_2014_Q3</td>\n",
       "      <td>ML-2_Kati_colu_2014</td>\n",
       "      <td>ML-2_Kati_colu_2014_08</td>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>34 rows  57 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   sample_id partner_sample_id  contributor country  \\\n",
       "0   VBS25302-5561STDY9810072           MA11-38  Austin Burt    Mali   \n",
       "1   VBS25303-5561STDY9810073           MA11-39  Austin Burt    Mali   \n",
       "2   VBS25304-5561STDY9810074           MA11-40  Austin Burt    Mali   \n",
       "3   VBS25307-5561STDY9810077           MA11-43  Austin Burt    Mali   \n",
       "4   VBS25308-5561STDY9810078           MA11-44  Austin Burt    Mali   \n",
       "5   VBS25309-5561STDY9810079           MA11-45  Austin Burt    Mali   \n",
       "6   VBS25311-5561STDY9810081           MA11-47  Austin Burt    Mali   \n",
       "7                   AZ0292-C            MA7-36  Austin Burt    Mali   \n",
       "8                   AZ0293-C            MA7-37  Austin Burt    Mali   \n",
       "9                   AZ0294-C             MA7-4  Austin Burt    Mali   \n",
       "10                  AZ0295-C            MA7-41  Austin Burt    Mali   \n",
       "11                  AZ0296-C            MA7-44  Austin Burt    Mali   \n",
       "12                  AZ0297-C             MA7-7  Austin Burt    Mali   \n",
       "13                  AZ0298-C             MA7-8  Austin Burt    Mali   \n",
       "14                  AZ0332-C           MA12-12  Austin Burt    Mali   \n",
       "15                  AZ0333-C            MA12-8  Austin Burt    Mali   \n",
       "16                  AZ0284-C             MA7-1  Austin Burt    Mali   \n",
       "17                  AZ0287-C            MA7-17  Austin Burt    Mali   \n",
       "18                  AZ0289-C            MA7-27  Austin Burt    Mali   \n",
       "19                  AZ0290-C            MA7-28  Austin Burt    Mali   \n",
       "20                  AZ0291-C            MA7-30  Austin Burt    Mali   \n",
       "21                  AZ0334-C            MA13-1  Austin Burt    Mali   \n",
       "22                  AZ0339-C           MA13-29  Austin Burt    Mali   \n",
       "23                  AZ0340-C           MA13-33  Austin Burt    Mali   \n",
       "24                  AZ0341-C           MA13-39  Austin Burt    Mali   \n",
       "25                  AZ0342-C           MA13-43  Austin Burt    Mali   \n",
       "26                  AZ0343-C           MA13-45  Austin Burt    Mali   \n",
       "27                  AZ0344-C            MA13-5  Austin Burt    Mali   \n",
       "28                 AZ0425-CW           MA10-46  Austin Burt    Mali   \n",
       "29                 AZ0433-CW           MA11-13  Austin Burt    Mali   \n",
       "30                 AZ0434-CW           MA11-14  Austin Burt    Mali   \n",
       "31                 AZ0416-CW           MA10-32  Austin Burt    Mali   \n",
       "32                 AZ0444-CW           MA11-28  Austin Burt    Mali   \n",
       "33                 AZ0459-CW            MA11-9  Austin Burt    Mali   \n",
       "\n",
       "          location  year  month  latitude  longitude sex_call  ...  \\\n",
       "0   Tieneguebougou  2014      8     12.81      -8.08        F  ...   \n",
       "1   Tieneguebougou  2014      8     12.81      -8.08        F  ...   \n",
       "2   Tieneguebougou  2014      8     12.81      -8.08        F  ...   \n",
       "3   Tieneguebougou  2014      8     12.81      -8.08        F  ...   \n",
       "4   Tieneguebougou  2014      8     12.81      -8.08        F  ...   \n",
       "5   Tieneguebougou  2014      8     12.81      -8.08        F  ...   \n",
       "6   Tieneguebougou  2014      8     12.81      -8.08        F  ...   \n",
       "7       Kababougou  2014      8     12.89      -8.15        F  ...   \n",
       "8       Kababougou  2014      8     12.89      -8.15        F  ...   \n",
       "9       Kababougou  2014      8     12.89      -8.15        F  ...   \n",
       "10      Kababougou  2014      8     12.89      -8.15        F  ...   \n",
       "11      Kababougou  2014      8     12.89      -8.15        F  ...   \n",
       "12      Kababougou  2014      8     12.89      -8.15        F  ...   \n",
       "13      Kababougou  2014      8     12.89      -8.15        F  ...   \n",
       "14      Ouassorola  2014      8     12.90      -8.16        F  ...   \n",
       "15      Ouassorola  2014      8     12.90      -8.16        F  ...   \n",
       "16      Kababougou  2014      8     12.89      -8.15        F  ...   \n",
       "17      Kababougou  2014      8     12.89      -8.15        F  ...   \n",
       "18      Kababougou  2014      8     12.89      -8.15        F  ...   \n",
       "19      Kababougou  2014      8     12.89      -8.15        F  ...   \n",
       "20      Kababougou  2014      8     12.89      -8.15        F  ...   \n",
       "21      Ouassorola  2014      8     12.90      -8.16        F  ...   \n",
       "22      Ouassorola  2014      8     12.90      -8.16        F  ...   \n",
       "23      Ouassorola  2014      8     12.90      -8.16        F  ...   \n",
       "24      Ouassorola  2014      8     12.90      -8.16        F  ...   \n",
       "25      Ouassorola  2014      8     12.90      -8.16        F  ...   \n",
       "26      Ouassorola  2014      8     12.90      -8.16        F  ...   \n",
       "27      Ouassorola  2014      8     12.90      -8.16        F  ...   \n",
       "28  Tieneguebougou  2014      8     12.81      -8.08        F  ...   \n",
       "29  Tieneguebougou  2014      8     12.81      -8.08        F  ...   \n",
       "30  Tieneguebougou  2014      8     12.81      -8.08        F  ...   \n",
       "31  Tieneguebougou  2014      8     12.81      -8.08        F  ...   \n",
       "32  Tieneguebougou  2014      8     12.81      -8.08        F  ...   \n",
       "33  Tieneguebougou  2014      8     12.81      -8.08        F  ...   \n",
       "\n",
       "   admin1_name admin1_iso  admin2_name     taxon cohort_admin1_year  \\\n",
       "0   Koulikouro       ML-2         Kati  coluzzii     ML-2_colu_2014   \n",
       "1   Koulikouro       ML-2         Kati  coluzzii     ML-2_colu_2014   \n",
       "2   Koulikouro       ML-2         Kati  coluzzii     ML-2_colu_2014   \n",
       "3   Koulikouro       ML-2         Kati  coluzzii     ML-2_colu_2014   \n",
       "4   Koulikouro       ML-2         Kati  coluzzii     ML-2_colu_2014   \n",
       "5   Koulikouro       ML-2         Kati  coluzzii     ML-2_colu_2014   \n",
       "6   Koulikouro       ML-2         Kati  coluzzii     ML-2_colu_2014   \n",
       "7   Koulikouro       ML-2         Kati  coluzzii     ML-2_colu_2014   \n",
       "8   Koulikouro       ML-2         Kati  coluzzii     ML-2_colu_2014   \n",
       "9   Koulikouro       ML-2         Kati  coluzzii     ML-2_colu_2014   \n",
       "10  Koulikouro       ML-2         Kati  coluzzii     ML-2_colu_2014   \n",
       "11  Koulikouro       ML-2         Kati  coluzzii     ML-2_colu_2014   \n",
       "12  Koulikouro       ML-2         Kati  coluzzii     ML-2_colu_2014   \n",
       "13  Koulikouro       ML-2         Kati  coluzzii     ML-2_colu_2014   \n",
       "14  Koulikouro       ML-2         Kati  coluzzii     ML-2_colu_2014   \n",
       "15  Koulikouro       ML-2         Kati  coluzzii     ML-2_colu_2014   \n",
       "16  Koulikouro       ML-2         Kati  coluzzii     ML-2_colu_2014   \n",
       "17  Koulikouro       ML-2         Kati  coluzzii     ML-2_colu_2014   \n",
       "18  Koulikouro       ML-2         Kati  coluzzii     ML-2_colu_2014   \n",
       "19  Koulikouro       ML-2         Kati  coluzzii     ML-2_colu_2014   \n",
       "20  Koulikouro       ML-2         Kati  coluzzii     ML-2_colu_2014   \n",
       "21  Koulikouro       ML-2         Kati  coluzzii     ML-2_colu_2014   \n",
       "22  Koulikouro       ML-2         Kati  coluzzii     ML-2_colu_2014   \n",
       "23  Koulikouro       ML-2         Kati  coluzzii     ML-2_colu_2014   \n",
       "24  Koulikouro       ML-2         Kati  coluzzii     ML-2_colu_2014   \n",
       "25  Koulikouro       ML-2         Kati  coluzzii     ML-2_colu_2014   \n",
       "26  Koulikouro       ML-2         Kati  coluzzii     ML-2_colu_2014   \n",
       "27  Koulikouro       ML-2         Kati  coluzzii     ML-2_colu_2014   \n",
       "28  Koulikouro       ML-2         Kati  coluzzii     ML-2_colu_2014   \n",
       "29  Koulikouro       ML-2         Kati  coluzzii     ML-2_colu_2014   \n",
       "30  Koulikouro       ML-2         Kati  coluzzii     ML-2_colu_2014   \n",
       "31  Koulikouro       ML-2         Kati  coluzzii     ML-2_colu_2014   \n",
       "32  Koulikouro       ML-2         Kati  coluzzii     ML-2_colu_2014   \n",
       "33  Koulikouro       ML-2         Kati  coluzzii     ML-2_colu_2014   \n",
       "\n",
       "   cohort_admin1_month cohort_admin1_quarter   cohort_admin2_year  \\\n",
       "0    ML-2_colu_2014_08     ML-2_colu_2014_Q3  ML-2_Kati_colu_2014   \n",
       "1    ML-2_colu_2014_08     ML-2_colu_2014_Q3  ML-2_Kati_colu_2014   \n",
       "2    ML-2_colu_2014_08     ML-2_colu_2014_Q3  ML-2_Kati_colu_2014   \n",
       "3    ML-2_colu_2014_08     ML-2_colu_2014_Q3  ML-2_Kati_colu_2014   \n",
       "4    ML-2_colu_2014_08     ML-2_colu_2014_Q3  ML-2_Kati_colu_2014   \n",
       "5    ML-2_colu_2014_08     ML-2_colu_2014_Q3  ML-2_Kati_colu_2014   \n",
       "6    ML-2_colu_2014_08     ML-2_colu_2014_Q3  ML-2_Kati_colu_2014   \n",
       "7    ML-2_colu_2014_08     ML-2_colu_2014_Q3  ML-2_Kati_colu_2014   \n",
       "8    ML-2_colu_2014_08     ML-2_colu_2014_Q3  ML-2_Kati_colu_2014   \n",
       "9    ML-2_colu_2014_08     ML-2_colu_2014_Q3  ML-2_Kati_colu_2014   \n",
       "10   ML-2_colu_2014_08     ML-2_colu_2014_Q3  ML-2_Kati_colu_2014   \n",
       "11   ML-2_colu_2014_08     ML-2_colu_2014_Q3  ML-2_Kati_colu_2014   \n",
       "12   ML-2_colu_2014_08     ML-2_colu_2014_Q3  ML-2_Kati_colu_2014   \n",
       "13   ML-2_colu_2014_08     ML-2_colu_2014_Q3  ML-2_Kati_colu_2014   \n",
       "14   ML-2_colu_2014_08     ML-2_colu_2014_Q3  ML-2_Kati_colu_2014   \n",
       "15   ML-2_colu_2014_08     ML-2_colu_2014_Q3  ML-2_Kati_colu_2014   \n",
       "16   ML-2_colu_2014_08     ML-2_colu_2014_Q3  ML-2_Kati_colu_2014   \n",
       "17   ML-2_colu_2014_08     ML-2_colu_2014_Q3  ML-2_Kati_colu_2014   \n",
       "18   ML-2_colu_2014_08     ML-2_colu_2014_Q3  ML-2_Kati_colu_2014   \n",
       "19   ML-2_colu_2014_08     ML-2_colu_2014_Q3  ML-2_Kati_colu_2014   \n",
       "20   ML-2_colu_2014_08     ML-2_colu_2014_Q3  ML-2_Kati_colu_2014   \n",
       "21   ML-2_colu_2014_08     ML-2_colu_2014_Q3  ML-2_Kati_colu_2014   \n",
       "22   ML-2_colu_2014_08     ML-2_colu_2014_Q3  ML-2_Kati_colu_2014   \n",
       "23   ML-2_colu_2014_08     ML-2_colu_2014_Q3  ML-2_Kati_colu_2014   \n",
       "24   ML-2_colu_2014_08     ML-2_colu_2014_Q3  ML-2_Kati_colu_2014   \n",
       "25   ML-2_colu_2014_08     ML-2_colu_2014_Q3  ML-2_Kati_colu_2014   \n",
       "26   ML-2_colu_2014_08     ML-2_colu_2014_Q3  ML-2_Kati_colu_2014   \n",
       "27   ML-2_colu_2014_08     ML-2_colu_2014_Q3  ML-2_Kati_colu_2014   \n",
       "28   ML-2_colu_2014_08     ML-2_colu_2014_Q3  ML-2_Kati_colu_2014   \n",
       "29   ML-2_colu_2014_08     ML-2_colu_2014_Q3  ML-2_Kati_colu_2014   \n",
       "30   ML-2_colu_2014_08     ML-2_colu_2014_Q3  ML-2_Kati_colu_2014   \n",
       "31   ML-2_colu_2014_08     ML-2_colu_2014_Q3  ML-2_Kati_colu_2014   \n",
       "32   ML-2_colu_2014_08     ML-2_colu_2014_Q3  ML-2_Kati_colu_2014   \n",
       "33   ML-2_colu_2014_08     ML-2_colu_2014_Q3  ML-2_Kati_colu_2014   \n",
       "\n",
       "       cohort_admin2_month   cohort_admin2_quarter  \n",
       "0   ML-2_Kati_colu_2014_08  ML-2_Kati_colu_2014_Q3  \n",
       "1   ML-2_Kati_colu_2014_08  ML-2_Kati_colu_2014_Q3  \n",
       "2   ML-2_Kati_colu_2014_08  ML-2_Kati_colu_2014_Q3  \n",
       "3   ML-2_Kati_colu_2014_08  ML-2_Kati_colu_2014_Q3  \n",
       "4   ML-2_Kati_colu_2014_08  ML-2_Kati_colu_2014_Q3  \n",
       "5   ML-2_Kati_colu_2014_08  ML-2_Kati_colu_2014_Q3  \n",
       "6   ML-2_Kati_colu_2014_08  ML-2_Kati_colu_2014_Q3  \n",
       "7   ML-2_Kati_colu_2014_08  ML-2_Kati_colu_2014_Q3  \n",
       "8   ML-2_Kati_colu_2014_08  ML-2_Kati_colu_2014_Q3  \n",
       "9   ML-2_Kati_colu_2014_08  ML-2_Kati_colu_2014_Q3  \n",
       "10  ML-2_Kati_colu_2014_08  ML-2_Kati_colu_2014_Q3  \n",
       "11  ML-2_Kati_colu_2014_08  ML-2_Kati_colu_2014_Q3  \n",
       "12  ML-2_Kati_colu_2014_08  ML-2_Kati_colu_2014_Q3  \n",
       "13  ML-2_Kati_colu_2014_08  ML-2_Kati_colu_2014_Q3  \n",
       "14  ML-2_Kati_colu_2014_08  ML-2_Kati_colu_2014_Q3  \n",
       "15  ML-2_Kati_colu_2014_08  ML-2_Kati_colu_2014_Q3  \n",
       "16  ML-2_Kati_colu_2014_08  ML-2_Kati_colu_2014_Q3  \n",
       "17  ML-2_Kati_colu_2014_08  ML-2_Kati_colu_2014_Q3  \n",
       "18  ML-2_Kati_colu_2014_08  ML-2_Kati_colu_2014_Q3  \n",
       "19  ML-2_Kati_colu_2014_08  ML-2_Kati_colu_2014_Q3  \n",
       "20  ML-2_Kati_colu_2014_08  ML-2_Kati_colu_2014_Q3  \n",
       "21  ML-2_Kati_colu_2014_08  ML-2_Kati_colu_2014_Q3  \n",
       "22  ML-2_Kati_colu_2014_08  ML-2_Kati_colu_2014_Q3  \n",
       "23  ML-2_Kati_colu_2014_08  ML-2_Kati_colu_2014_Q3  \n",
       "24  ML-2_Kati_colu_2014_08  ML-2_Kati_colu_2014_Q3  \n",
       "25  ML-2_Kati_colu_2014_08  ML-2_Kati_colu_2014_Q3  \n",
       "26  ML-2_Kati_colu_2014_08  ML-2_Kati_colu_2014_Q3  \n",
       "27  ML-2_Kati_colu_2014_08  ML-2_Kati_colu_2014_Q3  \n",
       "28  ML-2_Kati_colu_2014_08  ML-2_Kati_colu_2014_Q3  \n",
       "29  ML-2_Kati_colu_2014_08  ML-2_Kati_colu_2014_Q3  \n",
       "30  ML-2_Kati_colu_2014_08  ML-2_Kati_colu_2014_Q3  \n",
       "31  ML-2_Kati_colu_2014_08  ML-2_Kati_colu_2014_Q3  \n",
       "32  ML-2_Kati_colu_2014_08  ML-2_Kati_colu_2014_Q3  \n",
       "33  ML-2_Kati_colu_2014_08  ML-2_Kati_colu_2014_Q3  \n",
       "\n",
       "[34 rows x 57 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Load sample metadata for this cohort.\n",
    "\n",
    "df_samples = ag3.sample_metadata(sample_query=cohort['sample_query'])\n",
    "df_samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "6f5e51ed",
   "metadata": {
    "tags": [
     "remove-input",
     "remove-output"
    ]
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>year</th>\n",
       "      <th>month</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2014</td>\n",
       "      <td>8</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   year  month  count\n",
       "0  2014      8     34"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Determine collection dates.\n",
    "\n",
    "df_collection_dates = (\n",
    "    df_samples\n",
    "    .groupby([\"year\", \"month\"])\n",
    "    .size()\n",
    "    .reset_index()\n",
    "    .rename(columns={0: 'count'})\n",
    ")\n",
    "df_collection_dates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "391e6e57",
   "metadata": {
    "tags": [
     "remove-input",
     "remove-output"
    ]
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('August', 'August')"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Determine first and last collection months.\n",
    "\n",
    "min_month = df_collection_dates[\"month\"].min()\n",
    "max_month = df_collection_dates[\"month\"].max()\n",
    "\n",
    "if min_month < 0:\n",
    "    start_month = end_month = None\n",
    "else:\n",
    "    start_month = pd.to_datetime(min_month, format=\"%m\").month_name()\n",
    "    end_month = pd.to_datetime(max_month, format=\"%m\").month_name()\n",
    "    \n",
    "start_month, end_month"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "92155ecb",
   "metadata": {
    "tags": [
     "remove-input",
     "remove-output"
    ]
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>location</th>\n",
       "      <th>longitude</th>\n",
       "      <th>latitude</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Tieneguebougou</td>\n",
       "      <td>-8.08</td>\n",
       "      <td>12.81</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Kababougou</td>\n",
       "      <td>-8.15</td>\n",
       "      <td>12.89</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Ouassorola</td>\n",
       "      <td>-8.16</td>\n",
       "      <td>12.90</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          location  longitude  latitude\n",
       "0   Tieneguebougou      -8.08     12.81\n",
       "7       Kababougou      -8.15     12.89\n",
       "14      Ouassorola      -8.16     12.90"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Determine unique collection locations.\n",
    "\n",
    "df_locations = df_samples[[\"location\", \"longitude\", \"latitude\"]].drop_duplicates()\n",
    "df_locations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "b71db202",
   "metadata": {
    "tags": [
     "remove-input",
     "remove-output"
    ]
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>Sample set</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Contributor</th>\n",
       "      <th>Study</th>\n",
       "      <th>Data release</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">Austin Burt</th>\n",
       "      <th>TODO</th>\n",
       "      <th>Ag3.4</th>\n",
       "      <td>1191-VO-MULTI-OLOUGHLIN-VMF00106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ag1000G</th>\n",
       "      <th>Ag3.0</th>\n",
       "      <td>AG1000G-ML-A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                        Sample set\n",
       "Contributor Study   Data release                                  \n",
       "Austin Burt TODO    Ag3.4         1191-VO-MULTI-OLOUGHLIN-VMF00106\n",
       "            Ag1000G Ag3.0                             AG1000G-ML-A"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Extract provenance information about the samples.\n",
    "\n",
    "df_contributors = df_samples[[\"release\", \"sample_set\", \"contributor\"]].drop_duplicates()\n",
    "df_contributors[\"study\"] = df_contributors.apply(\n",
    "    lambda v: \"Ag1000G\" if v[\"sample_set\"].startswith(\"AG1000G\") else \"TODO\",\n",
    "    axis=\"columns\"\n",
    ")\n",
    "df_contributors[\"release\"] = df_contributors[\"release\"].apply(\n",
    "    lambda v: f\"Ag{v}\"\n",
    ")\n",
    "df_contributors.rename(columns={\n",
    "    \"contributor\": \"Contributor\",\n",
    "    \"study\": \"Study\",\n",
    "    \"release\": \"Data release\",\n",
    "    \"sample_set\": \"Sample set\",\n",
    "}, inplace=True)\n",
    "df_contributors.set_index([\"Contributor\", \"Study\", \"Data release\"], inplace=True)\n",
    "df_contributors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "3baa1f0d",
   "metadata": {
    "tags": [
     "remove-input"
    ]
   },
   "outputs": [
    {
     "data": {
      "text/markdown": [
       "This cohort comprises 27 samples from the \n",
       "*coluzzii* taxon, collected from 3 locations within the administrative \n",
       "division of Kati, Koulikouro, Mali. Collections were made in August 2014."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Build a paragraph with summary information about the samples in this cohort.\n",
    "\n",
    "n_locations = len(df_locations)\n",
    "\n",
    "summary_info = f\"\"\"This cohort comprises {cohort['cohort_size']:,} samples from the \n",
    "*{cohort['taxon']}* taxon, collected from {n_locations} locations within the administrative \n",
    "division of {cohort['admin2_name']}, {cohort['admin1_name']}, {cohort['country']}.\"\"\"\n",
    "\n",
    "if start_month and start_month == end_month:\n",
    "    summary_info += f\"\"\" Collections were made in {start_month} {cohort['year']}.\"\"\"\n",
    "elif start_month:\n",
    "    summary_info += f\"\"\" Collections were made between {start_month} and {end_month} in {cohort['year']}.\"\"\"\n",
    "else:\n",
    "    summary_info += f\"\"\" Collections were made in {cohort['year']}.\"\"\"\n",
    "    \n",
    "display(Markdown(summary_info))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3ff2d7fb",
   "metadata": {},
   "source": [
    "## Selection scans"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "7b10b224",
   "metadata": {
    "tags": [
     "remove-input",
     "remove-output"
    ]
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cohort_id</th>\n",
       "      <th>contig</th>\n",
       "      <th>gcenter</th>\n",
       "      <th>pcenter</th>\n",
       "      <th>delta_i</th>\n",
       "      <th>stat_max</th>\n",
       "      <th>gpos_max</th>\n",
       "      <th>ppos_max</th>\n",
       "      <th>focus_gstart</th>\n",
       "      <th>focus_gstop</th>\n",
       "      <th>...</th>\n",
       "      <th>skew</th>\n",
       "      <th>decay_left</th>\n",
       "      <th>decay_right</th>\n",
       "      <th>baseline</th>\n",
       "      <th>aic</th>\n",
       "      <th>bic</th>\n",
       "      <th>rss</th>\n",
       "      <th>constant_aic</th>\n",
       "      <th>bottom</th>\n",
       "      <th>top</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "      <td>2RL</td>\n",
       "      <td>56.921394</td>\n",
       "      <td>28460698</td>\n",
       "      <td>3781</td>\n",
       "      <td>0.199588</td>\n",
       "      <td>56.942336</td>\n",
       "      <td>28471168</td>\n",
       "      <td>56.851603</td>\n",
       "      <td>56.956290</td>\n",
       "      <td>...</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.279165</td>\n",
       "      <td>0.139583</td>\n",
       "      <td>0.026712</td>\n",
       "      <td>-19901.648084</td>\n",
       "      <td>-19873.507928</td>\n",
       "      <td>0.127284</td>\n",
       "      <td>-16119.799861</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "      <td>2RL</td>\n",
       "      <td>81.883266</td>\n",
       "      <td>40941634</td>\n",
       "      <td>2402</td>\n",
       "      <td>0.117284</td>\n",
       "      <td>81.947584</td>\n",
       "      <td>40973792</td>\n",
       "      <td>81.798233</td>\n",
       "      <td>81.957842</td>\n",
       "      <td>...</td>\n",
       "      <td>0.094657</td>\n",
       "      <td>0.340132</td>\n",
       "      <td>0.298303</td>\n",
       "      <td>0.025335</td>\n",
       "      <td>-16777.165198</td>\n",
       "      <td>-16750.235943</td>\n",
       "      <td>0.048725</td>\n",
       "      <td>-14375.109909</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "      <td>2RL</td>\n",
       "      <td>120.868338</td>\n",
       "      <td>64204801</td>\n",
       "      <td>2899</td>\n",
       "      <td>0.927984</td>\n",
       "      <td>120.809912</td>\n",
       "      <td>64175587</td>\n",
       "      <td>120.537650</td>\n",
       "      <td>121.050210</td>\n",
       "      <td>...</td>\n",
       "      <td>0.431277</td>\n",
       "      <td>1.322752</td>\n",
       "      <td>0.727485</td>\n",
       "      <td>0.013613</td>\n",
       "      <td>-7636.750854</td>\n",
       "      <td>-7610.095552</td>\n",
       "      <td>10.209972</td>\n",
       "      <td>-4737.336791</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "      <td>2RL</td>\n",
       "      <td>166.443566</td>\n",
       "      <td>86992415</td>\n",
       "      <td>3944</td>\n",
       "      <td>0.244170</td>\n",
       "      <td>166.489816</td>\n",
       "      <td>87015539</td>\n",
       "      <td>166.239307</td>\n",
       "      <td>166.545701</td>\n",
       "      <td>...</td>\n",
       "      <td>0.499957</td>\n",
       "      <td>0.817035</td>\n",
       "      <td>0.408542</td>\n",
       "      <td>0.022162</td>\n",
       "      <td>-17271.475352</td>\n",
       "      <td>-17243.569303</td>\n",
       "      <td>0.291891</td>\n",
       "      <td>-13327.413252</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "      <td>2RL</td>\n",
       "      <td>166.443566</td>\n",
       "      <td>86992415</td>\n",
       "      <td>3944</td>\n",
       "      <td>0.244170</td>\n",
       "      <td>166.489816</td>\n",
       "      <td>87015539</td>\n",
       "      <td>166.239946</td>\n",
       "      <td>166.545378</td>\n",
       "      <td>...</td>\n",
       "      <td>0.499983</td>\n",
       "      <td>0.814481</td>\n",
       "      <td>0.407250</td>\n",
       "      <td>0.022272</td>\n",
       "      <td>-17253.716883</td>\n",
       "      <td>-17225.818489</td>\n",
       "      <td>0.290156</td>\n",
       "      <td>-13309.165267</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "      <td>3RL</td>\n",
       "      <td>57.094257</td>\n",
       "      <td>28547129</td>\n",
       "      <td>4759</td>\n",
       "      <td>0.198903</td>\n",
       "      <td>57.130386</td>\n",
       "      <td>28565193</td>\n",
       "      <td>56.992045</td>\n",
       "      <td>57.203479</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.047854</td>\n",
       "      <td>0.408846</td>\n",
       "      <td>0.436888</td>\n",
       "      <td>0.021920</td>\n",
       "      <td>-18207.499649</td>\n",
       "      <td>-18179.857990</td>\n",
       "      <td>0.103724</td>\n",
       "      <td>-13448.108228</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "      <td>3RL</td>\n",
       "      <td>141.740603</td>\n",
       "      <td>73052958</td>\n",
       "      <td>2787</td>\n",
       "      <td>0.112483</td>\n",
       "      <td>141.740607</td>\n",
       "      <td>73052959</td>\n",
       "      <td>141.714888</td>\n",
       "      <td>141.765025</td>\n",
       "      <td>...</td>\n",
       "      <td>0.037245</td>\n",
       "      <td>0.102862</td>\n",
       "      <td>0.097686</td>\n",
       "      <td>0.022081</td>\n",
       "      <td>-21668.292875</td>\n",
       "      <td>-21640.584459</td>\n",
       "      <td>0.019087</td>\n",
       "      <td>-18880.892901</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ML-2_Kati_colu_2014_Q3</td>\n",
       "      <td>X</td>\n",
       "      <td>30.324885</td>\n",
       "      <td>15162443</td>\n",
       "      <td>4916</td>\n",
       "      <td>0.927984</td>\n",
       "      <td>30.302520</td>\n",
       "      <td>15151260</td>\n",
       "      <td>30.186625</td>\n",
       "      <td>30.547477</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.343509</td>\n",
       "      <td>0.553039</td>\n",
       "      <td>0.890367</td>\n",
       "      <td>0.031082</td>\n",
       "      <td>-11414.210712</td>\n",
       "      <td>-11386.399904</td>\n",
       "      <td>5.075485</td>\n",
       "      <td>-6498.114251</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8 rows  32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                cohort_id contig     gcenter   pcenter  delta_i  stat_max  \\\n",
       "0  ML-2_Kati_colu_2014_Q3    2RL   56.921394  28460698     3781  0.199588   \n",
       "1  ML-2_Kati_colu_2014_Q3    2RL   81.883266  40941634     2402  0.117284   \n",
       "2  ML-2_Kati_colu_2014_Q3    2RL  120.868338  64204801     2899  0.927984   \n",
       "3  ML-2_Kati_colu_2014_Q3    2RL  166.443566  86992415     3944  0.244170   \n",
       "4  ML-2_Kati_colu_2014_Q3    2RL  166.443566  86992415     3944  0.244170   \n",
       "0  ML-2_Kati_colu_2014_Q3    3RL   57.094257  28547129     4759  0.198903   \n",
       "1  ML-2_Kati_colu_2014_Q3    3RL  141.740603  73052958     2787  0.112483   \n",
       "0  ML-2_Kati_colu_2014_Q3      X   30.324885  15162443     4916  0.927984   \n",
       "\n",
       "     gpos_max  ppos_max  focus_gstart  focus_gstop  ...      skew  decay_left  \\\n",
       "0   56.942336  28471168     56.851603    56.956290  ...  0.500000    0.279165   \n",
       "1   81.947584  40973792     81.798233    81.957842  ...  0.094657    0.340132   \n",
       "2  120.809912  64175587    120.537650   121.050210  ...  0.431277    1.322752   \n",
       "3  166.489816  87015539    166.239307   166.545701  ...  0.499957    0.817035   \n",
       "4  166.489816  87015539    166.239946   166.545378  ...  0.499983    0.814481   \n",
       "0   57.130386  28565193     56.992045    57.203479  ... -0.047854    0.408846   \n",
       "1  141.740607  73052959    141.714888   141.765025  ...  0.037245    0.102862   \n",
       "0   30.302520  15151260     30.186625    30.547477  ... -0.343509    0.553039   \n",
       "\n",
       "   decay_right  baseline           aic           bic        rss  constant_aic  \\\n",
       "0     0.139583  0.026712 -19901.648084 -19873.507928   0.127284 -16119.799861   \n",
       "1     0.298303  0.025335 -16777.165198 -16750.235943   0.048725 -14375.109909   \n",
       "2     0.727485  0.013613  -7636.750854  -7610.095552  10.209972  -4737.336791   \n",
       "3     0.408542  0.022162 -17271.475352 -17243.569303   0.291891 -13327.413252   \n",
       "4     0.407250  0.022272 -17253.716883 -17225.818489   0.290156 -13309.165267   \n",
       "0     0.436888  0.021920 -18207.499649 -18179.857990   0.103724 -13448.108228   \n",
       "1     0.097686  0.022081 -21668.292875 -21640.584459   0.019087 -18880.892901   \n",
       "0     0.890367  0.031082 -11414.210712 -11386.399904   5.075485  -6498.114251   \n",
       "\n",
       "   bottom  top  \n",
       "0       0    1  \n",
       "1       0    1  \n",
       "2       0    1  \n",
       "3       0    1  \n",
       "4       0    1  \n",
       "0       0    1  \n",
       "1       0    1  \n",
       "0       0    1  \n",
       "\n",
       "[8 rows x 32 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# load signals to overlay on H12 plots\n",
    "import pandas as pd\n",
    "\n",
    "df_signals_contigs = []\n",
    "for contig in contigs:\n",
    "\n",
    "    try:\n",
    "        df_signals_contig = pd.read_csv(here() / \"build\" / analysis_version / \"h12-signal-detection\" / f\"{cohort_id}_{contig}.csv\")\n",
    "    except pd.errors.EmptyDataError:\n",
    "        df_signals_contig = pd.DataFrame()\n",
    "    df_signals_contigs.append(df_signals_contig)  \n",
    "    \n",
    "df_signals = pd.concat(df_signals_contigs)\n",
    "\n",
    "# Add extra columns to help with overlaying signals on plots.\n",
    "df_signals[\"bottom\"] = 0\n",
    "df_signals[\"top\"] = 1\n",
    "\n",
    "df_signals"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "212846ca",
   "metadata": {
    "tags": [
     "remove-input"
    ]
   },
   "outputs": [],
   "source": [
    "# load window sizes \n",
    "h12_calibration_dir = f\"build/{analysis_version}/h12-calibration\"\n",
    "with open(here() / h12_calibration_dir / f\"{cohort_id}.yaml\") as h12_calibration_file:\n",
    "    h12_calibration_params = yaml.safe_load(h12_calibration_file)\n",
    "h12_window_size = h12_calibration_params[\"h12_window_size\"]\n",
    "g123_calibration_dir = f\"build/{analysis_version}/g123-calibration\"\n",
    "with open(here() / g123_calibration_dir / f\"{cohort_id}.yaml\") as g123_calibration_file:\n",
    "    g123_calibration_params = yaml.safe_load(g123_calibration_file)\n",
    "g123_window_size = g123_calibration_params[\"g123_window_size\"]\n",
    "\n",
    "if cohort.taxon == 'arabiensis':\n",
    "    phasing_analysis = 'arab'\n",
    "else:\n",
    "    phasing_analysis = 'gamb_colu'\n",
    "\n",
    "ihs_window_size = 100\n",
    "\n",
    "def plot_h12_g123_ihs_tracks(\n",
    "        contig, \n",
    "        df_signals,\n",
    "        sizing_mode='stretch_width', \n",
    "        show=False, \n",
    "        width=800, \n",
    "        track_height=150,\n",
    "        genes_height=90,\n",
    "    ):\n",
    "\n",
    "    sample_query = cohort[\"sample_query\"]\n",
    "    \n",
    "    fig1 = ag3.plot_h12_gwss_track(\n",
    "        contig=contig, \n",
    "        window_size=h12_window_size, \n",
    "        analysis=phasing_analysis, \n",
    "        sample_sets=sample_sets,\n",
    "        sample_query=sample_query, \n",
    "        min_cohort_size=min_cohort_size,\n",
    "        max_cohort_size=max_cohort_size,\n",
    "        sizing_mode=sizing_mode,\n",
    "        show=show,\n",
    "        width=width,\n",
    "        height=track_height,\n",
    "    )\n",
    "    fig1.xaxis.visible = False\n",
    "        \n",
    "    if not df_signals.empty:\n",
    "        df = df_signals.query(\"contig == @contig\")\n",
    "        center_xs = [np.array([row.pcenter, row.pcenter]) for idx, row in df.iterrows()]\n",
    "        center_ys = [np.array([0, 1]) for idx, row in df.iterrows()]\n",
    "        source = bkmod.ColumnDataSource(data={\n",
    "            'cohort': df.cohort_id,\n",
    "            'chromosome': df.contig,\n",
    "            'score': df.delta_i.astype(int),\n",
    "            'peak_start': df.span2_pstart,\n",
    "            'peak_stop': df.span2_pstop,\n",
    "            'focus_start': df.focus_pstart,\n",
    "            'focus_stop': df.focus_pstop,    \n",
    "            'center_xs': center_xs,\n",
    "            'center_ys': center_ys,\n",
    "            'bottom': df.bottom ,\n",
    "            'top': df.top,\n",
    "            })\n",
    "        quad = fig1.quad(bottom='bottom', top='top', left='peak_start', right='peak_stop', \n",
    "              source=source, color=\"gray\", alpha=.1, line_width=1)\n",
    "        quad2 = fig1.quad(bottom='bottom', top='top', left='focus_start', right='focus_stop', \n",
    "              source=source, color=\"red\", alpha=.4)\n",
    "        glyph = bkmod.MultiLine(xs='center_xs', ys='center_ys', line_color='red', line_width=2, line_alpha=0.8)\n",
    "        fig1.add_glyph(source, glyph)\n",
    "    \n",
    "        hover = bkmod.HoverTool(tooltips=[\n",
    "            (\"Cohort\", '@cohort'),\n",
    "            (\"Score\", '@score'),\n",
    "            (\"Focus\", \"@focus_start{,} - @focus_stop{,}\"),\n",
    "        ], renderers=[quad])\n",
    "\n",
    "        fig1.add_tools(hover)\n",
    "    \n",
    "    fig2 = ag3.plot_g123_gwss_track(\n",
    "        contig=contig, \n",
    "        window_size=g123_window_size, \n",
    "        sites=phasing_analysis,\n",
    "        site_mask=phasing_analysis, \n",
    "        sample_sets=sample_sets,\n",
    "        sample_query=sample_query, \n",
    "        min_cohort_size=min_cohort_size,\n",
    "        max_cohort_size=max_cohort_size,\n",
    "        sizing_mode=sizing_mode,\n",
    "        width=width,\n",
    "        height=track_height,\n",
    "        show=show,\n",
    "        title=\"\",\n",
    "        x_range=fig1.x_range,\n",
    "    )\n",
    "    fig2.xaxis.visible = False\n",
    "\n",
    "    fig3 = ag3.plot_ihs_gwss_track(\n",
    "        contig=contig, \n",
    "        window_size=ihs_window_size, \n",
    "        analysis=phasing_analysis, \n",
    "        sample_sets=sample_sets,\n",
    "        sample_query=sample_query, \n",
    "        min_cohort_size=min_cohort_size,\n",
    "        max_cohort_size=max_cohort_size,\n",
    "        sizing_mode=sizing_mode,\n",
    "        width=width,\n",
    "        height=track_height,\n",
    "        show=show,\n",
    "        title=\"\",\n",
    "        x_range=fig1.x_range,\n",
    "    )\n",
    "    fig3.xaxis.visible = False\n",
    "\n",
    "\n",
    "    fig4 = ag3.plot_genes(\n",
    "        region=contig, \n",
    "        show=show,\n",
    "        sizing_mode=sizing_mode,\n",
    "        width=width,\n",
    "        height=genes_height,\n",
    "        x_range=fig1.x_range\n",
    "        )\n",
    "                        \n",
    "    fig = bklay.gridplot(\n",
    "        [fig1, fig2, fig3, fig4],\n",
    "        ncols=1,\n",
    "        toolbar_location=\"above\",\n",
    "        merge_tools=True,\n",
    "        sizing_mode=sizing_mode,\n",
    "    )\n",
    "    return fig "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "4a1cdce3",
   "metadata": {
    "tags": [
     "remove-input",
     "full-width"
    ]
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<h3>Chromosome 2RL</h3>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                     \r"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Load haplotypes:   0%|          | 0/252 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                      \r"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mCacheMiss\u001b[0m                                 Traceback (most recent call last)",
      "File \u001b[0;32m~/.local/lib/python3.10/site-packages/malariagen_data/anoph/g123.py:201\u001b[0m, in \u001b[0;36mAnophelesG123Analysis.g123_gwss\u001b[0;34m(self, contig, window_size, sites, site_mask, sample_sets, sample_query, sample_query_options, min_cohort_size, max_cohort_size, random_seed, inline_array, chunks)\u001b[0m\n\u001b[1;32m    200\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 201\u001b[0m     results \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mresults_cache_get\u001b[49m\u001b[43m(\u001b[49m\u001b[43mname\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mname\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mparams\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mparams\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    203\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m CacheMiss:\n",
      "File \u001b[0;32m~/.local/lib/python3.10/site-packages/malariagen_data/util.py:1138\u001b[0m, in \u001b[0;36mcheck_types.<locals>.check_types_wrapper\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m   1137\u001b[0m             \u001b[38;5;28;01mraise\u001b[39;00m error \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m-> 1138\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mf\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/.local/lib/python3.10/site-packages/malariagen_data/anoph/base.py:575\u001b[0m, in \u001b[0;36mAnophelesBase.results_cache_get\u001b[0;34m(self, name, params)\u001b[0m\n\u001b[1;32m    573\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m np\u001b[38;5;241m.\u001b[39mload(legacy_results_path)\n\u001b[0;32m--> 575\u001b[0m \u001b[38;5;28;01mraise\u001b[39;00m CacheMiss\n",
      "\u001b[0;31mCacheMiss\u001b[0m: ",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[15], line 5\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m contig \u001b[38;5;129;01min\u001b[39;00m contigs:\n\u001b[1;32m      3\u001b[0m     display(HTML(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m<h3>Chromosome \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mcontig\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m</h3>\u001b[39m\u001b[38;5;124m\"\u001b[39m))\n\u001b[0;32m----> 5\u001b[0m     fig \u001b[38;5;241m=\u001b[39m \u001b[43mplot_h12_g123_ihs_tracks\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m      6\u001b[0m \u001b[43m        \u001b[49m\u001b[43mcontig\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcontig\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m      7\u001b[0m \u001b[43m        \u001b[49m\u001b[43mdf_signals\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdf_signals\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m      8\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m;\n\u001b[1;32m     10\u001b[0m     bkplt\u001b[38;5;241m.\u001b[39mshow(fig)\n",
      "Cell \u001b[0;32mIn[14], line 77\u001b[0m, in \u001b[0;36mplot_h12_g123_ihs_tracks\u001b[0;34m(contig, df_signals, sizing_mode, show, width, track_height, genes_height)\u001b[0m\n\u001b[1;32m     69\u001b[0m     hover \u001b[38;5;241m=\u001b[39m bkmod\u001b[38;5;241m.\u001b[39mHoverTool(tooltips\u001b[38;5;241m=\u001b[39m[\n\u001b[1;32m     70\u001b[0m         (\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mCohort\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124m@cohort\u001b[39m\u001b[38;5;124m'\u001b[39m),\n\u001b[1;32m     71\u001b[0m         (\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mScore\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124m@score\u001b[39m\u001b[38;5;124m'\u001b[39m),\n\u001b[1;32m     72\u001b[0m         (\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mFocus\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m@focus_start\u001b[39m\u001b[38;5;124m{\u001b[39m\u001b[38;5;124m,} - @focus_stop\u001b[39m\u001b[38;5;124m{\u001b[39m\u001b[38;5;124m,}\u001b[39m\u001b[38;5;124m\"\u001b[39m),\n\u001b[1;32m     73\u001b[0m     ], renderers\u001b[38;5;241m=\u001b[39m[quad])\n\u001b[1;32m     75\u001b[0m     fig1\u001b[38;5;241m.\u001b[39madd_tools(hover)\n\u001b[0;32m---> 77\u001b[0m fig2 \u001b[38;5;241m=\u001b[39m \u001b[43mag3\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mplot_g123_gwss_track\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m     78\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcontig\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcontig\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\n\u001b[1;32m     79\u001b[0m \u001b[43m    \u001b[49m\u001b[43mwindow_size\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mg123_window_size\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\n\u001b[1;32m     80\u001b[0m \u001b[43m    \u001b[49m\u001b[43msites\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mphasing_analysis\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     81\u001b[0m \u001b[43m    \u001b[49m\u001b[43msite_mask\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mphasing_analysis\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\n\u001b[1;32m     82\u001b[0m \u001b[43m    \u001b[49m\u001b[43msample_sets\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msample_sets\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     83\u001b[0m \u001b[43m    \u001b[49m\u001b[43msample_query\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msample_query\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\n\u001b[1;32m     84\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmin_cohort_size\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmin_cohort_size\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     85\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmax_cohort_size\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmax_cohort_size\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     86\u001b[0m \u001b[43m    \u001b[49m\u001b[43msizing_mode\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msizing_mode\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     87\u001b[0m \u001b[43m    \u001b[49m\u001b[43mwidth\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mwidth\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     88\u001b[0m \u001b[43m    \u001b[49m\u001b[43mheight\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mtrack_height\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     89\u001b[0m \u001b[43m    \u001b[49m\u001b[43mshow\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mshow\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     90\u001b[0m \u001b[43m    \u001b[49m\u001b[43mtitle\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m     91\u001b[0m \u001b[43m    \u001b[49m\u001b[43mx_range\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mfig1\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mx_range\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     92\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     93\u001b[0m fig2\u001b[38;5;241m.\u001b[39mxaxis\u001b[38;5;241m.\u001b[39mvisible \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mFalse\u001b[39;00m\n\u001b[1;32m     95\u001b[0m fig3 \u001b[38;5;241m=\u001b[39m ag3\u001b[38;5;241m.\u001b[39mplot_ihs_gwss_track(\n\u001b[1;32m     96\u001b[0m     contig\u001b[38;5;241m=\u001b[39mcontig, \n\u001b[1;32m     97\u001b[0m     window_size\u001b[38;5;241m=\u001b[39mihs_window_size, \n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    108\u001b[0m     x_range\u001b[38;5;241m=\u001b[39mfig1\u001b[38;5;241m.\u001b[39mx_range,\n\u001b[1;32m    109\u001b[0m )\n",
      "File \u001b[0;32m~/.local/lib/python3.10/site-packages/malariagen_data/util.py:1138\u001b[0m, in \u001b[0;36mcheck_types.<locals>.check_types_wrapper\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m   1136\u001b[0m             error \u001b[38;5;241m=\u001b[39m \u001b[38;5;167;01mTypeError\u001b[39;00m(message)\n\u001b[1;32m   1137\u001b[0m             \u001b[38;5;28;01mraise\u001b[39;00m error \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m-> 1138\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mf\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/.local/lib/python3.10/site-packages/malariagen_data/anoph/g123.py:340\u001b[0m, in \u001b[0;36mAnophelesG123Analysis.plot_g123_gwss_track\u001b[0;34m(self, contig, window_size, sites, site_mask, sample_sets, sample_query, sample_query_options, min_cohort_size, max_cohort_size, random_seed, title, sizing_mode, width, height, show, x_range, output_backend, inline_array, chunks)\u001b[0m\n\u001b[1;32m    309\u001b[0m \u001b[38;5;129m@check_types\u001b[39m\n\u001b[1;32m    310\u001b[0m \u001b[38;5;129m@doc\u001b[39m(\n\u001b[1;32m    311\u001b[0m     summary\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mPlot G123 GWSS data.\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    338\u001b[0m ) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m gplt_params\u001b[38;5;241m.\u001b[39mfigure:\n\u001b[1;32m    339\u001b[0m     \u001b[38;5;66;03m# compute G123\u001b[39;00m\n\u001b[0;32m--> 340\u001b[0m     x, g123 \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mg123_gwss\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    341\u001b[0m \u001b[43m        \u001b[49m\u001b[43mcontig\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcontig\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    342\u001b[0m \u001b[43m        \u001b[49m\u001b[43msites\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msites\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    343\u001b[0m \u001b[43m        \u001b[49m\u001b[43msite_mask\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msite_mask\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    344\u001b[0m \u001b[43m        \u001b[49m\u001b[43mwindow_size\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mwindow_size\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    345\u001b[0m \u001b[43m        \u001b[49m\u001b[43mmin_cohort_size\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmin_cohort_size\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    346\u001b[0m \u001b[43m        \u001b[49m\u001b[43mmax_cohort_size\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmax_cohort_size\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    347\u001b[0m \u001b[43m        \u001b[49m\u001b[43msample_query\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msample_query\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    348\u001b[0m \u001b[43m        \u001b[49m\u001b[43msample_query_options\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msample_query_options\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    349\u001b[0m \u001b[43m        \u001b[49m\u001b[43msample_sets\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msample_sets\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    350\u001b[0m \u001b[43m        \u001b[49m\u001b[43mrandom_seed\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrandom_seed\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    351\u001b[0m \u001b[43m        \u001b[49m\u001b[43minline_array\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43minline_array\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    352\u001b[0m \u001b[43m        \u001b[49m\u001b[43mchunks\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mchunks\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    353\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    355\u001b[0m     \u001b[38;5;66;03m# determine X axis range\u001b[39;00m\n\u001b[1;32m    356\u001b[0m     x_min \u001b[38;5;241m=\u001b[39m x[\u001b[38;5;241m0\u001b[39m]\n",
      "File \u001b[0;32m~/.local/lib/python3.10/site-packages/malariagen_data/util.py:1138\u001b[0m, in \u001b[0;36mcheck_types.<locals>.check_types_wrapper\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m   1136\u001b[0m             error \u001b[38;5;241m=\u001b[39m \u001b[38;5;167;01mTypeError\u001b[39;00m(message)\n\u001b[1;32m   1137\u001b[0m             \u001b[38;5;28;01mraise\u001b[39;00m error \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m-> 1138\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mf\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/.local/lib/python3.10/site-packages/malariagen_data/anoph/g123.py:204\u001b[0m, in \u001b[0;36mAnophelesG123Analysis.g123_gwss\u001b[0;34m(self, contig, window_size, sites, site_mask, sample_sets, sample_query, sample_query_options, min_cohort_size, max_cohort_size, random_seed, inline_array, chunks)\u001b[0m\n\u001b[1;32m    201\u001b[0m     results \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mresults_cache_get(name\u001b[38;5;241m=\u001b[39mname, params\u001b[38;5;241m=\u001b[39mparams)\n\u001b[1;32m    203\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m CacheMiss:\n\u001b[0;32m--> 204\u001b[0m     results \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_g123_gwss\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    205\u001b[0m \u001b[43m        \u001b[49m\u001b[43minline_array\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43minline_array\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mchunks\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mchunks\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mparams\u001b[49m\n\u001b[1;32m    206\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    207\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mresults_cache_set(name\u001b[38;5;241m=\u001b[39mname, params\u001b[38;5;241m=\u001b[39mparams, results\u001b[38;5;241m=\u001b[39mresults)\n\u001b[1;32m    209\u001b[0m x \u001b[38;5;241m=\u001b[39m results[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mx\u001b[39m\u001b[38;5;124m\"\u001b[39m]\n",
      "File \u001b[0;32m~/.local/lib/python3.10/site-packages/malariagen_data/anoph/g123.py:122\u001b[0m, in \u001b[0;36mAnophelesG123Analysis._g123_gwss\u001b[0;34m(self, contig, sites, site_mask, window_size, sample_sets, sample_query, sample_query_options, min_cohort_size, max_cohort_size, random_seed, inline_array, chunks)\u001b[0m\n\u001b[1;32m    106\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m_g123_gwss\u001b[39m(\n\u001b[1;32m    107\u001b[0m     \u001b[38;5;28mself\u001b[39m,\n\u001b[1;32m    108\u001b[0m     \u001b[38;5;241m*\u001b[39m,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    120\u001b[0m     chunks,\n\u001b[1;32m    121\u001b[0m ):\n\u001b[0;32m--> 122\u001b[0m     gt, pos \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_load_data_for_g123\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    123\u001b[0m \u001b[43m        \u001b[49m\u001b[43mcontig\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcontig\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    124\u001b[0m \u001b[43m        \u001b[49m\u001b[43msites\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msites\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    125\u001b[0m \u001b[43m        \u001b[49m\u001b[43msite_mask\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msite_mask\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    126\u001b[0m \u001b[43m        \u001b[49m\u001b[43msample_sets\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msample_sets\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    127\u001b[0m \u001b[43m        \u001b[49m\u001b[43msample_query\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msample_query\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    128\u001b[0m \u001b[43m        \u001b[49m\u001b[43msample_query_options\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msample_query_options\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    129\u001b[0m \u001b[43m        \u001b[49m\u001b[43mmin_cohort_size\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmin_cohort_size\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    130\u001b[0m \u001b[43m        \u001b[49m\u001b[43mmax_cohort_size\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmax_cohort_size\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    131\u001b[0m \u001b[43m        \u001b[49m\u001b[43mrandom_seed\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrandom_seed\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    132\u001b[0m \u001b[43m        \u001b[49m\u001b[43minline_array\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43minline_array\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    133\u001b[0m \u001b[43m        \u001b[49m\u001b[43mchunks\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mchunks\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    134\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    136\u001b[0m     \u001b[38;5;28;01mwith\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_spinner(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mCompute G123\u001b[39m\u001b[38;5;124m\"\u001b[39m):\n\u001b[1;32m    137\u001b[0m         g123 \u001b[38;5;241m=\u001b[39m allel\u001b[38;5;241m.\u001b[39mmoving_statistic(gt, statistic\u001b[38;5;241m=\u001b[39mgarud_g123, size\u001b[38;5;241m=\u001b[39mwindow_size)\n",
      "File \u001b[0;32m~/.local/lib/python3.10/site-packages/malariagen_data/anoph/g123.py:52\u001b[0m, in \u001b[0;36mAnophelesG123Analysis._load_data_for_g123\u001b[0;34m(self, contig, sites, site_mask, sample_sets, sample_query, sample_query_options, min_cohort_size, max_cohort_size, random_seed, inline_array, chunks)\u001b[0m\n\u001b[1;32m     37\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m_load_data_for_g123\u001b[39m(\n\u001b[1;32m     38\u001b[0m     \u001b[38;5;28mself\u001b[39m,\n\u001b[1;32m     39\u001b[0m     \u001b[38;5;241m*\u001b[39m,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     50\u001b[0m     chunks,\n\u001b[1;32m     51\u001b[0m ):\n\u001b[0;32m---> 52\u001b[0m     ds_snps \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msnp_calls\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m     53\u001b[0m \u001b[43m        \u001b[49m\u001b[43mregion\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcontig\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     54\u001b[0m \u001b[43m        \u001b[49m\u001b[43msample_query\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msample_query\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     55\u001b[0m \u001b[43m        \u001b[49m\u001b[43msample_query_options\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msample_query_options\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     56\u001b[0m \u001b[43m        \u001b[49m\u001b[43msample_sets\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msample_sets\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     57\u001b[0m \u001b[43m        \u001b[49m\u001b[43msite_mask\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msite_mask\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     58\u001b[0m \u001b[43m        \u001b[49m\u001b[43mmin_cohort_size\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmin_cohort_size\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     59\u001b[0m \u001b[43m        \u001b[49m\u001b[43mmax_cohort_size\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmax_cohort_size\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     60\u001b[0m \u001b[43m        \u001b[49m\u001b[43mrandom_seed\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrandom_seed\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     61\u001b[0m \u001b[43m        \u001b[49m\u001b[43minline_array\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43minline_array\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     62\u001b[0m \u001b[43m        \u001b[49m\u001b[43mchunks\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mchunks\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     63\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     65\u001b[0m     \u001b[38;5;28;01mwith\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_dask_progress(desc\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mLoad genotypes\u001b[39m\u001b[38;5;124m\"\u001b[39m):\n\u001b[1;32m     66\u001b[0m         gt \u001b[38;5;241m=\u001b[39m ds_snps[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mcall_genotype\u001b[39m\u001b[38;5;124m\"\u001b[39m]\u001b[38;5;241m.\u001b[39mdata\u001b[38;5;241m.\u001b[39mcompute()\n",
      "File \u001b[0;32m~/.local/lib/python3.10/site-packages/malariagen_data/util.py:1138\u001b[0m, in \u001b[0;36mcheck_types.<locals>.check_types_wrapper\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m   1136\u001b[0m             error \u001b[38;5;241m=\u001b[39m \u001b[38;5;167;01mTypeError\u001b[39;00m(message)\n\u001b[1;32m   1137\u001b[0m             \u001b[38;5;28;01mraise\u001b[39;00m error \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m-> 1138\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mf\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/.local/lib/python3.10/site-packages/malariagen_data/anoph/snp_data.py:977\u001b[0m, in \u001b[0;36mAnophelesSnpData.snp_calls\u001b[0;34m(***failed resolving arguments***)\u001b[0m\n\u001b[1;32m    974\u001b[0m site_mask_prepped \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_prep_optional_site_mask_param(site_mask\u001b[38;5;241m=\u001b[39msite_mask)\n\u001b[1;32m    975\u001b[0m \u001b[38;5;28;01mdel\u001b[39;00m site_mask\n\u001b[0;32m--> 977\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_snp_calls\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m    978\u001b[0m \u001b[43m    \u001b[49m\u001b[43mregions\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mregions\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    979\u001b[0m \u001b[43m    \u001b[49m\u001b[43msample_sets\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msample_sets_prepped\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    980\u001b[0m \u001b[43m    \u001b[49m\u001b[43msample_query\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msample_query\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    981\u001b[0m \u001b[43m    \u001b[49m\u001b[43msample_query_options\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msample_query_options\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    982\u001b[0m \u001b[43m    \u001b[49m\u001b[43msample_indices\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msample_indices_prepped\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    983\u001b[0m \u001b[43m    \u001b[49m\u001b[43msite_mask\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msite_mask_prepped\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    984\u001b[0m \u001b[43m    \u001b[49m\u001b[43msite_class\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msite_class\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    985\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcohort_size\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcohort_size\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    986\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmin_cohort_size\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmin_cohort_size\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    987\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmax_cohort_size\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mmax_cohort_size\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    988\u001b[0m \u001b[43m    \u001b[49m\u001b[43mrandom_seed\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mrandom_seed\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    989\u001b[0m \u001b[43m    \u001b[49m\u001b[43minline_array\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43minline_array\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    990\u001b[0m \u001b[43m    \u001b[49m\u001b[43mchunks\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mchunks\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m    991\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/.local/lib/python3.10/site-packages/malariagen_data/anoph/snp_data.py:1089\u001b[0m, in \u001b[0;36mAnophelesSnpData._snp_calls\u001b[0;34m(self, regions, sample_sets, sample_query, sample_query_options, sample_indices, site_mask, site_class, cohort_size, min_cohort_size, max_cohort_size, random_seed, inline_array, chunks)\u001b[0m\n\u001b[1;32m   1071\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m_snp_calls\u001b[39m(\n\u001b[1;32m   1072\u001b[0m     \u001b[38;5;28mself\u001b[39m,\n\u001b[1;32m   1073\u001b[0m     \u001b[38;5;241m*\u001b[39m,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m   1087\u001b[0m ):\n\u001b[1;32m   1088\u001b[0m     \u001b[38;5;66;03m# Get SNP calls and concatenate multiple sample sets and/or regions.\u001b[39;00m\n\u001b[0;32m-> 1089\u001b[0m     ds \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_cached_snp_calls\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m   1090\u001b[0m \u001b[43m        \u001b[49m\u001b[43mregions\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mregions\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1091\u001b[0m \u001b[43m        \u001b[49m\u001b[43msample_sets\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msample_sets\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1092\u001b[0m \u001b[43m        \u001b[49m\u001b[43msite_mask\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msite_mask\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1093\u001b[0m \u001b[43m        \u001b[49m\u001b[43msite_class\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43msite_class\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1094\u001b[0m \u001b[43m        \u001b[49m\u001b[43minline_array\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43minline_array\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1095\u001b[0m \u001b[43m        \u001b[49m\u001b[43mchunks\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mchunks\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1096\u001b[0m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1098\u001b[0m     \u001b[38;5;66;03m# Handle sample selection.\u001b[39;00m\n\u001b[1;32m   1099\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m sample_query \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n",
      "File \u001b[0;32m~/.local/lib/python3.10/site-packages/malariagen_data/anoph/snp_data.py:1060\u001b[0m, in \u001b[0;36mAnophelesSnpData._cached_snp_calls\u001b[0;34m(self, regions, sample_sets, site_mask, site_class, inline_array, chunks)\u001b[0m\n\u001b[1;32m   1058\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m site_mask \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m   1059\u001b[0m     \u001b[38;5;28;01mwith\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_spinner(desc\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mApply site filters\u001b[39m\u001b[38;5;124m\"\u001b[39m):\n\u001b[0;32m-> 1060\u001b[0m         ds \u001b[38;5;241m=\u001b[39m \u001b[43mdask_compress_dataset\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m   1061\u001b[0m \u001b[43m            \u001b[49m\u001b[43mds\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1062\u001b[0m \u001b[43m            \u001b[49m\u001b[43mindexer\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43mf\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mvariant_filter_pass_\u001b[39;49m\u001b[38;5;132;43;01m{\u001b[39;49;00m\u001b[43msite_mask\u001b[49m\u001b[38;5;132;43;01m}\u001b[39;49;00m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1063\u001b[0m \u001b[43m            \u001b[49m\u001b[43mdim\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mDIM_VARIANT\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1064\u001b[0m \u001b[43m        \u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1066\u001b[0m \u001b[38;5;66;03m# Add call_genotype_mask.\u001b[39;00m\n\u001b[1;32m   1067\u001b[0m ds[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mcall_genotype_mask\u001b[39m\u001b[38;5;124m\"\u001b[39m] \u001b[38;5;241m=\u001b[39m ds[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mcall_genotype\u001b[39m\u001b[38;5;124m\"\u001b[39m] \u001b[38;5;241m<\u001b[39m \u001b[38;5;241m0\u001b[39m\n",
      "File \u001b[0;32m~/.local/lib/python3.10/site-packages/malariagen_data/util.py:312\u001b[0m, in \u001b[0;36mdask_compress_dataset\u001b[0;34m(ds, indexer, dim)\u001b[0m\n\u001b[1;32m    307\u001b[0m \u001b[38;5;28;01massert\u001b[39;00m indexer\u001b[38;5;241m.\u001b[39mshape[\u001b[38;5;241m0\u001b[39m] \u001b[38;5;241m==\u001b[39m ds\u001b[38;5;241m.\u001b[39msizes[dim]\n\u001b[1;32m    309\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(indexer, da\u001b[38;5;241m.\u001b[39mArray):\n\u001b[1;32m    310\u001b[0m     \u001b[38;5;66;03m# temporarily compute the indexer once, to avoid multiple reads from\u001b[39;00m\n\u001b[1;32m    311\u001b[0m     \u001b[38;5;66;03m# the underlying data\u001b[39;00m\n\u001b[0;32m--> 312\u001b[0m     indexer_computed \u001b[38;5;241m=\u001b[39m \u001b[43mindexer\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcompute\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    313\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m    314\u001b[0m     \u001b[38;5;28;01massert\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(indexer, np\u001b[38;5;241m.\u001b[39mndarray)\n",
      "File \u001b[0;32m~/envs/selection-atlas/lib/python3.10/site-packages/dask/base.py:310\u001b[0m, in \u001b[0;36mDaskMethodsMixin.compute\u001b[0;34m(self, **kwargs)\u001b[0m\n\u001b[1;32m    286\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mcompute\u001b[39m(\u001b[38;5;28mself\u001b[39m, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs):\n\u001b[1;32m    287\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124;03m\"\"\"Compute this dask collection\u001b[39;00m\n\u001b[1;32m    288\u001b[0m \n\u001b[1;32m    289\u001b[0m \u001b[38;5;124;03m    This turns a lazy Dask collection into its in-memory equivalent.\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    308\u001b[0m \u001b[38;5;124;03m    dask.compute\u001b[39;00m\n\u001b[1;32m    309\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[0;32m--> 310\u001b[0m     (result,) \u001b[38;5;241m=\u001b[39m \u001b[43mcompute\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtraverse\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    311\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m result\n",
      "File \u001b[0;32m~/envs/selection-atlas/lib/python3.10/site-packages/dask/base.py:595\u001b[0m, in \u001b[0;36mcompute\u001b[0;34m(traverse, optimize_graph, scheduler, get, *args, **kwargs)\u001b[0m\n\u001b[1;32m    592\u001b[0m     keys\u001b[38;5;241m.\u001b[39mappend(x\u001b[38;5;241m.\u001b[39m__dask_keys__())\n\u001b[1;32m    593\u001b[0m     postcomputes\u001b[38;5;241m.\u001b[39mappend(x\u001b[38;5;241m.\u001b[39m__dask_postcompute__())\n\u001b[0;32m--> 595\u001b[0m results \u001b[38;5;241m=\u001b[39m \u001b[43mschedule\u001b[49m\u001b[43m(\u001b[49m\u001b[43mdsk\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mkeys\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    596\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m repack([f(r, \u001b[38;5;241m*\u001b[39ma) \u001b[38;5;28;01mfor\u001b[39;00m r, (f, a) \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mzip\u001b[39m(results, postcomputes)])\n",
      "File \u001b[0;32m~/envs/selection-atlas/lib/python3.10/site-packages/dask/threaded.py:89\u001b[0m, in \u001b[0;36mget\u001b[0;34m(dsk, keys, cache, num_workers, pool, **kwargs)\u001b[0m\n\u001b[1;32m     86\u001b[0m     \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(pool, multiprocessing\u001b[38;5;241m.\u001b[39mpool\u001b[38;5;241m.\u001b[39mPool):\n\u001b[1;32m     87\u001b[0m         pool \u001b[38;5;241m=\u001b[39m MultiprocessingPoolExecutor(pool)\n\u001b[0;32m---> 89\u001b[0m results \u001b[38;5;241m=\u001b[39m \u001b[43mget_async\u001b[49m\u001b[43m(\u001b[49m\n\u001b[1;32m     90\u001b[0m \u001b[43m    \u001b[49m\u001b[43mpool\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msubmit\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     91\u001b[0m \u001b[43m    \u001b[49m\u001b[43mpool\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_max_workers\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     92\u001b[0m \u001b[43m    \u001b[49m\u001b[43mdsk\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     93\u001b[0m \u001b[43m    \u001b[49m\u001b[43mkeys\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     94\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcache\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mcache\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     95\u001b[0m \u001b[43m    \u001b[49m\u001b[43mget_id\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m_thread_get_id\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     96\u001b[0m \u001b[43m    \u001b[49m\u001b[43mpack_exception\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mpack_exception\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     97\u001b[0m \u001b[43m    \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m     98\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    100\u001b[0m \u001b[38;5;66;03m# Cleanup pools associated to dead threads\u001b[39;00m\n\u001b[1;32m    101\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m pools_lock:\n",
      "File \u001b[0;32m~/envs/selection-atlas/lib/python3.10/site-packages/dask/local.py:500\u001b[0m, in \u001b[0;36mget_async\u001b[0;34m(submit, num_workers, dsk, result, cache, get_id, rerun_exceptions_locally, pack_exception, raise_exception, callbacks, dumps, loads, chunksize, **kwargs)\u001b[0m\n\u001b[1;32m    498\u001b[0m \u001b[38;5;28;01mwhile\u001b[39;00m state[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mwaiting\u001b[39m\u001b[38;5;124m\"\u001b[39m] \u001b[38;5;129;01mor\u001b[39;00m state[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mready\u001b[39m\u001b[38;5;124m\"\u001b[39m] \u001b[38;5;129;01mor\u001b[39;00m state[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mrunning\u001b[39m\u001b[38;5;124m\"\u001b[39m]:\n\u001b[1;32m    499\u001b[0m     fire_tasks(chunksize)\n\u001b[0;32m--> 500\u001b[0m     \u001b[38;5;28;01mfor\u001b[39;00m key, res_info, failed \u001b[38;5;129;01min\u001b[39;00m \u001b[43mqueue_get\u001b[49m\u001b[43m(\u001b[49m\u001b[43mqueue\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241m.\u001b[39mresult():\n\u001b[1;32m    501\u001b[0m         \u001b[38;5;28;01mif\u001b[39;00m failed:\n\u001b[1;32m    502\u001b[0m             exc, tb \u001b[38;5;241m=\u001b[39m loads(res_info)\n",
      "File \u001b[0;32m~/envs/selection-atlas/lib/python3.10/site-packages/dask/local.py:137\u001b[0m, in \u001b[0;36mqueue_get\u001b[0;34m(q)\u001b[0m\n\u001b[1;32m    136\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mqueue_get\u001b[39m(q):\n\u001b[0;32m--> 137\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mq\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/envs/selection-atlas/lib/python3.10/queue.py:171\u001b[0m, in \u001b[0;36mQueue.get\u001b[0;34m(self, block, timeout)\u001b[0m\n\u001b[1;32m    169\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m timeout \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m    170\u001b[0m     \u001b[38;5;28;01mwhile\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_qsize():\n\u001b[0;32m--> 171\u001b[0m         \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mnot_empty\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mwait\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    172\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m timeout \u001b[38;5;241m<\u001b[39m \u001b[38;5;241m0\u001b[39m:\n\u001b[1;32m    173\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mtimeout\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m must be a non-negative number\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "File \u001b[0;32m~/envs/selection-atlas/lib/python3.10/threading.py:320\u001b[0m, in \u001b[0;36mCondition.wait\u001b[0;34m(self, timeout)\u001b[0m\n\u001b[1;32m    318\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:    \u001b[38;5;66;03m# restore state no matter what (e.g., KeyboardInterrupt)\u001b[39;00m\n\u001b[1;32m    319\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m timeout \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[0;32m--> 320\u001b[0m         \u001b[43mwaiter\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43macquire\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    321\u001b[0m         gotit \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mTrue\u001b[39;00m\n\u001b[1;32m    322\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "for contig in contigs:\n",
    "    \n",
    "    display(HTML(f\"<h3>Chromosome {contig}</h3>\"))\n",
    "    \n",
    "    fig = plot_h12_g123_ihs_tracks(\n",
    "        contig=contig,\n",
    "        df_signals=df_signals,\n",
    "    );\n",
    "\n",
    "    bkplt.show(fig)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "518be38c",
   "metadata": {},
   "source": [
    "## Sampling information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2f8e30da",
   "metadata": {
    "tags": [
     "remove-input"
    ]
   },
   "outputs": [],
   "source": [
    "from ipyleaflet import Map, Marker, basemaps\n",
    "from ipywidgets import HTML\n",
    "\n",
    "center = cohort[['latitude', 'longitude']].to_list()\n",
    "m = Map(center=center, zoom=9, basemap=basemaps.OpenTopoMap)\n",
    "\n",
    "df = df_samples[['latitude', 'longitude', 'taxon']].groupby(['latitude', 'longitude', 'taxon']).size().to_frame().rename(columns={0: 'count'}).reset_index()\n",
    "\n",
    "for coh_id, row in df.iterrows():\n",
    "    lat, long = row[['latitude', 'longitude']]\n",
    "    \n",
    "    if row['taxon'] == 'gambiae':\n",
    "        color= 'red'\n",
    "    elif row['taxon'] == 'coluzzii':\n",
    "        color='cadetblue'\n",
    "    elif row['taxon'] == 'arabiensis':\n",
    "        color='lightgreen'\n",
    "    else: \n",
    "        color='gray'\n",
    "\n",
    "    marker = Marker(location=(lat, long), draggable=False, opacity=0.7, color=color)\n",
    "    m.add_layer(marker);\n",
    "\n",
    "    message2 = HTML()\n",
    "    message2.value = f\"n = {row['count']}\"\n",
    "    marker.popup = message2\n",
    "\n",
    "display(m)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5297e05e-4973-435f-8878-b16a1d404e55",
   "metadata": {
    "tags": [
     "remove-input"
    ]
   },
   "outputs": [],
   "source": [
    "\n",
    "if min_month >= 0:\n",
    "    # For this cohort we have month data, so show a breakdown of sample\n",
    "    # numbers by location and month.\n",
    "    \n",
    "    # Construct a pivot table counting samples.\n",
    "    df_loc_dt = df_samples.pivot_table(\n",
    "        index=[\"location\", \"longitude\", \"latitude\"],\n",
    "        columns=\"month\",\n",
    "        values=\"sample_id\",\n",
    "        aggfunc=\"count\",\n",
    "        fill_value=0,\n",
    "    )\n",
    "\n",
    "    # Tidy up the columns using a multi index.\n",
    "    df_loc_dt.reset_index(inplace=True)\n",
    "    cols = pd.MultiIndex.from_tuples(\n",
    "        [(\"Location\", \"Name\"), (\"Location\", \"Longitude\"), (\"Location\", \"Latitude\")] + \n",
    "        [(\"Date\", pd.to_datetime(x, format=\"%m\").month_name()) \n",
    "         for x in df_loc_dt.columns[3:]], \n",
    "    )\n",
    "    df_loc_dt.columns = cols\n",
    "    \n",
    "else:\n",
    "    # For this cohort we do not have month data, so show a breakdown of sample\n",
    "    # numbers by location only.\n",
    "    \n",
    "    # Construct a pivot table counting samples.\n",
    "    df_loc_dt = df_samples.groupby([\"location\", \"longitude\", \"latitude\"]).agg({\"sample_id\": \"count\"})\n",
    "\n",
    "    # Tidy up the columns using a multi index.\n",
    "    df_loc_dt.reset_index(inplace=True)\n",
    "    cols = pd.MultiIndex.from_tuples(\n",
    "        [(\"Location\", \"Name\"), (\"Location\", \"Longitude\"), (\"Location\", \"Latitude\"), (\"Date\", cohort.year)]\n",
    "    )\n",
    "    df_loc_dt.columns = cols\n",
    "\n",
    "# Style the table.\n",
    "df_loc_dt_styled = (\n",
    "    df_loc_dt\n",
    "    .style\n",
    "    .format(precision=3)\n",
    "    .set_caption(\"Number of samples collected.\")\n",
    "    .hide()\n",
    ")\n",
    "display(df_loc_dt_styled)    "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d53794f4",
   "metadata": {},
   "source": [
    "## Diagnostics\n",
    "### H12 calibration"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d1c173f3",
   "metadata": {
    "tags": [
     "remove-input"
    ]
   },
   "outputs": [],
   "source": [
    "display(Markdown(f\"Selected window size: **{h12_window_size:,}**\"))\n",
    "\n",
    "window_sizes = (100, 200, 500, 1000, 2000, 5000, 10000, 20000)\n",
    "\n",
    "ag3.plot_h12_calibration(\n",
    "    contig=h12_calibration_contig,\n",
    "    analysis=phasing_analysis,\n",
    "    sample_sets=sample_sets,\n",
    "    sample_query=cohort['sample_query'],\n",
    "    min_cohort_size=min_cohort_size,\n",
    "    max_cohort_size=max_cohort_size,\n",
    "    window_sizes=window_sizes,\n",
    ");\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8d836cf6",
   "metadata": {},
   "source": [
    "### G123 Calibration"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "63f8c0b4",
   "metadata": {
    "tags": [
     "remove-input"
    ]
   },
   "outputs": [],
   "source": [
    "display(Markdown(f\"Selected window size: **{g123_window_size:,}**\"))\n",
    "\n",
    "ag3.plot_g123_calibration(\n",
    "    contig=h12_calibration_contig,\n",
    "    sites=phasing_analysis,\n",
    "    site_mask=phasing_analysis,\n",
    "    sample_sets=sample_sets,\n",
    "    sample_query=cohort['sample_query'],\n",
    "    min_cohort_size=min_cohort_size,\n",
    "    max_cohort_size=max_cohort_size,\n",
    "    window_sizes=window_sizes,\n",
    ");"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ecff988c",
   "metadata": {},
   "source": [
    "## Data sources"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9e1bfa5b",
   "metadata": {
    "tags": [
     "remove-input"
    ]
   },
   "outputs": [],
   "source": [
    "df_sources_style = (\n",
    "    df_contributors\n",
    "    .style\n",
    "    .set_caption(\"MalariaGEN Vector Observatory partners, studies and sample sets contributing data for this cohort.\")\n",
    ")\n",
    "df_sources_style"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "42bc048b",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "celltoolbar": "Tags",
  "kernelspec": {
   "display_name": "selection-atlas",
   "language": "python",
   "name": "selection-atlas"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
